diff --git a/test/package/test_dependency_hooks.py b/test/package/test_dependency_hooks.py
index 6a4a239..b3bc1e8 100644
--- a/test/package/test_dependency_hooks.py
+++ b/test/package/test_dependency_hooks.py
@@ -5,6 +5,9 @@ from io import BytesIO
 from torch.package import PackageExporter
 from torch.testing._internal.common_utils import run_tests
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
 try:
     from .common import PackageTestCase
