diff --git a/test/package/test_resources.py b/test/package/test_resources.py
index b37290a..a3c9369 100644
--- a/test/package/test_resources.py
+++ b/test/package/test_resources.py
@@ -8,6 +8,9 @@ from unittest import skipIf
 from torch.package import PackageExporter, PackageImporter
 from torch.testing._internal.common_utils import run_tests
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
 try:
     from .common import PackageTestCase
