diff --git a/test/package/test_digraph.py b/test/package/test_digraph.py
index c6c9de0..d0e9741 100644
--- a/test/package/test_digraph.py
+++ b/test/package/test_digraph.py
@@ -3,6 +3,9 @@
 from torch.package._digraph import DiGraph
 from torch.testing._internal.common_utils import run_tests
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
 try:
     from .common import PackageTestCase
