diff --git a/test/distributed/_composable/test_replicate_with_compiler.py b/test/distributed/_composable/test_replicate_with_compiler.py
index 839bbcd..ccefbb8 100644
--- a/test/distributed/_composable/test_replicate_with_compiler.py
+++ b/test/distributed/_composable/test_replicate_with_compiler.py
@@ -1,3 +1,8 @@
+import torch_npu.testing
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["oncall: distributed"]
 
 import contextlib
@@ -36,9 +41,10 @@ from torch.testing._internal.common_utils import run_tests, skipIfRocm
 from torch.testing._internal.distributed.fake_pg import FakeStore
 from torch.testing._internal.inductor_utils import HAS_GPU
 from torch.utils.checkpoint import checkpoint
+HAS_GPU = True
 
-
-device_type = str(get_devtype())
+# device_type = str(get_devtype())
+device_type = "npu"
 
 DIM = 2000
 
@@ -96,10 +102,11 @@ class ReplicateTest(MultiProcessInductorTestCase):
         checkpoint: bool = False,
         device: Union[str, torch.device],
     ):
+        torch.npu.set_device(self.rank)
         self.create_pg(device)
         torch._dynamo.config.optimize_ddp = "python_reducer"
         torch.manual_seed(123)
-        model = Net(checkpoint=checkpoint).to(device)
+        model = Net(checkpoint=checkpoint)
         input = torch.randn([1, DIM], device=device)
 
         compiled_replicate_model = replicate(deepcopy(model))
