diff --git a/test/distributed/_composable/test_composability/test_2d_composability.py b/test/distributed/_composable/test_composability/test_2d_composability.py
index cbd06e13c1d..2fb6fda9bed 100644
--- a/test/distributed/_composable/test_composability/test_2d_composability.py
+++ b/test/distributed/_composable/test_composability/test_2d_composability.py
@@ -58,6 +58,8 @@ from torch.testing._internal.distributed._tensor.common_dtensor import (
     MLPModule,
     ModelArgs,
     Transformer,
+from torch_npu.testing.common_distributed import with_comms
+    # with_comms,
 )
 from torch.testing._internal.distributed.checkpoint_utils import with_temp_dir
 
@@ -121,6 +123,7 @@ class TestFullyShard2DTraining(FSDPTestContinuous):
         # Prefer to test with >=4 GPUs, but for 2 GPUs, use 2-way TP
         dp_size = 2 if self.world_size > 2 else 1
         return init_device_mesh(
+            "npu", (dp_size, self.world_size // dp_size), mesh_dim_names=("dp", "tp")
             device_type,
             (dp_size, self.world_size // dp_size),
             mesh_dim_names=("dp", "tp"),
@@ -434,6 +437,7 @@ class TestFullyShard2DStateDict(DTensorContinuousTestBase):
     def test_fully_shard_tp_2d_set_full_state_dict(self):
         dummy_model = SimpleModel().to(device_type)
         mesh_2d = init_device_mesh(
+            "npu",
             device_type,
             (2, self.world_size // 2),
             mesh_dim_names=("dp", "tp"),
