diff --git a/test/profiler/test_cpp_thread.py b/test/profiler/test_cpp_thread.py
index 9dbecf9..97273a8 100644
--- a/test/profiler/test_cpp_thread.py
+++ b/test/profiler/test_cpp_thread.py
@@ -9,6 +9,9 @@ import torch.utils.cpp_extension
 from torch._environment import is_fbcode
 from torch.testing._internal.common_utils import IS_WINDOWS, run_tests, TestCase
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
 if is_fbcode():
     import caffe2.test.profiler_test_cpp_thread_lib as cpp  # @manual=//caffe2/test:profiler_test_cpp_thread_lib
