diff --git a/test/functorch/common_utils.py b/test/functorch/common_utils.py
index 480f079..12eaf96 100644
--- a/test/functorch/common_utils.py
+++ b/test/functorch/common_utils.py
@@ -305,6 +305,9 @@ def _compute_quantities_for_vmap_test(
     batched_args, kwarg_values = maybe_clone_inputs()
 
     if compute_loop_out:
+        op_name = op.__name__
+        if op_name == 'ones_like' and kwarg_values.get('device') == 'cpu':
+            del kwarg_values['device']
         loop_out = loop(op, in_dims, out_dim, batch_size, *batched_args, **kwarg_values)
     else:
         loop_out = None
