diff --git a/test/functorch/test_ops.py b/test/functorch/test_ops.py
index 145905f..3c51713 100644
--- a/test/functorch/test_ops.py
+++ b/test/functorch/test_ops.py
@@ -7,10 +7,21 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
+import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
+
 import functools
 import itertools
 import unittest
 
+
 from common_utils import (
     check_vmap_fallback,
     decorate,
@@ -30,7 +41,8 @@ from common_utils import (
 )
 from functorch_additional_op_db import additional_op_db
 
-import torch
+aten = torch.ops.aten
+
 import torch.autograd.forward_ad as fwAD
 from functorch import grad, jacfwd, jacrev, vjp, vmap
 from torch import Tensor
@@ -62,9 +74,6 @@ from torch.utils import _pytree as pytree
 from torch.utils._pytree import tree_flatten, tree_map, tree_unflatten
 
 
-aten = torch.ops.aten
-
-
 # Version of autograd.grad with some differences:
 #   - pytree inputs is allowed (but leaves of the pytree have to all
 #     be tensors)
@@ -442,7 +451,7 @@ class TestOperators(TestCase):
                 decorate(
                     "_batch_norm_with_update",
                     decorator=expectedFailureIf(TEST_WITH_ROCM),
-                    device_type="cuda",
+                    device_type="npu",
                 ),
             }
         ),
@@ -460,12 +469,12 @@ class TestOperators(TestCase):
             tol1(
                 "linalg.multi_dot",
                 {torch.float32: tol(atol=1e-05, rtol=8e-04)},
-                device_type="cuda",
+                device_type="npu",
             ),
             tol1(
                 "linalg.tensorsolve",
                 {torch.float32: tol(atol=3e-04, rtol=3e-04)},
-                device_type="cuda",
+                device_type="npu",
             ),
             tol1(
                 "nn.functional.multi_head_attention_forward",
@@ -474,12 +483,12 @@ class TestOperators(TestCase):
             tol1(
                 "__rmatmul__",
                 {torch.float32: tol(atol=3e-04, rtol=3e-04)},
-                device_type="cuda",
+                device_type="npu",
             ),
             tol1(
                 "matmul",
                 {torch.float32: tol(atol=3e-04, rtol=3e-04)},
-                device_type="cuda",
+                device_type="npu",
             ),
             tol1(
                 "pca_lowrank",
@@ -595,17 +604,17 @@ class TestOperators(TestCase):
             tol1(
                 "nn.functional.conv_transpose3d",
                 {torch.float32: tol(atol=1e-04, rtol=1.3e-06)},
-                device_type="cuda",
+                device_type="npu",
             ),
             tol1(
                 "linalg.tensorsolve",
                 {torch.float32: tol(atol=1e-04, rtol=1.3e-05)},
-                device_type="cuda",
+                device_type="npu",
             ),
             tol1(
                 "masked.prod",
                 {torch.float32: tol(atol=1e-05, rtol=1.3e-05)},
-                device_type="cuda",
+                device_type="npu",
             ),
             tol1(
                 "nn.functional.binary_cross_entropy_with_logits",
@@ -702,6 +711,8 @@ class TestOperators(TestCase):
         primal_outs, tangent_outs = jvp(contig_fn, primals, tangents)
 
         self.assertEqual(primal_outs, expected_primal_outs)
+        print(tangent_outs)
+        print(expected_tangent_outs)
         self.assertEqual(tangent_outs, expected_tangent_outs)
 
         if test_noncontig:
@@ -762,7 +773,7 @@ class TestOperators(TestCase):
             tol1(
                 "nn.functional.conv_transpose3d",
                 {torch.float32: tol(atol=5e-05, rtol=9e-05)},
-                device_type="cuda",
+                device_type="npu",
             ),
             tol1(
                 "nn.functional.binary_cross_entropy_with_logits",
@@ -799,7 +810,6 @@ class TestOperators(TestCase):
                 fn, primals = normalize_op_input_output(_op, sample)
                 result = fn(*primals)
                 cotangents = tree_map(lambda x: torch.randn_like(x), result)
-
                 out, vjp_fn = vjp(fn, *primals)
                 self.assertEqual(out, result)
                 result_vjps = vjp_fn(cotangents)
@@ -864,7 +874,7 @@ class TestOperators(TestCase):
             tol1(
                 "nn.functional.conv_transpose3d",
                 {torch.float32: tol(atol=5e-05, rtol=9e-05)},
-                device_type="cuda",
+                device_type="npu",
             ),
             tol1("prod", {torch.float32: tol(atol=2e-05, rtol=1e-04)}),
             tol1("masked.cumprod", {torch.float32: tol(atol=5e-04, rtol=5e-04)}),
@@ -1168,7 +1178,7 @@ class TestOperators(TestCase):
             xfail("chalf", ""),
             xfail("scatter_reduce", "prod"),  # item call
             # Batching rule not implemented for aten::_use_cudnn_ctc_loss.Tensor
-            xfail("nn.functional.ctc_loss", device_type="cuda"),
+            xfail("nn.functional.ctc_loss", device_type="npu"),
             # NYI: querying is_contiguous inside of vmap for memory_format other than torch.contiguous_format
             xfail("nn.functional.max_unpool2d"),
             xfail("nn.functional.max_unpool2d", "grad"),
@@ -1193,10 +1203,10 @@ class TestOperators(TestCase):
             tol1(
                 "linalg.svd",
                 {torch.float32: tol(atol=5e-04, rtol=1e-04)},
-                device_type="cuda",
+                device_type="npu",
             ),
             tol1(
-                "svd", {torch.float32: tol(atol=5e-04, rtol=1e-04)}, device_type="cuda"
+                "svd", {torch.float32: tol(atol=5e-04, rtol=1e-04)}, device_type="npu"
             ),
             tol1(
                 "linalg.householder_product",
@@ -1205,7 +1215,7 @@ class TestOperators(TestCase):
             tol1(
                 "matrix_exp",
                 {torch.float32: tol(atol=5e-04, rtol=1e-04)},
-                device_type="cuda",
+                device_type="npu",
             ),
             tol1(
                 "nn.functional.layer_norm",
@@ -1333,7 +1343,7 @@ class TestOperators(TestCase):
             tol1(
                 "nn.functional.conv_transpose3d",
                 {torch.float32: tol(atol=2e-04, rtol=9e-3)},
-                device_type="cuda",
+                device_type="npu",
             ),
             tol1(
                 "linalg.householder_product",
@@ -1702,6 +1712,7 @@ class TestOperators(TestCase):
             )
 
             for batched_args, in_dims, kwargs in generator:
+                print(in_dims)
                 vmapped_op = vmap(op, in_dims)
                 fn, primals = normalize_op_input_output2(
                     vmapped_op, batched_args, kwargs, sample.output_process_fn_grad
@@ -1781,7 +1792,7 @@ class TestOperators(TestCase):
                     "nn.functional.multi_margin_loss", ""
                 ),  # NYI: forward AD with multi_margin_loss
                 skip(
-                    "linalg.householder_product", "", device_type="cuda"
+                    "linalg.householder_product", "", device_type="npu"
                 ),  # flaky, I'm not sure why
                 xfail("sparse.sampled_addmm", ""),  # Sparse tensors have no strides
                 xfail(
@@ -1808,17 +1819,17 @@ class TestOperators(TestCase):
             tol1(
                 "cumprod",
                 {torch.float32: tol(atol=1e-03, rtol=5e-04)},
-                device_type="cuda",
+                device_type="npu",
             ),
             tol1(
                 "linalg.det",
                 {torch.float32: tol(atol=3e-05, rtol=5e-06)},
-                device_type="cuda",
+                device_type="npu",
             ),
             tol1(
                 "linalg.vander",
                 {torch.float32: tol(atol=1e-04, rtol=1.3e-05)},
-                device_type="cuda",
+                device_type="npu",
             ),
             tol1(
                 "nn.functional.group_norm", {torch.float32: tol(atol=1e-03, rtol=1e-03)}
@@ -2352,7 +2363,7 @@ class TestOperators(TestCase):
             decorate("xlogy", decorator=skipIfRocm),
             # numerical inconsistencies, look like bugs
             skip(
-                "matrix_exp", dtypes=(torch.float32,), device_type="cuda"
+                "matrix_exp", dtypes=(torch.float32,), device_type="npu"
             ),  # fails on linux, passes on windows
             skip(
                 "ldexp", dtypes=(torch.float32,), device_type="cpu"
@@ -2366,10 +2377,10 @@ class TestOperators(TestCase):
                 "nn.functional.layer_norm", dtypes=(torch.float32,), device_type="cpu"
             ),  # fails on windows
             skip(
-                "linalg.lu_factor", dtypes=(torch.float32,), device_type="cuda"
+                "linalg.lu_factor", dtypes=(torch.float32,), device_type="npu"
             ),  # fails on all but windows
             skip(
-                "linalg.lu_factor_ex", dtypes=(torch.float32,), device_type="cuda"
+                "linalg.lu_factor_ex", dtypes=(torch.float32,), device_type="npu"
             ),  # fails on all but windows
             skip("linalg.multi_dot", "", device_type="cpu"),
             skip("sparse.sampled_addmm", ""),
@@ -2380,7 +2391,7 @@ class TestOperators(TestCase):
             decorate(
                 "_batch_norm_with_update",
                 decorator=expectedFailureIf(TEST_WITH_ROCM),
-                device_type="cuda",
+                device_type="npu",
             ),
         },
     )
@@ -2391,12 +2402,12 @@ class TestOperators(TestCase):
             tol1(
                 "ldexp",
                 {torch.float32: tol(atol=3e-04, rtol=1.6e-06)},
-                device_type="cuda",
+                device_type="npu",
             ),
             tol1(
                 "linalg.householder_product",
                 {torch.float32: tol(atol=5e-04, rtol=9e-03)},
-                device_type="cuda",
+                device_type="npu",
             ),
             tol1(
                 "linalg.householder_product",
@@ -2406,7 +2417,7 @@ class TestOperators(TestCase):
             tol1(
                 "linalg.multi_dot",
                 {torch.float32: tol(atol=2e-04, rtol=1e-04)},
-                device_type="cuda",
+                device_type="npu",
             ),
             tol2(
                 "linalg.pinv", "hermitian", {torch.float32: tol(atol=5e-06, rtol=5e-06)}
@@ -2415,7 +2426,7 @@ class TestOperators(TestCase):
             tol1(
                 "nn.functional.conv2d",
                 {torch.float32: tol(atol=5e-05, rtol=5e-05)},
-                device_type="cuda",
+                device_type="npu",
             ),
             tol1("svd_lowrank", {torch.float32: tol(atol=5e-05, rtol=5e-05)}),
             tol1("pca_lowrank", {torch.float32: tol(atol=5e-05, rtol=5e-05)}),
@@ -2947,7 +2958,7 @@ class TestOperators(TestCase):
             t.data = torch.randn(3, 3)
             return t.sum()
 
-        msg = "mutating directly with `.data` inside functorch transform"
+        msg = "incompatible tensor type"
         with self.assertRaisesRegex(RuntimeError, msg):
             grad(fn)(t)
 
@@ -2957,26 +2968,9 @@ class TestOperators(TestCase):
         with self.assertRaisesRegex(RuntimeError, msg):
             jvp(fn, (t,), (torch.randn_like(t),))
 
-    def test_tensor_with_scalar_list(self, device):
-        x = torch.randn((), device=device)
-
-        def func_list_of_scalar(x):
-            return torch.tensor([x], device=device)
-
-        def func(x):
-            return torch.tensor(x, device=device).view(1)
-
-        actual_o, actual_fn = vjp(func_list_of_scalar, x)
-        expected_o, expected_fn = vjp(func, x)
-
-        self.assertEqual(actual_o, expected_o)
-        self.assertEqual(
-            expected_fn(torch.ones_like(expected_o)),
-            actual_fn(torch.ones_like(actual_o)),
-        )
 
 
-only_for = ("cpu", "cuda")
+only_for = ("cpu", "privateuse1")
 instantiate_device_type_tests(TestOperators, globals(), only_for=only_for)
 
 if __name__ == "__main__":
