diff --git a/test/functorch/test_aotdispatch.py b/test/functorch/test_aotdispatch.py
index bcf99f6..a5f6442 100644
--- a/test/functorch/test_aotdispatch.py
+++ b/test/functorch/test_aotdispatch.py
@@ -15,9 +15,17 @@ from functools import partial, wraps
 from typing import Any, Callable, Optional, Union
 from unittest.mock import patch
 
-from common_utils import decorate, decorateForModules, skip, skipOps, xfail
-
 import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
+
+from common_utils import decorate, decorateForModules, skip, skipOps, xfail
 import torch._dynamo as torchdynamo
 import torch.nn as nn
 import torch.utils._pytree as pytree
@@ -2663,7 +2671,7 @@ def forward(self, arg0_1, arg1_1):
         self.assertTrue("as_strided_scatter" in str(fw_graph_overlap1.code))
         self.assertTrue("as_strided_scatter" in str(fw_graph_overlap2.code))
 
-    @unittest.skipIf(not torch.cuda.is_available(), "CUDA is unavailable")
+    @unittest.skipIf(not torch_npu.npu.is_available(), "NPU is unavailable")
     def test_mem_leak_from_save_for_bw(self):
         # See a full diagnosis at this issue: https://github.com/pytorch/pytorch/issues/94990
         # Note [Detaching saved tensors in AOTAutograd]
@@ -2683,12 +2691,12 @@ def forward(self, arg0_1, arg1_1):
 
         f_compiled = aot_function(f, nop)
         inps = [
-            torch.ones(8, 8, device="cuda", requires_grad=True),
-            torch.ones(1, 4, 1, device="cuda", requires_grad=True),
+            torch.ones(8, 8, device="npu", requires_grad=True),
+            torch.ones(1, 4, 1, device="npu", requires_grad=True),
         ]
-        mem_before = torch.cuda.memory_allocated()
+        mem_before = torch_npu.npu.memory_allocated()
         f_compiled(*inps)
-        mem_after = torch.cuda.memory_allocated()
+        mem_after = torch_npu.npu.memory_allocated()
         self.assertTrue(mem_after == mem_before)
 
     def test_output_aliases_multiple_inputs_get_correct_one(self):
@@ -2993,14 +3001,14 @@ def forward(self, primals_1, primals_2, primals_3):
     return (as_strided_scatter, add_2, view_2, unsqueeze_1)""",
         )  # noqa: B950
 
-    @unittest.skipIf(not torch.cuda.is_available(), "CUDA is unavailable")
+    @unittest.skipIf(not torch_npu.npu.is_available(), "NPU is unavailable")
     def test_synthetic_base_base_attribute_is_none(self):
         def f(a, b):
             a.add_(1)
             return a + b
 
         def inp_callable():
-            base = torch.ones(4, 4, device="cuda")
+            base = torch.ones(4, 4, device="npu")
             # detach() so that none of the inputs have a ._base attribute.
             a = base[0].detach()
             b = base[1].detach()
@@ -3428,14 +3436,14 @@ def forward(self, tangents_1):
 
         self.verify_aot_autograd(f, [torch.randn(3)])
 
-    @unittest.skipIf(not torch.cuda.is_available(), "CUDA is unavailable")
+    @unittest.skipIf(not torch_npu.npu.is_available(), "NPU is unavailable")
     def test_autocast_disable_guard(self):
         with torch._C._DisableAutocast():
-            x = torch.rand([4, 4]).cuda()
+            x = torch.rand([4, 4]).npu()
             y = x @ x
             self.assertEqual(y.dtype, torch.float32)
 
-    @unittest.skipIf(not torch.cuda.is_available(), "CUDA is unavailable")
+    @unittest.skipIf(not torch_npu.npu.is_available(), "NPU is unavailable")
     def test_nonidempotent_amp(self):
         def f(self_s_emb, add_3):
             einsum_2 = torch.functional.einsum("ah,th->t", self_s_emb, add_3)
@@ -3443,21 +3451,21 @@ def forward(self, tangents_1):
             return (log_softmax_2,)
 
         args = [
-            torch.rand((1, 256), dtype=torch.float32, device="cuda"),
-            torch.rand((30, 256), dtype=torch.float16, device="cuda"),
+            torch.rand((1, 256), dtype=torch.float32, device="npu"),
+            torch.rand((30, 256), dtype=torch.float16, device="npu"),
         ]
-        with torch.cuda.amp.autocast(enabled=True):
+        with torch_npu.npu.amp.autocast(enabled=True):
             self.verify_aot_autograd(f, args)
 
         args = [e.requires_grad_(True) for e in args]
-        with torch.cuda.amp.autocast(enabled=True):
+        with torch_npu.npu.amp.autocast(enabled=True):
             self.verify_aot_autograd(f, args)
 
-    @unittest.skipIf(not torch.cuda.is_available(), "CUDA is unavailable")
+    @unittest.skipIf(not torch_npu.npu.is_available(), "NPU is unavailable")
     @unittest.skipIf(not torch.backends.cudnn.is_available(), "CUDNN is unavailable")
     @skipIfRocm  # https://github.com/pytorch/pytorch/issues/96560
     def test_batch_norm_amp(self):
-        device = "cuda"
+        device = "npu"
         input_dtype = torch.float16
         param_dtype = torch.float32
         weight, bias = (
@@ -5320,17 +5328,17 @@ class TestPartitioning(AOTTestCase):
         aot_fn(x)
         self.assertTrue(inference_graph_cell[0] is not None)
 
-    @unittest.skipIf(not torch.cuda.is_available(), "CUDA is unavailable")
+    @unittest.skipIf(not torch_npu.npu.is_available(), "NPU is unavailable")
     @unittest.skipIf(not USE_TORCHVISION, "test requires torchvision")
     def test_autocast(self):
-        mod = torchvision.models.resnet18().cuda()
+        mod = torchvision.models.resnet18().npu()
         mod.train()
 
-        x = torch.randn(16, 3, 32, 32, device="cuda")
+        x = torch.randn(16, 3, 32, 32, device="npu")
         aot_mod = memory_efficient_fusion(mod)
 
         # Ensure that AOT Autograd works with AMP
-        with torch.cuda.amp.autocast(True):
+        with torch_npu.npu.amp.autocast(True):
             res = aot_mod(x)
         res.sum().backward()
 
@@ -5825,7 +5833,7 @@ class GradsNoForceContiguousContextManager(ContextDecorator):
         def foo2_meta(a):
             return a.clone()
 
-        for backend in ["CPU", "CUDA"]:
+        for backend in ["CPU", "NPU"]:
             self.lib.impl("foo", foo_impl, backend)
             self.lib.impl("foo2", foo2_impl, backend)
 
@@ -6749,7 +6757,7 @@ class TestEagerFusionModuleInfo(AOTTestCase):
 
 
 instantiate_parametrized_tests(TestAOTAutograd)
-only_for = "cpu"
+only_for = ['cpu']
 instantiate_device_type_tests(
     TestPythonKey,
     globals(),
