diff --git a/test/functorch/test_ac_logging.py b/test/functorch/test_ac_logging.py
index 03ddb7d..e00b763 100644
--- a/test/functorch/test_ac_logging.py
+++ b/test/functorch/test_ac_logging.py
@@ -1,5 +1,14 @@
 # Owner(s): ["module: functorch"]
 from unittest.mock import MagicMock, patch
+import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
 
 from torch._functorch._activation_checkpointing.ac_logging_utils import (
     create_activation_checkpointing_logging_structure_payload,
