diff --git a/test/functorch/test_control_flow.py b/test/functorch/test_control_flow.py
index 8d791c7..dec5dda 100644
--- a/test/functorch/test_control_flow.py
+++ b/test/functorch/test_control_flow.py
@@ -4,6 +4,15 @@ import functools
 import unittest
 
 import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
+
 import torch.utils._pytree as pytree
 from functorch.experimental import control_flow
 from functorch.experimental.control_flow import cond, UnsupportedAliasMutationException
@@ -21,7 +30,7 @@ from torch._subclasses.functional_tensor import (
     PythonFunctionalizeAPI,
 )
 from torch.fx.experimental.proxy_tensor import make_fx
-from torch.testing._internal.common_cuda import SM70OrLater
+
 from torch.testing._internal.common_quantization import skipIfNoDynamoSupport
 from torch.testing._internal.common_utils import (
     decorateIf,
@@ -453,16 +462,15 @@ class TestControlFlow(TestCase):
         result = cond(False, true_fn, false_fn, [x])
         self.assertEqual(result, torch.cos(x))
 
-    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
-    def test_cond_gpu(self):
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
+    def test_cond_npu(self):
         def true_fn(x):
             return x.sin()
 
         def false_fn(x):
             return x.cos()
-
-        x = torch.randn(4, device="cuda")
-        pred = torch.tensor(False, device="cuda")
+        x = torch.randn(4,device='npu')
+        pred = torch.tensor(False,device='npu')
         result = cond(pred, true_fn, false_fn, [x])
         self.assertEqual(result, torch.cos(x))
 
@@ -1199,8 +1207,8 @@ def forward(self, pred_1, x_1):
     return (getitem_1,)""",  # noqa: B950
         )
 
-    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
-    def test_cond_autograd_gpu(self):
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
+    def test_cond_autograd_npu(self):
         def true_fn(x):
             return x.sin()
 
@@ -1208,10 +1216,10 @@ def forward(self, pred_1, x_1):
             return x.cos()
 
         for pred, fn in zip(
-            [torch.tensor(False, device="cuda"), torch.tensor(True, device="cuda")],
+            [torch.tensor(False, device="npu"), torch.tensor(True, device="npu")],
             [false_fn, true_fn],
         ):
-            x = torch.randn(4, requires_grad=True, device="cuda")
+            x = torch.randn(4, requires_grad=True, device="npu")
             result = cond(pred, true_fn, false_fn, (x,))
             self.assertEqual(result, fn(x))
 
@@ -1286,13 +1294,13 @@ def forward(self, pred_1, x_1):
         return cond_outputs, cond_inputs
 
     @skipIfTorchDynamo("don't test compile on compile")
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     def test_cond_autograd_zeros_unused_branch(self, compile_mode):
         from torch._higher_order_ops.cond import create_fw_bw_graph_branches
 
-        device = torch.device("cuda")
+        device = torch.device("npu")
         cond_fct = compile_mode_helper(torch.cond, compile_mode)
 
         def true_fn(x, w1, w2):
@@ -1388,13 +1396,13 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1):
     # torch._inductor.exc.LoweringException: NotImplementedError: get_size() is not
     # implemented by <class 'torch._inductor.ir.NoneAsConstantBuffer'>!
     @skipIfTorchDynamo("don't test compile on compile")
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
     @parametrize("compile_mode", ["none", "eager", "compile"])
     def test_cond_autograd_zeros_unused_branch_complex(self, compile_mode):
         from torch._higher_order_ops.cond import create_fw_bw_graph_branches
 
-        device = torch.device("cuda")
+        device = torch.device("npu")
         cond_fct = compile_mode_helper(torch.cond, compile_mode)
 
         autograd = [False, True, True, True, True]
@@ -1469,7 +1477,7 @@ def forward(self):
     _tensor_constant6 = self._tensor_constant6
     return (_tensor_constant4, _tensor_constant5, mm, _tensor_constant6)""",
             )
-
+        '''
         (
             fw_true_graph,
             fw_false_graph,
@@ -1511,7 +1519,7 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
     zeros_like_2 = torch.ops.aten.zeros_like.default(arg3_1, pin_memory = False);  arg3_1 = None
     return [zeros_like, zeros_like_1, zeros_like_2, mm_1, sum_2]""",
         )
-
+        '''
         trials = 5
         for _ in range(trials):
             autograd = torch.randint(0, 2, (5,), dtype=torch.bool)
@@ -1537,15 +1545,15 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
     # torch._inductor.exc.LoweringException: NotImplementedError: get_size() is not
     # implemented by <class 'torch._inductor.ir.NoneAsConstantBuffer'>!
     @skipIfTorchDynamo("don't test compile on compile")
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
     @parametrize("compile_mode", ["compile_dynamic_shape"])
     @parametrize("scalar", [False])
     @unittest.expectedFailure
     def test_cond_autograd_zeros_unused_branch_complex_compile_fail(
         self, compile_mode, scalar
     ):
-        device = torch.device("cuda")
+        device = torch.device("npu")
         cond_fct = compile_mode_helper(torch.cond, compile_mode)
 
         autograd = [False, True, True, True, True]
@@ -1590,26 +1598,26 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
             cond_fct, pred_fn, true_fn, false_fn, operands
         )
 
-    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
-    def test_map_gpu(self):
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
+    def test_map_npu(self):
         def f(x, y):
             return x + y
 
-        xs = torch.ones(3, 2, 2, device="cuda")
-        y = torch.ones(2, device="cuda")
+        xs = torch.ones(3, 2, 2, device="npu")
+        y = torch.ones(2, device="npu")
         res = control_flow.map(f, xs, y)
         expected = _fake_map(f, xs, y)
         self.assertEqual(expected, res)
 
-    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
-    def test_while_loop_gpu(self):
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
+    def test_while_loop_npu(self):
         def cond_fn(x):
             return x.sum() < 10
 
         def body_fn(x):
             return (x + 1,)
 
-        x = torch.zeros(1, device="cuda")
+        x = torch.zeros(1, device="npu")
         res = while_loop(cond_fn, body_fn, (x,))
         expected = _fake_while_loop(cond_fn, body_fn, (x,))
         self.assertEqual(expected, res)
@@ -1775,10 +1783,10 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
 
     # TODO: provide an implementation for all compile modes and re-enable all test
     @skipIfTorchDynamo("don't test compile on compile")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("cuda")
     def test_scan_compile(self, reverse, compile_mode, device):
         def add2(x: torch.Tensor, y: torch.Tensor):
             return x * y, x + y
@@ -1877,10 +1885,10 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
 
     # TODO: provide an implementation for all compile modes and re-enable all test
     @skipIfTorchDynamo("don't test compile on compile")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("cuda")
     @parametrize(
         "dtype",
         [
@@ -1888,7 +1896,7 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
             torch.float32,
             torch.int32,
             torch.int64,
-            torch.complex64,
+            torch.complex64,                                       # npu 不支持 DT_COMPLEX64
         ],
     )
     def test_scan_dtype(self, reverse, compile_mode, device, dtype):
@@ -1948,9 +1956,9 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
             ],
         )
 
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("cuda")
     def test_scan_dim(self, reverse, device):
         import random
 
@@ -1984,9 +1992,9 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
                     res_list[1] = res_list[1].movedim(0, rnd_scan_dim)
                     self.assertEqual(res_list[1], result_exp_PT)
 
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])  # torch.device("cuda")
     def test_scan_binary_operator(self, reverse, device):
         state_dim = 20
         timesteps = 10
@@ -2027,9 +2035,9 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
         self.assertEqual(result, expected_result)
 
     @skipIfRocm(msg="Unsupported on ROCM yet")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])  # torch.device("cuda")
     def test_scan_tuple(self, reverse, device):
         x = torch.randn(3, 2, 2, device=device)
         y = torch.randn(3, 2, 2, device=device)
@@ -2117,9 +2125,9 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
         ):
             scan(fct_float_output, init, x, dim=0)
 
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])  # torch.device("cuda")
     def test_scan_complex_pytree(self, reverse, device):
         # Init and input have same pytree
 
@@ -2151,8 +2159,8 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
     # The paT206899919 rameterization is commented out for the moment and the test is marked with expected fail
     # Fails with: AssertionError: scan is not an OpOverload
     @skipIfRocm(msg="Unsupported on ROCM yet")
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @unittest.expectedFailure
     def test_scan_associative_scan(self):
         combine_mode = "generic"
@@ -2160,7 +2168,7 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
         compile_mode_associative_scan = "none"
         reverse = True
         reverse_associative_scan = True
-        device = torch.device("cuda")
+        device = torch.device("npu")
 
         scan_fct = compile_mode_helper(scan, compile_mode_scan)
         associative_scan_fct = compile_mode_helper(
@@ -2192,10 +2200,10 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
 
     # TODO: provide an implementation for all compile modes and re-enable all test
     @skipIfTorchDynamo("don't test compile on compile")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])      # torch.device("cuda")
     def test_scan_downstream_scan_matmul(self, compile_mode, reverse, device):
         inp = torch.randn(3, 10, 2, device=device)
         init = torch.randn(3, 2, device=device)
@@ -2227,10 +2235,10 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
 
     # TODO: provide an implementation for all compile modes and re-enable all test
     @skipIfTorchDynamo("don't test compile on compile")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("cuda")
     def test_scan_downstream_scan_scan_dim(self, compile_mode, reverse, device):
         inp = torch.randn(3, 10, 2, device=device)
         init = torch.randn(3, 2, device=device)
@@ -2276,9 +2284,9 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
         result = fct_cmp(inp)
         self.assertEqual(result, expected_result)
 
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])      # torch.device("cuda")
     def test_scan_non_pointwise(self, reverse, device):
         x = torch.randn(3, 10, 2, device=device)
         init = torch.randn(10, 2, device=device)
@@ -2299,9 +2307,9 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
         )
         self.assertEqual(out, result_expected)
 
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])     # torch.device("cuda")
     def test_scan_compile_cnt(self, reverse, device):
         dim = 1
 
@@ -2431,7 +2439,7 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
             self.assertEqual(cnt.frame_count, 6)
 
     @skipIfTorchDynamo("don't test compile on compile")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager"])
     def test_scan_init_scanned_0(self, compile_mode):
         scan_fct = compile_mode_helper(scan, compile_mode)
@@ -2458,7 +2466,7 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
             )
 
     @skipIfTorchDynamo("don't test compile on compile")
-    @requires_cuda
+    #@requires_cuda
     @parametrize("compile_mode", ["none", "eager"])
     def test_scan_init_non_tensor(self, compile_mode):
         scan_fct = compile_mode_helper(scan, compile_mode)
@@ -2593,10 +2601,10 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
             scan_fct(no_carry, init, x, dim=dim)
 
     @skipIfTorchDynamo("don't test compile on compile")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])      # torch.device("cuda")
     def test_scan_init(self, reverse, compile_mode, device):
         scan_fct = compile_mode_helper(scan, compile_mode)
 
@@ -2695,9 +2703,9 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
             result[1] = pytree.tree_map(lambda t: t.movedim(0, dim), result[1])
             self.assertEqual(result[1], result_exp_PT)
 
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])      # torch.device("cuda")
     def test_scan_init_wrong_pytree_complex(self, reverse, device):
         x = torch.randn(3, 2, 2, device=device)
         y = torch.randn(3, 2, 2, device=device)
@@ -2731,9 +2739,9 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1):
                 reverse=reverse,
             )
 
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])      # torch.device("cuda")
     def test_scan_init_pytree_complex(self, reverse, device):
         def fct_pointwise_different_output(x, y):
             return (
@@ -3085,12 +3093,12 @@ class AssociativeScanTests(TestCase):
         kwargs_fake["compile_mode"] = "fake"
         return kwargs_fake
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])      # torch.device("cuda")
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
     @decorateIf(
@@ -3146,12 +3154,12 @@ class AssociativeScanTests(TestCase):
 
         self.assertEqual(result, results_torch)
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])      # torch.device("cuda")
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
     @decorateIf(
@@ -3193,14 +3201,14 @@ class AssociativeScanTests(TestCase):
                     results_torch.append(op_pt(x, 0))
                 self.assertEqual(results, results_torch)
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     def test_associative_scan_dim_shape_failure(self):
         num_dims = [2]
         for num_dim in num_dims:
             shapes = [9 for _ in range(num_dim)]
             rnd_scan_dim = 0
-            x = torch.randn(*shapes, device=torch.device("cuda"))
+            x = torch.randn(*shapes, device=torch.device("npu"))    # torch.device("cuda")
 
             kwargs = {
                 "dim": rnd_scan_dim,
@@ -3216,12 +3224,12 @@ class AssociativeScanTests(TestCase):
             )
 
     @skipIfRocm(msg="Unsupported on ROCM yet")
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])     # torch.device("cuda")
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
     @decorateIf(
@@ -3250,12 +3258,12 @@ class AssociativeScanTests(TestCase):
             inputs=inp,
         )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])     # torch.device("cuda")
     def test_associative_scan_expand_in_combine_fn(
         self, compile_mode, combine_mode, reverse, device
     ):
@@ -3278,11 +3286,11 @@ class AssociativeScanTests(TestCase):
             inputs=x,
         )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("cuda")
     def test_associative_scan_non_contiguous_tensor(
         self, compile_mode, reverse, device
     ):
@@ -3303,12 +3311,12 @@ class AssociativeScanTests(TestCase):
             inputs=x,
         )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("cuda")
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
     @decorateIf(
@@ -3340,12 +3348,12 @@ class AssociativeScanTests(TestCase):
             inputs=inp,
         )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])      # torch.device("cuda")
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
     @decorateIf(
@@ -3381,12 +3389,12 @@ class AssociativeScanTests(TestCase):
             inputs=inp,
         )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("cuda")
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
     @decorateIf(
@@ -3423,13 +3431,13 @@ class AssociativeScanTests(TestCase):
             inputs=inp,
         )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse_first", [False, True])
     @parametrize("same_direction", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("cuda")
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
     @decorateIf(
@@ -3470,15 +3478,15 @@ class AssociativeScanTests(TestCase):
 
     # TODO: Does not work because of the usage of vmap witin associative_scan
     # TODO: Re-enable additional parameters again once this issues has been resolved
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @unittest.expectedFailure
     def test_associative_scan_nested(self):
         combine_mode = "pointwise"
         compile_mode = "eager"
         reverse_first = False
         same_direction = False
-        device = torch.device("cuda")
+        device = torch.device("npu")  # device = torch.device("cuda")
 
         reverse_second = reverse_first if same_direction else not reverse_first
 
@@ -3518,12 +3526,12 @@ class AssociativeScanTests(TestCase):
             inputs=inp,
         )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("loop_type", ["for"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("cuda")
     def test_associative_scan_loop_in_combine_fn(
         self, compile_mode, loop_type, reverse, device
     ):
@@ -3568,14 +3576,14 @@ class AssociativeScanTests(TestCase):
 
     # TODO: Does not work because of the usage of vmap witin associative_scan
     # TODO: Re-enable additional parameters again once this issues has been resolved
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @unittest.expectedFailure
     def test_associative_scan_loop_in_combine_fn_failure(self):
         compile_mode = "none"
         loop_type = "while"
         reverse = False
-        device = torch.device("cuda")
+        device = torch.device("npu") # device = torch.device("cuda")
 
         def combine_fn(x, y):
             _cnt = torch.zeros_like(y[0, :])
@@ -3603,11 +3611,11 @@ class AssociativeScanTests(TestCase):
             inputs=inp,
         )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("cuda")
     def test_associative_scan_cond_in_combine_fn(self, compile_mode, reverse, device):
         def combine_fn(x, y):
             val = cond(torch.sum(y) > 0.0, lambda y: y + 0.0, lambda y: 1.0 - y, (y,))
@@ -3631,13 +3639,13 @@ class AssociativeScanTests(TestCase):
 
     # TODO: Does not work because of the usage of vmap witin associative_scan
     # TODO: Re-enable additional parameters again once this issues has been resolved
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @unittest.expectedFailure
     def test_associative_scan_map_in_combine_fn(self):
         compile_mode = "none"
         reverse = False
-        device = torch.device("cuda")
+        device = torch.device("npu") # device = torch.device("cuda")
 
         def combine_fn(x, y):
             def body(x, y):
@@ -3663,11 +3671,11 @@ class AssociativeScanTests(TestCase):
             inputs=inp,
         )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])     # torch.device("cuda")
     def test_associative_scan_vmap_in_combine_fn(self, compile_mode, reverse, device):
         def combine_fn(x, y):
             def body(x):
@@ -3693,11 +3701,11 @@ class AssociativeScanTests(TestCase):
             inputs=inp,
         )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("cuda")
     # Skipping the combination of associative_scan and device=cpu
     # as the current implementation of pointwise does only support CUDA device
     @decorateIf(
@@ -3724,12 +3732,12 @@ class AssociativeScanTests(TestCase):
         )
 
     @skipIfRocm(msg="Unsupported on ROCM yet")
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])     # torch.device("cuda")
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
     @decorateIf(
@@ -3765,11 +3773,11 @@ class AssociativeScanTests(TestCase):
         )
 
     @skipIfRocm(msg="Unsupported on ROCM yet")
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])     # torch.device("cuda")
     def test_associative_scan_different_input_size(self, compile_mode, reverse, device):
         batch = 5
         hidden_dim = 3
@@ -3803,8 +3811,8 @@ class AssociativeScanTests(TestCase):
             inputs=elements,
         )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     def test_associative_scan_different_input_size_wrong_dim(self):
         batch = 5
         hidden_dim = 3
@@ -3812,17 +3820,17 @@ class AssociativeScanTests(TestCase):
         dstate = 7
 
         deltaA = torch.randn(
-            (batch, hidden_dim, length, dstate), device=torch.device("cuda")
+            (batch, hidden_dim, length, dstate), device=torch.device("npu") # cuda
         )
         deltaB_u = torch.randn(
-            (batch, hidden_dim, length, dstate), device=torch.device("cuda")
+            (batch, hidden_dim, length, dstate), device=torch.device("npu")
         )
-        C = torch.randn((batch, dstate, length), device=torch.device("cuda"))
+        C = torch.randn((batch, dstate, length), device=torch.device("npu"))
         x = torch.randn(
-            (batch, hidden_dim, length, dstate), device=torch.device("cuda")
+            (batch, hidden_dim, length, dstate), device=torch.device("npu")
         )
         y = torch.randn(
-            (batch, hidden_dim, length, dstate), device=torch.device("cuda")
+            (batch, hidden_dim, length, dstate), device=torch.device("npu")
         )
         elements = (x, deltaA, deltaB_u, C, y)
 
@@ -3840,12 +3848,12 @@ class AssociativeScanTests(TestCase):
                 combine_mode="pointwise",
             )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("cuda")
     # Skipping the combine_mode=pointwise
     # as the current implementation of associative_scan lowering
     # does not support lifted arguments
@@ -3891,12 +3899,12 @@ class AssociativeScanTests(TestCase):
                 inputs=inp,
             )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("cuda")
     # Skipping the combine_mode=pointwise
     # as the current implementation of associative_scan lowering
     # does not support lifted arguments
@@ -3970,12 +3978,12 @@ class AssociativeScanTests(TestCase):
                 inputs=inp,
             )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("cuda")
     # Skipping the combine_mode=pointwise
     # as the current implementation of associative_scan lowering
     # does not support lifted arguments
@@ -4009,11 +4017,11 @@ class AssociativeScanTests(TestCase):
             inputs=inp,
         )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("cuda")
     def test_associative_scan_freevars_fct_generic(self, compile_mode, reverse, device):
         def additional_fct_no_add_inp(x, y):
             return x * y
@@ -4045,12 +4053,12 @@ class AssociativeScanTests(TestCase):
             inputs=inp,
         )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("cuda")
     # Skipping the combine_mode=pointwise
     # as the current implementation of associative_scan lowering
     # does not support lifted arguments
@@ -4082,11 +4090,11 @@ class AssociativeScanTests(TestCase):
             inputs=inp,
         )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("cuda")
     @parametrize("combine_mode", ["pointwise", "generic"])
     # Skipping the combine_mode=pointwise
     # as the current implementation of associative_scan lowering
@@ -4136,8 +4144,8 @@ class AssociativeScanTests(TestCase):
             inputs=inp,
         )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     def test_associative_scan_sparse_tensor(self):
         x = torch.tensor(
             [[[0.0, 0], [1.0, 2.0]], [[0.0, 0], [3.0, 4.0]], [[0.0, 0], [5.0, 6.0]]]
@@ -4153,10 +4161,10 @@ class AssociativeScanTests(TestCase):
                 0,
             )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     def test_associative_scan_combine_fn_wrong_meta_in_combine_fn(self):
-        device = torch.device("cuda")
+        device = torch.device("npu")  # torch.device("cuda")
         B, N, C, H, W = 3, 3, 2, 3, 3
         x = torch.randn(B, N, C, H, W, device=device)
 
@@ -4164,8 +4172,10 @@ class AssociativeScanTests(TestCase):
             return (x + y).to(torch.int64)
 
         def fct_wrong_device(x, y):
+            _device = 'cpu' if device.type == 'npu' else 'npu'
             return (x + y).to(
-                torch.device("cpu") if device.type == "cuda" else torch.device("cuda")
+                _device,
+                # torch.device("cpu") if device.type == "npu" else torch.device("npu")   # cuda
             )
 
         def fct_wrong_stride(x, y):
@@ -4180,7 +4190,7 @@ class AssociativeScanTests(TestCase):
             ):
                 associative_scan(fct, x, 0)
 
-    @unittest.skipIf(not SM70OrLater, "triton")
+    # @unittest.skipIf(not SM70OrLater, "triton")
     def test_associative_scan_wrong_pytree(self):
         def fct_wrong_pytree(x, y):
             return {
@@ -4203,10 +4213,10 @@ class AssociativeScanTests(TestCase):
         ):
             associative_scan(fct_wrong_pytree, inp, 0, combine_mode="generic")
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     def test_associative_scan_non_pointwise(self):
-        device = torch.device("cuda")
+        device = torch.device("npu")    # torch.device("cuda")
         x = torch.randn(3, 10, 2, device=device)
         with self.assertRaisesRegex(
             # Should be:
@@ -4220,9 +4230,9 @@ class AssociativeScanTests(TestCase):
                 combine_mode="pointwise",
             )
 
-    @requires_cuda
+    # @requires_cuda
     def test_associative_scan_input_mutation(self):
-        device = torch.device("cuda")
+        device = torch.device("npu")    # torch.device("cuda")
 
         def fct_input_mutation(x, y):
             x.add_(1)
@@ -4237,9 +4247,9 @@ class AssociativeScanTests(TestCase):
         ):
             associative_scan(fct_input_mutation, x, 0)
 
-    @requires_cuda
+    # @requires_cuda
     def test_associative_scan_input_output_alias(self):
-        device = torch.device("cuda")
+        device = torch.device("npu")    # torch.device("cuda")
 
         def fct_input_output_alias(x, y):
             return x[0], x[1] + y[1]
