diff --git a/test/dynamo/cpython/3_13/test_set.py b/test/dynamo/cpython/3_13/test_set.py
index 1d80fccca5b..fa64c2e99fe 100644
--- a/test/dynamo/cpython/3_13/test_set.py
+++ b/test/dynamo/cpython/3_13/test_set.py
@@ -9,6 +9,8 @@
 
 import sys
 import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 import torch._dynamo.test_case
 import unittest
 from torch._dynamo.test_case import CPythonTestCase
