diff --git a/test/dynamo/cpython/3_13/test_generator_stop.py b/test/dynamo/cpython/3_13/test_generator_stop.py
index e3ff8d346a7..c5c49ab89c6 100644
--- a/test/dynamo/cpython/3_13/test_generator_stop.py
+++ b/test/dynamo/cpython/3_13/test_generator_stop.py
@@ -11,6 +11,8 @@ from __future__ import generator_stop
 
 import sys
 import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 import torch._dynamo.test_case
 import unittest
 from torch._dynamo.test_case import CPythonTestCase
