diff --git a/test/distributed/checkpoint/test_fsdp_tp_checkpoint_conversion.py b/test/distributed/checkpoint/test_fsdp_tp_checkpoint_conversion.py
index 8e49edf..d76b6e2 100644
--- a/test/distributed/checkpoint/test_fsdp_tp_checkpoint_conversion.py
+++ b/test/distributed/checkpoint/test_fsdp_tp_checkpoint_conversion.py
@@ -1,3 +1,8 @@
+import torch_npu.testing
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["oncall: distributed"]
 import torch
 import torch.distributed.checkpoint as dist_cp
@@ -16,9 +21,10 @@ from torch.testing._internal.distributed._tensor.common_dtensor import (
     DTensorTestBase,
     MLPModule,
     skip_if_lt_x_gpu,
-    with_comms,
+    # with_comms,
 )
 from torch.testing._internal.distributed.checkpoint_utils import with_temp_dir
+from torch_npu.testing.common_distributed import with_comms
 
 
 # TODO: modularize this test and add test for checkpoint conversion in both direction.
