diff --git a/test/distributed/_pycute/test_typing.py b/test/distributed/_pycute/test_typing.py
index 61f50c08a1a..434f4b32916 100644
--- a/test/distributed/_pycute/test_typing.py
+++ b/test/distributed/_pycute/test_typing.py
@@ -38,7 +38,8 @@ Unit tests for _pycute.typing
 """
 
 import logging
-
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 from torch.distributed._pycute import *
 from torch.testing._internal.common_utils import run_tests, TestCase
 
