diff --git a/test/cpp_extensions/open_registration_extension/torch_openreg/tests/test_autograd.py b/test/cpp_extensions/open_registration_extension/torch_openreg/tests/test_autograd.py
index 38fa1d77bef..4b44a64debe 100644
--- a/test/cpp_extensions/open_registration_extension/torch_openreg/tests/test_autograd.py
+++ b/test/cpp_extensions/open_registration_extension/torch_openreg/tests/test_autograd.py
@@ -4,6 +4,8 @@ import os
 
 import psutil
 import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 from torch.testing._internal.common_utils import (
     run_tests,
     skipIfMPS,
