diff --git a/torch/distributed/distributed_c10d.py b/torch/distributed/distributed_c10d.py
index 05986c191a2..f2f6d4e5f43 100644
--- a/torch/distributed/distributed_c10d.py
+++ b/torch/distributed/distributed_c10d.py
@@ -304,10 +304,14 @@ class Backend(str):  # noqa: SLOT000
         "cuda": NCCL,
         "xpu": XCCL,
         "mps": GLOO,
+        MPI: ["cpu", "npu"],
+        UCC: ["cpu", "npu"],
+        HCCL: ["npu"],
+        GLOO: ["cpu", "npu"],
+        "npu": HCCL,
     }
 
     backend_capability: dict[str, list[str]] = {
-        GLOO: ["cpu", "cuda"],
         NCCL: ["cuda"],
         XCCL: ["xpu"],
         UCC: ["cpu", "cuda"],
@@ -1465,7 +1469,7 @@ def get_default_backend_for_device(device: str | torch.device) -> str:
         device_str = device.type
     else:
         device_str = torch.device(device).type
-
+    device_str = 'npu'
     backend = Backend.default_device_backend_map.get(device_str)
     if backend is None:
         raise ValueError(f"Default backend not registered for device : {device}")
