diff --git a/test/package/test_repackage.py b/test/package/test_repackage.py
index 0e21d7012f5..3d963b41b46 100644
--- a/test/package/test_repackage.py
+++ b/test/package/test_repackage.py
@@ -5,6 +5,9 @@ from io import BytesIO
 from torch.package import PackageExporter, PackageImporter, sys_importer
 from torch.testing._internal.common_utils import run_tests
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
 try:
     from .common import PackageTestCase
