diff --git a/test/package/test_model.py b/test/package/test_model.py
index 09b10a1..ce847de 100644
--- a/test/package/test_model.py
+++ b/test/package/test_model.py
@@ -8,6 +8,9 @@ import torch
 from torch.package import PackageExporter, PackageImporter, sys_importer
 from torch.testing._internal.common_utils import IS_FBCODE, IS_SANDCASTLE, run_tests
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
 try:
     from torchvision.models import resnet18
