diff --git a/test/package/test_package_fx.py b/test/package/test_package_fx.py
index 9976766..2afb906 100644
--- a/test/package/test_package_fx.py
+++ b/test/package/test_package_fx.py
@@ -12,6 +12,9 @@ from torch.package import (
 )
 from torch.testing._internal.common_utils import run_tests
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
 try:
     from .common import PackageTestCase
