diff --git a/test/package/test_directory_reader.py b/test/package/test_directory_reader.py
index 85d01b0..c418094 100644
--- a/test/package/test_directory_reader.py
+++ b/test/package/test_directory_reader.py
@@ -16,6 +16,9 @@ from torch.testing._internal.common_utils import (
     run_tests,
 )
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
 try:
     from torchvision.models import resnet18
