diff --git a/test/package/test_package_script.py b/test/package/test_package_script.py
index 13c2426..3ee4edb 100644
--- a/test/package/test_package_script.py
+++ b/test/package/test_package_script.py
@@ -13,6 +13,9 @@ from torch.testing._internal.common_utils import (
     skipIfTorchDynamo,
 )
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
 try:
     from .common import PackageTestCase
