diff --git a/test/nn/test_module_hooks.py b/test/nn/test_module_hooks.py
index c9c29f0..df09b0c 100644
--- a/test/nn/test_module_hooks.py
+++ b/test/nn/test_module_hooks.py
@@ -12,6 +12,8 @@ from tempfile import NamedTemporaryFile
 from typing import Any
 
 import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 import torch.nn as nn
 from torch.testing._internal.common_nn import _create_basic_net, NNTestCase
 from torch.testing._internal.common_utils import (
