diff --git a/test/nn/test_load_state_dict.py b/test/nn/test_load_state_dict.py
index 8ce1f03c0a8..70cf8fa22d8 100644
--- a/test/nn/test_load_state_dict.py
+++ b/test/nn/test_load_state_dict.py
@@ -5,6 +5,7 @@ from copy import deepcopy
 from itertools import product
 
 import torch
+import torch_npu
 import torch.nn as nn
 from torch.testing._internal.common_nn import NNTestCase
 from torch.testing._internal.common_utils import (
@@ -74,6 +75,7 @@ class TestLoadStateDict(NNTestCase):
         net.block = block
         net.add_module("empty", None)
         conv1_bias_dtype = block.conv1.bias.dtype
+        net.npu()
 
         state_dict = net.state_dict()
         state_dict.update(
@@ -100,6 +102,9 @@ class TestLoadStateDict(NNTestCase):
             self.assertEqual(len(incompatible_keys.missing_keys), 0)
             self.assertEqual(len(incompatible_keys.unexpected_keys), 0)
             self.assertNotIn("Incompatible", str(incompatible_keys))
+            self.assertEqual(net.linear1.weight.device.type, "npu")
+            self.assertEqual(net.block.conv1.bias.device.type, "npu")
+            self.assertEqual(net.bn.running_mean.device.type, "npu")
             self.assertEqual(net.linear1.weight, sd["linear1.weight"])
             self.assertEqual(net.block.conv1.bias, sd["block.conv1.bias"])
             self.assertEqual(net.bn.running_mean, sd["bn.running_mean"])
@@ -152,6 +157,9 @@ class TestLoadStateDict(NNTestCase):
             "nonexistent_key": torch.rand(3),
         }
         net.load_state_dict(state_dict, strict=False)
+        self.assertEqual(net.linear1.weight.device.type, "npu")
+        self.assertEqual(net.block.conv1.bias.device.type, "npu")
+        self.assertEqual(net.bn.running_mean.device.type, "npu")
         self.assertEqual(net.linear1.weight, state_dict["linear1.weight"])
         self.assertEqual(net.block.conv1.bias, state_dict["block.conv1.bias"])
         self.assertEqual(net.bn.running_mean, state_dict["bn.running_mean"])
@@ -170,15 +178,17 @@ class TestLoadStateDict(NNTestCase):
         # BatchNormNd
         # Added num_batches_tracked buffer at version 2. For state dict with
         # earlier versions or no versions, it should provide default value of 0.
-        bn = nn.BatchNorm2d(3)
+        bn = nn.BatchNorm2d(3).npu()
         state_dict = bn.state_dict()
         del state_dict["num_batches_tracked"]
         state_dict._metadata[""]["version"] = 1  # version 1
         bn.load_state_dict(state_dict)
+        self.assertEqual(bn.num_batches_tracked.device.type, "npu")
         self.assertEqual(bn.num_batches_tracked.dtype, torch.long)
         self.assertEqual(bn.num_batches_tracked.item(), 0)
         del state_dict._metadata[""]["version"]  # no version
         bn.load_state_dict(state_dict)
+        self.assertEqual(bn.num_batches_tracked.device.type, "npu")
         self.assertEqual(bn.num_batches_tracked.dtype, torch.long)
         self.assertEqual(bn.num_batches_tracked.item(), 0)
 
@@ -188,6 +198,7 @@ class TestLoadStateDict(NNTestCase):
         model = base_module
         for _ in range(3):
             model = nn.Sequential(*[deepcopy(model) for _ in range(10)])
+        model.npu()
 
         def hook_fn(
             module,
@@ -204,6 +215,7 @@ class TestLoadStateDict(NNTestCase):
 
         model[0][0].register_load_state_dict_pre_hook(hook_fn)
         model.load_state_dict(model.state_dict(), strict=True)
+        self.assertEqual(next(model.parameters()).device.type, "npu")
 
     # fails swapping as LSTM installs weak references on the parameters
     @swap([False])
@@ -245,7 +257,7 @@ class TestLoadStateDict(NNTestCase):
                 self.param.data.copy_(state_dict[prefix + "serialized"] - 1)
 
         # use sequential to verify nesting
-        m = nn.Sequential(CustomState())
+        m = nn.Sequential(CustomState()).npu()
         with torch.no_grad():
             m[0].param[0] = 10
             m[0].sub.weight[0, 0] = 555
@@ -254,9 +266,11 @@ class TestLoadStateDict(NNTestCase):
         self.assertIn("0.sub.weight", state_dict)
         self.assertNotIn("0.param", state_dict)
         del m
-        mm = nn.Sequential(CustomState())
+        mm = nn.Sequential(CustomState()).npu()
         self.assertEqual(mm[0].param[0].item(), 1)
         mm.load_state_dict(state_dict)
+        self.assertEqual(mm[0].param.device.type, "npu")
+        self.assertEqual(mm[0].sub.weight.device.type, "npu")
         self.assertEqual(mm[0].param[0].item(), 10)
         self.assertEqual(mm[0].sub.weight[0, 0].item(), 555)
 
@@ -398,14 +412,16 @@ class TestLoadStateDict(NNTestCase):
             def forward(self, input):
                 return self.bn(self.fc1(input))
 
-        net = MyModule()
+        net = MyModule().npu()
         state_dict = net.state_dict()
         # loading should be ok if stride is different
-        state_dict["fc1.weight"] = torch.randn(3, 5).transpose(0, 1)
-        net2 = MyModule()
+        state_dict["fc1.weight"] = torch.randn(3, 5).npu().transpose(0, 1)
+        net2 = MyModule().npu()
         net2.load_state_dict(state_dict, strict=False, assign=True)
+        self.assertEqual(net2.fc1.weight.device.type, "npu")
+        self.assertEqual(net2.fc1.weight, state_dict["fc1.weight"])
 
-        state_dict["fc1.weight"] = torch.randn(2, 4)
+        state_dict["fc1.weight"] = torch.randn(2, 4).npu()
         with self.assertRaisesRegex(
             RuntimeError, "size mismatch for fc1.weight: copying a param with shape"
         ):
@@ -430,12 +446,12 @@ class TestLoadStateDict(NNTestCase):
                 super().__init__()
                 self.fc1 = torch.nn.Linear(5, 10)
 
-        m = MyModule()
+        m = MyModule().npu()
 
         # Unexpected key & strict = True
         with self.assertRaisesRegex(RuntimeError, "Unexpected key"):
             state_dict = m.state_dict()
-            state_dict["fc1.bad_suffix"] = torch.randn(5, 10)
+            state_dict["fc1.bad_suffix"] = torch.randn(5, 10).npu()
             m.load_state_dict(state_dict)
 
         # Unexpected key & strict = False
@@ -445,7 +461,7 @@ class TestLoadStateDict(NNTestCase):
         # Unexpected key whose prefix matches a valid key & strict = True
         with self.assertRaisesRegex(RuntimeError, "Unexpected key"):
             state_dict = m.state_dict()
-            state_dict["fc1.weight.bad_suffix"] = torch.randn(5, 10)
+            state_dict["fc1.weight.bad_suffix"] = torch.randn(5, 10).npu()
             m.load_state_dict(state_dict)
 
         # Unexpected key whose prefix matches a valid key & strict = False
