diff --git a/test/nn/test_module_hooks.py b/test/nn/test_module_hooks.py
index 72e3665..2639527 100644
--- a/test/nn/test_module_hooks.py
+++ b/test/nn/test_module_hooks.py
@@ -12,6 +12,7 @@ from tempfile import NamedTemporaryFile
 from typing import Any

 import torch
+import torch_npu
 import torch.nn as nn
 from torch.testing._internal.common_nn import _create_basic_net, NNTestCase
 from torch.testing._internal.common_utils import (
@@ -934,11 +935,11 @@ class TestModuleGlobalHooks(TestCase):
     def test_module_global_hooks(self):
         module = nn.Sigmoid

-        module_1 = module()
-        module_2 = module()
-        module_3 = module()
+        module_1 = module().npu()
+        module_2 = module().npu()
+        module_3 = module().npu()

-        input = torch.ones(5, 5, requires_grad=True)
+        input = torch.ones(5, 5, device="npu", requires_grad=True)

         counter = {"forwards": 0, "backwards": 0}

@@ -946,15 +947,15 @@ class TestModuleGlobalHooks(TestCase):
             self.assertIsInstance(input, tuple)
             self.assertTrue(isinstance(output, torch.Tensor))
             self.assertTrue(isinstance(h_module, module))
-            self.assertEqual(input[0], torch.ones(5, 5))
-            self.assertEqual(output, torch.empty(5, 5).fill_(1 / (1 + 1 / math.e)))
+            self.assertEqual(input[0], torch.ones(5, 5, device="npu"))
+            self.assertEqual(output, torch.empty(5, 5, device="npu").fill_(1 / (1 + 1 / math.e)))
             counter["forwards"] += inc

         def bw_hook(inc, h_module, grad_input, grad_output):
             self.assertIsInstance(grad_input, tuple)
             self.assertIsInstance(grad_output, tuple)
             self.assertTrue(isinstance(h_module, module))
-            self.assertEqual(grad_output[0], torch.ones(5, 5) * 2)
+            self.assertEqual(grad_output[0], torch.ones(5, 5, device="npu") * 2)
             counter["backwards"] += inc

         test_fwd = nn.modules.module.register_module_forward_hook(
@@ -977,13 +978,13 @@ class TestModuleGlobalHooks(TestCase):
         self.assertEqual(counter["forwards"], 6)
         self.assertEqual(counter["backwards"], 0)

-        output_1.backward(torch.ones(5, 5) * 2, retain_graph=True)
-        output_2.backward(torch.ones(5, 5) * 2, retain_graph=False)
-        output_3.backward(torch.ones(5, 5) * 2, retain_graph=False)
+        output_1.backward(torch.ones(5, 5, device="npu") * 2, retain_graph=True)
+        output_2.backward(torch.ones(5, 5, device="npu") * 2, retain_graph=False)
+        output_3.backward(torch.ones(5, 5, device="npu") * 2, retain_graph=False)
         self.assertEqual(counter["forwards"], 6)
         self.assertEqual(counter["backwards"], 3)

-        output_1.backward(torch.ones(5, 5) * 2, retain_graph=True)
+        output_1.backward(torch.ones(5, 5, device="npu") * 2, retain_graph=True)
         self.assertEqual(counter["forwards"], 6)
         self.assertEqual(counter["backwards"], 4)

@@ -1001,19 +1002,19 @@ class TestModuleGlobalHooks(TestCase):
             lambda *args: bw_hook(2, *args)
         )

-        module_1(input).backward(torch.ones(5, 5) * 2)
+        module_1(input).backward(torch.ones(5, 5, device="npu") * 2)
         self.assertEqual(counter["forwards"], 18)
         self.assertEqual(counter["backwards"], 7)

         test2_bwd.remove()

-        module_2(input).backward(torch.ones(5, 5) * 2)
+        module_2(input).backward(torch.ones(5, 5, device="npu") * 2)
         self.assertEqual(counter["forwards"], 21)
         self.assertEqual(counter["backwards"], 8)

         test2_fwd.remove()

-        module_3(input).backward(torch.ones(5, 5) * 2)
+        module_3(input).backward(torch.ones(5, 5, device="npu") * 2)
         self.assertEqual(counter["forwards"], 22)
         self.assertEqual(counter["backwards"], 9)

@@ -1021,14 +1022,14 @@ class TestModuleGlobalHooks(TestCase):
         test_bwd.remove()

     def test_module_global_hook_invalid_outputs(self):
-        module = nn.Sigmoid()
-        input = torch.randn(5, 5, requires_grad=True)
+        module = nn.Sigmoid().npu()
+        input = torch.randn(5, 5, device="npu", requires_grad=True)

         def bw_fail1(self, grad_input, grad_output):
             return grad_input[:-1]

         def bw_fail2(self, grad_input, grad_output):
-            return grad_input + (torch.randn(2, 2),)
+            return grad_input + (torch.randn(2, 2, device="npu"),)

         with nn.modules.module.register_module_backward_hook(bw_fail1):
             with self.assertRaisesRegex(RuntimeError, "got 0, but expected 1"):
@@ -1039,8 +1040,8 @@ class TestModuleGlobalHooks(TestCase):
                 module(input).sum().backward()

     def test_module_backward_global_hook_writeable(self):
-        module = nn.Sigmoid()
-        input = torch.randn(5, 5, requires_grad=True)
+        module = nn.Sigmoid().npu()
+        input = torch.randn(5, 5, device="npu", requires_grad=True)
         sig_x = torch.sigmoid(input)

         def bw_hook(module, grad_input, grad_output):
@@ -1051,14 +1052,14 @@ class TestModuleGlobalHooks(TestCase):
             return tuple(gi * 2 for gi in grad_input)

         nn.modules.module.register_module_backward_hook(bw_hook)
-        module(input).backward(torch.ones(5, 5))
+        module(input).backward(torch.ones(5, 5, device="npu"))
         expected_grad = sig_x * (1 - sig_x) * 2
         self.assertEqual(input.grad, expected_grad)

     @skipIfTorchDynamo("TorchDynamo does not work well with hooks")
     def test_module_global_forward_preforward_hook_writeable(self):
-        module = nn.Sigmoid()
-        input = torch.randn(5, 5, requires_grad=True)
+        module = nn.Sigmoid().npu()
+        input = torch.randn(5, 5, device="npu", requires_grad=True)
         sig_x = torch.sigmoid(input)

         def forward_pre_hook(m, input):
@@ -1072,7 +1073,7 @@ class TestModuleGlobalHooks(TestCase):
         output = module(input)
         expected_res = -torch.sigmoid(torch.nn.functional.relu(input))
         self.assertEqual(output, expected_res)
-        output.backward(torch.ones(5, 5) * 2, retain_graph=True)
+        output.backward(torch.ones(5, 5, device="npu") * 2, retain_graph=True)
         mask = input > 0
         expected_grad = -sig_x * (1 - sig_x) * 2 * mask
         self.assertEqual(input.grad, expected_grad)
@@ -1149,7 +1150,7 @@ class TestModuleGlobalHooks(TestCase):

     @skipIfTorchDynamo("TorchDynamo does not work well with hooks")
     def test_global_and_local_hooks_order(self):
-        module = nn.Sigmoid()
+        module = nn.Sigmoid().npu()

         global_forward_pre_called = False
         local_forward_pre_called = False
@@ -1194,7 +1195,7 @@ class TestModuleGlobalHooks(TestCase):
             local_backward_called = True
             return input

-        input = torch.randn(5, 5, requires_grad=True)
+        input = torch.randn(5, 5, device="npu", requires_grad=True)
         nn.modules.module.register_module_forward_pre_hook(global_forward_pre_hook)
         module.register_forward_pre_hook(local_forward_pre_hook)
         nn.modules.module.register_module_forward_hook(global_forward_hook)
@@ -1210,7 +1211,7 @@ class TestModuleGlobalHooks(TestCase):
             and global_forward_pre_called
         )

-        output.backward(torch.ones(5, 5), retain_graph=True)
+        output.backward(torch.ones(5, 5, device="npu"), retain_graph=True)
         self.assertTrue(local_backward_called and global_backward_called)

     @skipIfTorchDynamo("TorchDynamo does not work well with hooks")
@@ -1224,13 +1225,13 @@ class TestModuleGlobalHooks(TestCase):
             out = out + kwargs["bias"]
             return out

-        model = KwargModel()
+        model = KwargModel().npu()
         nn.modules.module.register_module_forward_hook(
             kwarg_global_forward_hook,
             with_kwargs=True,
         )
-        x: torch.Tensor = torch.randn(10, 20)
-        bias: torch.Tensor = torch.randn(10, 20)
+        x: torch.Tensor = torch.randn(10, 20, device="npu")
+        bias: torch.Tensor = torch.randn(10, 20, device="npu")
         out = model(x, bias=bias)
         self.assertEqual(out, x + 2 * bias, rtol=0, atol=1e-5)

@@ -1687,6 +1688,7 @@ class TestModuleHookNN(NNTestCase):
             )
             try:
                 l, n, s = _create_basic_net()
+                s = s.npu()
                 for b in s.buffers():
                     self.assertTrue(getattr(b, "registered", False))
             finally:
@@ -1706,6 +1708,7 @@ class TestModuleHookNN(NNTestCase):
             )
             try:
                 l, n, s = _create_basic_net()
+                s = s.npu()
                 for m in s.modules():
                     self.assertTrue(getattr(m, "registered", False))
             finally:
@@ -1726,6 +1729,7 @@ class TestModuleHookNN(NNTestCase):
             )
             try:
                 l, n, s = _create_basic_net()
+                s = s.npu()
                 for p in s.parameters():
                     self.assertTrue(getattr(p, "registered", False))
             finally:
