diff --git a/test/nn/test_embedding.py b/test/nn/test_embedding.py
index 8f6847f18f5..87a7ea45a95 100644
--- a/test/nn/test_embedding.py
+++ b/test/nn/test_embedding.py
@@ -1,10 +1,18 @@
 # Owner(s): ["module: nn"]
 import itertools
 import random
+    @dtypesIfPRIVATEUSE1(
+            self.assertEqual(embedding.weight.device.type, "npu")
+            # self.assertEqual(embedding.weight.device.type, "cuda")
 import unittest
 from itertools import product
 
 import torch
+from torch_npu.contrib import transfer_to_npu
+import torch_npu
+    dtypesIfPRIVATEUSE1,
+from torch_npu.contrib import transfer_to_npu
+import torch_npu
 import torch.nn as nn
 import torch.nn.functional as F
 from torch.testing._internal.common_device_type import (
@@ -323,7 +331,7 @@ class TestEmbeddingNNDeviceType(NNTestCase):
             with self.assertRaisesRegex(RuntimeError, "'weight' must be 2-D"):
                 torch.nn.functional.embedding(indices, weight)
 
-    @dtypesIfCUDA(torch.float16, torch.float64)
+    @dtypesIfPRIVATEUSE1(torch.float16, torch.float64)
     @dtypesIfXPU(torch.float16, torch.float64)
     @dtypes(torch.float64)
     def test_embedding_backward(self, device, dtype):
@@ -344,6 +352,7 @@ class TestEmbeddingNNDeviceType(NNTestCase):
         self.assertEqual(embedding.weight.grad._indices(), tensor)
         self.assertEqual(embedding.weight.grad._values(), ones)
 
+    @dtypesIfPRIVATEUSE1(
         embedding.zero_grad()
         embedding(tensor[0]).sum().backward()
         embedding(tensor[0]).sum().backward()
@@ -408,7 +417,7 @@ class TestEmbeddingNNDeviceType(NNTestCase):
         expected_grad = torch.ones((2, 2, 4), device=device, dtype=dtype)
         self.assertEqual(jvp, expected_grad)
 
-    @dtypesIfCUDA(
+    @dtypesIfPRIVATEUSE1(
         *(
             (torch.float, torch.double, torch.bfloat16, torch.half)
             if TEST_WITH_ROCM
@@ -507,7 +516,7 @@ class TestEmbeddingNNDeviceType(NNTestCase):
     @skipIfTorchDynamo("see https://github.com/pytorch/pytorch/pull/95621")
     @onlyNativeDeviceTypes
     @dtypes(torch.float32, torch.float64)
-    @dtypesIfCUDA(torch.half, torch.bfloat16)
+    @dtypesIfPRIVATEUSE1(torch.half, torch.bfloat16)
     @dtypesIfXPU(torch.half, torch.bfloat16)
     def test_embedding_bag_1D_padding_idx(self, device, dtype):
         num_features = 3
@@ -653,9 +662,9 @@ class TestEmbeddingNNDeviceType(NNTestCase):
                     rtol = None
                 self.assertEqual(
                     weights.grad, weights_check.grad, msg=msg, atol=atol, rtol=rtol
+    @onlyOn(["npu", "xpu"])
                 )
 
-    @onlyOn(["cuda", "xpu"])
     @dtypes(
         *(
             (torch.float, torch.double, torch.bfloat16, torch.half)
@@ -793,11 +802,11 @@ class TestEmbeddingNNDeviceType(NNTestCase):
                 f"Expected non-zero gradient for index {idx}",
             )
 
-    @onlyOn(["cuda", "xpu"])
+    @largeTensorTest("80GB", device="npu")
+    @onlyOn(["npu", "xpu"])
     @dtypes(
         torch.bfloat16,
     )
-    @largeTensorTest("80GB", device="cuda")
     @largeTensorTest("80GB", device="xpu")
     def test_embedding_backward_large_batch_overflow(self, device, dtype):
         """
@@ -880,7 +889,7 @@ class TestEmbeddingNNDeviceType(NNTestCase):
     # against torch.nn.functional.embedding followed by a reduction.
     @onlyNativeDeviceTypes
     @dtypes(torch.float32, torch.float64)
-    @dtypesIfCUDA(torch.half, torch.bfloat16)
+    @dtypesIfPRIVATEUSE1(torch.half, torch.bfloat16)
     @dtypesIfXPU(torch.half, torch.bfloat16)
     def test_embedding_bag_2D_padding_idx(self, device, dtype):
         # Use a Python implementation of embedding_bag with padding_idx support
@@ -1248,7 +1257,7 @@ class TestEmbeddingNNDeviceType(NNTestCase):
             (torch.half, torch.bfloat16, torch.float, torch.double),
         )
     )
-    @dtypesIfCUDA(
+    @dtypesIfPRIVATEUSE1(
         *itertools.product(
             (torch.int, torch.long),
             (torch.int, torch.long),
@@ -1300,6 +1309,7 @@ class TestEmbeddingNNDeviceType(NNTestCase):
                 rtol=0,
             )
             if trainable_scale:
+    @dtypesIfPRIVATEUSE1(
                 ref_per_sample_weights_grad = torch.empty_like(per_sample_weights)
                 self.assertEqual(
                     per_sample_weights.grad,
@@ -1321,7 +1331,10 @@ class TestEmbeddingNNDeviceType(NNTestCase):
             (torch.float, torch.double, torch.half, torch.bfloat16),
         )
     )
-    @dtypesIfCUDA(
+    @dtypesIfPRIVATEUSE1(
+    @dtypesIfPRIVATEUSE1(
+    @dtypesIfPRIVATEUSE1(
+    @dtypesIfPRIVATEUSE1(
         *itertools.product(
             (torch.int, torch.long),
             (torch.int, torch.long),
