diff --git a/test/nn/test_pooling.py b/test/nn/test_pooling.py
index 5b032e055db..a71dff50d44 100644
--- a/test/nn/test_pooling.py
+++ b/test/nn/test_pooling.py
@@ -11,6 +11,8 @@ from functools import partial, reduce
 from itertools import repeat
 
 import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 import torch.nn as nn
 import torch.nn.functional as F
 from torch import inf, nan
@@ -26,7 +28,7 @@ from torch.testing._internal.common_device_type import (
     instantiate_device_type_tests,
     largeTensorTest,
     onlyCPU,
-    onlyCUDA,
+    onlyPRIVATEUSE1,
     onlyNativeDeviceTypes,
     TEST_WITH_ROCM,
 )
@@ -310,7 +312,7 @@ class TestPoolingNN(NNTestCase):
                     self, device, dtype, torch.nn.AdaptiveMaxPool2d, torch.channels_last
                 )
 
-    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    # @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
     @largeTensorTest("12GB", device="cuda")
     def test_adaptive_pooling_avg_nhwc_launch_config_backward(self):
         input = torch.randint(
@@ -335,7 +337,7 @@ class TestPoolingNN(NNTestCase):
         self.assertEqual(out, ref_out)
         self.assertEqual(input.grad, ref_input.grad)
 
-    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    # @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
     @largeTensorTest("12GB", device="cuda")
     def test_adaptive_pooling_avg_nhwc_launch_config_forward(self):
         input = torch.randint(
@@ -354,7 +356,7 @@ class TestPoolingNN(NNTestCase):
         self.assertTrue(ref_out.is_contiguous())
         self.assertEqual(out, ref_out)
 
-    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    # @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
     def test_adaptive_avg_pooling_overflow(self):
         input = torch.randint(
             -256, 256, (20, 32, 256, 256), dtype=torch.half, device="cuda"
@@ -364,7 +366,7 @@ class TestPoolingNN(NNTestCase):
         self.assertFalse(torch.isinf(out).any())
         self.assertFalse(torch.isnan(out).any())
 
-    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    # @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
     def test_adaptive_avg_pooling_nhwc_overflow(self):
         input = torch.randint(
             -256, 256, (20, 32, 256, 256), dtype=torch.half, device="cuda"
@@ -746,7 +748,7 @@ class TestPoolingNNDeviceType(NNTestCase):
         self.assertEqual(inp.grad, torch.zeros_like(inp))
         self.assertEqual(unpool_out, torch.zeros_like(unpool_out))
 
-    @slowTest
+    # @slowTest
     @onlyNativeDeviceTypes
     @parametrize_test(
         "module_name,module_size,output_size,test_index,should_error",
@@ -1114,7 +1116,7 @@ torch.cuda.synchronize()
         helper(10, 512, 31, 31, 3, stride=2)
         helper(1, 129, 8, 8, 3, stride=2)
 
-    @onlyCPU
+    # @onlyCPU
     @dtypes(torch.float, torch.double)
     def test_max_pool1d_corner_cases(self, device, dtype):
         def check(x, args, expected):
@@ -1215,7 +1217,7 @@ torch.cuda.synchronize()
         check(tensor, 3, 2, 1, 2, ceil_mode=True)
         check(tensor.transpose(1, 2), 3, 2, 1, 2, ceil_mode=True)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @gcIfJetson
     def test_max_pool2d(self, device):
         def helper(n, c, h, w, ks):
@@ -1396,7 +1398,7 @@ torch.cuda.synchronize()
         helper(1, 79, 4, 4, 4, 3, stride=2)
         helper(0, 79, 4, 4, 4, 3, stride=2)
 
-    @onlyCPU
+    # @onlyCPU
     @dtypes(torch.half, torch.bfloat16)
     def test_max_pool_bfloat16_half(self, device, dtype):
         def helper(shape, kernel_size, stride, memory_format, dtype):
@@ -1436,7 +1438,7 @@ torch.cuda.synchronize()
         helper((4, 10, 3, 8, 8), 3, 1, torch.contiguous_format, dtype)
         helper((4, 10, 8, 8, 8), 7, 1, torch.channels_last_3d, dtype)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @gcIfJetson
     def test_max_pool2d_indices(self, device):
         def helper(n, c, h, w, ks):
@@ -1490,7 +1492,7 @@ torch.cuda.synchronize()
                 indices,
             )
 
-    @onlyCPU
+    # @onlyCPU
     @dtypes(torch.half, torch.bfloat16)
     def test_avg_pool2d_reduced_floating(self, device, dtype):
         def helper(n, c, h, w, kernel_size, stride, memory_format):
@@ -1635,7 +1637,7 @@ torch.cuda.synchronize()
             helper(4, 8, 9, 14, (2, 2), (1, 1), (1, 1), (2, 2), contig, device)
             helper(4, 8, 11, 11, (4, 4), (2, 2), (2, 2), (2, 2), contig, device)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     def test_pool3d_size_one_feature_dim(self, device):
         # Tests crazy strides for feature dim of size 1
         x = torch.randn(7, 1, 5, 3, 2, device=device)
@@ -1654,7 +1656,7 @@ torch.cuda.synchronize()
             out_x = fn(x)
             self.assertEqual(out_y, out_x.to(device), msg=test)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @largeTensorTest("18GB")
     @largeTensorTest("180GB", "cpu")
     def test_pool3d_large_size_int64(self, device):
@@ -1677,7 +1679,7 @@ torch.cuda.synchronize()
         self.assertEqual(y, ref_y, exact_dtype=False)
         self.assertEqual(x.grad, ref_x.grad, exact_dtype=False)
 
-    @onlyCUDA
+    # @onlyPRIVATEUSE1
     def test_AvgPool3d_backward_after_cat_dim1_device(self, device):
         # x has to have batch_size 1 to test contiguous checks
         x = torch.randn(1, 3, 4, 4, 4, device=device, requires_grad=True)
@@ -2056,7 +2058,7 @@ torch.cuda.synchronize()
                 # check if the output shape was still computed correctly
                 self.assertEqual(x.shape[2], res.shape[2])
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @largeTensorTest("6GB")
     def test_pooling_large(self, device):
         def helper(pool):
@@ -2094,7 +2096,7 @@ torch.cuda.synchronize()
                         # some implementations do not support dilation
                         fn(x, 6, stride=2, padding=0)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     def test_pooling_bfloat16(self, device):
         _test_bfloat16_ops(
             self,
@@ -2145,7 +2147,7 @@ torch.cuda.synchronize()
             F.max_pool3d(x, kernel_size=(1, 1, 1)).sum().backward()
             self.assertEqual(x.grad, torch.ones_like(x.grad))
 
-    @slowTest
+    # @slowTest
     def test_adaptive_pool_odd_size(self, device):
         # See https://github.com/pytorch/pytorch/issues/81409
         Ih, Iw, Oh, Ow = 5873, 3693, 3527, 2219
