diff --git a/test/nn/test_multihead_attention.py b/test/nn/test_multihead_attention.py
index c041966..2a5d34e 100644
--- a/test/nn/test_multihead_attention.py
+++ b/test/nn/test_multihead_attention.py
@@ -5,6 +5,8 @@ import unittest
 import unittest.mock as mock
 
 import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 import torch.nn as nn
 from torch.nn import MultiheadAttention
 from torch.testing._internal.common_device_type import (
