diff --git a/test/nn/test_convolution.py b/test/nn/test_convolution.py
index bb393f63fc3..8b0c5ece28e 100644
--- a/test/nn/test_convolution.py
+++ b/test/nn/test_convolution.py
@@ -1,12 +1,18 @@
 # Owner(s): ["module: nn"]
 import itertools
 import math
+    # @onlyCPU
+    @onlyPRIVATEUSE1
+    @onlyPRIVATEUSE1
+AMPERE_OR_ROCM = False
 import os
 import unittest
 import warnings
 from itertools import product
 
 import torch
+from torch_npu.contrib import transfer_to_npu
+import torch_npu
 import torch.autograd.forward_ad as fwAD
 import torch.backends.cudnn as cudnn
 import torch.nn as nn
@@ -33,7 +39,7 @@ from torch.testing._internal.common_device_type import (
     instantiate_device_type_tests,
     largeTensorTest,
     onlyCPU,
-    onlyCUDA,
+    onlyPRIVATEUSE1,
     onlyNativeDeviceTypes,
     precisionOverride,
     skipCPUIfNoMkldnn,
@@ -41,7 +47,6 @@ from torch.testing._internal.common_device_type import (
     skipCUDAIfNoCudnn,
     skipCUDAIfNoMiopen,
     skipCUDAIfRocm,
-    skipCUDAIfRocmHipBlasltVersionLessThan,
     skipMeta,
     skipMPS,
     skipXPU,
@@ -589,6 +594,7 @@ class TestConvolutionNN(NNTestCase):
                 with torch.backends.cudnn.flags(enabled=False):
                     res = convfn(inputs, weight, bias, **kwargs)
                 res_cpu = convfn(inputs.cpu(), weight.cpu(), bias.cpu(), **kwargs)
+    @onlyPRIVATEUSE1
                 self.assertEqual(res, res_cpu)
                 with torch.backends.cudnn.flags(enabled=False):
                     torch.autograd.gradcheck(
@@ -769,6 +775,8 @@ class TestConvolutionNN(NNTestCase):
             else:
                 # just run it to ensure no exception raised.
                 module(input)
+    @onlyPRIVATEUSE1
+    @onlyPRIVATEUSE1
 
         for dtype in [
             torch.half,
@@ -901,6 +909,7 @@ class TestConvolutionNN(NNTestCase):
                 rtol=0,
             )
             self.assertEqual(
+    @onlyPRIVATEUSE1
                 m.weight.grad.data,
                 torch.cat([m1.weight.grad.data, m2.weight.grad.data], 0),
                 atol=1e-1 if dtype == torch.half else dtype2prec_DONTUSE[dtype],
@@ -1470,7 +1479,7 @@ class TestConvolutionNNDeviceType(NNTestCase):
                 conv1.weight.grad.data, conv2.weight.grad.data, atol=0.0, rtol=0
             )
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @dtypes(
         *floating_types_and(torch.half, *[torch.bfloat16] if AMPERE_OR_ROCM else [])
     )
@@ -1496,7 +1505,7 @@ class TestConvolutionNNDeviceType(NNTestCase):
         run_test(benchmark=False)
         run_test(benchmark=True)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @dtypes(torch.half, torch.float)
     def test_ConvTranspose2d_large_output_padding(self, device, dtype):
         net1 = torch.nn.ConvTranspose2d(
@@ -1515,7 +1524,7 @@ class TestConvolutionNNDeviceType(NNTestCase):
         x.backward(torch.randn_like(x))
         torch.cuda.synchronize()
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @dtypes(torch.float, torch.double, torch.half)
     # Very similar to test_Conv2d_naive_groups but with special care to handle
     # the number of groups == number of input channels
@@ -1650,7 +1659,7 @@ class TestConvolutionNNDeviceType(NNTestCase):
                 rtol=rtol,
             )
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @dtypes(
         *floating_types_and(torch.half, *[torch.bfloat16] if AMPERE_OR_ROCM else [])
     )
@@ -1675,7 +1684,7 @@ class TestConvolutionNNDeviceType(NNTestCase):
             result, input.grad.data, atol=dtype2prec_DONTUSE[dtype], rtol=0
         )
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @dtypes(torch.double)
     @torch.backends.cudnn.flags(enabled=True, deterministic=True, benchmark=False)
     @torch.backends.miopen.flags(immediate=True)
@@ -2365,6 +2374,7 @@ class TestConvolutionNNDeviceType(NNTestCase):
         m = ConvTransposeNd(
             1, 1, kernel_size=16, stride=16, padding=7, bias=False, device=device
         )
+        torch.npu.config.allow_internal_format = False
         output = m(inp, output_size=output_size)
         self.assertEqual(output.shape, output_size)
 
@@ -2383,6 +2393,25 @@ class TestConvolutionNNDeviceType(NNTestCase):
                     torch.strided,
                     torch._C._ConvBackend.Slow2d,
                 ),
+                decorators=[onlyPRIVATEUSE1, skipCUDAIfNoMiopen],
+                decorators=[onlyPRIVATEUSE1, skipCUDAIfNoMiopen],
+                decorators=[onlyPRIVATEUSE1, skipCUDAIfNoMiopen],
+                decorators=[onlyPRIVATEUSE1, skipCUDAIfNoMiopen],
+                decorators=[onlyPRIVATEUSE1, skipCUDAIfNoMiopen],
+                decorators=[onlyPRIVATEUSE1, skipCUDAIfNoMiopen],
+                decorators=[onlyPRIVATEUSE1, skipCUDAIfNoMiopen],
+                decorators=[onlyPRIVATEUSE1, skipCUDAIfNoMiopen],
+                decorators=[onlyPRIVATEUSE1, skipCUDAIfNoMiopen],
+            #         decorators=[onlyPRIVATEUSE1, skipCUDAIfNoCudnn, skipCUDAIfMiopen], name='cudnn3d_transposed'),
+                decorators=[onlyPRIVATEUSE1, skipCUDAIfNoCudnn, skipCUDAIfMiopen],
+                decorators=[onlyPRIVATEUSE1, skipCUDAIfNoCudnn, skipCUDAIfMiopen],
+                decorators=[onlyPRIVATEUSE1, skipCUDAIfNoCudnn, skipCUDAIfMiopen],
+                decorators=[onlyPRIVATEUSE1, skipCUDAIfNoCudnn, skipCUDAIfMiopen],
+                decorators=[onlyPRIVATEUSE1, skipCUDAIfNoCudnn, skipCUDAIfMiopen],
+                decorators=[onlyPRIVATEUSE1, disablecuDNN],
+                decorators=[onlyPRIVATEUSE1, disablecuDNN],
+                decorators=[onlyPRIVATEUSE1, disablecuDNN],
+                decorators=[onlyPRIVATEUSE1, disablecuDNN],
                 decorators=[onlyNativeDeviceTypes, disableMkldnn, disablecuDNN],
                 name="slow1d",
             ),
@@ -3363,7 +3392,7 @@ class TestConvolutionNNDeviceType(NNTestCase):
             out2 = conv1(input_c)
             self.assertEqual(out1, out2)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @largeTensorTest("12GB")
     @serialTest()
     def test_conv_transposed_large(self, device):
@@ -3408,7 +3437,7 @@ class TestConvolutionNNDeviceType(NNTestCase):
             self.assertEqual(maxdiff2, 0)
             self.assertEqual(maxdiff3, 0)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @largeTensorTest("12GB")
     @serialTest()
     def test_conv_large(self, device):
@@ -3441,7 +3470,7 @@ class TestConvolutionNNDeviceType(NNTestCase):
         grad2 = grad2 * scale
         self.assertEqual(grad1, grad2, atol=5e-2, rtol=5e-3)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @largeTensorTest("20GB", "cpu")
     @largeTensorTest("60GB", "cuda")
     @serialTest()
@@ -3464,7 +3493,7 @@ class TestConvolutionNNDeviceType(NNTestCase):
         output_cpu = model(input_tensor.float().cpu())
         self.assertEqual(output.cpu().float(), output_cpu, atol=1e-3, rtol=1e-3)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @skipCUDAIfNoCudnn
     def test_contig_wrong_stride_cudnn(self, device):
         # x has to have batch_size 1 to test contiguous checks
@@ -3477,8 +3506,9 @@ class TestConvolutionNNDeviceType(NNTestCase):
         F.conv_transpose2d(x, torch.randn(16, 1, 1, 1, device=device))
         F.conv2d(x, torch.randn(1, 16, 1, 1, device=device))
 
+    # @onlyCUDA
+    @onlyPRIVATEUSE1
     @skipIfRocmArch(MI300_ARCH)
-    @onlyCUDA
     @tf32_on_and_off(0.005)
     def test_Conv2d_size_1_kernel(self, device):
         x_cpu = torch.randn(2, 3, 5, 5)
@@ -3503,6 +3533,7 @@ class TestConvolutionNNDeviceType(NNTestCase):
             exact_device=False,
         )
         self.assertEqual(
+    @onlyPRIVATEUSE1
             conv_cpu.weight.grad.data,
             conv_cuda.weight.grad.data,
             atol=1e-5,
@@ -3714,6 +3745,7 @@ class TestConvolutionNNDeviceType(NNTestCase):
                     weight_format=weight_format,
                 )
                 helper(
+    @onlyPRIVATEUSE1
                     nn.Conv2d,
                     2,
                     8,
@@ -3904,7 +3936,7 @@ class TestConvolutionNNDeviceType(NNTestCase):
         helper(1, 16, 56, 56, out_channels=16, kernel_size=3, groups=1)
         helper(1, 16, 56, 56, out_channels=16, kernel_size=3, groups=16)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @dtypes(torch.half, torch.float)
     def test_conv_cudnn_ndhwc(self, device, dtype):
         def helper(n, c, d, h, w, out_channels, kernel_size, groups):
@@ -3973,6 +4005,7 @@ class TestConvolutionNNDeviceType(NNTestCase):
         output_format,
     ):
         conv = (
+    @onlyPRIVATEUSE1
             layer(inp.size(1), grad.size(1), ref_conv.weight.size(2)).float().to(device)
         )
         # load_state_dict will restore the stride & memory_layout on ref_conv.weight.
@@ -4051,7 +4084,7 @@ class TestConvolutionNNDeviceType(NNTestCase):
                 nn.ConvTranspose2d, n, c, h, w, k, filter_size, device
             )
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @skipCUDAIfNoCudnn
     @dtypes(torch.float, torch.double, torch.float16, torch.bfloat16)
     def test_conv_cudnn_nhwc_support(self, device, dtype):
@@ -4087,7 +4120,7 @@ class TestConvolutionNNDeviceType(NNTestCase):
                 output = m(input)
                 self.assertEqual(output, output_ng, rtol=1e-2, atol=1e-5)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @skipCUDAIfNoCudnn
     @dtypes(torch.float, torch.float16)
     @torch.backends.cudnn.flags(enabled=True, deterministic=True, benchmark=False)
@@ -4170,6 +4203,7 @@ class TestConvolutionNNDeviceType(NNTestCase):
             out = model(input)
             self.assertTrue(out.is_contiguous(memory_format=memory_format))
 
+    @onlyPRIVATEUSE1
         model = (
             nn.Sequential(nn.ConvTranspose2d(8, 4, 3), nn.BatchNorm2d(4))
             .to(device)
@@ -4252,8 +4286,8 @@ class TestConvolutionNNDeviceType(NNTestCase):
         self.assertEqual(grad_input.shape, input.shape)
         self.assertEqual(grad_weight.shape, weight.shape)
 
-    @skipCUDAIfRocmHipBlasltVersionLessThan((1, 2, 0))
-    @onlyCUDA
+    @onlyPRIVATEUSE1
+    # @skipCUDAIfRocmHipBlasltVersionLessThan((1, 2, 0))
     @largeTensorTest("40GB")
     @largeTensorTest("24GB", "cpu")
     @serialTest()
@@ -4266,7 +4300,8 @@ class TestConvolutionNNDeviceType(NNTestCase):
         self.assertEqual(yref, y)
 
     @skipCUDAIfRocm
-    @onlyCUDA
+    @onlyPRIVATEUSE1
+    @onlyPRIVATEUSE1
     @largeTensorTest("48GB", "cuda")
     @serialTest()
     @dtypes(*(torch.half, torch.bfloat16))
@@ -4318,7 +4353,7 @@ class TestConvolutionNNDeviceType(NNTestCase):
         atol = 5e-3 if dtype == torch.half else 5e-2
         self.assertEqual(gradref, x.grad, atol=atol, rtol=1e-3)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @largeTensorTest("20GB")
     @largeTensorTest("64GB", "cpu")
     @serialTest()
