diff --git a/test/distributed/launcher/test_run.py b/test/distributed/launcher/test_run.py
index f71bffd..b5195a5 100644
--- a/test/distributed/launcher/test_run.py
+++ b/test/distributed/launcher/test_run.py
@@ -1,3 +1,8 @@
+import torch_npu.testing
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 #!/usr/bin/env python3
 # Owner(s): ["oncall: r2p"]
 
@@ -244,6 +249,9 @@ class ElasticLaunchTest(TestCase):
         world_size = nnodes * expected_number
         # make sure all the workers ran
         # each worker touches a file with its global rank as the name
+
+        print("111111111111", world_size)
+        print("22222222222222", set(os.listdir(self.test_dir)))
         self.assertSetEqual(
             {str(i) for i in range(world_size)}, set(os.listdir(self.test_dir))
         )
