diff --git a/test/distributed/test_device_mesh.py b/test/distributed/test_device_mesh.py
index e6d8a09..67e6605 100644
--- a/test/distributed/test_device_mesh.py
+++ b/test/distributed/test_device_mesh.py
@@ -1,3 +1,8 @@
+import torch_npu.testing
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Copyright (c) Meta Platforms, Inc. and affiliates
 # Owner(s): ["oncall: distributed"]
 import os
@@ -28,10 +33,12 @@ from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
 from torch.testing._internal.common_utils import run_tests
 from torch.testing._internal.distributed._tensor.common_dtensor import (
     DTensorTestBase,
-    with_comms,
+    # with_comms,
 )
 from torch.testing._internal.distributed.fake_pg import FakeStore
 from torch.utils._typing_utils import not_none
+from torch_npu.testing.common_distributed import with_comms
+
 
 
 def _get_device_type(world_size):
@@ -40,7 +47,7 @@ def _get_device_type(world_size):
         and torch.cuda.device_count() >= world_size
         and is_nccl_available()
     ):
-        device_type = "cuda"
+        device_type = "npu"
     else:
         device_type = "cpu"
     return device_type
@@ -77,6 +84,7 @@ class DeviceMeshTest(DTensorTestBase):
 
     @skip_if_lt_x_gpu(4)
     def test_init_process_group(self):
+        torch.npu.set_device(self.rank)
         device_type = _get_device_type(self.world_size)
         mesh_tensor = torch.arange(4).reshape(2, 2)
         self.assertTrue(not is_initialized())
@@ -92,7 +100,7 @@ class DeviceMeshTest(DTensorTestBase):
         with self.assertRaises(ValueError):
             DeviceMesh(self.device_type, mesh)
 
-    @with_comms()
+    @with_comms
     def test_2d_mesh_non_eager_init_subgroup(self):
         mesh_shape = (2, self.world_size // 2)
         mesh_2d = init_device_mesh(self.device_type, mesh_shape)
@@ -102,7 +110,7 @@ class DeviceMeshTest(DTensorTestBase):
 
     # TODO: need to refactor the other tests in this file to test both
     # eager_init=True and eager_init=False scenarios.
-    @with_comms(eager_init=True)
+    @with_comms
     def test_2d_mesh_eager_init_subgroup(self):
         mesh_shape = (2, self.world_size // 2)
         mesh_2d = init_device_mesh(self.device_type, mesh_shape)
@@ -114,7 +122,7 @@ class DeviceMeshTest(DTensorTestBase):
             self.assertEqual(mesh_2d.get_group(0).bound_device_id.index, curr_device)
             self.assertEqual(mesh_2d.get_group(1).bound_device_id.index, curr_device)
 
-    @with_comms()
+    @with_comms
     def test_get_group_and_get_all_groups(self):
         mesh_shape = (2, self.world_size // 2)
         mesh_2d = init_device_mesh(
@@ -218,6 +226,8 @@ class DeviceMeshTest(DTensorTestBase):
     def test_from_group_with_global_pg(self):
         # Simple test: check `from_group` from a mesh pg vs. directly
         # initializing via `init_device_mesh`
+        self.device_type = 'npu'
+
         ref_global_mesh = init_device_mesh(self.device_type, (self.world_size,))
         mesh_pg = ref_global_mesh.get_group()
         global_mesh = DeviceMesh.from_group(mesh_pg, self.device_type)
@@ -357,6 +367,8 @@ class DeviceMeshTestNDim(DTensorTestBase):
 
     @with_comms
     def test_device_mesh_parent_child_hash(self):
+        self.device_type = 'npu'
+
         mesh_2d = init_device_mesh(
             self.device_type, (2, self.world_size // 2), mesh_dim_names=("DP", "TP")
         )
@@ -429,11 +441,13 @@ class DeviceMeshTestNDim(DTensorTestBase):
         ):
             self.assertEqual(ref_ranks, ranks)
 
-    @with_comms()
+    @with_comms
     def test_from_group_with_mesh_shape_2d(self):
         """Tests ``from_group`` when passing ``mesh_shape`` as 2D."""
         # Consider the following scenario where the process group has been created,
         # but we need to create the 2D HSDP mesh from it later in the program.
+        self.device_type = 'npu'
+
         mesh_shape = (2, 4)
         mesh_dim_names = ("dp_replicate", "dp_shard")
         ref_mesh = init_device_mesh(
@@ -489,6 +503,8 @@ class InitDeviceMeshTest(DTensorTestBase):
 
     @with_comms
     def test_init_device_mesh(self):
+        self.device_type = 'npu'
+        
         mesh_shape = (2, 4)
         mesh_dim_names = ("DP", "TP")
         ref_mesh = DeviceMesh(
@@ -702,7 +718,7 @@ class TestDeviceMeshGetItem(DTensorTestBase):
         cp_tp_mesh._flatten("dummy")
         self.assertEqual(mesh_3d["dummy"].mesh_dim_names[0], "dummy")
 
-    @with_comms(eager_init=True)
+    @with_comms
     def test_flatten_mesh_4d(self):
         mesh_shape = (2, 2, 2, 1)
         mesh_dim_names = ("dp_replicate", "dp_shard", "cp", "tp")
