diff --git a/test/distributed/test_nccl.py b/test/distributed/test_nccl.py
index 78db1c1aa9a..90d373fa666 100644
--- a/test/distributed/test_nccl.py
+++ b/test/distributed/test_nccl.py
@@ -4,6 +4,8 @@ import os
 import sys
 
 import torch
+from torch_npu.contrib import transfer_to_npu
+import torch_npu
 import torch.cuda
 import torch.cuda.nccl as nccl
 import torch.distributed as c10d
@@ -30,7 +32,8 @@ from torch.testing._internal.common_utils import (
     TEST_WITH_ROCM,
     TestCase,
 )
-
+TEST_CUDA = True
+TEST_MULTIGPU = True
 
 # load_tests from common_utils is used to automatically filter tests for
 # sharding on sandcastle. This line silences flake warnings
@@ -43,10 +46,10 @@ if not TEST_CUDA:
 
 
 datatypes = [torch.float]
-if (
-    TEST_CUDA and c10d.is_nccl_available() and nccl.version() >= (2, 10)
-) or TEST_WITH_ROCM:
-    datatypes.append(torch.bfloat16)
+# if (
+#     TEST_CUDA and c10d.is_nccl_available() and nccl.version() >= (2, 10)
+# ) or TEST_WITH_ROCM:
+datatypes.append(torch.bfloat16)
 
 # Broadcast (and alltoall) support float8, while reduce and allreduce do not support float8 currently
 broadcast_dtypes = (
