diff --git a/test/distributed/test_cupy_as_tensor.py b/test/distributed/test_cupy_as_tensor.py
index 63b290e2e8e..e7140a6ce34 100644
--- a/test/distributed/test_cupy_as_tensor.py
+++ b/test/distributed/test_cupy_as_tensor.py
@@ -1,11 +1,14 @@
 # Owner(s): ["oncall: distributed"]
 
 # To run:
+device_type = "npu"
 # python test/distributed/test_cupy_as_tensor.py
 
 from dataclasses import dataclass
 
 import torch
+from torch_npu.contrib import transfer_to_npu
+import torch_npu
 from torch.multiprocessing.reductions import reduce_tensor
 from torch.testing._internal.common_cuda import SM100OrLater
 from torch.testing._internal.common_distributed import (
