diff --git a/test/distributed/test_dist2.py b/test/distributed/test_dist2.py
index fd44be2d06d..2fe233e6560 100644
--- a/test/distributed/test_dist2.py
+++ b/test/distributed/test_dist2.py
@@ -5,6 +5,8 @@ import unittest
 from datetime import timedelta
 
 import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 import torch.distributed as dist
 import torch.distributed._dist2 as dist2
 from torch.testing._internal.common_distributed import (
@@ -287,7 +289,7 @@ class ProcessGroupGlooTest(Dist2MultiProcessTestCase):
 class ProcessGroupNCCLTest(Dist2MultiProcessTestCase):
     @property
     def device(self) -> torch.device:
-        return torch.device("cuda", self.rank)
+        return torch.device("npu", self.rank)
 
     @requires_nccl()
     @skip_if_lt_x_gpu(2)
