diff --git a/test/distributed/test_c10d_common.py b/test/distributed/test_c10d_common.py
index ca38c54..cb6e638 100644
--- a/test/distributed/test_c10d_common.py
+++ b/test/distributed/test_c10d_common.py
@@ -1,3 +1,8 @@
+import torch_npu.testing
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["oncall: distributed"]
 import torch_npu.testing
 import torch_npu
@@ -1907,6 +1912,11 @@ class ProcessGroupWithDispatchedCollectivesTests(MultiProcessTestCase):
                 world_size=self.world_size,
                 store=store,
             )
+            
+            # On NPU, nccl is replaced by hccl and the process group reports name "custom"
+            if (backend == dist.Backend.NCCL and getattr(torch, "npu", None) is not None and torch.npu.is_available()):
+                excepted_backend = "custom"
+
             pg = c10d._get_default_group()
             self.assertEqual(pg.rank(), self.rank)
             self.assertEqual(pg.size(), self.world_size)