diff --git a/test/distributed/test_c10d_gloo.py b/test/distributed/test_c10d_gloo.py
index aec59c2..aab09c1 100644
--- a/test/distributed/test_c10d_gloo.py
+++ b/test/distributed/test_c10d_gloo.py
@@ -1,3 +1,8 @@
+import torch_npu.testing
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["oncall: distributed"]
 
 import copy
@@ -1797,11 +1802,11 @@ class DistributedDataParallelTest(
             world_size=self.world_size,
             rank=self.rank,
         )
-        device = torch.device(f"cuda:{self.rank}")
+        device = torch.device(f"npu:{self.rank}")
         local_shard_metadata = ShardMetadata(
             shard_offsets=[(self.rank % 2) * 5, 0],
             shard_sizes=[5, 10],
-            placement=f"rank:{self.rank}/cuda:{self.rank}",
+            placement=f"rank:{self.rank}/npu:{self.rank}",
         )
         local_shards = [Shard(torch.randn(5, 10, device=device), local_shard_metadata)]
         st = init_from_local_shards(local_shards, [10, 10])
