diff --git a/test/distributed/test_run.py b/test/distributed/test_run.py
index be10c3a78ba..bd861ee5e8e 100644
--- a/test/distributed/test_run.py
+++ b/test/distributed/test_run.py
@@ -13,6 +13,8 @@ from unittest.mock import MagicMock, patch
 import torch.distributed.run as run
 from torch.distributed.launcher.api import launch_agent, LaunchConfig
 from torch.testing._internal.common_utils import run_tests, TestCase
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 
 
 class RunTest(TestCase):
