diff --git a/test/distributed/test_nvshmem_triton.py b/test/distributed/test_nvshmem_triton.py
index a7555760458..bdc230070f7 100644
--- a/test/distributed/test_nvshmem_triton.py
+++ b/test/distributed/test_nvshmem_triton.py
@@ -1,6 +1,7 @@
 # Owner(s): ["oncall: distributed"]
 # To run:
 # python test/distributed/test_nvshmem_triton.py
+device_type = "npu"
 
 import sys
 
@@ -19,6 +20,8 @@ if TEST_WITH_ROCM or not torch.backends.cuda.is_built():
 import triton.language as tl
 
 import torch.distributed as dist
+from torch_npu.contrib import transfer_to_npu
+import torch_npu
 import torch.distributed._symmetric_memory as symm_mem
 import torch.distributed._symmetric_memory._nvshmem_triton as nvshmem
 from torch._inductor.runtime.triton_compat import triton
