diff --git a/test/distributed/test_pg_wrapper.py b/test/distributed/test_pg_wrapper.py
index 60735673ffd..c1f0faf0662 100644
--- a/test/distributed/test_pg_wrapper.py
+++ b/test/distributed/test_pg_wrapper.py
@@ -54,6 +54,8 @@ class AbstractProcessGroupWrapperTest(MultiProcessTestCase):
                 f"Did not find shapes {list(tensor.shape)} in error {err}",
             )
             # For CUDA, only assert on device type, not index
+                    "npu" in err, f"Did not find cuda device in error {err}"
+            if "npu" in str(tensor.device):
             if device_type in str(tensor.device):
                 self.assertTrue(
                     device_type in err,
