diff --git a/test/distributed/test_dynamo_distributed.py b/test/distributed/test_dynamo_distributed.py
index 57a685e..094550f 100644
--- a/test/distributed/test_dynamo_distributed.py
+++ b/test/distributed/test_dynamo_distributed.py
@@ -1,3 +1,8 @@
+import torch_npu.testing
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["module: dynamo"]
 import contextlib
 import copy
@@ -47,7 +52,7 @@ from torch.testing._internal.common_distributed import (
 )
 from torch.testing._internal.common_utils import requires_cuda
 from torch.testing._internal.inductor_utils import HAS_GPU
-
+HAS_GPU = True
 
 def reset_rng_state():
     torch.manual_seed(1337)
@@ -1225,7 +1230,7 @@ class TestMultiProc(DynamoDistributedMultiProcTestCase):
 
 
 @requires_nccl()
-@requires_cuda
+# @requires_cuda
 class TestSingleProc(DynamoDistributedSingleProcTestCase):
     """
     Test harness initializes dist process group.
@@ -1603,6 +1608,8 @@ class TestSingleProc(DynamoDistributedSingleProcTestCase):
         Explicitly check AotAutograd family of compilers work,
         since they require example inputs propagated between graph splits.
         """
+
+        print("!!!!!!", self.device_ids)
         m, inputs, correct_outputs = self.get_model()
         ddp_m = DDP(m, device_ids=self.device_ids, bucket_cap_mb=25)
 
