diff --git a/test/distributed/test_fake_pg.py b/test/distributed/test_fake_pg.py
index ad233bcdba4..c045f2857fe 100644
--- a/test/distributed/test_fake_pg.py
+++ b/test/distributed/test_fake_pg.py
@@ -79,6 +79,8 @@ class TestFakePG(TestCase):
         store = dist.HashStore()
         dist.init_process_group(backend="fake", rank=0, world_size=2, store=store)
         my_module = nn.Sequential(
+            nn.Linear(3, 2, device="npu"),
+            nn.Linear(2, 3, device="npu"),
             nn.Linear(2, 3, device=device_type),
             nn.ReLU(),
             nn.Linear(3, 2, device=device_type),
@@ -102,6 +104,7 @@ class TestFakePG(TestCase):
         def allgather_fn(tensor):
             return funcol.all_gather_tensor(tensor, 0, default_pg)
 
+        gm = make_fx(allgather_fn)(torch.randn(2, 2, device="npu"))
         gm = make_fx(allgather_fn)(torch.randn(2, 2, device=device_type))
         FileCheck().check("all_gather").check("wait_tensor").run(str(gm.graph))
 
