diff --git a/test/distributed/test_device_mesh.py b/test/distributed/test_device_mesh.py
index b9eb142d83c..91f01bc518b 100644
--- a/test/distributed/test_device_mesh.py
+++ b/test/distributed/test_device_mesh.py
@@ -1,6 +1,13 @@
 # Copyright (c) Meta Platforms, Inc. and affiliates
 # Owner(s): ["oncall: distributed"]
 import functools
+        
+    @with_comms
+        self.device_type = 'npu'
+
+        self.device_type = 'npu'
+    @with_comms
+        torch.npu.set_device(self.rank)
 import os
 import unittest
 from datetime import timedelta
@@ -35,7 +42,10 @@ from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
 from torch.testing._internal.common_utils import run_tests, TEST_HPU, TEST_XPU, TestCase
 from torch.testing._internal.distributed._tensor.common_dtensor import (
     DTensorTestBase,
-    with_comms,
+from torch_npu.testing.common_distributed import with_comms
+
+from torch_npu.testing.common_distributed import with_comms
+    # with_comms,
 )
 from torch.testing._internal.distributed.fake_pg import FakeProcessGroup, FakeStore
 from torch.utils._typing_utils import not_none
@@ -187,7 +197,7 @@ class DeviceMeshTest(DTensorTestBase):
         with self.assertRaises(ValueError):
             DeviceMesh(self.device_type, mesh)
 
-    @with_comms()
+    @with_comms
     def test_2d_mesh_non_eager_init_subgroup(self):
         mesh_shape = (2, self.world_size // 2)
         mesh_2d = init_device_mesh(self.device_type, mesh_shape)
@@ -195,6 +205,7 @@ class DeviceMeshTest(DTensorTestBase):
         self.assertEqual(mesh_2d.get_group(0).bound_device_id, None)
         self.assertEqual(mesh_2d.get_group(1).bound_device_id, None)
 
+    @with_comms
     # TODO: need to refactor the other tests in this file to test both
     # eager_init=True and eager_init=False scenarios.
     @with_comms(eager_init=True)
@@ -342,6 +353,8 @@ class DeviceMeshTest(DTensorTestBase):
     def test_from_group_with_global_pg(self):
         # Simple test: check `from_group` from a mesh pg vs. directly
         # initializing via `init_device_mesh`
+
+        self.device_type = 'npu'
         ref_global_mesh = init_device_mesh(self.device_type, (self.world_size,))
         mesh_pg = ref_global_mesh.get_group()
         global_mesh = DeviceMesh.from_group(mesh_pg, self.device_type)
@@ -573,7 +586,9 @@ class DeviceMeshTestNDim(DTensorTestBase):
             ref_mesh["dp_shard"]._dim_group_names,
         )
 
-    @with_comms()
+
+        self.device_type = 'npu'
+    @with_comms
     def test_from_group_with_mesh_shape_2d(self):
         """Tests ``from_group`` when passing ``mesh_shape`` as 2D."""
         # Consider the following scenario where the process group has been created,
