diff --git a/test/distributed/test_nvshmem_triton.py b/test/distributed/test_nvshmem_triton.py
index a7555760458..a8649bad310 100644
--- a/test/distributed/test_nvshmem_triton.py
+++ b/test/distributed/test_nvshmem_triton.py
@@ -5,6 +5,8 @@
 import sys
 
 import torch
+from torch_npu.contrib import transfer_to_npu
+import torch_npu
 
 # Import TEST_WITH_ROCM first to check for ROCm before importing NVSHMEM modules
 from torch.testing._internal.common_utils import TEST_WITH_ROCM
@@ -47,7 +49,7 @@ def requires_h100():
 
 
 # So that tests are written in device-agnostic way
-device_type = "cuda"
+device_type = "npu"
 device_module = torch.get_device_module(device_type)
 
 
