diff --git a/test/test_optim.py b/test/test_optim.py
index 2dcf3fa..23b650d 100644
--- a/test/test_optim.py
+++ b/test/test_optim.py
@@ -12,6 +12,8 @@ from optim.test_optim import TestDifferentiableOptimizer  # noqa: F401
 from optim.test_swa_utils import TestSWAUtils  # noqa: F401
 
 import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 from torch.nn import Parameter
 from torch.optim import Optimizer, SGD
 from torch.optim.lr_scheduler import ReduceLROnPlateau
@@ -112,7 +114,7 @@ class TestOptimRenewed(TestCase):
         * Grads can also be None, empty, or zero-valued, and this should not disrupt training.
     """
 
-    @onlyCPU
+    # @onlyCPU
     @optims(optim_db)
     def test_optim_infos_do_not_specify_global_cliquey_kwargs(
         self, device, dtype, optim_info
@@ -906,7 +908,7 @@ class TestOptimRenewed(TestCase):
                     actual = new_p_state[k]
                     self.assertEqual(og_p_state[k], actual, rtol=rtol, atol=atol)
 
-    @onlyCUDA
+    # @onlyCUDA
     @optims(
         [optim for optim in optim_db if "foreach" in optim.supported_impls],
         dtypes=[torch.float64],
@@ -934,7 +936,7 @@ class TestOptimRenewed(TestCase):
             finally:
                 torch.set_default_dtype(old_default_dtype)
 
-    @onlyCUDA
+    # @onlyCUDA
     @largeTensorTest("72GB", "cuda")
     @optims(
         [optim for optim in optim_db if "foreach" in optim.supported_impls],
@@ -949,7 +951,7 @@ class TestOptimRenewed(TestCase):
             optimizer = optim_cls(params, foreach=True, **optim_input.kwargs)
             optimizer.step()
 
-    @onlyCUDA
+    # @onlyCUDA
     @optims(
         [optim for optim in optim_db if "foreach" in optim.supported_impls],
         dtypes=[torch.float32],
@@ -1118,7 +1120,7 @@ class TestOptimRenewed(TestCase):
             optimizer = optim_cls(params, fused=True, **optim_input.kwargs)
             optimizer.step()
 
-    @onlyCUDA
+    # @onlyCUDA
     @optims(
         [optim for optim in optim_db if "fused" in optim.supported_impls],
         dtypes=[torch.float32],
@@ -2120,7 +2122,7 @@ class TestOptimRenewed(TestCase):
             res2 = optim_neg_inf.step(closure)
             self.assertEqual(type(res1), type(res2))
 
-    @onlyCUDA
+    # @onlyCUDA
     @optims(
         [
             optim
@@ -2168,7 +2170,7 @@ class TestOptimRenewed(TestCase):
                 optimizers.append(optimizer)
         self._compare_between(inpts, models, optimizers)
 
-    @onlyCUDA
+    # @onlyCUDA
     @optims(
         [
             o
