diff --git a/test/test_legacy_vmap.py b/test/test_legacy_vmap.py
index 882838b..2ff255b 100644
--- a/test/test_legacy_vmap.py
+++ b/test/test_legacy_vmap.py
@@ -13,6 +13,9 @@ from torch._vmap_internals import vmap
 from torch.testing._internal.common_device_type import instantiate_device_type_tests
 from torch.testing._internal.common_utils import run_tests, skipIfTorchDynamo, TestCase
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
 FALLBACK_REGEX = r"There is a performance drop"
 
