diff --git a/test/test_autocast.py b/test/test_autocast.py
index 19e05dd..0579881 100644
--- a/test/test_autocast.py
+++ b/test/test_autocast.py
@@ -11,6 +11,9 @@ from torch.testing._internal.common_device_type import expectedFailureMPSPre14
 from torch.testing._internal.common_utils import run_tests, skipIfTorchDynamo, TestCase
 from torch.utils._python_dispatch import TorchDispatchMode
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
 class TestAutocastCPU(TestAutocast):
     def setUp(self):
@@ -210,9 +213,11 @@ class WeightDTypeCastCounterMode(TorchDispatchMode):
 
     def __torch_dispatch__(self, func, types, args=(), kwargs=None):
         if (
-            func is torch.ops.aten._to_copy.default
+            (func is torch.ops.aten._to_copy.default
             and args[0] is self.weight
-            and kwargs["dtype"] is torch.float16
+            and kwargs["dtype"] is torch.float16)
+            or (func is torch.ops.npu._npu_dtype_cast.default
+            and args[0] is self.weight)
         ):
             self.dtype_cast_counter += 1
         return func(*args, **kwargs)
