diff --git a/test/test_ops_fwd_gradients.py b/test/test_ops_fwd_gradients.py
index 1eeb481..e96bcf3 100644
--- a/test/test_ops_fwd_gradients.py
+++ b/test/test_ops_fwd_gradients.py
@@ -19,7 +19,8 @@ from torch.testing._internal.common_utils import (
     TestGradients,
     unMarkDynamoStrictTest,
 )
-
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 
 # TODO: mitigate flaky issue on macOS https://github.com/pytorch/pytorch/issues/66033
 # AFAIK, c10::ThreadPool looks correct in the way it uses condition_variable wait. The
