diff --git a/test/test_cuda_sanitizer.py b/test/test_cuda_sanitizer.py
index 6d2ecc3..f720458 100644
--- a/test/test_cuda_sanitizer.py
+++ b/test/test_cuda_sanitizer.py
@@ -10,7 +10,7 @@ import torch.cuda._sanitizer as csan
 from torch.cuda._sanitizer import DataPtr, EventId, StreamId
 from torch.testing._internal.common_utils import NoTest, run_tests, TEST_CUDA, TestCase
 from torch.testing._internal.two_tensor import TwoTensor
-
+from torch_npu.contrib import transfer_to_npu
 
 if not TEST_CUDA:
     print("CUDA not available, skipping tests", file=sys.stderr)
