diff --git a/test/test_proxy_tensor.py b/test/test_proxy_tensor.py
index 26131d5..02704bb 100644
--- a/test/test_proxy_tensor.py
+++ b/test/test_proxy_tensor.py
@@ -32,7 +32,7 @@ import re
 
 import functools
 import itertools
-
+from torch_npu.contrib import transfer_to_npu
 aten = torch.ops.aten
 
 HAS_CUDA = torch.cuda.is_available()
@@ -2148,7 +2148,7 @@ class TestProxyTensorOpInfo(TestCase):
         _test_make_fx_helper(self, device, dtype, op, "symbolic", out=True)
 
 
-only_for = ("cpu")
+only_for = ("cpu",)
 instantiate_device_type_tests(TestProxyTensorOpInfo, globals(), only_for=only_for)
 
 
