diff --git a/test/test_tensorexpr_pybind.py b/test/test_tensorexpr_pybind.py
index 0c39fd6..2bdbfb9 100644
--- a/test/test_tensorexpr_pybind.py
+++ b/test/test_tensorexpr_pybind.py
@@ -7,7 +7,8 @@ import torch._C._te as te
 from torch.testing._internal.common_utils import run_tests
 from torch.testing._internal.jit_utils import JitTestCase
 import unittest
-
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 LLVM_ENABLED = torch._C._llvm_enabled()
 
 
