diff --git a/test/test_xpu.py b/test/test_xpu.py
index 4208bf6..243fe44 100644
--- a/test/test_xpu.py
+++ b/test/test_xpu.py
@@ -490,7 +490,7 @@ print(torch.xpu.device_count())
         del a
 
     @skipXPUIf(
-        int(torch.version.xpu) < 20250000,
+        torch.version.xpu is None or int(torch.version.xpu) < 20250000,
         "Test requires SYCL compiler version 2025.0.0 or newer.",
     )
     def test_mem_get_info(self):
