diff --git a/test/test_per_overload_api.py b/test/test_per_overload_api.py
index e47c55f..d3e4605 100644
--- a/test/test_per_overload_api.py
+++ b/test/test_per_overload_api.py
@@ -2,9 +2,9 @@
 import copy
 
 import torch
+from torch_npu.contrib import transfer_to_npu
 from torch.testing._internal.common_utils import run_tests, TestCase
 
-
 class TestPerOverloadAPI(TestCase):
     def test_basics_opoverloadpacket(self):
         # add is ony used as an example here. It is ok to update the test
