diff --git a/test/test_monitor.py b/test/test_monitor.py
index cf9cecc..ac92236 100644
--- a/test/test_monitor.py
+++ b/test/test_monitor.py
@@ -19,6 +19,9 @@ from torch.monitor import (
 )
 from torch.testing._internal.common_utils import run_tests, skipIfTorchDynamo, TestCase
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
 class TestMonitor(TestCase):
     def test_interval_stat(self) -> None:
