diff --git a/test/test_ops_gradients.py b/test/test_ops_gradients.py
index a0adc4c..a903ae7 100644
--- a/test/test_ops_gradients.py
+++ b/test/test_ops_gradients.py
@@ -24,7 +24,7 @@ from torch.testing._internal.hop_db import hop_db
 _gradcheck_ops = partial(
     ops, dtypes=OpDTypes.supported, allowed_dtypes=[torch.double, torch.cdouble]
 )
-
+from torch_npu.contrib import transfer_to_npu
 
 @unMarkDynamoStrictTest
 class TestBwdGradients(TestGradients):
