diff --git a/test/test_stateless.py b/test/test_stateless.py
index 9838729..ea43289 100644
--- a/test/test_stateless.py
+++ b/test/test_stateless.py
@@ -13,6 +13,7 @@ from torch.testing._internal.common_cuda import TEST_MULTIGPU
 from torch.testing._internal.common_utils import run_tests, TestCase, parametrize, instantiate_parametrized_tests, \
     subtest
 
+from torch_npu.contrib import transfer_to_npu
 
 class MockModule(torch.nn.Module):
     def __init__(self) -> None:
@@ -104,6 +105,7 @@ class TestStatelessFunctionalAPI(TestCase):
         subtest(stateless.functional_call, "stateless")
     ])
     def test_functional_call_with_jit(self, functional_call):
+        # 这个测试用例是jit不能使用transfer_to_npu需要单独跑
         module = MockModule()
         jit_module = torch.jit.script(module)
         with self.assertRaisesRegex(
