diff --git a/test/test_mkldnn_fusion.py b/test/test_mkldnn_fusion.py
index 4cb2786..8384f9a 100644
--- a/test/test_mkldnn_fusion.py
+++ b/test/test_mkldnn_fusion.py
@@ -4,6 +4,8 @@ import unittest
 from typing import NamedTuple
 
 import torch
+import torch_npu
+#from torch_npu.contrib import transfer_to_npu
 from torch import nn
 
 from torch.testing._internal.common_utils import run_tests, skipIfTorchDynamo
