diff --git a/test/test_import_stats.py b/test/test_import_stats.py
index bebd291..c31c5aa 100644
--- a/test/test_import_stats.py
+++ b/test/test_import_stats.py
@@ -2,6 +2,8 @@
 
 from torch.testing._internal.common_utils import TestCase, run_tests
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 
 # these tests could eventually be changed to fail if the import/init
 # time is greater than a certain threshold, but for now we just use them
