diff --git a/test/test_view_ops.py b/test/test_view_ops.py
index 2e113a9..6f3f588 100644
--- a/test/test_view_ops.py
+++ b/test/test_view_ops.py
@@ -129,7 +129,7 @@ class TestViewOps(TestCase):
             return False
         # Note: only validates storage on native device types
         # because some accelerators, like XLA, do not expose storage
-        if base.device.type == "cpu" or base.device.type == "cuda":
+        if base.device.type == "cpu" or base.device.type == "npu":
             if base.untyped_storage().data_ptr() != other.untyped_storage().data_ptr():
                 return False
 
@@ -1256,7 +1256,7 @@ class TestOldViewOps(TestCase):
             ):
                 src.flatten(2, 0)
 
-    # TODO: update to work on CUDA, too
+    # TODO: update to work on NPU, too
     @onlyCPU
     def test_narrow(self, device):
         x = torch.tensor([[0, 1, 2], [3, 4, 5], [6, 7, 8]])
@@ -1271,7 +1271,7 @@ class TestOldViewOps(TestCase):
         self.assertEqual(x.narrow(-1, -1, 1), torch.tensor([[2], [5], [8]]))
         self.assertEqual(x.narrow(-2, -1, 1), torch.tensor([[6, 7, 8]]))
 
-    # TODO: update to work on CUDA, too
+    # TODO: update to work on NPU, too
     @onlyCPU
     def test_narrow_tensor(self, device):
         x = torch.tensor([[0, 1, 2], [3, 4, 5], [6, 7, 8]])
@@ -1283,7 +1283,7 @@ class TestOldViewOps(TestCase):
         with self.assertRaises(Exception):
             x.narrow(0, torch.tensor([0, 1]), 1)
 
-    # TODO: make work on CUDA, too
+    # TODO: make work on NPU, too
     @onlyCPU
     def test_t(self, device):
         # Test 0D tensors
@@ -1379,7 +1379,7 @@ class TestOldViewOps(TestCase):
         with self.assertRaisesRegex(RuntimeError, error_regex):
             tensor.chunk(-2)
 
-    # TODO: make work on CUDA, too
+    # TODO: make work on NPU, too
     @skipIfTorchDynamo("TorchDynamo fails with unknown reason")
     @onlyCPU
     def test_unsqueeze(self, device) -> None:
