diff --git a/test/test_itt.py b/test/test_itt.py
index 99841e1..425bc7b 100644
--- a/test/test_itt.py
+++ b/test/test_itt.py
@@ -4,6 +4,10 @@ import torch
 import unittest
 from torch.testing._internal.common_utils import TestCase, run_tests, load_tests
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
+
 # load_tests from common_utils is used to automatically filter tests for
 # sharding on sandcastle. This line silences flake warnings
 load_tests = load_tests
