diff --git a/test/test_foreach.py b/test/test_foreach.py
index 8531c3a..b3e436f 100644
--- a/test/test_foreach.py
+++ b/test/test_foreach.py
@@ -1,5 +1,11 @@
 # Owner(s): ["module: mta"]
 # ruff: noqa: F841
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 import itertools
 import os
 import random
@@ -9,7 +15,6 @@ import weakref
 from contextlib import nullcontext
 from numbers import Number
 
-import torch
 from torch.testing import make_tensor
 from torch.testing._comparison import default_tolerances
 from torch.testing._internal.common_cuda import _get_torch_cuda_version, TEST_MULTIGPU
