diff --git a/test/test_kernel_launch_checks.py b/test/test_kernel_launch_checks.py
index 10479e7..e014d16 100644
--- a/test/test_kernel_launch_checks.py
+++ b/test/test_kernel_launch_checks.py
@@ -5,6 +5,9 @@ from torch.testing._internal.check_kernel_launches import (
     check_cuda_kernel_launches, check_code_for_cuda_kernel_launches
 )
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
 class AlwaysCheckCudaLaunchTest(TestCase):
     def test_check_code(self):
