diff --git a/test/test_license.py b/test/test_license.py
index 516cb78..a88bd5a 100644
--- a/test/test_license.py
+++ b/test/test_license.py
@@ -14,6 +14,10 @@ try:
 except ImportError:
     create_bundled = None
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
+
 license_file = "third_party/LICENSES_BUNDLED.txt"
 starting_txt = "The PyTorch repository and source distributions bundle"
 site_packages = os.path.dirname(os.path.dirname(torch.__file__))
