diff --git a/test/test_quantization.py b/test/test_quantization.py
index 61a8e31..7a257a4 100644
--- a/test/test_quantization.py
+++ b/test/test_quantization.py
@@ -2,7 +2,6 @@
 
 import logging
 from torch.testing._internal.common_utils import run_tests
-
 # Quantization core tests. These include tests for
 # - quantized kernels
 # - quantized functional operators
@@ -13,9 +12,9 @@ from torch.testing._internal.common_utils import run_tests
 # 1. Quantized Kernels
 # TODO: merge the different quantized op tests into one test class
 from quantization.core.test_quantized_op import TestQuantizedOps  # noqa: F401
+from quantization.core.test_quantized_op import TestQuantizedConv  # noqa: F401
 from quantization.core.test_quantized_op import TestQNNPackOps  # noqa: F401
 from quantization.core.test_quantized_op import TestQuantizedLinear  # noqa: F401
-from quantization.core.test_quantized_op import TestQuantizedConv  # noqa: F401
 from quantization.core.test_quantized_op import TestDynamicQuantizedOps  # noqa: F401
 from quantization.core.test_quantized_op import TestComparatorOps  # noqa: F401
 from quantization.core.test_quantized_op import TestPadding  # noqa: F401
@@ -78,8 +77,8 @@ except ImportError as e:
     # In FBCode we separate FX out into a separate target for the sake of dev
     # velocity. These are covered by a separate test target `quantization_fx`
     logging.warning(e)
-
-# PyTorch 2 Export Quantization
+#
+# # PyTorch 2 Export Quantization
 try:
     # To be moved to compiler side later
     from quantization.pt2e.test_graph_utils import TestGraphUtils  # noqa: F401
@@ -153,13 +152,13 @@ try:
 except ImportError as e:
     logging.warning(e)
 
-# Experimental functionality
+# # Experimental functionality
 try:
     from quantization.core.experimental.test_bits import TestBitsCPU  # noqa: F401
 except ImportError as e:
     logging.warning(e)
 try:
-    from quantization.core.experimental.test_bits import TestBitsCUDA  # noqa: F401
+    from quantization.core.experimental.test_bits import TestBitsPRIVATEUSE1  # noqa: F401
 except ImportError as e:
     logging.warning(e)
 try:
@@ -167,7 +166,7 @@ try:
 except ImportError as e:
     logging.warning(e)
 try:
-    from quantization.core.experimental.test_float8 import TestFloat8DtypeCUDA  # noqa: F401
+    from quantization.core.experimental.test_float8 import TestFloat8DtypePRIVATEUSE1  # noqa: F401
 except ImportError as e:
     logging.warning(e)
 try:
