diff --git a/test/test_cpp_api_parity.py b/test/test_cpp_api_parity.py
index 2193243..da8af12 100644
--- a/test/test_cpp_api_parity.py
+++ b/test/test_cpp_api_parity.py
@@ -1,6 +1,5 @@
 # Owner(s): ["module: cpp"]
 
-
 import os
 
 from cpp_api_parity import (
@@ -20,7 +19,7 @@ import torch.testing._internal.common_utils as common
 # NOTE: turn this on if you want to print source code of all C++ tests (e.g. for debugging purpose)
 PRINT_CPP_SOURCE = False
 
-devices = ["cpu", "cuda"]
+devices = ["cpu", "npu"]
 
 PARITY_TABLE_PATH = os.path.join(
     os.path.dirname(__file__), "cpp_api_parity", "parity-tracker.md"
