diff --git a/test/test_schema_check.py b/test/test_schema_check.py
index 9e1d6a6..a00cfb2 100644
--- a/test/test_schema_check.py
+++ b/test/test_schema_check.py
@@ -14,6 +14,9 @@ from torch.utils._python_dispatch import TorchDispatchMode
 from torch.testing._internal.common_methods_invocations import op_db
 from torch.testing._internal.jit_utils import JitTestCase
 from torch.testing._internal.common_device_type import ops, OpDTypes, instantiate_device_type_tests
+
+#import torch_npu
+#from torch_npu.contrib import transfer_to_npu
 pytorch_test_dir = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
 sys.path.append(pytorch_test_dir)
 
