diff --git a/test/test_serialization.py b/test/test_serialization.py
index 94632b1..74b6b60 100644
--- a/test/test_serialization.py
+++ b/test/test_serialization.py
@@ -62,6 +62,9 @@ from torch.testing._internal.common_utils import (
 from torch.testing._internal.two_tensor import TwoTensor  # noqa: F401
 from torch.utils._import_utils import import_dill
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
 if not IS_WINDOWS:
     from mmap import MAP_PRIVATE, MAP_SHARED
@@ -513,7 +516,7 @@ class SerializationMixin:
             with self.assertRaisesRegex(
                     RuntimeError,
                     f"`{compressed_indices_name}[[]..., 0[]] == 0` is not satisfied."):
-                y = torch.load(f)
+                y = torch.load(f, weights_only=False)
 
     def test_serialization_sparse_csr_invalid(self):
         self._test_serialization_sparse_compressed_invalid(
