diff --git a/test/inductor/test_codegen_triton.py b/test/inductor/test_codegen_triton.py
index a44e63e0bc7..ae54c84aa38 100644
--- a/test/inductor/test_codegen_triton.py
+++ b/test/inductor/test_codegen_triton.py
@@ -17,6 +17,7 @@ from torch._inductor.graph import GraphLowering
 from torch._inductor.test_case import TestCase as InductorTestCase
 from torch._inductor.utils import run_and_get_code
 from torch._inductor.virtualized import V
+import torch_npu._inductor
 from torch.testing._internal.inductor_utils import (
     GPU_TYPE,
     HAS_CPU,
@@ -261,5 +262,4 @@ class TestCodegenTriton(InductorTestCase):
 if __name__ == "__main__":
     from torch._inductor.test_case import run_tests
 
-    if HAS_CPU or HAS_GPU:
-        run_tests("sympy")
+    run_tests("sympy")
