diff --git a/test/inductor/test_decompose_mem_bound_mm.py b/test/inductor/test_decompose_mem_bound_mm.py
index 14775a7..9199d3f 100644
--- a/test/inductor/test_decompose_mem_bound_mm.py
+++ b/test/inductor/test_decompose_mem_bound_mm.py
@@ -1,8 +1,15 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: inductor"]
 
 import logging
 
-import torch
 import torch._inductor
 from torch._dynamo.utils import counters
 from torch._inductor.fx_passes.decompose_mem_bound_mm import check_device
@@ -16,6 +23,7 @@ from torch.testing._internal.common_utils import (
 )
 from torch.testing._internal.inductor_utils import GPU_TYPE, HAS_CUDA
 from torch.testing._internal.triton_utils import requires_gpu
+import torch_npu._inductor
 
 
 class MyModule(torch.nn.Module):
@@ -47,7 +55,6 @@ class MyModule3(torch.nn.Module):
         return output
 
 
-@requires_gpu
 @skipIfXpu(
     msg="Intel GPU has not enabled decompose_mem_bound_mm PASS in "
     "torch/_inductor/fx_passes/decompose_mem_bound_mm.py"