diff --git a/test/inductor/test_aot_inductor_package.py b/test/inductor/test_aot_inductor_package.py
index 6ae1dc02084..f9f68f6c2e4 100644
--- a/test/inductor/test_aot_inductor_package.py
+++ b/test/inductor/test_aot_inductor_package.py
@@ -1,3 +1,11 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: inductor"]
 import copy
 import functools
@@ -14,7 +22,6 @@ from pathlib import Path
 
 from parameterized import parameterized_class
 
-import torch
 import torch._inductor.config
 from torch._inductor.codecache import get_kernel_bin_format, WritableTempFile
 from torch._inductor.package import load_package, package_aoti
@@ -34,6 +41,7 @@ from torch.testing._internal.common_cuda import (
 )
 from torch.testing._internal.common_utils import IS_FBCODE, TEST_CUDA
 from torch.testing._internal.inductor_utils import GPU_TYPE, HAS_GPU
+import torch_npu._inductor
 
 
 def skipif(predicate: Callable[[str, bool], bool], reason: str):
@@ -1060,6 +1068,5 @@ class TestAOTInductorPackage(TestCase):
 
 if __name__ == "__main__":
     from torch._inductor.test_case import run_tests
-
-    if HAS_GPU or sys.platform == "darwin":
-        run_tests(needs="filelock")
+    
+    run_tests(needs="filelock")
