diff --git a/test/inductor/test_inplace_padding.py b/test/inductor/test_inplace_padding.py
index 80cb86e..019d9ee 100644
--- a/test/inductor/test_inplace_padding.py
+++ b/test/inductor/test_inplace_padding.py
@@ -1,9 +1,16 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: inductor"]
 import os
 import sys
 import unittest
 
-import torch
 from torch import nn
 from torch._dynamo.utils import same
 from torch._inductor.test_case import run_tests, TestCase
@@ -28,6 +35,7 @@ from torch._inductor import config as inductor_config
 
 
 aten = torch.ops.aten
+import torch_npu._inductor
 
 
 def num_inplace_padding():
@@ -258,5 +266,4 @@ class InplacePaddingTest(TestCase):
 
 
 if __name__ == "__main__":
-    if HAS_GPU:
-        run_tests()
+    run_tests()