diff --git a/test/inductor/test_triton_kernels.py b/test/inductor/test_triton_kernels.py
index 7f5e1faf0af..143fa747c90 100644
--- a/test/inductor/test_triton_kernels.py
+++ b/test/inductor/test_triton_kernels.py
@@ -1,3 +1,11 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: inductor"]
 # ruff: noqa: F841
 # flake8: noqa: E731
@@ -310,6 +318,7 @@ def forward(self, x_1, output_1):
             self.assertFalse(
                 torch._functionalize_are_all_mutations_hidden_from_autograd(x_func.elem)
             )
+import torch_npu._inductor
 
         # triton kernel mutation only
         with FakeTensorMode():
@@ -438,7 +447,6 @@ def forward(self, x_1, output_1):
         eager_result = f(t.clone())[0]
         self.assertEqual(eager_result, compiled_result)
 
-    @requires_gpu
     @common_utils.parametrize("dynamic", [False, True])
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     def test_triton_kernel_with_views(self, dynamic, backend):
@@ -471,7 +479,6 @@ def forward(self, x_1, output_1):
         self.assertEqual(2 * t_view, compiled_func(t).view(16))
         self.assertEqual(2 * t, compiled_func(t))
 
-    @requires_gpu
     def test_no_nan_kernels(self):
         @triton.jit
         def add_one_kernel(
@@ -527,7 +534,6 @@ def forward(self, x_1, output_1):
             self.assertEqual(output_code.count('float("nan")'), 0)
             self.assertEqual(output_code.count("float('nan')"), 0)
 
-    @requires_gpu
     @common_utils.parametrize("grad_fn", [torch.no_grad, torch.enable_grad])
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     def test_triton_kernel_with_grad_option(self, grad_fn, backend):
@@ -543,7 +549,6 @@ def forward(self, x_1, output_1):
         compiled_func = torch.compile(call_triton, backend=backend, fullgraph=True)
         self.assertEqual(2 * t, compiled_func(t))
 
-    @requires_gpu
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     def test_triton_kernel_inner_triton_function(self, backend):
         def f(x: torch.Tensor):
@@ -574,7 +579,6 @@ def forward(self, x_1, output_1):
         # TODO(oulgen): NYI - Support this
         # self.assertEqual(t * t, compiled_func(t))
 
-    @requires_gpu
     @common_utils.parametrize("grad", [False, True])
     @common_utils.parametrize("dynamic", [False, True])
     @inductor_config.patch("implicit_fallbacks", False)
@@ -629,7 +633,6 @@ def forward(self, x_1, output_1):
                 code,
             )
 
-    @requires_gpu
     def test_triton_kernel_caching(self):
         from torch._inductor.utils import run_and_get_code
 
@@ -658,7 +661,6 @@ def forward(self, x_1, output_1):
         self.assertEqual(test, 5 * torch.ones(5, device=GPU_TYPE))
         self.assertTrue("add_kernel_autotuned_1.run" not in code)
 
-    @requires_gpu
     def test_triton_kernel_caching_duplicate(self):
         from torch._inductor.utils import run_and_get_code
 
@@ -707,7 +709,6 @@ def forward(self, x_1, output_1):
         self.assertTrue(self._kernel_launched_in_code("pass_kernel_0", code))
         self.assertTrue(self._kernel_launched_in_code("pass_kernel_1", code))
 
-    @requires_gpu
     def test_triton_kernel_various_args(self):
         @triton.autotune(
             configs=[triton.Config({"BLOCK_SIZE": 128})],
@@ -743,7 +744,6 @@ def forward(self, x_1, output_1):
         # Make sure this does not crash
         call_triton(output)
 
-    @requires_gpu
     def test_triton_kernel_dependancies(self):
         def call_triton(
             x: torch.Tensor,
@@ -764,7 +764,6 @@ def forward(self, x_1, output_1):
         compiled_result = torch.compile(call_triton)(t1, t2)
         self.assertEqual(torch_result, compiled_result)
 
-    @requires_gpu
     def test_triton_kernel_reinplace_inplaceable_pass(self):
         def call_triton(
             x: torch.Tensor,
@@ -783,7 +782,6 @@ def forward(self, x_1, output_1):
         compiled_result = torch.compile(call_triton)(t1, t2)
         self.assertEqual(torch_result, compiled_result)
 
-    @requires_gpu
     @common_utils.parametrize("grad", [False, True])
     def test_triton_kernel_multi_kernel(self, grad):
         @triton.jit
@@ -912,7 +910,6 @@ def forward(self, x_1, output_1):
         # reset back
         CONSTANT_C = prev_c
 
-    @requires_gpu
     @common_utils.parametrize("grad", [False, True])
     @common_utils.parametrize("dynamic", [False, True])
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
@@ -946,7 +943,6 @@ def forward(self, x_1, output_1):
         output2 = torch.zeros_like(t1, requires_grad=grad)
         self.assertEqual(compiled_func(t1, t2, output2), torch_add)
 
-    @requires_gpu
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     @inductor_config.patch("unsafe_ignore_unsupported_triton_autotune_args", True)
     def test_triton_kernel_autotune_with_unsupported_args(self, backend):
@@ -966,7 +962,6 @@ def forward(self, x_1, output_1):
         compiled_add = compiled_func(t1, t2)
         self.assertEqual(compiled_add, torch_add)
 
-    @requires_gpu
     @common_utils.parametrize("grad", [False, True])
     @common_utils.parametrize("dynamic", [False, True])
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
@@ -1073,7 +1068,6 @@ def forward(self, x_1, output_1):
             result = test(t2, t3)
             self.assertEqual(result, torch_add)
 
-    @requires_gpu
     @common_utils.parametrize("grad", [False, True])
     @common_utils.parametrize("dynamic", [False, True])
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
@@ -1135,7 +1129,6 @@ def forward(self, x_1, output_1):
         o6 = torch.zeros_like(t1, requires_grad=grad)
         self.assertEqual(compiled_func(t1, t2, o6, 2, 200), torch_add)
 
-    @requires_gpu
     def test_triton_kernel_mutation_not_mark_dirty(self):
         @torch.compile
         def f(x):
@@ -1149,7 +1142,6 @@ def forward(self, x_1, output_1):
         f(x_cloned)
         out.sum().backward()
 
-    @requires_gpu
     @inductor_config.patch("allow_buffer_reuse", True)
     def test_triton_kernel_inputs_buffer_reuse(self):
         def _mul2(x):
@@ -1197,7 +1189,6 @@ def forward(self, x_1, output_1):
         )
         self.assertEqual(num_bufs_reused, 3)
 
-    @requires_gpu
     def test_triton_kernel_matmul_tracking(self):
         @triton.jit
         def ones_kernel(x_ptr, n_elements, BLOCK_SIZE: "tl.constexpr"):
@@ -1219,7 +1210,6 @@ def forward(self, x_1, output_1):
         python_out = torch.mm(torch.ones(4, 4, device=GPU_TYPE), x) + 10
         self.assertEqual(torch_out, python_out)
 
-    @requires_gpu
     def test_triton_kernel_strided_input(self):
         def f(inp):
             # left has strides [256, 1]
@@ -1246,7 +1236,6 @@ def forward(self, x_1, output_1):
     @inductor_config.patch(
         triton_kernel_default_layout_constraint="needs_fixed_stride_order"
     )
-    @requires_gpu
     def test_layout_constraint_needs_fixed_stride_order(self):
         # Construct a custom op whose output strides are (1, 2)
         @torch.library.custom_op("mylib::weird_op_with_lowering", mutates_args={})
@@ -1303,7 +1292,6 @@ def forward(self, x_1, output_1):
         compiled_inductor_out = compiled_inductor_f(x)
         self.assertEqual(compiled_inductor_out, eager_out)
 
-    @requires_gpu
     def test_triton_kernel_strided_input_nonzero_offset(self):
         def f(inp):
             # right has strides [256, 1] and storage offset 128
@@ -1327,7 +1315,6 @@ def forward(self, x_1, output_1):
         compiled_out = torch.compile(f)(inp)
         self.assertEqual(compiled_out, eager_out)
 
-    @requires_gpu
     def test_triton_kernel_slice_and_view_input(self):
         def f(inp):
             # left has strides [256, 1]
@@ -1355,7 +1342,6 @@ def forward(self, x_1, output_1):
         compiled_out = torch.compile(f)(inp)
         self.assertEqual(compiled_out, eager_out)
 
-    @requires_gpu
     def test_triton_kernel_fallback(self):
         def f(x, y):
             out = torch.zeros_like(x)
@@ -1386,7 +1372,6 @@ def forward(self, x_1, output_1):
         compiled_out = torch.compile(f)(x, y)
         self.assertEqual(compiled_out, eager_out)
 
-    @requires_gpu
     def test_triton_kernel_out_of_order(self):
         @triton.jit
         def add_kernel(
@@ -1417,7 +1402,6 @@ def forward(self, x_1, output_1):
         compiled_out = torch.compile(f)(x, y)
         self.assertEqual(compiled_out, eager_out)
 
-    @requires_gpu
     @dynamo_config.patch(capture_dynamic_output_shape_ops=True)
     @dynamo_config.patch(capture_scalar_outputs=True)
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
@@ -1712,7 +1696,6 @@ def forward(self, x_1, output_1):
 
         self.assertEqual(compiled_out, eager_out)
 
-    @requires_gpu
     @common_utils.parametrize("dynamic", [False, True])
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     def test_triton_kernel_triton_dtype(self, dynamic, backend):
@@ -1917,7 +1900,6 @@ def forward(self, x_1, output_1):
         self.assertEqual(eager_out, expected_out)
         self.assertEqual(compiled_out, expected_out)
 
-    @requires_gpu
     @common_utils.parametrize("dynamic", [False, True])
     @common_utils.parametrize("tma_version", ["new", "old"])
     def test_on_device_tma(self, dynamic, tma_version):
@@ -2403,7 +2385,6 @@ def forward(self, arg0_1, arg1_1):
         self.assertEqual(eager_out, expected_out)
         self.assertEqual(compiled_out, expected_out)
 
-    @requires_gpu
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     def test_triton_kernel_num_ctas(self, backend):
         @triton.jit
@@ -2460,7 +2441,6 @@ def forward(self, arg0_1, arg1_1):
         x = torch.randn(4, device=GPU_TYPE)
         f(x, x)
 
-    @requires_gpu
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     @common_utils.parametrize("autotune_at_compile_time", [True, False])
     def test_triton_kernel_restore_value(self, backend, autotune_at_compile_time):
@@ -4406,7 +4386,6 @@ class CustomOpTests(torch._inductor.test_case.TestCase):
                 self.assertEqual(f(x, other), f_compile(x, other))
                 self.assertTrue(called)
 
-    @requires_gpu
     @common_utils.parametrize("dynamic", [False, True])
     @common_utils.parametrize("autotune", [False, True])
     def test_capture_triton_special_kwargs(self, dynamic, autotune):
@@ -4714,7 +4693,6 @@ class CustomOpTests(torch._inductor.test_case.TestCase):
 
         self.assertEqual(y + increment, x)
 
-    @requires_gpu
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     def test_triton_single_autotune(self, backend):
         @triton.autotune(
@@ -4859,7 +4837,6 @@ class CustomOpTests(torch._inductor.test_case.TestCase):
             self.assertTrue(records["capture_kwargs"])
             self.assertTrue(records["capture_named_args"])
 
-    @requires_gpu
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     @common_utils.parametrize("with_perf_model", [True, False])
     def test_triton_kernel_prune_configs_by_recompile(self, backend, with_perf_model):
