diff --git a/test/inductor/test_inductor_utils.py b/test/inductor/test_inductor_utils.py
index 12468a0..f6f5748 100644
--- a/test/inductor/test_inductor_utils.py
+++ b/test/inductor/test_inductor_utils.py
@@ -1,15 +1,23 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: inductor"]
 
 import functools
 import logging
 
-import torch
 from torch._inductor.runtime.benchmarking import benchmarker
 from torch._inductor.test_case import run_tests, TestCase
 from torch._inductor.utils import do_bench_using_profiling
 
 
 log = logging.getLogger(__name__)
+import torch_npu._inductor
 
 
 class TestBench(TestCase):
