diff --git a/test/inductor/test_op_dtype_prop.py b/test/inductor/test_op_dtype_prop.py
index 28b2b3b..ff3f5ce 100644
--- a/test/inductor/test_op_dtype_prop.py
+++ b/test/inductor/test_op_dtype_prop.py
@@ -1,10 +1,17 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: inductor"]
 import importlib
 import os
 import re
 import sys
 
-import torch
 from torch._dynamo.utils import disable_cache_limit
 from torch._inductor import config
 from torch._inductor.codegen.triton import OpDtypeSupport
@@ -52,6 +59,7 @@ pointwise_ops = [
     for op in op_db
     if op.name in unique_pointwise_op_names and "reduction" not in op.variant_test_name
 ]
+import torch_npu._inductor
 
 
 class TestCase(InductorTestCase):
@@ -80,7 +88,6 @@ class TestCase(InductorTestCase):
             out_c = torch.compile(run)(op.get_op(), args, kwargs)
             self.assertEqual(out, out_c)
 
-    @requires_gpu()
     @parametrize("upcast_to_fp32", [False, True])
     @config.patch("triton.use_block_ptr", True)
     def test_codegen_upcast_to_fp32(self, upcast_to_fp32):
@@ -95,7 +102,6 @@ class TestCase(InductorTestCase):
             fp32_cast_in_code = "to(tl.float32)" in code
             self.assertEqual(fp32_cast_in_code, upcast_to_fp32)
 
-        @requires_gpu()
         @parametrize("input_shape", [(32, 32), (32, 128), (256, 32)])
         @parametrize(
             "reduction_func",
@@ -169,7 +175,6 @@ class TestCase(InductorTestCase):
             self.assertIn(torch.float32, supported_dtypes)
             self.assertIn(torch.float64, supported_dtypes)
 
-    @requires_gpu()
     @parametrize("op_name", OpDtypeSupport.supported_dtypes)
     @parametrize("load_upcast_to_fp32", [False, True])
     @parametrize("input_dtype", [torch.float16, torch.bfloat16])
@@ -289,5 +294,4 @@ instantiate_device_type_tests(TestCase, globals(), only_for=("cuda",))
 if __name__ == "__main__":
     from torch._inductor.test_case import run_tests
 
-    if HAS_GPU:
-        run_tests(needs="filelock")
+    run_tests(needs="filelock")
