diff --git a/test/inductor/test_ordered_set.py b/test/inductor/test_ordered_set.py
index b057fe3..a1618b9 100644
--- a/test/inductor/test_ordered_set.py
+++ b/test/inductor/test_ordered_set.py
@@ -1,3 +1,11 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: inductor"]
 # ruff: noqa: F841
 # flake8: noqa
@@ -15,6 +23,7 @@ from test import support
 
 from torch.testing._internal.common_utils import TestCase
 from torch.utils._ordered_set import OrderedSet
+import torch_npu._inductor
 
 
 class PassThru(Exception):
diff --git a/test/inductor/test_pad_mm.py b/test/inductor/test_pad_mm.py
index ce52376..ac193a0 100644
--- a/test/inductor/test_pad_mm.py
+++ b/test/inductor/test_pad_mm.py
@@ -1,7 +1,14 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: inductor"]
 import unittest
 
-import torch
 import torch._inductor.config as inductor_config
 from torch._dynamo.testing import rand_strided
 from torch._dynamo.utils import counters
@@ -17,6 +24,7 @@ from torch._inductor.utils import fresh_inductor_cache, is_big_gpu, run_and_get_
 from torch.testing import FileCheck
 from torch.testing._internal.common_utils import skipIfRocm
 from torch.testing._internal.inductor_utils import HAS_CUDA
+import torch_npu._inductor
 
 
 class PadMMTest(TestCase):
@@ -540,5 +548,4 @@ class PadMMTest(TestCase):
 
 
 if __name__ == "__main__":
-    if HAS_CUDA:
-        run_tests()
+    run_tests()
