diff --git a/test/inductor/test_extension_backend.py b/test/inductor/test_extension_backend.py
index 5b1e406..d29326c 100644
--- a/test/inductor/test_extension_backend.py
+++ b/test/inductor/test_extension_backend.py
@@ -1,9 +1,16 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: inductor"]
 import os
 import sys
 import unittest
 
-import torch
 import torch._dynamo
 import torch.utils.cpp_extension
 from torch._C import FileCheck
@@ -45,7 +52,7 @@ except unittest.SkipTest:
     if __name__ == "__main__":
         sys.exit(0)
     raise
-
+import torch_npu._inductor
 
 run_and_get_cpp_code = test_torchinductor.run_and_get_cpp_code
 TestCase = test_torchinductor.TestCase