diff --git a/test/inductor/test_custom_lowering.py b/test/inductor/test_custom_lowering.py
index 478b5768f51..dedad1a8493 100644
--- a/test/inductor/test_custom_lowering.py
+++ b/test/inductor/test_custom_lowering.py
@@ -1,9 +1,16 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: inductor"]
 
 from functools import partial
 from unittest import skipIf
 
-import torch
 from torch._inductor import config
 from torch._inductor.ir import Pointwise
 from torch._inductor.lowering import make_fallback, make_pointwise, register_lowering
@@ -16,7 +23,7 @@ from torch.testing._internal.inductor_utils import (
     HAS_GPU,
     requires_gpu,
 )
-
+import torch_npu._inductor
 
 # These tests check issues for lowerings that aren't in the main pytorch repo
 class TestCustomLowering(InductorTestCase):
@@ -195,7 +202,6 @@ class TestCustomLowering(InductorTestCase):
             fn(inp, offsets, max_seq_len), fn_opt(inp, offsets, max_seq_len)
         )
 
-    @requires_gpu()
     @skipIf(GPU_TYPE == "mps", "Not applicable to MPS")
     def test_jagged_to_padded_dense_zero_size(self):
         # Previously, the masking was being completely stripped for the
@@ -217,7 +223,6 @@ class TestCustomLowering(InductorTestCase):
             fn(inp, offsets, max_seq_len), fn_opt(inp, offsets, max_seq_len)
         )
 
-    @requires_gpu()
     @skipIfRocm
     @skipIfXpu(msg="`tl.inline_asm_elementwise` is not yet supported on Intel GPUs")
     @skipIf(GPU_TYPE == "mps", "Not applicable to MPS")
@@ -232,7 +237,6 @@ class TestCustomLowering(InductorTestCase):
         b = fn_opt(inp)
         self.assertEqual(a, b)
 
-    @requires_gpu()
     @skipIfXpu(msg="`tl.inline_asm_elementwise` is not yet supported on Intel GPUs")
     @skipIf(GPU_TYPE == "mps", "Not applicable to MPS")
     def test_multi_inp_asm(self):
@@ -262,5 +266,4 @@ class TestCustomLowering(InductorTestCase):
 if __name__ == "__main__":
     from torch._inductor.test_case import run_tests
 
-    if HAS_CPU or HAS_GPU:
-        run_tests(needs="filelock")
+    run_tests(needs="filelock")
