diff --git a/test/inductor/test_codegen_triton.py b/test/inductor/test_codegen_triton.py
index a44e63e0bc7..e59c30bc5f1 100644
--- a/test/inductor/test_codegen_triton.py
+++ b/test/inductor/test_codegen_triton.py
@@ -5,6 +5,8 @@ import unittest
 import sympy
 
 import torch
+from torch_npu.contrib import transfer_to_npu
+import torch_npu
 import torch._inductor.config as inductor_config
 from torch._inductor.codegen import triton_utils
 from torch._inductor.codegen.common import CSEVariable, SizeArg, TensorArg
@@ -261,5 +263,4 @@ class TestCodegenTriton(InductorTestCase):
 if __name__ == "__main__":
     from torch._inductor.test_case import run_tests
 
-    if HAS_CPU or HAS_GPU:
-        run_tests("sympy")
+    run_tests("sympy")
