diff --git a/test/inductor/test_mmdecomp.py b/test/inductor/test_mmdecomp.py
index 71c81e6..de0528e 100644
--- a/test/inductor/test_mmdecomp.py
+++ b/test/inductor/test_mmdecomp.py
@@ -1,17 +1,24 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: nn"]
 
 import math
 import unittest
 from typing import Union
 
-import torch
 from torch._inductor import config
 from torch.testing._internal.common_cuda import SM80OrLater
 from torch.testing._internal.common_device_type import instantiate_device_type_tests
 from torch.testing._internal.common_nn import NNTestCase
 from torch.testing._internal.common_utils import IS_WINDOWS, parametrize, run_tests
 from torch.testing._internal.inductor_utils import GPU_TYPE, HAS_GPU
-
+import torch_npu._inductor
 
 default_atol = {
     torch.float16: 1e-3,
