diff --git a/test/inductor/test_torchinductor_codegen_config_overrides.py b/test/inductor/test_torchinductor_codegen_config_overrides.py
index 930a74557aa..611460fd056 100644
--- a/test/inductor/test_torchinductor_codegen_config_overrides.py
+++ b/test/inductor/test_torchinductor_codegen_config_overrides.py
@@ -19,6 +19,7 @@ from torch.testing._internal.inductor_utils import (
     HAS_GPU,
     requires_gpu,
 )
+import torch_npu._inductor
 
 
 importlib.import_module("filelock")
@@ -91,7 +92,6 @@ class CodegenInductorTest(InductorTestCase):
         else:
             self.count_code(reinterpret_call, code, 2)
 
-    @requires_gpu()
     @skipIf(GPU_TYPE == "mps", "Triton is not available for MPS")
     def test_cse_make_block_ptr_reduction(self):
         def func(a, b):
@@ -177,5 +177,4 @@ class CodegenInductorTest(InductorTestCase):
 if __name__ == "__main__":
     from torch._inductor.test_case import run_tests
 
-    if HAS_GPU or HAS_CPU:
-        run_tests(needs="filelock")
+    run_tests(needs="filelock")
