diff --git a/test/inductor/test_triton_heuristics.py b/test/inductor/test_triton_heuristics.py
index 8b1a56e..1ed720d 100644
--- a/test/inductor/test_triton_heuristics.py
+++ b/test/inductor/test_triton_heuristics.py
@@ -1,9 +1,16 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: inductor"]
 
 import sys
 import unittest
 
-import torch
 from torch._dynamo.testing import rand_strided
 from torch._inductor.utils import clone_preserve_strides
 from torch.testing._internal.common_utils import IS_LINUX, skipIfXpu
@@ -37,10 +44,12 @@ from torch._inductor.runtime.triton_heuristics import (
     triton_config,
 )
 from torch._inductor.test_case import run_tests, TestCase
+import torch_npu._inductor
 
+NPU_TYPE = "npu"
 
 class TestTritonHeuristics(TestCase):
-    device_type = GPU_TYPE
+    device_type = NPU_TYPE
 
     def test_triton_config(self):
         """
@@ -74,9 +83,9 @@ class TestTritonHeuristics(TestCase):
         s1 = 8
 
         args = [
-            torch.rand([2, 4], device=GPU_TYPE),
-            torch.rand([2], device=GPU_TYPE),
-            torch.rand([s0, s1], device=GPU_TYPE),
+            torch.rand([2, 4], device=NPU_TYPE),
+            torch.rand([2], device=NPU_TYPE),
+            torch.rand([s0, s1], device=NPU_TYPE),
         ]
         torch._dynamo.mark_dynamic(args[-1], 0)
         foo_c = torch.compile(forward)
@@ -84,9 +93,9 @@ class TestTritonHeuristics(TestCase):
         self.assertEqual(forward(*args), foo_c(*args))
 
         args = [
-            torch.rand([2, 4], device=GPU_TYPE),
-            torch.rand([2], device=GPU_TYPE),
-            torch.rand([s0, s1], device=GPU_TYPE),
+            torch.rand([2, 4], device=NPU_TYPE),
+            torch.rand([2], device=NPU_TYPE),
+            torch.rand([s0, s1], device=NPU_TYPE),
         ]
         self.assertEqual(forward(*args), foo_c(*args))
 
@@ -114,7 +123,7 @@ class TestTritonHeuristics(TestCase):
 
         triton_meta = {
             "signature": {"in_ptr0": "*fp32", "out_ptr0": "*fp32", "xnumel": "i32"},
-            "device": DeviceProperties.create(torch.device("cuda")),
+            "device": DeviceProperties.create(torch.device(NPU_TYPE)),
             "constants": {},
             "configs": [
                 AttrsDescriptorWrapper(divisible_by_16=(0, 1, 2), equal_to_1=())
@@ -156,7 +165,7 @@ class TestTritonHeuristics(TestCase):
             CachingAutotuner(**args)
 
     def test_autotune_hints_to_configs(self):
-        device_props = DeviceProperties.create(torch.device(GPU_TYPE))
+        device_props = DeviceProperties.create(torch.device(NPU_TYPE))
         device_props = device_props._replace(warp_size=8)
 
         hints = {AutotuneHint.ONE_ELEMENT_PER_THREAD}
@@ -205,7 +214,7 @@ class TestArgumentCloneAndRestore(TestCase):
         """
         M = 2
         N = 2**29 // 4
-        out = rand_strided((M, N), (N + pad, 1), device=GPU_TYPE)
+        out = rand_strided((M, N), (N + pad, 1), device=NPU_TYPE)
         if with_offset:
             out = out[:, 1:]
         return out
@@ -268,5 +277,4 @@ class TestArgumentCloneAndRestore(TestCase):
 
 
 if __name__ == "__main__":
-    if IS_LINUX and HAS_GPU:
-        run_tests()
+    run_tests()
