diff --git a/test/inductor/test_cudagraph_trees.py b/test/inductor/test_cudagraph_trees.py
index 8c040da1188..b3a7ba5ccc7 100644
--- a/test/inductor/test_cudagraph_trees.py
+++ b/test/inductor/test_cudagraph_trees.py
@@ -12,7 +12,6 @@ import warnings
 from collections import defaultdict
 from collections.abc import Mapping, Sequence
 
-import torch
 import torch._dynamo.config as dynamo_config
 import torch.nn as nn
 from torch._dynamo.backends.debugging import aot_eager_decomp_partition_with_mode
@@ -51,12 +50,10 @@ if IS_WINDOWS and IS_CI:
     sys.stderr.write(
         "Windows CI does not have necessary dependencies for test_torchinductor yet\n"
     )
-    if __name__ == "__main__":
-        sys.exit(0)
     raise unittest.SkipTest("requires sympy/functorch/filelock")
-
 importlib.import_module("functorch")
 importlib.import_module("filelock")
+    run_tests(needs="filelock")
 
 
 aten = torch.ops.aten
@@ -64,6 +61,7 @@ requires_multigpu = functools.partial(
     unittest.skipIf, not TEST_MULTIGPU, "requires multiple cuda devices"
 )
 from io import StringIO
+import torch_npu._inductor
 
 from torch._library.opaque_object import OpaqueBase, register_opaque_type
 
