diff --git a/test/inductor/test_select_algorithm.py b/test/inductor/test_select_algorithm.py
index 2d2443e..5225c22 100644
--- a/test/inductor/test_select_algorithm.py
+++ b/test/inductor/test_select_algorithm.py
@@ -1,8 +1,15 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: inductor"]
 import functools
 from unittest.mock import patch
 
-import torch
 import torch._dynamo.config as dynamo_config
 import torch._inductor.config as inductor_config
 import torch._inductor.select_algorithm as select_algorithm
@@ -17,6 +24,7 @@ from torch.testing._internal.inductor_utils import GPU_TYPE, HAS_GPU
 
 
 aten = torch.ops.aten
+import torch_npu._inductor
 
 
 def patches(fn):
@@ -368,5 +376,4 @@ class TestSelectAlgorithm(TestCase):
 
 
 if __name__ == "__main__":
-    if IS_LINUX and HAS_GPU and is_big_gpu():
-        run_tests()
+    run_tests()