diff --git a/test/inductor/test_auto_functionalize.py b/test/inductor/test_auto_functionalize.py
index 7033d36..fd000d5 100644
--- a/test/inductor/test_auto_functionalize.py
+++ b/test/inductor/test_auto_functionalize.py
@@ -1,10 +1,17 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: functionalization"]
 
 import unittest
 
 import numpy as np
 
-import torch
 import torch._dynamo.testing
 import torch._inductor.config as inductor_config
 import torch._inductor.test_case
@@ -14,6 +21,7 @@ from torch import Tensor
 from torch._dynamo.testing import CompileCounterWithBackend
 from torch._higher_order_ops.auto_functionalize import try_use_slice
 from torch.testing._internal.logging_utils import logs_to_string
+import torch_npu._inductor
 
 
 class AutoFunctionalizeTests(torch._inductor.test_case.TestCase):
 