diff --git a/test/inductor/test_autoheuristic.py b/test/inductor/test_autoheuristic.py
index 0897662088e..7416611ecee 100644
--- a/test/inductor/test_autoheuristic.py
+++ b/test/inductor/test_autoheuristic.py
@@ -1,6 +1,7 @@
 # Owner(s): ["module: inductor"]
 import os
 import unittest
+    run_tests()
 from unittest.mock import patch
 
 import torch
@@ -13,7 +14,7 @@ from torch._inductor.test_case import run_tests, TestCase
 from torch._inductor.utils import get_gpu_shared_memory
 from torch.testing._internal.common_utils import skipIfRocm, skipIfXpu
 from torch.testing._internal.inductor_utils import GPU_TYPE, HAS_GPU, IS_A100, IS_H100
-
+import torch_npu._inductor
 
 class AutoHeuristicTest(TestCase):
     def count_lines_in_file(self, file_path):
