diff --git a/test/inductor/test_compiled_optimizers.py b/test/inductor/test_compiled_optimizers.py
index 7124296..505fd82 100644
--- a/test/inductor/test_compiled_optimizers.py
+++ b/test/inductor/test_compiled_optimizers.py
@@ -1,3 +1,11 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: inductor"]
 
 import sys
@@ -7,7 +15,6 @@ from contextlib import ExitStack
 from copy import deepcopy
 from typing import NamedTuple
 
-import torch
 import torch._inductor
 import torch._inductor.cudagraph_trees
 import torch.optim.lr_scheduler
@@ -62,6 +69,7 @@ from torch.testing._internal.inductor_utils import (
     has_triton,
 )
 from torch.testing._internal.triton_utils import requires_cuda, requires_gpu
+import torch_npu._inductor
 
 
 def get_inputs(optim):
@@ -517,7 +525,6 @@ def make_test(
 
 def make_recompile_test(optim_cls, closure=None, kernel_count=2, **kwargs):
     @config.patch("score_fusion_memory_threshold", 1)
-    @requires_gpu
     def test_fn(self):
         torch._dynamo.reset()
         torch._inductor.metrics.reset()
@@ -714,7 +721,6 @@ class CompiledOptimizerTests(TestCase):
         SGD, kernel_count=1, lr=0.01, foreach=True
     )
 
-    @requires_gpu
     def test_static_address_finalizer(self):
         import gc
 
@@ -777,7 +783,6 @@ class CompiledOptimizerTests(TestCase):
         self.assertEqual(actual_steps, expected_steps)
 
     # Basic shampoo test to verify we support compiling the various ops without error
-    @requires_gpu
     def test_basic_shampoo(self):
         param_buf = torch.rand((1024, 128))
         param_buf_c = param_buf.detach().clone()
@@ -846,7 +851,6 @@ class CompiledOptimizerTests(TestCase):
 
         self.assertEqual(compiled_fn(params_c), shampoo_functional_basic(params))
 
-    @requires_gpu
     def test_closure_graph_break(self):
         param = torch.rand(
             2, 3, dtype=torch.float32, device=GPU_TYPE, requires_grad=True
@@ -889,7 +893,6 @@ class CompiledOptimizerTests(TestCase):
 
     # compile a large foreach op and verify
     # that the time taken is within an expected range
-    @requires_gpu
     def test_compile_time_smoketest(self):
         import time
 
@@ -980,5 +983,4 @@ instantiate_device_type_tests(
 if __name__ == "__main__":
     from torch._inductor.test_case import run_tests
 
-    if HAS_CPU or HAS_GPU:
-        run_tests(needs="filelock")
+    run_tests(needs="filelock")
