diff --git a/test/inductor/test_inductor_utils.py b/test/inductor/test_inductor_utils.py
index 2871a579fe5..af0f130b65f 100644
--- a/test/inductor/test_inductor_utils.py
+++ b/test/inductor/test_inductor_utils.py
@@ -4,6 +4,8 @@ import functools
 import logging
 
 import torch
+from torch_npu.contrib import transfer_to_npu
+import torch_npu
 from torch._inductor.runtime.benchmarking import benchmarker
 from torch._inductor.test_case import run_tests, TestCase
 from torch._inductor.utils import do_bench_using_profiling
