diff --git a/test/inductor/test_snode_runtime.py b/test/inductor/test_snode_runtime.py
index e002a61..b0d1d53 100644
--- a/test/inductor/test_snode_runtime.py
+++ b/test/inductor/test_snode_runtime.py
@@ -1,9 +1,16 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: inductor"]
 
 import contextlib
 from unittest import skipIf
 
-import torch
 import torch.distributed as dist
 from torch._inductor import config, metrics
 from torch._inductor.comm_analysis import estimate_nccl_collective_runtime
@@ -17,6 +24,7 @@ from torch.testing._internal.inductor_utils import GPU_TYPE, HAS_GPU
 aten = torch.ops.aten
 c10d = torch.ops.c10d_functional
 _c10d = torch.ops._c10d_functional
+import torch_npu._inductor
 
 
 def compile_but_use_eager(gm, example_inputs):
@@ -371,5 +379,4 @@ class TestCommAnalysis(TestCase):
 if __name__ == "__main__":
     from torch._inductor.test_case import run_tests
 
-    if HAS_GPU:
-        run_tests(needs="filelock")
+    run_tests(needs="filelock")