diff --git a/test/inductor/test_cpu_select_algorithm.py b/test/inductor/test_cpu_select_algorithm.py
index b9912f3..dbcc7a1 100644
--- a/test/inductor/test_cpu_select_algorithm.py
+++ b/test/inductor/test_cpu_select_algorithm.py
@@ -1,3 +1,11 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["oncall: cpu inductor"]
 import contextlib
 import functools
@@ -6,7 +14,6 @@ import unittest
 from typing import Optional
 from unittest.mock import patch
 
-import torch
 import torch._dynamo.config
 import torch._dynamo.config as dynamo_config
 import torch._inductor.config as inductor_config
@@ -47,6 +54,7 @@ check_model = test_torchinductor.check_model
 set_num_threads = test_cpu_repro.set_num_threads
 
 aten = torch.ops.aten
+import torch_npu._inductor
 
 
 def patches(fn):
