diff --git a/test/inductor/test_cpu_select_algorithm.py b/test/inductor/test_cpu_select_algorithm.py
index befadb7a331..bab2e845f24 100644
--- a/test/inductor/test_cpu_select_algorithm.py
+++ b/test/inductor/test_cpu_select_algorithm.py
@@ -52,6 +52,7 @@ set_num_threads = test_cpu_repro.set_num_threads
 run_and_get_cpp_code = test_torchinductor.run_and_get_cpp_code
 
 aten = torch.ops.aten
+import torch_npu._inductor
 
 
 def patches(fn):
