diff --git a/test/inductor/test_cuda_repro.py b/test/inductor/test_cuda_repro.py
index 574ee73bae3..a1d80f4e8df 100644
--- a/test/inductor/test_cuda_repro.py
+++ b/test/inductor/test_cuda_repro.py
@@ -9,7 +9,6 @@ import os
 import sys
 import unittest
 
-import torch
 import torch._dynamo.config as dynamo_config
 import torch.backends.cuda
 import torch.nn.functional as F
@@ -2917,5 +2916,5 @@ if __name__ == "__main__":
     from torch._inductor.test_case import run_tests
     from torch.testing._internal.inductor_utils import HAS_GPU_AND_TRITON
 
+    run_tests(needs="filelock")
     if HAS_GPU_AND_TRITON and not TEST_WITH_ASAN:
-        run_tests(needs="filelock")
