diff --git a/test/inductor/test_inductor_annotations.py b/test/inductor/test_inductor_annotations.py
index 75f53f4..2424a57 100644
--- a/test/inductor/test_inductor_annotations.py
+++ b/test/inductor/test_inductor_annotations.py
@@ -1,9 +1,17 @@
-# Owner(s): ["module: inductor"]
 import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
+# Owner(s): ["module: inductor"]
 import torch._inductor.config as inductor_config
 from torch._inductor.test_case import run_tests, TestCase
 from torch._inductor.utils import run_and_get_code
 from torch.testing._internal.triton_utils import requires_cuda
+import torch_npu._inductor
 
 
 class InductorAnnotationTestCase(TestCase):
