diff --git a/test/inductor/test_triton_kernels.py b/test/inductor/test_triton_kernels.py
index a5f360b..ae2ded8 100644
--- a/test/inductor/test_triton_kernels.py
+++ b/test/inductor/test_triton_kernels.py
@@ -1,3 +1,11 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: inductor"]
 # ruff: noqa: F841
 # flake8: noqa: E731
@@ -5,7 +13,6 @@
 import functools
 import logging
 
-import torch
 import torch._dynamo.testing
 import torch._inductor.test_case
 import torch.utils._pytree as pytree
@@ -61,6 +68,7 @@ if HAS_GPU:
                 fast_dividef,
                 fast_dividef as my_fast_dividef,
             )
+import torch_npu._inductor
 
     def _triton_get_ast_equal_to_str(params):
         try:
@@ -83,7 +91,6 @@ class KernelTests(torch._inductor.test_case.TestCase):
             return f"launchKernel({kernel_name}" in code
         return f"{kernel_name}.run(" in code
 
-    @requires_gpu
     def test_triton_kernel_with_kernel_param(self):
         @triton.jit
         def pass_kernel(kernel):
@@ -99,7 +106,6 @@ class KernelTests(torch._inductor.test_case.TestCase):
         # No need to assert anything, the goal is to make sure dynamo does
         # not crash
 
-    @requires_gpu
     def test_triton_kernel_higher_order_func(self):
         from torch._higher_order_ops.triton_kernel_wrap import kernel_side_table
 
@@ -150,7 +156,6 @@ class KernelTests(torch._inductor.test_case.TestCase):
         # Make sure it is NOT modified
         self.assertEqual(output, torch.zeros_like(t1))
 
-    @requires_gpu
     def test_triton_kernel_functionalize(self):
         from functorch import make_fx
         from torch._higher_order_ops.triton_kernel_wrap import kernel_side_table
@@ -204,7 +209,6 @@ def forward(self, x_1, output_1):
     return getitem_1""",
         )
 
-    @requires_gpu
     def test_triton_kernel_mutation_type(self):
         from torch._higher_order_ops.triton_kernel_wrap import kernel_side_table
         from torch._subclasses.fake_tensor import FakeTensorMode
@@ -274,7 +278,6 @@ def forward(self, x_1, output_1):
                 torch._functionalize_are_all_mutations_hidden_from_autograd(x_func.elem)
             )
 
-    @requires_gpu
     @common_utils.parametrize("dynamic", [False, True])
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     def test_triton_kernel_with_views(self, dynamic, backend):
@@ -307,7 +310,6 @@ def forward(self, x_1, output_1):
         self.assertEqual(2 * t_view, compiled_func(t).view(16))
         self.assertEqual(2 * t, compiled_func(t))
 
-    @requires_gpu
     def test_no_nan_kernels(self):
         @triton.jit
         def add_one_kernel(
@@ -363,7 +365,6 @@ def forward(self, x_1, output_1):
             self.assertEqual(output_code.count('float("nan")'), 0)
             self.assertEqual(output_code.count("float('nan')"), 0)
 
-    @requires_gpu
     @common_utils.parametrize("grad_fn", [torch.no_grad, torch.enable_grad])
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     def test_triton_kernel_with_grad_option(self, grad_fn, backend):
@@ -379,7 +380,6 @@ def forward(self, x_1, output_1):
         compiled_func = torch.compile(call_triton, backend=backend, fullgraph=True)
         self.assertEqual(2 * t, compiled_func(t))
 
-    @requires_gpu
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     def test_triton_kernel_inner_triton_function(self, backend):
         def f(x: torch.Tensor):
@@ -410,7 +410,6 @@ def forward(self, x_1, output_1):
         # TODO(oulgen): NYI - Support this
         # self.assertEqual(t * t, compiled_func(t))
 
-    @requires_gpu
     @common_utils.parametrize("grad", [False, True])
     @common_utils.parametrize("dynamic", [False, True])
     @inductor_config.patch("implicit_fallbacks", False)
@@ -465,7 +464,6 @@ def forward(self, x_1, output_1):
                 code,
             )
 
-    @requires_gpu
     def test_triton_kernel_caching(self):
         from torch._inductor.utils import run_and_get_code
 
@@ -494,7 +492,6 @@ def forward(self, x_1, output_1):
         self.assertEqual(test, 5 * torch.ones(5, device=GPU_TYPE))
         self.assertTrue("add_kernel_autotuned_1.run" not in code)
 
-    @requires_gpu
     def test_triton_kernel_caching_duplicate(self):
         from torch._inductor.utils import run_and_get_code
 
@@ -543,7 +540,6 @@ def forward(self, x_1, output_1):
         self.assertTrue(self._kernel_launched_in_code("pass_kernel_0", code))
         self.assertTrue(self._kernel_launched_in_code("pass_kernel_1", code))
 
-    @requires_gpu
     def test_triton_kernel_various_args(self):
         @triton.autotune(
             configs=[triton.Config({"BLOCK_SIZE": 128})],
@@ -579,7 +575,6 @@ def forward(self, x_1, output_1):
         # Make sure this does not crash
         call_triton(output)
 
-    @requires_gpu
     def test_triton_kernel_dependancies(self):
         def call_triton(
             x: torch.Tensor,
@@ -600,7 +595,6 @@ def forward(self, x_1, output_1):
         compiled_result = torch.compile(call_triton)(t1, t2)
         self.assertEqual(torch_result, compiled_result)
 
-    @requires_gpu
     def test_triton_kernel_reinplace_inplaceable_pass(self):
         def call_triton(
             x: torch.Tensor,
@@ -619,7 +613,6 @@ def forward(self, x_1, output_1):
         compiled_result = torch.compile(call_triton)(t1, t2)
         self.assertEqual(torch_result, compiled_result)
 
-    @requires_gpu
     @common_utils.parametrize("grad", [False, True])
     def test_triton_kernel_multi_kernel(self, grad):
         @triton.jit
@@ -696,7 +689,6 @@ def forward(self, x_1, output_1):
         self.assertEqual(float_result, result)
         self.assertEqual(int_result, resulti)
 
-    @requires_gpu
     @skipIfXpu
     def test_triton_kernel_constants(self):
         @triton.jit
@@ -747,7 +739,6 @@ def forward(self, x_1, output_1):
         # reset back
         CONSTANT_C = prev_c
 
-    @requires_gpu
     @common_utils.parametrize("grad", [False, True])
     @common_utils.parametrize("dynamic", [False, True])
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
@@ -781,7 +772,6 @@ def forward(self, x_1, output_1):
         output2 = torch.zeros_like(t1, requires_grad=grad)
         self.assertEqual(compiled_func(t1, t2, output2), torch_add)
 
-    @requires_gpu
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     @inductor_config.patch("unsafe_ignore_unsupported_triton_autotune_args", True)
     def test_triton_kernel_autotune_with_unsupported_args(self, backend):
@@ -801,7 +791,6 @@ def forward(self, x_1, output_1):
         compiled_add = compiled_func(t1, t2)
         self.assertEqual(compiled_add, torch_add)
 
-    @requires_gpu
     @common_utils.parametrize("grad", [False, True])
     @common_utils.parametrize("dynamic", [False, True])
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
@@ -841,7 +830,6 @@ def forward(self, x_1, output_1):
         output2 = torch.zeros_like(t1, requires_grad=grad)
         self.assertEqual(compiled_func(t1, t2, output2), torch_result)
 
-    @requires_gpu
     @common_utils.parametrize("dynamic", [False, True])
     def test_triton_kernel_tracing(self, dynamic):
         def call_triton_add(
@@ -905,7 +893,6 @@ def forward(self, x_1, output_1):
             result = test(t2, t3)
             self.assertEqual(result, torch_add)
 
-    @requires_gpu
     @common_utils.parametrize("grad", [False, True])
     @common_utils.parametrize("dynamic", [False, True])
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
@@ -967,7 +954,6 @@ def forward(self, x_1, output_1):
         o6 = torch.zeros_like(t1, requires_grad=grad)
         self.assertEqual(compiled_func(t1, t2, o6, 2, 200), torch_add)
 
-    @requires_gpu
     def test_triton_kernel_mutation_not_mark_dirty(self):
         @torch.compile
         def f(x):
@@ -981,7 +967,6 @@ def forward(self, x_1, output_1):
         f(x_cloned)
         out.sum().backward()
 
-    @requires_gpu
     @inductor_config.patch("allow_buffer_reuse", True)
     def test_triton_kernel_inputs_buffer_reuse(self):
         def _mul2(x):
@@ -1022,7 +1007,6 @@ def forward(self, x_1, output_1):
         )
         self.assertEqual(num_bufs_reused, 3)
 
-    @requires_gpu
     def test_triton_kernel_matmul_tracking(self):
         @triton.jit
         def ones_kernel(x_ptr, n_elements, BLOCK_SIZE: "tl.constexpr"):
@@ -1044,7 +1028,6 @@ def forward(self, x_1, output_1):
         python_out = torch.mm(torch.ones(4, 4, device=GPU_TYPE), x) + 10
         self.assertEqual(torch_out, python_out)
 
-    @requires_gpu
     def test_triton_kernel_strided_input(self):
         def f(inp):
             # left has strides [256, 1]
@@ -1071,7 +1054,6 @@ def forward(self, x_1, output_1):
     @inductor_config.patch(
         triton_kernel_default_layout_constraint="needs_fixed_stride_order"
     )
-    @requires_gpu
     def test_layout_constraint_needs_fixed_stride_order(self):
         # Construct a custom op whose output strides are (1, 2)
         @torch.library.custom_op("mylib::weird_op_with_lowering", mutates_args={})
@@ -1128,7 +1110,6 @@ def forward(self, x_1, output_1):
         compiled_inductor_out = compiled_inductor_f(x)
         self.assertEqual(compiled_inductor_out, eager_out)
 
-    @requires_gpu
     def test_triton_kernel_strided_input_nonzero_offset(self):
         def f(inp):
             # right has strides [256, 1] and storage offset 128
@@ -1152,7 +1133,6 @@ def forward(self, x_1, output_1):
         compiled_out = torch.compile(f)(inp)
         self.assertEqual(compiled_out, eager_out)
 
-    @requires_gpu
     def test_triton_kernel_slice_and_view_input(self):
         def f(inp):
             # left has strides [256, 1]
@@ -1180,7 +1160,6 @@ def forward(self, x_1, output_1):
         compiled_out = torch.compile(f)(inp)
         self.assertEqual(compiled_out, eager_out)
 
-    @requires_gpu
     def test_triton_kernel_fallback(self):
         def f(x, y):
             out = torch.zeros_like(x)
@@ -1197,7 +1176,6 @@ def forward(self, x_1, output_1):
         compiled_out = torch.compile(f)(x, y)
         self.assertEqual(compiled_out, eager_out)
 
-    @requires_gpu
     def test_triton_kernel_out_of_order(self):
         @triton.jit
         def add_kernel(
@@ -1228,7 +1206,6 @@ def forward(self, x_1, output_1):
         compiled_out = torch.compile(f)(x, y)
         self.assertEqual(compiled_out, eager_out)
 
-    @requires_gpu
     @dynamo_config.patch(capture_dynamic_output_shape_ops=True)
     @dynamo_config.patch(capture_scalar_outputs=True)
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
@@ -1261,7 +1238,6 @@ def forward(self, x_1, output_1):
         compiled_out = torch.compile(f, fullgraph=True, backend=backend)(x)
         self.assertEqual(compiled_out, eager_out)
 
-    @requires_gpu
     @common_utils.parametrize("dynamic", [False, True])
     def test_triton_kernel_equal_to_1_arg(self, dynamic):
         @triton.jit
@@ -1310,7 +1286,6 @@ def forward(self, x_1, output_1):
                 self.assertTrue(_triton_get_ast_equal_to_str((3,)) in sources[0])
         self.assertEqual(compiled_out, eager_out)
 
-    @requires_gpu
     @common_utils.parametrize("dynamic", [False, True])
     def test_triton_kernel_equal_to_1_float_arg(self, dynamic):
         def f(x, y):
@@ -1340,7 +1315,6 @@ def forward(self, x_1, output_1):
             self.assertTrue(_triton_get_ast_equal_to_str(()) in sources[0])
         self.assertEqual(compiled_out, eager_out)
 
-    @requires_gpu
     @skipIfRocm
     def test_triton_kernel_with_imported_symbol(self):
         @triton.jit
@@ -1372,7 +1346,6 @@ def forward(self, x_1, output_1):
 
         self.assertEqual(compiled_out, eager_out)
 
-    @requires_gpu
     @skipIfRocm
     def test_triton_kernel_with_imported_symbol_with_custom_name(self):
         @triton.jit
@@ -1404,7 +1377,6 @@ def forward(self, x_1, output_1):
 
         self.assertEqual(compiled_out, eager_out)
 
-    @requires_gpu
     @common_utils.parametrize("size", [4, 16])
     @common_utils.parametrize("dynamic", [False, True])
     def test_triton_kernel_different_shapes(self, size, dynamic):
@@ -1445,7 +1417,6 @@ def forward(self, x_1, output_1):
 
         self.assertEqual(compiled_out, eager_out)
 
-    @requires_gpu
     @common_utils.parametrize("dynamic", [False, True])
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     def test_triton_kernel_triton_dtype(self, dynamic, backend):
@@ -1489,7 +1460,6 @@ def forward(self, x_1, output_1):
             )(*args)
             self.assertEqual(compiled_out, eager_out)
 
-    @requires_gpu
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     def test_triton_kernel_special_kwargs_with_autotune(self, backend):
         @triton.autotune(
@@ -1534,7 +1504,6 @@ def forward(self, x_1, output_1):
         x = torch.randn(4, device=GPU_TYPE)
         f(x, x)
 
-    @requires_gpu
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     def test_triton_kernel_empty_autotune_config_dict(self, backend):
         @triton.autotune(
@@ -1578,7 +1547,6 @@ def forward(self, x_1, output_1):
         x = torch.randn(4, device=GPU_TYPE)
         f(x, x)
 
-    @requires_gpu
     @common_utils.parametrize("autotune", [False, True])
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     def test_triton_kernel_special_params(self, autotune, backend):
@@ -1645,7 +1613,6 @@ def forward(self, x_1, output_1):
         self.assertEqual(eager_out, expected_out)
         self.assertEqual(compiled_out, expected_out)
 
-    @requires_gpu
     @common_utils.parametrize("dynamic", [False, True])
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     def test_triton_kernel_multiple_outputs(self, dynamic, backend):
@@ -1686,7 +1653,6 @@ def forward(self, x_1, output_1):
         self.assertEqual(out2, x + y + 1)
         self.assertEqual(out3, z**2)
 
-    @requires_gpu
     @unittest.skipIf(not has_triton_tma(), "requires Triton TMA support")
     @common_utils.parametrize("dynamic", [False, True])
     def test_tma_capture_and_functionalize(self, dynamic):
@@ -1754,7 +1720,6 @@ def forward(self, arg0_1, arg1_1):
     return (getitem,)""",
             )
 
-    @requires_gpu
     @unittest.skipIf(not has_triton_tma(), "requires Triton TMA support")
     @common_utils.parametrize("after_data_ptr", [False, True])
     @common_utils.parametrize("after_create_desc", [False, True])
@@ -1803,7 +1768,6 @@ def forward(self, arg0_1, arg1_1):
         self.assertEqual(eager_out, expected_out)
         self.assertEqual(compiled_out, expected_out)
 
-    @requires_gpu
     @unittest.skipIf(not has_triton_tma(), "requires Triton TMA support")
     @common_utils.parametrize("dynamic", [False, True])
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
@@ -1848,7 +1812,6 @@ def forward(self, arg0_1, arg1_1):
         self.assertEqual(eager_out, expected_out)
         self.assertEqual(compiled_out, expected_out)
 
-    @requires_gpu
     @unittest.skipIf(not has_triton_tma(), "requires Triton TMA support")
     def test_tma_descriptor_dedup(self):
         def f(a):
@@ -1896,7 +1859,6 @@ def forward(self, arg0_1, arg1_1):
         # 2 calls: one for two inputs (dedupped), one for the output
         self.assertEqual(code.count("create_1d_tma_descriptor("), 2)
 
-    @requires_gpu
     @unittest.skipIf(not has_triton_tma(), "requires Triton TMA support")
     @common_utils.parametrize("dynamic", [False, True])
     @common_utils.parametrize("backend", ["eager", "aot_eager"])
@@ -1948,7 +1910,6 @@ def forward(self, arg0_1, arg1_1):
         self.assertEqual(eager_out, expected_out)
         self.assertEqual(compiled_out, expected_out)
 
-    @requires_gpu
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     def test_triton_kernel_num_ctas(self, backend):
         @triton.jit
@@ -1966,7 +1927,6 @@ def forward(self, arg0_1, arg1_1):
             x = torch.randn(4, device=GPU_TYPE)
             f(x)
 
-    @requires_gpu
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     def test_triton_kernel_special_kwargs_without_autotune(self, backend):
         @triton.jit
@@ -2005,7 +1965,6 @@ def forward(self, arg0_1, arg1_1):
         x = torch.randn(4, device=GPU_TYPE)
         f(x, x)
 
-    @requires_gpu
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     @common_utils.parametrize("autotune_at_compile_time", [True, False])
     def test_triton_kernel_restore_value(self, backend, autotune_at_compile_time):
@@ -2052,7 +2011,6 @@ def forward(self, arg0_1, arg1_1):
         # make sure x was restored after autotuning
         torch.testing.assert_close(x, prev + 1)
 
-    @requires_gpu
     @parametrize("dtype", (torch.float16, torch.float32, torch.float64))
     def test_triton_kernel_float64_constant(self, dtype):
         def f(x):
@@ -2110,7 +2068,6 @@ def forward(self, arg0_1, arg1_1):
             self.assertEqual(out_e[1], out_c[1])
 
     # TODO enable this test case on XPU.
-    @requires_gpu
     def test_i64_input(self):
         # The i64 "seed" input needs to be marked as "i64", not "i32".
         @triton.jit
@@ -2150,7 +2107,6 @@ def forward(self, arg0_1, arg1_1):
 
         self.assertTrue(((res < 2) & (res >= 0)).all().item())
 
-    @requires_gpu
     @parametrize("wrapped", [False, True])
     @parametrize("autotune", [False, True])
     def test_constexpr_dynamic_shapes(self, wrapped, autotune):
@@ -2220,7 +2176,6 @@ def forward(self, arg0_1, arg1_1):
         res2 = fn_c(x2)
         self.assertEqual(x2 * x2, res2)
 
-    @requires_gpu
     def test_triton_kernel_none_args(self):
         # https://github.com/pytorch/pytorch/issues/115344
         @triton.autotune(
@@ -2265,7 +2220,6 @@ def forward(self, arg0_1, arg1_1):
         sin_triton_compiled(None, out_compiled)
         self.assertEqual(out, out_compiled)
 
-    @requires_gpu
     def test_triton_kernel_global_constexpr(self):
         @triton.jit
         def triton_(in_ptr, out_ptr, BLOCK_SIZE: tl.constexpr):
@@ -2292,7 +2246,6 @@ def forward(self, arg0_1, arg1_1):
         actual = torch.compile(fn)(x)
         self.assertEqual(expected, actual)
 
-    @requires_gpu
     @unittest.skipIf(
         not triton_version_uses_attrs_dict(),
         "Test is only valid for new triton versions where attrs is represented by a raw dict",
@@ -2343,7 +2296,6 @@ def forward(self, arg0_1, arg1_1):
 
 
 def make_mutation_test(fn):
-    @requires_gpu
     def test_fn(self):
         from torch._higher_order_ops.triton_kernel_wrap import identify_mutated_tensors
 
@@ -2517,7 +2469,6 @@ class MutationTests(torch._inductor.test_case.TestCase):
             expected,
         )
 
-    @requires_gpu
     def test_triton_kernel_inference_mode(self):
         def f(x, y, out):
             n_elements = x.numel()
@@ -3224,7 +3175,6 @@ if HAS_GPU:
 class CustomOpTests(torch._inductor.test_case.TestCase):
     """Tests for custom ops wrapping triton kernels"""
 
-    @requires_gpu
     @common_utils.parametrize("autotuned", [False, True])
     @common_utils.parametrize("dynamic", [False, True])
     def test_add_kernel(self, autotuned, dynamic):
@@ -3265,7 +3215,6 @@ class CustomOpTests(torch._inductor.test_case.TestCase):
         self.assertNotIn(libname, code)
         self.assertNotIn(opname, code)
 
-    @requires_gpu
     @dynamo_config.patch("recompile_limit", 1)
     def test_triton_dynamic_grid_no_recompile(self):
         libname = "my_cool_namespace"
@@ -3328,7 +3277,6 @@ class CustomOpTests(torch._inductor.test_case.TestCase):
         expected = torch.empty_like(x)
         self.assertEqual(out, expected)
 
-    @requires_gpu
     def test_wrap_triton_disabled_in_triton_op(self):
         import triton  # @manual
         import triton.language as tl  # @manual
@@ -3384,7 +3332,6 @@ class CustomOpTests(torch._inductor.test_case.TestCase):
         self.assertEqual(status[-1], False)
         self.assertEqual(z, (x + y) * 2)
 
-    @requires_gpu
     def test_preserves_strides(self):
         import triton
         import triton.language as tl
@@ -3453,7 +3400,6 @@ class CustomOpTests(torch._inductor.test_case.TestCase):
             self.assertEqual(f(x, other), f_compile(x, other))
             self.assertTrue(called)
 
-    @requires_gpu
     @common_utils.parametrize("dynamic", [False, True])
     @common_utils.parametrize("autotune", [False, True])
     def test_capture_triton_special_kwargs(self, dynamic, autotune):
@@ -3514,7 +3460,6 @@ class CustomOpTests(torch._inductor.test_case.TestCase):
         self.assertEqual(gm(x, x), x + x)
 
     @skipIfWindows(msg="AOTI/Cpp_Wrapper have not enabled on Windows")
-    @requires_gpu
     @inductor_config.patch("cpp_wrapper", True)
     @inductor_config.patch("triton.autotune_at_compile_time", True)
     def test_autotune_unbacked(self):
@@ -3642,7 +3587,6 @@ class CustomOpTests(torch._inductor.test_case.TestCase):
 
     # Triton 3.2.0 adds the required flags to the Autotuner object for this test
     # PR: https://github.com/triton-lang/triton/pull/5092
-    @requires_gpu
     def test_autotune_no_pre_or_post_hook_user_defined(self):
         from triton.runtime.autotuner import Autotuner
 
@@ -3706,7 +3650,6 @@ class CustomOpTests(torch._inductor.test_case.TestCase):
             add_compiled = torch.compile(add, mode="reduce-overhead", fullgraph=True)
             add_compiled(x, y).mean()
 
-    @requires_gpu
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     @common_utils.parametrize("autotune_at_compile_time", [True, False])
     def test_triton_kernel_reset_to_zero(self, backend, autotune_at_compile_time):
@@ -3759,7 +3702,6 @@ class CustomOpTests(torch._inductor.test_case.TestCase):
 
         self.assertEqual(y + increment, x)
 
-    @requires_gpu
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     def test_triton_single_autotune(self, backend):
         @triton.autotune(
@@ -3811,7 +3753,6 @@ class CustomOpTests(torch._inductor.test_case.TestCase):
             )
             add_compiled(x, y).mean()
 
-    @requires_gpu
     @common_utils.parametrize("non_strict", [True, False])
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     @common_utils.parametrize("with_perf_model", [True, False])
@@ -3904,7 +3845,6 @@ class CustomOpTests(torch._inductor.test_case.TestCase):
             self.assertTrue(records["capture_kwargs"])
             self.assertTrue(records["capture_named_args"])
 
-    @requires_gpu
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     @common_utils.parametrize("with_perf_model", [True, False])
     def test_triton_kernel_prune_configs_by_recompile(self, backend, with_perf_model):
@@ -3997,7 +3937,6 @@ class CustomOpTests(torch._inductor.test_case.TestCase):
         self.assertEqual(counter.op_count, 2)
 
     # see: https://github.com/triton-lang/triton/blob/67ea999935f4511a535a25bdecb27e79e3c3af41/python/test/unit/language/test_decorator.py#L31
-    @requires_gpu
     @common_utils.parametrize("non_strict", [True, False])
     @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
     @common_utils.parametrize("autotune_at_compile_time", [True, False])