diff --git a/test/inductor/test_smoke.py b/test/inductor/test_smoke.py
index 895e8ba..6e8eee2 100644
--- a/test/inductor/test_smoke.py
+++ b/test/inductor/test_smoke.py
@@ -1,12 +1,20 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: inductor"]
 import logging
 import unittest
 
-import torch
 import torch._logging
 from torch._inductor.test_case import TestCase
 from torch.testing._internal.common_utils import IS_LINUX
 from torch.testing._internal.inductor_utils import GPU_TYPE, HAS_CUDA, HAS_GPU
+import torch_npu._inductor
 
 
 class MLP(torch.nn.Module):
@@ -26,7 +34,7 @@ def _test_f(x):
 
 
 class SmokeTest(TestCase):
-    @unittest.skipIf(not HAS_GPU, "Triton is not available")
+    # @unittest.skipIf(not HAS_GPU, "Triton is not available")
     def test_mlp(self):
         torch._logging.set_logs(
             dynamo=logging.DEBUG, inductor=logging.DEBUG, aot=logging.DEBUG
@@ -39,7 +47,7 @@ class SmokeTest(TestCase):
         # set back to defaults
         torch._logging.set_logs()
 
-    @unittest.skipIf(not HAS_GPU, "Triton is not available")
+    # @unittest.skipIf(not HAS_GPU, "Triton is not available")
     def test_compile_decorator(self):
         @torch.compile
         def foo(x):
@@ -61,6 +69,4 @@ class SmokeTest(TestCase):
 if __name__ == "__main__":
     from torch._inductor.test_case import run_tests
 
-    if IS_LINUX and HAS_GPU:
-        if (not HAS_CUDA) or torch.cuda.get_device_properties(0).major <= 5:
-            run_tests()
+    run_tests()