diff --git a/test/inductor/test_cutlass_backend.py b/test/inductor/test_cutlass_backend.py
index a57d2603559..4123ead1dd3 100644
--- a/test/inductor/test_cutlass_backend.py
+++ b/test/inductor/test_cutlass_backend.py
@@ -1,6 +1,7 @@
 # Owner(s): ["module: inductor"]
 import gc
 import itertools
+import torch_npu._inductor
 import logging
 import math
 import os
@@ -29,7 +30,6 @@ try:
 except ImportError:
     from .test_aot_inductor_utils import AOTIRunnerUtil
 
-import torch
 import torch._inductor.codecache
 import torch.version
 from torch._dynamo import config as dynamo_config
