diff --git a/test/inductor/test_metrics.py b/test/inductor/test_metrics.py
index cf8c941..28336d1 100644
--- a/test/inductor/test_metrics.py
+++ b/test/inductor/test_metrics.py
@@ -1,12 +1,19 @@
-# Owner(s): ["module: inductor"]
 import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
+# Owner(s): ["module: inductor"]
 from torch._inductor import config, metrics
 from torch._inductor.test_case import run_tests, TestCase
 from torch._inductor.utils import collect_defined_kernels
 from torch._inductor.wrapper_benchmark import get_kernel_category_by_source_code
 from torch.testing._internal.common_device_type import largeTensorTest
 from torch.testing._internal.inductor_utils import GPU_TYPE, HAS_GPU
-
+import torch_npu._inductor
 
 example_kernel = """
 @triton_heuristics.reduction(
@@ -116,5 +123,4 @@ class TestMetrics(TestCase):
 
 
 if __name__ == "__main__":
-    if HAS_GPU:
-        run_tests()
+    run_tests()
