diff --git a/test/inductor/test_config.py b/test/inductor/test_config.py
index 2c3dc9d..02de240 100644
--- a/test/inductor/test_config.py
+++ b/test/inductor/test_config.py
@@ -1,12 +1,20 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: inductor"]
 import math
 import unittest
 
-import torch
 from torch._dynamo.utils import counters
 from torch._inductor import config
 from torch._inductor.test_case import run_tests, TestCase
 from torch.testing._internal.inductor_utils import HAS_CPU, HAS_TRITON
+import torch_npu._inductor
 
 
 def dummy_fn(x):