diff --git a/test/inductor/test_ordered_set.py b/test/inductor/test_ordered_set.py
index b057fe3..a1618b9 100644
--- a/test/inductor/test_ordered_set.py
+++ b/test/inductor/test_ordered_set.py
@@ -1,3 +1,11 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: inductor"]
 # ruff: noqa: F841
 # flake8: noqa
@@ -15,6 +23,7 @@ from test import support

 from torch.testing._internal.common_utils import TestCase
 from torch.utils._ordered_set import OrderedSet
+import torch_npu._inductor


 class PassThru(Exception):