diff --git a/test/inductor/test_minifier_isolate.py b/test/inductor/test_minifier_isolate.py
index f1862b65f9b..8a153efa901 100644
--- a/test/inductor/test_minifier_isolate.py
+++ b/test/inductor/test_minifier_isolate.py
@@ -1,7 +1,7 @@
 # Owner(s): ["module: inductor"]
 import unittest
-
 import torch._inductor.config as inductor_config
+import torch_npu._inductor
 from torch._dynamo.test_minifier_common import MinifierTestBase
 from torch.testing._internal.common_utils import (
     IS_JETSON,
