diff --git a/test/inductor/test_cooperative_reductions.py b/test/inductor/test_cooperative_reductions.py
index 469ceec..758a29b 100644
--- a/test/inductor/test_cooperative_reductions.py
+++ b/test/inductor/test_cooperative_reductions.py
@@ -1,10 +1,17 @@
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.utils import _dynamo
+_dynamo.use_jit_script = True
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+
 # Owner(s): ["module: inductor"]
 import unittest
 from typing import Any
 
 import sympy
 
-import torch
 import torch._inductor
 from torch._inductor import config
 from torch._inductor.choices import InductorChoices
@@ -18,6 +25,7 @@ from torch.testing._internal.common_utils import (
     parametrize,
 )
 from torch.testing._internal.inductor_utils import HAS_CUDA
+import torch_npu._inductor
 
 
 class TestingHeuristics(InductorChoices):
@@ -301,5 +309,4 @@ class TestFixedConfigs(TestCase):
 if __name__ == "__main__":
     from torch._dynamo.test_case import run_tests
 
-    if HAS_CUDA:
-        run_tests(needs="filelock")
+    run_tests(needs="filelock")
