diff --git a/test/mobile/test_quantize_fx_lite_script_module.py b/test/mobile/test_quantize_fx_lite_script_module.py
index 30cd464..1632a0a 100644
--- a/test/mobile/test_quantize_fx_lite_script_module.py
+++ b/test/mobile/test_quantize_fx_lite_script_module.py
@@ -1,5 +1,4 @@
 # Owner(s): ["oncall: mobile"]
-
 import torch
 import torch.ao.nn.quantized as nnq
 import torch.nn as nn
@@ -13,7 +12,7 @@ from torch.testing._internal.common_quantization import (
     NodeSpec as ns,
     QuantizationLiteTestCase,
 )
-
+from torch.testing._internal.common_utils import run_tests, TestCase
 
 class TestLiteFuseFx(QuantizationLiteTestCase):
     # Tests from:
