diff --git a/test/lazy/test_extract_compiled_graph.py b/test/lazy/test_extract_compiled_graph.py
index 79359dd..25106b0 100644
--- a/test/lazy/test_extract_compiled_graph.py
+++ b/test/lazy/test_extract_compiled_graph.py
@@ -1,5 +1,6 @@
 # Owner(s): ["oncall: jit"]
-
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 import unittest
 
 from torch._lazy.ts_backend import init as init_ts_backend
