diff --git a/torch/testing/_internal/common_utils.py b/torch/testing/_internal/common_utils.py
index 193d625..5e3ae27 100644
--- a/torch/testing/_internal/common_utils.py
+++ b/torch/testing/_internal/common_utils.py
@@ -38,6 +38,7 @@ import time
 import types
 import unittest
 import warnings
+import torch_npu
 from collections.abc import Mapping, Sequence
 from contextlib import closing, contextmanager
 from copy import deepcopy
@@ -1504,6 +1505,7 @@ MACOS_VERSION = float('.'.join(platform.mac_ver()[0].split('.')[:2]) or -1)
 TEST_XPU = torch.xpu.is_available()
 TEST_HPU = bool(hasattr(torch, "hpu") and torch.hpu.is_available())
 TEST_CUDA = torch.cuda.is_available()
+TEST_NPU = hasattr(torch, 'npu') and torch.npu.is_available()
 TEST_ACCELERATOR = LazyVal(lambda: torch.accelerator.is_available())  # type: ignore[call-arg]
 TEST_MULTIACCELERATOR = LazyVal(lambda: torch.accelerator.device_count() > 1)  # type: ignore[call-arg]
 custom_device_mod = getattr(torch, torch._C._get_privateuse1_backend_name(), None)
@@ -2600,6 +2602,26 @@ def to_gpu(obj, type_map=None):
         return deepcopy(obj)
 
 
+def to_npu(obj, type_map=None):
+    if type_map is None:
+        type_map = {}
+    if isinstance(obj, torch.Tensor):
+        assert obj.is_leaf
+        t = type_map.get(obj.dtype, obj.dtype)
+        with torch.no_grad():
+            res = obj.to(dtype=t, device="npu", copy=True)
+            res.requires_grad = obj.requires_grad
+        return res
+    elif torch.is_storage(obj):
+        return obj.new().resize_(obj.size()).copy_(obj)  # type: ignore[attr-defined, union-attr]
+    elif isinstance(obj, list):
+        return [to_npu(o, type_map) for o in obj]
+    elif isinstance(obj, tuple):
+        return tuple(to_npu(o, type_map) for o in obj)
+    else:
+        return deepcopy(obj)
+
+
 def get_function_arglist(func):
     return inspect.getfullargspec(func).args
 
@@ -2677,24 +2699,26 @@ class CudaNonDefaultStream:
         # to ensure CUDA tests do not use default stream by mistake.
         beforeDevice = torch.cuda.current_device()
         self.beforeStreams = []
+        import torch_npu
         for d in range(torch.cuda.device_count()):
             self.beforeStreams.append(torch.cuda.current_stream(d))
             deviceStream = torch.cuda.Stream(device=d)
             self.beforeStreams[-1].synchronize()
-            torch._C._cuda_setStream(stream_id=deviceStream.stream_id,
+            torch_npu._C._npu_setStream(stream_id=deviceStream.stream_id,
                                      device_index=deviceStream.device_index,
                                      device_type=deviceStream.device_type)
-        torch._C._cuda_setDevice(beforeDevice)
+        torch_npu._C._npu_setDevice(beforeDevice)
 
     def __exit__(self, exc_type, exc_value, traceback):
         # After completing CUDA test load previously active streams on all
         # CUDA devices.
         beforeDevice = torch.cuda.current_device()
+        import torch_npu
         for d in range(torch.cuda.device_count()):
-            torch._C._cuda_setStream(stream_id=self.beforeStreams[d].stream_id,
+            torch_npu._C._npu_setStream(stream_id=self.beforeStreams[d].stream_id,
                                      device_index=self.beforeStreams[d].device_index,
                                      device_type=self.beforeStreams[d].device_type)
-        torch._C._cuda_setDevice(beforeDevice)
+        torch_npu._C._npu_setDevice(beforeDevice)
 
 class CudaMemoryLeakCheck:
     def __init__(self, testcase, name=None):
