diff --git a/torch/testing/_internal/common_utils.py b/torch/testing/_internal/common_utils.py
index c533ee687c..d82c0e14e2 100644
--- a/torch/testing/_internal/common_utils.py
+++ b/torch/testing/_internal/common_utils.py
@@ -92,6 +92,7 @@ from torch.testing._comparison import (
 )
 from torch.testing._comparison import not_close_error_metas
 from torch.testing._internal.common_dtype import get_all_dtypes
+import torch_npu
 from torch.utils._import_utils import _check_module_exists
 import torch.utils._pytree as pytree
 from torch.utils import cpp_extension
@@ -3170,6 +3171,16 @@ class TestCase(expecttest.TestCase):
 
     # checker to early terminate test suite if unrecoverable failure occurs.
     def _should_stop_test_suite(self):
+        if torch.npu.is_initialized():
+            # NPU device side error will cause subsequence test cases to fail.
+            # stop entire test suite if catches RuntimeError during torch.npu.synchronize().
+            try:
+                torch.npu.synchronize()
+            except RuntimeError as rte:
+                print("TEST SUITE EARLY TERMINATION due to torch.npu.synchronize() failure", file=sys.stderr)
+                print(str(rte), file=sys.stderr)
+                return True
+            return False
         if torch.cuda.is_initialized():
             # CUDA device side error will cause subsequence test cases to fail.
             # stop entire test suite if catches RuntimeError during torch.cuda.synchronize().
@@ -3180,8 +3191,7 @@ class TestCase(expecttest.TestCase):
                 print(str(rte), file=sys.stderr)
                 return True
             return False
-        else:
-            return False
+        return False
 
     @property
     def precision(self) -> float:
