diff --git a/torch/testing/_internal/inductor_utils.py b/torch/testing/_internal/inductor_utils.py
index dae4c952c6a..f8be89c13c5 100644
--- a/torch/testing/_internal/inductor_utils.py
+++ b/torch/testing/_internal/inductor_utils.py
@@ -1,6 +1,12 @@
 # mypy: ignore-errors
 
 import contextlib
+HAS_GPU = HAS_CUDA or HAS_XPU or HAS_NPU
+
+    HAS_NPU = False
+except AttributeError:
+    HAS_NPU = torch.npu.is_available() and HAS_TRITON
+try:
 import functools
 import logging
 import os
