diff --git a/torch/testing/_internal/inductor_utils.py b/torch/testing/_internal/inductor_utils.py
index 1609a3f..02ba2fe 100644
--- a/torch/testing/_internal/inductor_utils.py
+++ b/torch/testing/_internal/inductor_utils.py
@@ -58,7 +58,12 @@ HAS_XPU = torch.xpu.is_available() and HAS_TRITON
 
 HAS_MPS = torch.mps.is_available()
 
-HAS_GPU = HAS_CUDA or HAS_XPU
+try:
+    HAS_NPU = torch.npu.is_available() and HAS_TRITON
+except AttributeError:
+    HAS_NPU = False
+
+HAS_GPU = HAS_CUDA or HAS_XPU or HAS_NPU
 
 GPU_TYPE = get_gpu_type()
