diff --git a/torch/testing/_internal/jit_utils.py b/torch/testing/_internal/jit_utils.py
index 299eb99..2d823f9 100644
--- a/torch/testing/_internal/jit_utils.py
+++ b/torch/testing/_internal/jit_utils.py
@@ -45,12 +45,14 @@ RUN_CUDA = torch.cuda.is_available()
 RUN_CUDA_MULTI_GPU = RUN_CUDA and torch.cuda.device_count() > 1
 RUN_CUDA_HALF = RUN_CUDA
 # HIP supports half, no version check necessary
-if torch.cuda.is_available() and not torch.version.hip:
-    CUDA_VERSION = torch._C._cuda_getCompiledVersion()
-    for d in range(torch.cuda.device_count()):
-        major = torch.cuda.get_device_capability(d)[0]
-        if (major < 6):
-            RUN_CUDA_HALF = False
+#if torch.cuda.is_available() and not torch.version.hip:
+    #import torch_npu
+    #CUDA_VERSION = torch._C._cuda_getCompiledVersion()
+    #CUDA_VERSION = torch_npu._C._npu_getCompiledVersion()
+    #for d in range(torch.cuda.device_count()):
+        #major = torch.cuda.get_device_capability(d)[0]
+        #if (major < 6):
+            #RUN_CUDA_HALF = False
 
 def execWrapper(code, glob, loc):
     exec(code, glob, loc)
