diff --git a/test/export/test_passes.py b/test/export/test_passes.py
index 56f56438776..8da86356beb 100644
--- a/test/export/test_passes.py
+++ b/test/export/test_passes.py
@@ -1382,12 +1382,12 @@ def forward(self, arg0_1):
         mod = Model()
         example_inputs = (torch.rand(1, 10, 4),)
         ep = export(mod, example_inputs, strict=True)
-        location = torch.device("cuda:0")
+        location = torch.device("npu:0")
         ep = move_to_device_pass(ep, location=location)
         gm = ep.module()
-        test_inputs = (torch.rand(1, 10, 4).to("cuda:0"),)
+        test_inputs = (torch.rand(1, 10, 4).to("npu:0"),)
         outputs = gm(*test_inputs)
-        self.assertEqual(outputs.device, torch.device("cuda:0"))
+        self.assertEqual(outputs.device, torch.device("npu:0"))
         # move it back to cpu
         location = "cpu"
         ep = move_to_device_pass(ep, location=location)
@@ -1396,12 +1396,12 @@ def forward(self, arg0_1):
         outputs = gm(*test_inputs)
         self.assertEqual(outputs.device, torch.device("cpu"))
         # move it to cuda:0 again
-        location = {"cpu": "cuda:0"}
+        location = {"cpu": "npu:0"}
         ep = move_to_device_pass(ep, location=location)
         gm = ep.module()
-        test_inputs = (torch.rand(1, 10, 4).to("cuda:0"),)
+        test_inputs = (torch.rand(1, 10, 4).to("npu:0"),)
         outputs = gm(*test_inputs)
-        self.assertEqual(outputs.device, torch.device("cuda:0"))
+        self.assertEqual(outputs.device, torch.device("npu:0"))
 
     @unittest.skipIf(not TEST_CUDA, "requires cuda")
     def test_move_device_example_inputs(self):
