diff --git a/test/export/test_export_opinfo.py b/test/export/test_export_opinfo.py
index b33aeb45438..7b92e509c51 100644
--- a/test/export/test_export_opinfo.py
+++ b/test/export/test_export_opinfo.py
@@ -14,7 +14,7 @@ from torch.testing._internal.common_device_type import (
     ops,
 )
 from torch.testing._internal.common_methods_invocations import (
-    onlyCUDA,
+    onlyPRIVATEUSE1,
     op_db,
     skip,
     skipOps,
@@ -45,7 +45,7 @@ export_failures = {
     xfail("tensor_split"),
 }
 
-# following are failing fake export on cuda device
+# following are failing fake export on npu device
 fake_export_failures = {
     xfail("geqrf"),
     xfail("histogram"),
@@ -84,7 +84,7 @@ def _test_export_helper(self, dtype, op):
     sample_inputs_itr = op.sample_inputs("cpu", dtype, requires_grad=False)
 
     mode = FakeTensorMode(allow_non_fake_inputs=True)
-    target_device = "cuda:0"
+    target_device = "npu:0"
 
     def to_fake_device(x):
         return x.to(target_device)
@@ -152,7 +152,7 @@ class TestExportOnFakeCuda(TestCase):
     # In CI, this test runs on a CUDA machine with cuda build
     # We set CUDA_VISIBLE_DEVICES="" to simulate a CPU machine with cuda build
     # Running this on all ops in op_db is too slow, so we only run on a selected subset
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @unittest.skipIf(
         IS_WINDOWS,
         'Subprocess with CUDA_VISIBLE_DEVICES="" imports op_db which triggers '
@@ -175,7 +175,7 @@ for op in ops:
 
     mode = FakeTensorMode(allow_non_fake_inputs=True)
 
-    target_device = "cuda:0"
+    target_device = "npu:0"
 
     def to_fake_device(x):
         return x.to(target_device)
@@ -238,12 +238,12 @@ def cuda_calls_behavior_unchanged():
 
     try:
         cpu_x = torch.randn(2)
-        cuda_x = cpu_x.to("cuda")
+        cuda_x = cpu_x.to("npu")
     except Exception as e:
         exception_count += 1
 
     try:
-        torch.randn(2, device="cuda")
+        torch.randn(2, device="npu")
     except Exception as e:
         exception_count += 1
 
@@ -271,9 +271,9 @@ cuda_calls_behavior_unchanged()
 cpu_x = torch.randn(2)
 with FakeTensorMode(allow_non_fake_inputs=True) as mode:
     cuda_x = mode.from_tensor(cpu_x)
-    cuda_x.fake_device = torch.device("cuda")
+    cuda_x.fake_device = torch.device("npu")
     cuda_y = cuda_x + cuda_x
-    assert cuda_y.device.type == "cuda"
+    assert cuda_y.device.type == "npu"
 
 # should fail again after exiting the fake mode, with the identical error message
 cuda_calls_behavior_unchanged()
@@ -291,7 +291,7 @@ cuda_calls_behavior_unchanged()
         self.assertEqual(r, "")
 
 
-instantiate_device_type_tests(TestExportOnFakeCuda, globals(), only_for="cuda")
+instantiate_device_type_tests(TestExportOnFakeCuda, globals(), only_for="npu")
 
 
 if __name__ == "__main__":
