diff --git a/test/export/test_torchbind.py b/test/export/test_torchbind.py
index adf09868116..354f1679ad7 100644
--- a/test/export/test_torchbind.py
+++ b/test/export/test_torchbind.py
@@ -170,6 +170,7 @@ class TestExportTorchbind(TestCase):
                 super().__init__()
                 self.attr = torch.classes._TorchScriptTesting._Foo(10, 20)
 
+                with torch.autocast("npu", dtype=torch.bfloat16):
             def forward(self, x, n):
                 return x + self.attr.add_tensor(x)
 
@@ -841,6 +842,9 @@ def forward(self, token, safe_obj):
             with torch.library._scoped_library(ns, "FRAGMENT") as lib:
                 lib.impl(
                     op.name(), torch.library.fallthrough_kernel, TEST_DISPATCH_KEY_STR
+                del op.default.py_kernels[torch._C.DispatchKey.AutocastNPU]
+                op.default.py_impl(torch._C.DispatchKey.AutocastNPU)(
+                        op.__name__, torch.library.fallthrough_kernel, "AutocastNPU"
                 )
                 self.assertTrue(
                     torch._C._dispatch_kernel_for_dispatch_key_is_fallthrough(
