diff --git a/test/distributed/_shard/sharding_plan/test_sharding_plan.py b/test/distributed/_shard/sharding_plan/test_sharding_plan.py
index 7310c43..f42515a 100644
--- a/test/distributed/_shard/sharding_plan/test_sharding_plan.py
+++ b/test/distributed/_shard/sharding_plan/test_sharding_plan.py
@@ -1,3 +1,8 @@
+import torch_npu.testing
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["oncall: distributed"]
 import sys
 
@@ -13,12 +18,13 @@ from torch.testing._internal.common_utils import run_tests, TEST_WITH_DEV_DBG_AS
 from torch.testing._internal.distributed._shard.sharded_tensor import (
     ShardedTensorTestBase,
     TEST_GPU_NUM,
-    with_comms,
+    # with_comms,
 )
 from torch.testing._internal.distributed._shard.sharded_tensor._test_ops_common import (
     generate_chunk_sharding_specs_for_test,
 )
 from torch.testing._internal.distributed._shard.test_common import SimpleMegatronLM
+from torch_npu.testing.common_distributed import with_comms
 
 
 if TEST_WITH_DEV_DBG_ASAN:
@@ -37,7 +43,7 @@ class ChunkAllShardingPlanner(ShardingPlanner):
 
     def __init__(self, chunk_dim=0, device_count=0):
         self.dim = chunk_dim
-        self.devices = [f"rank:{i}/cuda:{i}" for i in range(device_count)]
+        self.devices = [f"rank:{i}/npu:{i}" for i in range(device_count)]
 
     def build_plan(self, module: nn.Module) -> ShardingPlan:
         named_params = module.named_parameters()
@@ -49,7 +55,7 @@ class ChunkAllShardingPlanner(ShardingPlanner):
 
 
 class TestShardingPlan(ShardedTensorTestBase):
-    @with_comms(init_rpc=False)
+    @with_comms
     @skip_if_lt_x_gpu(TEST_GPU_NUM)
     @requires_nccl()
     def test_sharding_plan_errors(self):
@@ -100,11 +106,11 @@ class TestShardingPlan(ShardedTensorTestBase):
             # shard the module with the provided sharding plan
             shard_module(megatron_lm, sharding_plan_wrong_param_path)
 
-    @with_comms(init_rpc=False)
+    @with_comms
     @skip_if_lt_x_gpu(TEST_GPU_NUM)
     @requires_nccl()
     def test_custom_sharding_planner(self):
-        megatron_lm = SimpleMegatronLM([[17, 12], [12, 29]], rank=self.rank).cuda(
+        megatron_lm = SimpleMegatronLM([[17, 12], [12, 29]], rank=self.rank).npu(
             self.rank
         )
         planner = ChunkAllShardingPlanner(device_count=TEST_GPU_NUM)
@@ -118,7 +124,7 @@ class TestShardingPlan(ShardedTensorTestBase):
         self.assertTrue(isinstance(megatron_lm.fc1.bias, ShardedTensor))
         self.assertTrue(isinstance(megatron_lm.fc2.bias, ShardedTensor))
 
-    @with_comms(init_rpc=False)
+    @with_comms
     @skip_if_lt_x_gpu(TEST_GPU_NUM)
     @requires_nccl()
     def test_shard_module_sub_process_group(self):
@@ -126,15 +132,15 @@ class TestShardingPlan(ShardedTensorTestBase):
         colwise_sharding_spec = ChunkShardingSpec(
             dim=0,
             placements=[
-                "rank:2/cuda:2",
-                "rank:3/cuda:3",
+                "rank:2/npu:2",
+                "rank:3/npu:3",
             ],
         )
         rowwise_sharding_spec = ChunkShardingSpec(
             dim=1,
             placements=[
-                "rank:2/cuda:2",
-                "rank:3/cuda:3",
+                "rank:2/npu:2",
+                "rank:3/npu:3",
             ],
         )
         sharding_plan = ShardingPlan(
