diff --git a/test/torch_np/numpy_tests/core/test_indexing.py b/test/torch_np/numpy_tests/core/test_indexing.py
index ed402bd..7634e49 100644
--- a/test/torch_np/numpy_tests/core/test_indexing.py
+++ b/test/torch_np/numpy_tests/core/test_indexing.py
@@ -252,7 +252,7 @@ class TestIndexing(TestCase):
         a[b] = 1.0
         assert_equal(a, [[1.0, 1.0, 1.0]])
 
-    @skip(reason="NP_VER: fails on CI")
+    # @skip(reason="NP_VER: fails on CI")
     def test_boolean_assignment_value_mismatch(self):
         # A boolean assignment should fail when the shape of the values
         # cannot be broadcast to the subscription. (see also gh-3458)
