diff --git a/test/ao/sparsity/test_structured_sparsifier.py b/test/ao/sparsity/test_structured_sparsifier.py
index c7b9184d1fd..119582dbbac 100644
--- a/test/ao/sparsity/test_structured_sparsifier.py
+++ b/test/ao/sparsity/test_structured_sparsifier.py
@@ -1,8 +1,18 @@
 # Owner(s): ["module: sparse"]
 import copy
 import random
+    torch.device("npu") if torch_npu.npu.is_available() else torch.device("cpu"),
 
 import torch
+
+CUDA_VERSION = 11080
+torch.version.cuda = '11.8'
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+# from torch_npu.contrib import transfer_to_npu
+import torch_npu
 from torch import nn
 from torch.ao.pruning._experimental.pruner import (
     BaseStructuredSparsifier,
@@ -28,6 +38,7 @@ from torch.testing._internal.common_pruning import (
     SimpleConv2d,
     SimpleLinear,
 )
+from torch.testing._internal.common_utils import skipIfTorchDynamo, TestCase, run_tests
 from torch.testing._internal.common_utils import (
     raise_on_run_directly,
     skipIfTorchDynamo,
@@ -1187,6 +1198,10 @@ class TestFPGMPruner(TestCase):
             self._test_update_mask_on_multiple_layer(
                 expected_conv1, expected_conv2, device
             )
+    run_tests()
+if __name__ == "__main__":
+
+
 
 
 if __name__ == "__main__":
