diff --git a/test/distributed/optim/test_zero_redundancy_optimizer.py b/test/distributed/optim/test_zero_redundancy_optimizer.py
index 5cf402a..4159162 100644
--- a/test/distributed/optim/test_zero_redundancy_optimizer.py
+++ b/test/distributed/optim/test_zero_redundancy_optimizer.py
@@ -1,3 +1,7 @@
+import torch_npu.testing
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["oncall: distributed"]
 
 # Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
@@ -726,14 +730,16 @@ class TestZeroRedundancyOptimizerDistributed(TestZeroRedundancyOptimizer):
                 MIN_WORLD_SIZE,
             )
             return
-        BACKEND = dist.Backend.GLOO
+        # BACKEND = dist.Backend.GLOO
+        BACKEND = "cpu:gloo,npu:hccl"
         self.dist_init(self.rank, self.world_size, BACKEND)
         # Use GPU if enough are available, or fall back to CPU otherwise, which
         # is fine since Gloo backend supports both
-        if torch.cuda.is_available() and torch.cuda.device_count() >= self.world_size:
-            device = torch.device(self.rank)
-        else:
-            device = torch.device("cpu")
+        # if torch.cuda.is_available() and torch.cuda.device_count() >= self.world_size:
+        #     device = torch.device(self.rank)
+        # else:
+        #     device = torch.device("cpu")
+        device = torch.device(self.rank)
         # Create a new process group consisting of the even ranks to exercise
         # the case where the global and local ranks do not necessarily match
         subgroup_ranks = [r for r in range(self.world_size) if r % 2 == 0]
@@ -741,6 +747,7 @@ class TestZeroRedundancyOptimizerDistributed(TestZeroRedundancyOptimizer):
             ranks=subgroup_ranks,
             backend=BACKEND,
         )
+        torch.npu.set_device(self.rank)
         # Ranks not participating in the new process group are no longer needed
         if self.rank not in subgroup_ranks:
             return
@@ -975,8 +982,10 @@ class TestZeroRedundancyOptimizerDistributed(TestZeroRedundancyOptimizer):
         rank = self.rank
         world_size = self.world_size
         is_gpu = device.type == "cuda"
-        backend = _get_backend_for_tests() if is_gpu else dist.Backend.GLOO
+        # backend = _get_backend_for_tests() if is_gpu else dist.Backend.GLOO
+        backend = "hccl"
         self.dist_init(rank, world_size, backend)
+        torch.npu.set_device(self.rank)
 
         model = torch.nn.Sequential(
             torch.nn.Linear(2, 3),
