diff --git a/test/distributed/optim/test_zero_redundancy_optimizer.py b/test/distributed/optim/test_zero_redundancy_optimizer.py
index 283ac98bf5c..258987e058b 100644
--- a/test/distributed/optim/test_zero_redundancy_optimizer.py
+++ b/test/distributed/optim/test_zero_redundancy_optimizer.py
@@ -1,3 +1,7 @@
+import torch_npu.testing
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["oncall: distributed"]
 
 # Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
@@ -711,6 +715,7 @@ class TestZeroRedundancyOptimizerDistributed(TestZeroRedundancyOptimizer):
             ranks=subgroup_ranks,
             backend=self.backend(device.type),
         )
+        torch.npu.set_device(self.rank)
         # Ranks not participating in the new process group are no longer needed
         if self.rank not in subgroup_ranks:
             return
