diff --git a/test/jit/test_list_dict.py b/test/jit/test_list_dict.py
index 1949ec46557..febdaa21b96 100644
--- a/test/jit/test_list_dict.py
+++ b/test/jit/test_list_dict.py
@@ -1,3 +1,6 @@
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["oncall: jit"]
 # ruff: noqa: F841
 
@@ -1753,7 +1756,7 @@ class TestDict(JitTestCase):
 
         tester(pop, "a")
 
-        with self.assertRaisesRegexWithHighlight(RuntimeError, "KeyError", "x.pop"):
+        with self.assertRaisesRegexWithHighlight(RuntimeError, "RuntimeError", "x.pop"):
             torch.jit.script(pop)(self.dict(), "x")
 
         def default_pop(
@@ -1901,7 +1904,7 @@ class TestDict(JitTestCase):
             "KeyError",
             'x["dne"',  # codespell:ignore
         ):
-            missing_index({"item": 20, "other_item": 120})
+            missing_index({"dne": 20, "other_item": 120})
 
         code = dedent(
             """
