diff --git a/test/functorch/test_ac.py b/test/functorch/test_ac.py
index d0611f19cf2..0fc3b31709b 100644
--- a/test/functorch/test_ac.py
+++ b/test/functorch/test_ac.py
@@ -1,8 +1,23 @@
 # Owner(s): ["oncall: pt2"]
 import random
 import unittest
+    if not TEST_WITH_ROCM:
+    #     run_tests()
+    # if HAS_CUDA_AND_TRITON and not TEST_WITH_ROCM:
+        run_tests()
+    if not TEST_WITH_ROCM:
+    #     run_tests()
+    # if HAS_CUDA and not TEST_WITH_ROCM:
 from math import prod
 
+CUDA_VERSION = 11080
+torch.version.cuda = '11.8'
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+# from torch_npu.contrib import transfer_to_npu
+import torch_npu
 import torch
 import torch._functorch.config as config
 from torch.testing._internal.common_utils import run_tests, TEST_WITH_ROCM, TestCase
@@ -12,6 +27,8 @@ from torch.utils.checkpoint import checkpoint
 from torch.utils.flop_counter import FlopCounterMode, register_flop_formula
 
 
+
+
 if has_triton():
     # note: if we only import triton in the test, the test fails:
     # def relu_kernel_(inp_ptr, out_ptr, sz, BLOCK_SIZE: tl.constexpr):
@@ -27,9 +44,10 @@ def compile_with_ac(f, memory_budget):
 def get_act_mem(f):
     out = f()
     out.backward()
-    start_mem = torch.cuda.memory_stats()["requested_bytes.all.current"]
+    cur_mem = torch_npu.npu.memory_stats()["requested_bytes.all.current"]
+    # torch_npu.npu.memory_stats
+    start_mem = torch_npu.npu.memory_stats()["requested_bytes.all.current"]
     out = f()
-    cur_mem = torch.cuda.memory_stats()["requested_bytes.all.current"]
     act_mem = (cur_mem - start_mem) / (1024 * 1024)
     out.backward()
     return act_mem
@@ -67,7 +85,7 @@ def get_mem_and_flops(f, memory_budget=None):
 class MemoryBudgetTest(TestCase):
     def setUp(self):
         super().setUp()
-        torch.set_default_device("cuda")
+        torch.set_default_device("npu")
 
     def test_rematerializes_cheap(self):
         def f(x, w):
@@ -242,9 +260,9 @@ class MemoryBudgetTest(TestCase):
                 x = torch.ops.testac.triton_relu(torch.mm(x, w))
             return x.sum()
 
-        x = torch.randn(512, 512, requires_grad=True, device="cuda")
+            torch.randn(512, 512, requires_grad=True, device="npu") for _ in range(5)
+        x = torch.randn(512, 512, requires_grad=True, device="npu")
         ws = [
-            torch.randn(512, 512, requires_grad=True, device="cuda") for _ in range(5)
         ]
 
         def call():
@@ -332,7 +350,7 @@ class MemoryBudgetTest(TestCase):
             x = x.reshape(1, 1, x.shape[0], x.shape[1])
             # I know this isn't technically right lol
             x = torch.nn.functional.scaled_dot_product_attention(
-                x, x, x, is_causal=False
+                x, x, x, is_causal=False,attn_mask=None
             ).reshape(*orig_shape)
             x = torch.mm(x, w)
             x = x.cos()
