diff --git a/test/functorch/test_vmap.py b/test/functorch/test_vmap.py
index d552179..6a0cfbb 100644
--- a/test/functorch/test_vmap.py
+++ b/test/functorch/test_vmap.py
@@ -16,6 +16,15 @@ import unittest
 import warnings
 from collections import namedtuple, OrderedDict
 from unittest.case import skipIf
+import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
 
 from common_utils import (
     check_vmap_fallback,
@@ -36,9 +45,9 @@ from common_utils import (
 from functorch_additional_op_db import additional_op_db
 
 import functorch
-import torch
 import torch.nn.functional as F
-from functorch import grad, grad_and_value, jacfwd, jvp, vjp, vmap
+from functorch import grad, grad_and_value, jacfwd, jvp, vjp
+from torch import vmap
 from functorch.experimental import chunk_vmap
 from torch import Tensor
 from torch._C._functorch import reshape_dim_into, reshape_dim_outof
@@ -78,7 +87,6 @@ from torch.testing._internal.common_utils import (
 from torch.testing._internal.custom_op_db import custom_op_db
 from torch.utils import _pytree as pytree
 
-
 def get_platform_specific_sdpa():
     ret = [SDPBackend.MATH]
     if PLATFORM_SUPPORTS_FLASH_ATTENTION:
@@ -1200,9 +1208,9 @@ class TestVmapAPI(TestCase):
     def test_vmap_autocast_cpu(self):
         self._test_vmap_autocast("cpu")
 
-    @skipIf(not torch.cuda.is_available(), "CUDA is unavailable")
-    def test_vmap_autocast_cuda(self):
-        self._test_vmap_autocast("cuda")
+    @skipIf(not torch_npu.npu.is_available(), "NPU is unavailable")
+    def test_vmap_autocast_npu(self):
+        self._test_vmap_autocast("npu")
 
     def test_restore_vmap_pytree_input_output(self):
         def f(x, y):
@@ -1794,7 +1802,7 @@ class TestVmapOperators(Namespace.TestVmapBase):
         test(op, (getter([B0, 2], device), getter([B0], device, torch.double)))
         test(op, (getter([B0], device, torch.double), getter([B0, 2], device)))
 
-        if not torch.cuda.is_available():
+        if not torch_npu.npu.is_available():
             return
 
         # TODO(rzou): fix the following
@@ -3875,7 +3883,7 @@ class TestVmapBatchedGradient(Namespace.TestVmapBase):
     @parametrize("backend", PLATFORM_SPECIFIC_SDPA)
     def test_sdpa(self, device, backend):
         if device == "cpu":
-            raise unittest.SkipTest("This test is only for CUDA for now")
+            raise unittest.SkipTest("This test is only for NPU for now")
 
         def T(*args):
             return torch.randn(*args, dtype=torch.float16, device=device)
@@ -3930,7 +3938,7 @@ class TestVmapBatchedGradient(Namespace.TestVmapBase):
     @parametrize("randomness", ["error", "same", "different"])
     def test_randomness(self, device, randomness, backend):
         if device == "cpu":
-            raise unittest.SkipTest("This test is only for CUDA for now")
+            raise unittest.SkipTest("This test is only for NPU for now")
         backend_ctx = sdpa_kernel([backend])
         with backend_ctx:
             B = 4
@@ -4287,22 +4295,22 @@ class TestVmapOperatorsOpInfo(TestCase):
         xfail("cdouble"),
         xfail("cfloat"),
         xfail(
-            "jiterator_binary", device_type="cuda"
+            "jiterator_binary", device_type="npu"
         ),  # NYI: querying is_contiguous inside of vmap
         xfail(
-            "jiterator_binary_return_by_ref", device_type="cuda"
+            "jiterator_binary_return_by_ref", device_type="npu"
         ),  # NYI: querying is_contiguous inside of vmap
         xfail(
-            "jiterator_4inputs_with_extra_args", device_type="cuda"
+            "jiterator_4inputs_with_extra_args", device_type="npu"
         ),  # NYI: querying is_contiguous inside of vmap
         xfail(
             "equal", ""
         ),  # TypeError: object of type 'bool' has no len(); likely testrunner problem
         xfail(
-            "jiterator_unary", device_type="cuda"
+            "jiterator_unary", device_type="npu"
         ),  # NYI: querying is_contiguous inside of vmap
         xfail(
-            "jiterator_2inputs_2outputs", device_type="cuda"
+            "jiterator_2inputs_2outputs", device_type="npu"
         ),  # NYI: querying is_contiguous inside of vmap
         # ---------------------------------------------------------------------
         # TypeError: expected Tensor as element 0 in argument 0, but got NotImplementedType
@@ -4321,12 +4329,12 @@ class TestVmapOperatorsOpInfo(TestCase):
         xfail("nn.functional.one_hot"),
         # RuntimeError: Expected all tensors to be on the same device,
         # but found at least two devices, cuda:0 and cpu!
-        xfail("eq", device_type="cuda"),
-        xfail("ge", device_type="cuda"),
-        xfail("gt", device_type="cuda"),
-        xfail("le", device_type="cuda"),
-        xfail("lt", device_type="cuda"),
-        xfail("ne", device_type="cuda"),
+        xfail("eq", device_type="npu"),
+        xfail("ge", device_type="npu"),
+        xfail("gt", device_type="npu"),
+        xfail("le", device_type="npu"),
+        xfail("lt", device_type="npu"),
+        xfail("ne", device_type="npu"),
         # RuntimeError: aten::_flash_attention_forward hit the vmap fallback which is currently disabled
         xfail("torch.ops.aten._flash_attention_forward"),
     }
@@ -4343,14 +4351,14 @@ class TestVmapOperatorsOpInfo(TestCase):
             tol1(
                 "linalg.det",
                 {torch.float32: tol(atol=1e-04, rtol=1e-04)},
-                device_type="cuda",
+                device_type="npu",
             ),
             # The following is often flaky, but just on windows.
             # We should investigate if it's actually a problem or not.
             tol1(
                 "nn.functional.conv_transpose3d",
                 {torch.float32: tol(atol=1e-04, rtol=1e-02)},
-                device_type="cuda",
+                device_type="npu",
             ),
         ),
     )
@@ -4417,7 +4425,7 @@ class TestVmapOperatorsOpInfo(TestCase):
             tol1(
                 "linalg.det",
                 {torch.float32: tol(atol=1e-04, rtol=1e-04)},
-                device_type="cuda",
+                device_type="npu",
             ),
         ),
     )
@@ -4532,9 +4540,9 @@ class TestVmapOperatorsOpInfo(TestCase):
                 xfail("linalg.ldl_solve", "", device_type="cpu"),
                 xfail("chalf", ""),
                 xfail("clamp_max", ""),
-                xfail("jiterator_binary_return_by_ref", device_type="cuda"),
-                xfail("jiterator_unary", device_type="cuda"),
-                xfail("jiterator_2inputs_2outputs", device_type="cuda"),
+                xfail("jiterator_binary_return_by_ref", device_type="npu"),
+                xfail("jiterator_unary", device_type="npu"),
+                xfail("jiterator_2inputs_2outputs", device_type="npu"),
                 xfail("special.airy_ai"),
                 xfail("clamp_min", ""),
                 xfail("sparse.sampled_addmm"),
@@ -4547,8 +4555,8 @@ class TestVmapOperatorsOpInfo(TestCase):
                 xfail("index_reduce", "amax"),
                 xfail("special.laguerre_polynomial_l"),
                 xfail("special.hermite_polynomial_h"),
-                xfail("jiterator_binary", device_type="cuda"),
-                xfail("jiterator_4inputs_with_extra_args", device_type="cuda"),
+                xfail("jiterator_binary", device_type="npu"),
+                xfail("jiterator_4inputs_with_extra_args", device_type="npu"),
                 xfail("_segment_reduce", "lengths"),
                 xfail("lu_solve", ""),
                 xfail("special.hermite_polynomial_he"),
@@ -4563,10 +4571,11 @@ class TestVmapOperatorsOpInfo(TestCase):
                 xfail("bincount"),
                 # RuntimeError: Expected all tensors to be on the same device,
                 # but found at least two devices, cuda:0 and cpu!
-                xfail("ge", device_type="cuda"),
+                xfail("ge", device_type="npu"),
                 xfail(
                     "searchsorted"
                 ),  # aten::searchsorted.Scalar hit the vmap fallback which is currently disabled
+                skip("scatter_add", "", device_type="npu", dtypes=(torch.float32,)),
             }
         ),
     )
@@ -4815,7 +4824,7 @@ class TestVmapOperatorsOpInfo(TestCase):
         op = torch.ops.aten._convolution_double_backward
 
         generator = get_fallback_and_vmap_exhaustive(op, args, {})
-        is_cuda_sm86 = device.startswith("cuda") and torch.cuda.get_device_capability(
+        is_cuda_sm86 = device.startswith("npu") and torch.cuda.get_device_capability(
             0
         ) == (8, 6)
         atol, rtol = (1e-3, 1e-3) if is_cuda_sm86 else (1e-4, 1e-4)
@@ -4977,7 +4986,7 @@ class TestVmapOperatorsOpInfo(TestCase):
 
         self.vmap_outplace_test(f, (x, gy), {}, in_dims=(None, 0))
 
-    @onlyCUDA
+    # @onlyCUDA
     @parametrize("inplace", [True, False])
     def test_0d_tensor_index_put(self, device, inplace):
         def f(t, idx, v):
@@ -4985,7 +4994,7 @@ class TestVmapOperatorsOpInfo(TestCase):
             return fn(t, idx, v)
 
         N = 2
-        t = torch.zeros((N, 5), device="cuda")
+        t = torch.zeros((N, 5), device="npu")
         idx = torch.tensor([1, 3])
         v = torch.tensor(1, dtype=t.dtype, device="cpu")
 
@@ -5446,14 +5455,14 @@ class TestRandomness(TestCase):
             for i in range(B0):
                 expected = torch.randperm(10, **kwargs)
                 # RNG differs between eager and via dynamo trace on CUDA
-                if TEST_WITH_TORCHDYNAMO and torch.device(device).type == "cuda":
+                if TEST_WITH_TORCHDYNAMO and torch.device(device).type == "npu":
                     self._assert_all_slices_unique(vmap_result)
                 else:
                     self.assertEqual(vmap_result[i], expected)
         else:
             expected = torch.randperm(10, **kwargs)
             # RNG differs between eager and via dynamo trace on CUDA
-            if TEST_WITH_TORCHDYNAMO and torch.device(device).type == "cuda":
+            if TEST_WITH_TORCHDYNAMO and torch.device(device).type == "npu":
                 self._assert_all_slices_equal(vmap_result)
             else:
                 for i in range(B0):
@@ -5660,7 +5669,7 @@ class TestRandomness(TestCase):
 
                 self._assert_all_slices_unique(vmap_result)
                 # RNG differs between eager and via dynamo trace on CUDA
-                if not (TEST_WITH_TORCHDYNAMO and torch.device(device).type == "cuda"):
+                if not (TEST_WITH_TORCHDYNAMO and torch.device(device).type == "npu"):
                     self.assertEqual(expected, vmap_result)
                 return
 
@@ -5670,7 +5679,7 @@ class TestRandomness(TestCase):
             expected = op(passed, 0)
             self._assert_all_slices_equal(vmap_result)
             # RNG differs between eager and via dynamo trace on CUDA
-            if not (TEST_WITH_TORCHDYNAMO and torch.device(device).type == "cuda"):
+            if not (TEST_WITH_TORCHDYNAMO and torch.device(device).type == "npu"):
                 for i in range(B0):
                     self.assertEqual(expected, vmap_result[i])
 
@@ -6415,7 +6424,8 @@ class TestVmapNestedTensor(Namespace.TestVmapBase):
             vmap(vmap(vmap(f)))(x)
 
 
-only_for = ("cpu", "cuda")
+only_for = ("cpu", "privateuse1")
+# only_for = ("cpu", "gpu")
 instantiate_device_type_tests(TestVmapOperatorsOpInfo, globals(), only_for=only_for)
 
 instantiate_device_type_tests(
