diff --git a/test/functorch/test_ac.py b/test/functorch/test_ac.py
index c3bc905..94b4f6a 100644
--- a/test/functorch/test_ac.py
+++ b/test/functorch/test_ac.py
@@ -2,8 +2,16 @@
 import random
 import unittest
 from math import prod
-
 import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
+
 import torch._functorch.config as config
 from torch.testing._internal.common_utils import run_tests, TEST_WITH_ROCM, TestCase
 from torch.testing._internal.inductor_utils import HAS_CUDA
@@ -11,6 +19,8 @@ from torch.utils._triton import has_triton
 from torch.utils.flop_counter import FlopCounterMode, register_flop_formula
 
 
+
+
 if has_triton():
     # note: if we only import triton in the test, the test fails:
     # def relu_kernel_(inp_ptr, out_ptr, sz, BLOCK_SIZE: tl.constexpr):
@@ -26,9 +36,10 @@ def compile_with_ac(f, memory_budget):
 def get_act_mem(f):
     out = f()
     out.backward()
-    start_mem = torch.cuda.memory_stats()["requested_bytes.all.current"]
+    start_mem = torch_npu.npu.memory_stats()["requested_bytes.all.current"]
+    # torch_npu.npu.memory_stats
     out = f()
-    cur_mem = torch.cuda.memory_stats()["requested_bytes.all.current"]
+    cur_mem = torch_npu.npu.memory_stats()["requested_bytes.all.current"]
     act_mem = (cur_mem - start_mem) / (1024 * 1024)
     out.backward()
     return act_mem
@@ -66,7 +77,7 @@ def get_mem_and_flops(f, memory_budget=None):
 class MemoryBudgetTest(TestCase):
     def setUp(self):
         super().setUp()
-        torch.set_default_device("cuda")
+        torch.set_default_device("npu")
 
     def test_rematerializes_cheap(self):
         def f(x, w):
@@ -241,9 +252,9 @@ class MemoryBudgetTest(TestCase):
                 x = torch.ops.testac.triton_relu(torch.mm(x, w))
             return x.sum()
 
-        x = torch.randn(512, 512, requires_grad=True, device="cuda")
+        x = torch.randn(512, 512, requires_grad=True, device="npu")
         ws = [
-            torch.randn(512, 512, requires_grad=True, device="cuda") for _ in range(5)
+            torch.randn(512, 512, requires_grad=True, device="npu") for _ in range(5)
         ]
 
         def call():
@@ -331,7 +342,7 @@ class MemoryBudgetTest(TestCase):
             x = x.reshape(1, 1, x.shape[0], x.shape[1])
             # I know this isn't technically right lol
             x = torch.nn.functional.scaled_dot_product_attention(
-                x, x, x, is_causal=False
+                x, x, x, is_causal=False,attn_mask=None
             ).reshape(*orig_shape)
             x = torch.mm(x, w)
             x = x.cos()
@@ -377,5 +388,7 @@ class MemoryBudgetTest(TestCase):
 
 if __name__ == "__main__":
     # I'm using the cuda memory allocator to verify memory allocations
-    if HAS_CUDA and not TEST_WITH_ROCM:
-        run_tests()
+    # if HAS_CUDA and not TEST_WITH_ROCM:
+    #     run_tests()
+    if not TEST_WITH_ROCM:
+        run_tests()
\ No newline at end of file
