diff --git a/test/functorch/test_control_flow.py b/test/functorch/test_control_flow.py
index 6ed0b5841b4..82e30f0132d 100644
--- a/test/functorch/test_control_flow.py
+++ b/test/functorch/test_control_flow.py
@@ -4,6 +4,15 @@ import functools
 import unittest
 
 import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
+
 import torch.utils._pytree as pytree
 from functorch.experimental import control_flow
 from functorch.experimental.control_flow import cond
@@ -559,16 +568,15 @@ class TestControlFlow(TestCase):
         result = cond(False, true_fn, false_fn, [x])
         self.assertEqual(result, torch.cos(x))
 
-    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
-    def test_cond_gpu(self):
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
+    def test_cond_npu(self):
         def true_fn(x):
             return x.sin()
 
         def false_fn(x):
             return x.cos()
-
-        x = torch.randn(4, device="cuda")
-        pred = torch.tensor(False, device="cuda")
+        x = torch.randn(4,device='npu')
+        pred = torch.tensor(False,device='npu')
         result = cond(pred, true_fn, false_fn, [x])
         self.assertEqual(result, torch.cos(x))
 
@@ -1297,8 +1305,8 @@ def forward(self, pred_1, x_1):
     return (getitem_1,)""",  # noqa: B950
         )
 
-    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
-    def test_cond_autograd_gpu(self):
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
+    def test_cond_autograd_npu(self):
         def true_fn(x):
             return x.sin()
 
@@ -1306,10 +1314,10 @@ def forward(self, pred_1, x_1):
             return x.cos()
 
         for pred, fn in zip(
-            [torch.tensor(False, device="cuda"), torch.tensor(True, device="cuda")],
+            [torch.tensor(False, device="npu"), torch.tensor(True, device="npu")],
             [false_fn, true_fn],
         ):
-            x = torch.randn(4, requires_grad=True, device="cuda")
+            x = torch.randn(4, requires_grad=True, device="npu")
             result = cond(pred, true_fn, false_fn, (x,))
             self.assertEqual(result, fn(x))
 
@@ -1384,14 +1392,14 @@ def forward(self, pred_1, x_1):
         return cond_outputs, cond_inputs
 
     @skipIfTorchDynamo("don't test compile on compile")
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
     @parametrize("compile_mode", ["compile_dynamic_shape"])
     @parametrize("scalar", [False])
     def test_cond_autograd_zeros_unused_branch_complex_compile_fail(
         self, compile_mode, scalar
     ):
-        device = torch.device("cuda")
+        device = torch.device("npu")
         cond_fct = compile_mode_helper(torch.cond, compile_mode)
 
         autograd = [False, True, True, True, True]
@@ -1436,26 +1444,26 @@ def forward(self, pred_1, x_1):
             cond_fct, pred_fn, true_fn, false_fn, operands
         )
 
-    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
-    def test_map_gpu(self):
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
+    def test_map_npu(self):
         def f(x, y):
             return x + y
 
-        xs = torch.ones(3, 2, 2, device="cuda")
-        y = torch.ones(2, device="cuda")
+        xs = torch.ones(3, 2, 2, device="npu")
+        y = torch.ones(2, device="npu")
         res = control_flow.map(f, xs, y)
         expected = _fake_map(f, xs, y)
         self.assertEqual(expected, res)
 
-    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
-    def test_while_loop_gpu(self):
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
+    def test_while_loop_npu(self):
         def cond_fn(x):
             return x.sum() < 10
 
         def body_fn(x):
             return (x + 1,)
 
-        x = torch.zeros(1, device="cuda")
+        x = torch.zeros(1, device="npu")
         res = while_loop(cond_fn, body_fn, (x,))
         expected = _fake_while_loop(cond_fn, body_fn, (x,))
         self.assertEqual(expected, res)
@@ -1660,10 +1668,10 @@ def forward(self, pred_1, x_1):
 
     # TODO: provide an implementation for all compile modes and re-enable all test
     @skipIfTorchDynamo("don't test compile on compile")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_scan_compile(self, reverse, compile_mode, device, autograd):
         def add2(x: torch.Tensor, y: torch.Tensor):
@@ -1771,10 +1779,10 @@ def forward(self, pred_1, x_1):
 
     # TODO: provide an implementation for all compile modes and re-enable all test
     @skipIfTorchDynamo("don't test compile on compile")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("npu")
     @parametrize(
         "dtype",
         [
@@ -1782,7 +1790,7 @@ def forward(self, pred_1, x_1):
             torch.float32,
             torch.int32,
             torch.int64,
-            torch.complex64,
+            torch.complex64,                                       # npu 涓嶆敮鎸?DT_COMPLEX64
         ],
     )
     def test_scan_dtype(self, reverse, compile_mode, device, dtype):
@@ -1843,10 +1851,10 @@ def forward(self, pred_1, x_1):
         )
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_scan_dim(self, reverse, compile_mode, device, autograd):
         import random
@@ -1887,10 +1895,10 @@ def forward(self, pred_1, x_1):
                     self.check_autograd(result, result_exp, (init, x))
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_scan_binary_operator(self, reverse, compile_mode, device, autograd):
         state_dim = 20
@@ -1949,10 +1957,10 @@ def forward(self, pred_1, x_1):
             self.assertEqual(grads, expected_grads)
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_scan_tuple(self, reverse, compile_mode, device, autograd):
         x = torch.randn(3, 2, 2, device=device, requires_grad=autograd)
@@ -2052,10 +2060,10 @@ def forward(self, pred_1, x_1):
         ):
             scan(fct_float_output, init, x, dim=0)
 
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_scan_complex_pytree(self, reverse, compile_mode, device, autograd):
         # Init and input have same pytree
@@ -2092,15 +2100,15 @@ def forward(self, pred_1, x_1):
     # TODO: Does not work because of the usage of vmap within associative_scan
     # The paT206899919 rameterization is commented out for the moment and the test is marked with expected fail
     # Fails with: AssertionError: scan is not an OpOverload
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     def test_scan_associative_scan(self):
         combine_mode = "generic"
         compile_mode_scan = "compile"
         compile_mode_associative_scan = "none"
         reverse = True
         reverse_associative_scan = True
-        device = torch.device("cuda")
+        device = torch.device("npu")
 
         scan_fct = compile_mode_helper(scan, compile_mode_scan)
         associative_scan_fct = compile_mode_helper(
@@ -2132,10 +2140,10 @@ def forward(self, pred_1, x_1):
 
     # TODO: provide an implementation for all compile modes and re-enable all test
     @skipIfTorchDynamo("don't test compile on compile")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_scan_downstream_scan_matmul(self, compile_mode, reverse, device, autograd):
         inp = torch.randn(3, 10, 2, device=device, requires_grad=autograd)
@@ -2171,10 +2179,10 @@ def forward(self, pred_1, x_1):
 
     # TODO: provide an implementation for all compile modes and re-enable all test
     @skipIfTorchDynamo("don't test compile on compile")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_scan_downstream_scan_scan_dim(
         self, compile_mode, reverse, device, autograd
@@ -2227,10 +2235,10 @@ def forward(self, pred_1, x_1):
             self.check_autograd(result, expected_result, (init, init2, inp))
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_scan_non_pointwise(self, reverse, compile_mode, device, autograd):
         scan_fct = compile_mode_helper(scan, compile_mode)
@@ -2257,9 +2265,9 @@ def forward(self, pred_1, x_1):
         if autograd:
             self.check_autograd(result, expected_result, (init, x))
 
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])     # torch.device("npu")
     def test_scan_compile_cnt(self, reverse, device):
         dim = 1
 
@@ -2571,10 +2579,10 @@ def forward(self, pred_1, x_1):
             scan_fct(no_carry, init, x, dim=dim)
 
     @skipIfTorchDynamo("don't test compile on compile")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_scan_init(self, reverse, compile_mode, device, autograd):
         scan_fct = compile_mode_helper(scan, compile_mode)
@@ -2681,9 +2689,9 @@ def forward(self, pred_1, x_1):
         if autograd:
             self.check_autograd(result, result_exp, (init, x))
 
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])      # torch.device("npu")
     def test_scan_init_wrong_pytree_complex(self, reverse, device):
         x = torch.randn(3, 2, 2, device=device)
         y = torch.randn(3, 2, 2, device=device)
@@ -2718,10 +2726,10 @@ def forward(self, pred_1, x_1):
             )
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_scan_init_pytree_complex(self, reverse, compile_mode, device, autograd):
         def fct_pointwise_different_output(x, y):
@@ -2903,7 +2911,7 @@ class GraphModule(torch.nn.Module):
 
     @skipIfTorchDynamo("Graph is not captured by backend if test with dynamo")
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager"])
     @parametrize("autograd", [False, True])
     def test_scan_closure_RNN(self, compile_mode, autograd):
@@ -2978,13 +2986,13 @@ class GraphModule(torch.nn.Module):
             self.assertEqual(add_input_grads, expected_add_input_grads)
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
     @parametrize(
         "partial_grad", ["xs", "init", "additional_inputs", "complex", "random"]
     )
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     def test_scan_closure_RNN_partial_autograd(
         self, reverse, compile_mode, partial_grad, device
     ):
@@ -3062,11 +3070,11 @@ class GraphModule(torch.nn.Module):
                     params,
                 )
 
-    @requires_cuda
+    # @requires_cuda
     @skipIfTorchDynamo("not a dynamo test")
     @unittest.skipIf(not SM70OrLater, "triton")
     @parametrize("layers", [1, 2, 3])
-    @parametrize("device", ["cpu", "cuda"])
+    @parametrize("device", ["cpu", "npu"])
     @torch._dynamo.config.patch(capture_scalar_outputs=True)
     def test_scan_multiple_layers_gradient(self, layers, device):
         import torch.nn as nn
@@ -3220,10 +3228,10 @@ class GraphModule(torch.nn.Module):
             )
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_scan_closure_combine_fn_with_no_grad_init_carries_unequal_grad(
         self, reverse, compile_mode, device, autograd
@@ -3260,10 +3268,10 @@ class GraphModule(torch.nn.Module):
             self.check_autograd(res_req_grad_flat, res_exp_req_grad_flat, (x, h2))
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_scan_closure_combine_fn_with_no_grad_init_carries_equal_grad(
         self, reverse, compile_mode, device, autograd
@@ -3300,10 +3308,10 @@ class GraphModule(torch.nn.Module):
             self.check_autograd(res_req_grad_flat, res_exp_req_grad_flat, (x, h2))
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_scan_closure_combine_fn_with_no_grad_for_out(
         self, reverse, compile_mode, device, autograd
@@ -3329,10 +3337,10 @@ class GraphModule(torch.nn.Module):
             self.check_autograd(result[0], result_exp[0], (x, h1, h2))
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_scan_closure_combine_fn_with_no_grad_additional_inputs_partial(
         self, reverse, compile_mode, device, autograd
@@ -3364,10 +3372,10 @@ class GraphModule(torch.nn.Module):
             self.check_autograd(result[1], result_exp[1], (h, x, W_ih, b_ih))
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_scan_closure_combine_fn_with_no_grad_additional_inputs_all(
         self, reverse, compile_mode, device, autograd
@@ -3401,10 +3409,10 @@ class GraphModule(torch.nn.Module):
             self.check_autograd(result[1], result_exp[1], (h, x))
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_scan_closure_combine_fn_carries_ys_same_grad(
         self, reverse, compile_mode, device, autograd
@@ -3438,10 +3446,10 @@ class GraphModule(torch.nn.Module):
             self.check_autograd(result[1], result_exp[1], (h, x))
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_scan_closure_nested(self, reverse, compile_mode, device, autograd):
         scan_fct = compile_mode_helper(scan, compile_mode)
@@ -3603,9 +3611,9 @@ def forward(self, L_init_ : torch.Tensor, L_xs_ : torch.Tensor):
     return (carry, out_1)""",  # noqa: B950
         )
 
-    @requires_cuda
+    # @requires_cuda
     def test_scan_input_mutation(self):
-        device = torch.device("cuda")
+        device = torch.device("npu")
 
         def fct_input_mutation(x, y):
             x.add_(1)
@@ -3623,9 +3631,9 @@ def forward(self, L_init_ : torch.Tensor, L_xs_ : torch.Tensor):
         ):
             scan(fct_input_mutation, init, x, dim=0)
 
-    @requires_cuda
+    # @requires_cuda
     def test_scan_input_carry_alias(self):
-        device = torch.device("cuda")
+        device = torch.device("npu")
 
         def fct_input_output_alias(x, y):
             return (x[0], x[1] + y[1]), (x[1] + y[1] + 1, x[1] + y[1] + 2)
@@ -3644,9 +3652,9 @@ def forward(self, L_init_ : torch.Tensor, L_xs_ : torch.Tensor):
         ):
             scan(fct_input_output_alias, init, inp, dim=0)
 
-    @requires_cuda
+    # @requires_cuda
     def test_scan_input_output_alias(self):
-        device = torch.device("cuda")
+        device = torch.device("npu")
 
         def fct_input_output_alias(x, y):
             return (x[0] + 1, x[1] + y[1]), (x[1], x[1] + y[1] + 2)
@@ -3666,9 +3674,9 @@ def forward(self, L_init_ : torch.Tensor, L_xs_ : torch.Tensor):
             scan(fct_input_output_alias, init, inp, dim=0)
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     def test_scan_carry_carry_alias(self):
-        device = torch.device("cuda")
+        device = torch.device("npu")
 
         def fct_carry_carry_alias(x, y):
             c = x[0] + y[1]
@@ -3689,9 +3697,9 @@ def forward(self, L_init_ : torch.Tensor, L_xs_ : torch.Tensor):
             scan(fct_carry_carry_alias, init, inp, dim=0)
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     def test_scan_carry_output_alias(self):
-        device = torch.device("cuda")
+        device = torch.device("npu")
 
         def fct_carry_output_alias(x, y):
             c = x[0] + y[1]
@@ -3874,11 +3882,11 @@ class AssociativeScanTests(TestCase):
         return kwargs_fake
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
@@ -3957,11 +3965,11 @@ class AssociativeScanTests(TestCase):
         self.assertEqual(result, results_torch)
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
@@ -4012,15 +4020,15 @@ class AssociativeScanTests(TestCase):
                     results_torch.append(op_pt(x, 0))
                 self.assertEqual(results, results_torch)
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @unittest.expectedFailure
     def test_associative_scan_dim_shape_failure(self, compile_mode, combine_mode):
         num_dims = [2]
         for num_dim in num_dims:
             shapes = [9 for _ in range(num_dim)]
             rnd_scan_dim = 0
-            x = torch.randn(*shapes, device=torch.device("cuda"))
+            x = torch.randn(*shapes, device=torch.device("npu"))
 
             kwargs = {
                 "dim": rnd_scan_dim,
@@ -4036,11 +4044,11 @@ class AssociativeScanTests(TestCase):
             )
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
@@ -4079,10 +4087,10 @@ class AssociativeScanTests(TestCase):
         )
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_associative_scan_expand_in_combine_fn(
         self, compile_mode, reverse, device, autograd
@@ -4108,10 +4116,10 @@ class AssociativeScanTests(TestCase):
         )
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_associative_scan_non_contiguous_tensor(
         self, compile_mode, reverse, device, autograd
@@ -4140,11 +4148,11 @@ class AssociativeScanTests(TestCase):
         )
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
@@ -4314,11 +4322,11 @@ class GraphModule(torch.nn.Module):
         )
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
@@ -4362,11 +4370,11 @@ class GraphModule(torch.nn.Module):
         )
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
@@ -4411,12 +4419,12 @@ class GraphModule(torch.nn.Module):
         )
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse_first", [False, True])
     @parametrize("same_direction", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
@@ -4477,14 +4485,14 @@ class GraphModule(torch.nn.Module):
     # TODO: Does not work because of the usage of vmap within associative_scan
     # TODO: Re-enable additional parameters again once this issues has been resolved
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @unittest.expectedFailure
     def test_associative_scan_nested(self):
         combine_mode = "pointwise"
         compile_mode = "eager"
         reverse_first = False
         same_direction = False
-        device = torch.device("cuda")
+        device = torch.device("npu")
 
         reverse_second = reverse_first if same_direction else not reverse_first
 
@@ -4525,11 +4533,11 @@ class GraphModule(torch.nn.Module):
         )
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("loop_type", ["for"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_associative_scan_loop_in_combine_fn(
         self, compile_mode, loop_type, reverse, device, autograd
@@ -4577,13 +4585,13 @@ class GraphModule(torch.nn.Module):
     # TODO: Does not work because of the usage of vmap within associative_scan
     # TODO: Re-enable additional parameters again once this issues has been resolved
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @unittest.expectedFailure
     def test_associative_scan_loop_in_combine_fn_failure(self):
         compile_mode = "none"
         loop_type = "while"
         reverse = False
-        device = torch.device("cuda")
+        device = torch.device("npu")
 
         def combine_fn(x, y):
             _cnt = torch.zeros_like(y[0, :])
@@ -4612,10 +4620,10 @@ class GraphModule(torch.nn.Module):
         )
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     # Skipping the combination of compile_mode=compile_dynamic_shape
     # as the current implementation does not support lifted arguments
@@ -4653,12 +4661,12 @@ class GraphModule(torch.nn.Module):
     # TODO: Does not work because of the usage of vmap within associative_scan
     # TODO: Re-enable additional parameters again once this issues has been resolved
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @unittest.expectedFailure
     def test_associative_scan_map_in_combine_fn(self):
         compile_mode = "none"
         reverse = False
-        device = torch.device("cuda")
+        device = torch.device("npu")
 
         def combine_fn(x, y):
             def body(x, y):
@@ -4685,10 +4693,10 @@ class GraphModule(torch.nn.Module):
         )
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_associative_scan_vmap_in_combine_fn(
         self, compile_mode, reverse, device, autograd
@@ -4719,10 +4727,10 @@ class GraphModule(torch.nn.Module):
         )
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     # Skipping the combination of associative_scan and device=cpu
     # as the current implementation of pointwise does only support CUDA device
@@ -4751,11 +4759,11 @@ class GraphModule(torch.nn.Module):
         )
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
@@ -4798,10 +4806,10 @@ class GraphModule(torch.nn.Module):
         )
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     def test_associative_scan_different_input_size(self, compile_mode, reverse, device):
         batch = 5
         hidden_dim = 3
@@ -4835,8 +4843,8 @@ class GraphModule(torch.nn.Module):
             inputs=elements,
         )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     def test_associative_scan_different_input_size_wrong_dim(self):
         batch = 5
         hidden_dim = 3
@@ -4844,17 +4852,17 @@ class GraphModule(torch.nn.Module):
         dstate = 7
 
         deltaA = torch.randn(
-            (batch, hidden_dim, length, dstate), device=torch.device("cuda")
+            (batch, hidden_dim, length, dstate), device=torch.device("npu")
         )
         deltaB_u = torch.randn(
-            (batch, hidden_dim, length, dstate), device=torch.device("cuda")
+            (batch, hidden_dim, length, dstate), device=torch.device("npu")
         )
-        C = torch.randn((batch, dstate, length), device=torch.device("cuda"))
+        C = torch.randn((batch, dstate, length), device=torch.device("npu"))
         x = torch.randn(
-            (batch, hidden_dim, length, dstate), device=torch.device("cuda")
+            (batch, hidden_dim, length, dstate), device=torch.device("npu")
         )
         y = torch.randn(
-            (batch, hidden_dim, length, dstate), device=torch.device("cuda")
+            (batch, hidden_dim, length, dstate), device=torch.device("npu")
         )
         elements = (x, deltaA, deltaB_u, C, y)
 
@@ -4874,7 +4882,7 @@ class GraphModule(torch.nn.Module):
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     # Skipping the combine_mode=pointwise
     # as the current implementation of associative_scan lowering
@@ -4927,7 +4935,7 @@ class GraphModule(torch.nn.Module):
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     # Skipping the combine_mode=pointwise
     # as the current implementation of associative_scan lowering
@@ -5004,7 +5012,7 @@ class GraphModule(torch.nn.Module):
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     # Skipping the combine_mode=pointwise
     # as the current implementation of associative_scan lowering
@@ -5044,7 +5052,7 @@ class GraphModule(torch.nn.Module):
     @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     def test_associative_scan_freevars_fct_generic(
         self, compile_mode, reverse, device, autograd
@@ -5085,7 +5093,7 @@ class GraphModule(torch.nn.Module):
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("autograd", [False, True])
     # Skipping the combine_mode=pointwise
     # as the current implementation of associative_scan lowering
@@ -5123,7 +5131,7 @@ class GraphModule(torch.nn.Module):
     @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("autograd", [False, True])
     # Skipping the combine_mode=pointwise
@@ -5175,12 +5183,12 @@ class GraphModule(torch.nn.Module):
             autograd_param=None if not autograd else (*pytree.tree_leaves(inp),),
         )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])      # torch.device("npu")
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
     # Skipping the combination of combine_mode=pointwise and compile_mode=compile_dynamic_shape
@@ -5235,12 +5243,12 @@ class GraphModule(torch.nn.Module):
                 autograd_param=inp,
             )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("npu")
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
     # Skipping the combination of combine_mode=pointwise and compile_mode=compile_dynamic_shape
@@ -5284,7 +5292,7 @@ class GraphModule(torch.nn.Module):
             autograd_param=inp[0:1],
         )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
+    # @unittest.skipIf(not SM70OrLater, "triton")
     def test_associative_scan_sparse_tensor(self):
         x = torch.tensor(
             [[[0.0, 0], [1.0, 2.0]], [[0.0, 0], [3.0, 4.0]], [[0.0, 0], [5.0, 6.0]]]
@@ -5298,10 +5306,10 @@ class GraphModule(torch.nn.Module):
                 get_scan_combine_fn("add", True), x, 0, combine_mode="generic"
             )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     def test_associative_scan_combine_fn_wrong_meta_in_combine_fn(self):
-        device = torch.device("cuda")
+        device = torch.device("npu")  # torch.device("npu")
         B, N, C, H, W = 3, 3, 2, 3, 3
         x = torch.randn(B, N, C, H, W, device=device)
 
@@ -5309,8 +5317,10 @@ class GraphModule(torch.nn.Module):
             return (x + y).to(torch.int64)
 
         def fct_wrong_device(x, y):
+            _device = 'cpu' if device.type == 'npu' else 'npu'
             return (x + y).to(
-                torch.device("cpu") if device.type == "cuda" else torch.device("cuda")
+                _device,
+                # torch.device("cpu") if device.type == "npu" else torch.device("npu")   # cuda
             )
 
         def fct_wrong_stride(x, y):
@@ -5323,7 +5333,7 @@ class GraphModule(torch.nn.Module):
             ):
                 associative_scan(fct, x, 0)
 
-    @unittest.skipIf(not SM70OrLater, "triton")
+    # @unittest.skipIf(not SM70OrLater, "triton")
     def test_associative_scan_wrong_pytree(self):
         def fct_wrong_pytree(x, y):
             return {
@@ -5343,10 +5353,10 @@ class GraphModule(torch.nn.Module):
         ):
             associative_scan(fct_wrong_pytree, inp, 0, combine_mode="generic")
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @requires_cuda
     def test_associative_scan_non_pointwise(self):
-        device = torch.device("cuda")
+        device = torch.device("npu")    # torch.device("npu")
         x = torch.randn(3, 10, 2, device=device)
         with self.assertRaisesRegex(
             # Should be:
@@ -5360,9 +5370,9 @@ class GraphModule(torch.nn.Module):
                 combine_mode="pointwise",
             )
 
-    @requires_cuda
+    # @requires_cuda
     def test_associative_scan_input_mutation(self):
-        device = torch.device("cuda")
+        device = torch.device("npu")    # torch.device("npu")
 
         def fct_input_mutation(x, y):
             x.add_(1)
@@ -5379,9 +5389,9 @@ class GraphModule(torch.nn.Module):
         ):
             associative_scan(fct_input_mutation, x, 0)
 
-    @requires_cuda
+    # @requires_cuda
     def test_associative_scan_input_output_alias(self):
-        device = torch.device("cuda")
+        device = torch.device("npu")    # torch.device("npu")
 
         def fct_input_output_alias(x, y):
             return x[0], x[1] + y[1]
@@ -5400,9 +5410,9 @@ class GraphModule(torch.nn.Module):
             associative_scan(fct_input_output_alias, inp, 0)
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    # @requires_cuda
     def test_associative_scan_output_output_alias(self):
-        device = torch.device("cuda")
+        device = torch.device("npu")
 
         def fct_output_output_alias(x, y):
             c = x[0] + y[1]
@@ -5674,13 +5684,13 @@ def forward(self, L_pred_ : torch.Tensor, L_x_ : torch.Tensor):
             "graph_capture_record_stream_reuse:True"
         )
         try:
-            predicate = torch.tensor(True, device="cuda")
+            predicate = torch.tensor(True, device="npu")
 
             def true_fn():
-                return torch.zeros(8, device="cuda"), torch.zeros(8, device="cuda")
+                return torch.zeros(8, device="npu"), torch.zeros(8, device="npu")
 
             def false_fn():
-                return torch.zeros(8, device="cuda"), torch.zeros(8, device="cuda")
+                return torch.zeros(8, device="npu"), torch.zeros(8, device="npu")
 
             g = torch.cuda.CUDAGraph()
             with self.assertRaisesRegex(
@@ -6162,7 +6172,7 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1, arg6_1, arg7_1
         self.assertEqual(graph_module(*example_inputs), f(*example_inputs))
 
         if TEST_CUDA_GRAPH_CONDITIONAL_NODES:
-            pred = torch.tensor(example_inputs[0].shape[0] == 1, device="cuda")
+            pred = torch.tensor(example_inputs[0].shape[0] == 1, device="npu")
             _check_compile_cudagraph_backend(self, f_, [torch.ones(4, 5).cuda(), pred])
             _check_compile_many_backends_with_cudagraph(
                 self, f_, [torch.ones(4, 5).cuda(), pred]
@@ -9263,7 +9273,7 @@ class GraphModule(torch.nn.Module):
                     torch.compile(fn)(f, x)
 
     @requires_cuda
-    @parametrize("device", ["cuda", "cpu"])
+    @parametrize("device", ["npu", "cpu"])
     def test_cond_input_mutation(self, device):
         predicate_true = torch.tensor(True, device=device)
         predicate_false = torch.tensor(False, device=device)
@@ -9722,7 +9732,7 @@ class TestAutoFunctionalizeControlFlow(TestCase):
 
     @requires_cuda
     @unittest.skipIf(not SM70OrLater, "triton")
-    @parametrize("device", ["cuda", "cpu"])
+    @parametrize("device", ["npu", "cpu"])
     @parametrize("dynamic", [True, False])
     def test_cond_auto_functionalize_input_mutation(self, device, dynamic):
         class M(torch.nn.Module):
@@ -9740,7 +9750,7 @@ class TestAutoFunctionalizeControlFlow(TestCase):
             torch.randn(3, 4, requires_grad=True),
         )
         fw_gm = self.check(M, (x, y), device, dynamic)
-        if not TEST_WITH_CROSSREF and not dynamic and device == "cuda":
+        if not TEST_WITH_CROSSREF and not dynamic and device == "npu":
             self.assertExpectedInline(
                 normalize_gm(fw_gm.print_readable(print_output=False)),
                 """\
@@ -9777,7 +9787,7 @@ class <lambda>(torch.nn.Module):
 
     @requires_cuda
     @unittest.skipIf(not SM70OrLater, "triton")
-    @parametrize("device", ["cuda", "cpu"])
+    @parametrize("device", ["npu", "cpu"])
     @parametrize("dynamic", [True, False])
     def test_cond_auto_functionalize_buffer_mutation(self, device, dynamic):
         class M(torch.nn.Module):
@@ -9799,7 +9809,7 @@ class <lambda>(torch.nn.Module):
 
         p, x = torch.tensor(True), torch.randn(1, requires_grad=True)
         fw_gm = self.check(M, (p, x), device, dynamic)
-        if not TEST_WITH_CROSSREF and not dynamic and device == "cuda":
+        if not TEST_WITH_CROSSREF and not dynamic and device == "npu":
             self.assertExpectedInline(
                 normalize_gm(fw_gm.print_readable(print_output=False)),
                 """\
@@ -9843,7 +9853,7 @@ class <lambda>(torch.nn.Module):
 
     @requires_cuda
     @unittest.skipIf(not SM70OrLater, "triton")
-    @parametrize("device", ["cuda", "cpu"])
+    @parametrize("device", ["npu", "cpu"])
     @parametrize("dynamic", [True, False])
     def test_cond_auto_functionalize_union_input_mutation(self, device, dynamic):
         class M(torch.nn.Module):
@@ -9869,7 +9879,7 @@ class <lambda>(torch.nn.Module):
             torch.randn(1, requires_grad=False),
         )
         fw_gm = self.check(M, (x, y), device, dynamic)
-        if not TEST_WITH_CROSSREF and not dynamic and device == "cuda":
+        if not TEST_WITH_CROSSREF and not dynamic and device == "npu":
             self.assertExpectedInline(
                 normalize_gm(fw_gm.print_readable(print_output=False)),
                 """\
@@ -10288,7 +10298,7 @@ class TestControlFlowNN(TestCase):
             grads = [p.grad for p in model.parameters()]
             return (output, loss, grads)
 
-        x = torch.randn(16, device="cuda")
+        x = torch.randn(16, device="npu")
 
         _check_compile_many_backends_with_cudagraph(self, autograd_test, [x])
         _check_compile_cudagraph_backend(self, autograd_test, [x])
@@ -10301,14 +10311,14 @@ class TestControlFlowNN(TestCase):
 class TestControlFlowAndRNG(TestCase):
     @parametrize("rng_func", ["custom_generator", "default_generator"])
     def test_rng_with_conditional_nodes_errors(self, rng_func):
-        pred = torch.tensor(True, device="cuda")
-        x = torch.ones(10, dtype=torch.float32, device="cuda")
+        pred = torch.tensor(True, device="npu")
+        x = torch.ones(10, dtype=torch.float32, device="npu")
 
         if rng_func == "custom_generator":
             self.skipTest(
                 "randn() currently does not work with a generator argument in dynamo."
             )
-            generator = torch.Generator("cuda")
+            generator = torch.Generator("npu")
 
             def custom_generator(x):
                 return x + torch.randn(
@@ -10334,8 +10344,8 @@ class TestControlFlowAndRNG(TestCase):
             compiled_func(pred, x)
 
     def test_rng_outside_conditional_nodes_does_not_error(self):
-        pred = torch.tensor(True, device="cuda")
-        x = torch.ones(10, dtype=torch.float32, device="cuda")
+        pred = torch.tensor(True, device="npu")
+        x = torch.ones(10, dtype=torch.float32, device="npu")
 
         def func(pred, x):
             y = torch.cond(pred, lambda t: 2 * t, lambda t: 3 * t, [x])
