diff --git a/test/functorch/test_control_flow.py b/test/functorch/test_control_flow.py
index 6ed0b5841b4..6fb504fbad8 100644
--- a/test/functorch/test_control_flow.py
+++ b/test/functorch/test_control_flow.py
@@ -1,9 +1,85 @@
 # Owner(s): ["module: functorch"]
 import contextlib
 import functools
+        device = torch.device("npu")    # torch.device("cuda")
+    # @requires_cuda
+        device = torch.device("npu")    # torch.device("cuda")
+    # @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])     # torch.device("cuda")
+    # @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+        device = torch.device("npu") # device = torch.device("cuda")
+    # @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+        device = torch.device("npu")
+    # @requires_cuda
+        device = torch.device("npu")
+    # @requires_cuda
+        device = torch.device("npu")  # device = torch.device("cuda")
+    # @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    #@requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])     # torch.device("cuda")
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])  # torch.device("cuda")
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("cuda")
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
 import unittest
 
 import torch
+
+CUDA_VERSION = 11080
+torch.version.cuda = '11.8'
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+# from torch_npu.contrib import transfer_to_npu
+import torch_npu
 import torch.utils._pytree as pytree
 from functorch.experimental import control_flow
 from functorch.experimental.control_flow import cond
@@ -31,7 +107,7 @@ from torch._subclasses.functional_tensor import (
     PythonFunctionalizeAPI,
 )
 from torch.fx.experimental.proxy_tensor import make_fx
-from torch.testing._internal.common_cuda import SM70OrLater
+
 from torch.testing._internal.common_quantization import skipIfNoDynamoSupport
 from torch.testing._internal.common_utils import (
     decorateIf,
@@ -556,19 +632,20 @@ class TestControlFlow(TestCase):
             return x.cos()
 
         x = torch.randn(4)
+            x = torch.randn(4, requires_grad=True, device="npu")
+            [torch.tensor(False, device="npu"), torch.tensor(True, device="npu")],
         result = cond(False, true_fn, false_fn, [x])
         self.assertEqual(result, torch.cos(x))
-
-    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
-    def test_cond_gpu(self):
         def true_fn(x):
+        pred = torch.tensor(False,device='npu')
+        x = torch.randn(4,device='npu')
+    def test_cond_npu(self):
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
             return x.sin()
 
         def false_fn(x):
             return x.cos()
 
-        x = torch.randn(4, device="cuda")
-        pred = torch.tensor(False, device="cuda")
         result = cond(pred, true_fn, false_fn, [x])
         self.assertEqual(result, torch.cos(x))
 
@@ -614,6 +691,8 @@ def forward(self, pred_1, x_1):
     return (getitem_1,)""",  # noqa: B950
         )
 
+    def test_cond_autograd_npu(self):
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
     def test_cond_autograd_complex(self):
         def true_fn(x):
             return torch.abs((x**2).sin())
@@ -1300,6 +1379,8 @@ def forward(self, pred_1, x_1):
     @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
     def test_cond_autograd_gpu(self):
         def true_fn(x):
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
             return x.sin()
 
         def false_fn(x):
@@ -1321,6 +1402,13 @@ def forward(self, pred_1, x_1):
     def _test_cond_autograd(self, cond_fct, pred_fn, true_fn, false_fn, operands):
         from torch.fx.passes.shape_prop import _extract_tensor_metadata, TensorMetadata
 
+        x = torch.zeros(1, device="npu")
+    def test_while_loop_npu(self):
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
+        y = torch.ones(2, device="npu")
+        xs = torch.ones(3, 2, 2, device="npu")
+    def test_map_npu(self):
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
         # This is a helper function that extracts the metadata from the tensor and
         # sets the requires_grad flag to false. This is needed as we compare the
         # metadata of the operands and the gradients
@@ -1336,6 +1424,8 @@ def forward(self, pred_1, x_1):
                 metadata.qparams,
             )
             return metadata
+    @parametrize("device", ["cpu", "npu"])
+    # @requires_cuda
 
         # Comparison of FWD path
         cond_outputs = cond_fct(pred_fn(*operands), true_fn, false_fn, operands)
@@ -1384,8 +1474,9 @@ def forward(self, pred_1, x_1):
         return cond_outputs, cond_inputs
 
     @skipIfTorchDynamo("don't test compile on compile")
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
+        device = torch.device("npu")
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
+    # @unittest.skipIf(not SM70OrLater, "triton")
     @parametrize("compile_mode", ["compile_dynamic_shape"])
     @parametrize("scalar", [False])
     def test_cond_autograd_zeros_unused_branch_complex_compile_fail(
@@ -1695,7 +1786,13 @@ def forward(self, pred_1, x_1):
             if autograd:
                 self.check_autograd(result, result_exp, (init, x))
 
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])      # torch.device("cuda")
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])     # torch.device("npu")
+    # @requires_cuda
         # Jax Examples
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
         x = torch.arange(0, 4, device=device, dtype=torch.int64)
         init = torch.zeros(1, device=device, dtype=torch.int64)
         cumsum1 = scan_fct(
@@ -1750,6 +1847,7 @@ def forward(self, pred_1, x_1):
             0, 5, device=device, dtype=torch.float32, requires_grad=autograd
         )
         init = torch.ones(1, device=device, dtype=torch.float32, requires_grad=autograd)
+            torch.complex64,                                       # npu 不支持 DT_COMPLEX64
         result = scan_fct(
             get_scan_combine_fn("div", False),
             init,
@@ -1949,10 +2047,10 @@ def forward(self, pred_1, x_1):
             self.assertEqual(grads, expected_grads)
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
     @parametrize("autograd", [False, True])
     def test_scan_tuple(self, reverse, compile_mode, device, autograd):
         x = torch.randn(3, 2, 2, device=device, requires_grad=autograd)
@@ -2090,17 +2188,26 @@ def forward(self, pred_1, x_1):
             self.check_autograd(result, expected_result, (init, inp))
 
     # TODO: Does not work because of the usage of vmap within associative_scan
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+        device = torch.device("npu")
+    # @requires_cuda
+        device = torch.device("npu")
+    # @requires_cuda
+        device = torch.device("npu")
+    # @requires_cuda
     # The paT206899919 rameterization is commented out for the moment and the test is marked with expected fail
     # Fails with: AssertionError: scan is not an OpOverload
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+        device = torch.device("npu")
+    # @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
     def test_scan_associative_scan(self):
         combine_mode = "generic"
         compile_mode_scan = "compile"
         compile_mode_associative_scan = "none"
         reverse = True
         reverse_associative_scan = True
-        device = torch.device("cuda")
+        device = torch.device("npu")
 
         scan_fct = compile_mode_helper(scan, compile_mode_scan)
         associative_scan_fct = compile_mode_helper(
@@ -2571,10 +2678,10 @@ def forward(self, pred_1, x_1):
             scan_fct(no_carry, init, x, dim=dim)
 
     @skipIfTorchDynamo("don't test compile on compile")
-    @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
     @parametrize("autograd", [False, True])
     def test_scan_init(self, reverse, compile_mode, device, autograd):
         scan_fct = compile_mode_helper(scan, compile_mode)
@@ -2978,7 +3085,8 @@ class GraphModule(torch.nn.Module):
             self.assertEqual(add_input_grads, expected_add_input_grads)
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
     @parametrize(
@@ -3260,10 +3368,12 @@ class GraphModule(torch.nn.Module):
             self.check_autograd(res_req_grad_flat, res_exp_req_grad_flat, (x, h2))
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
     @parametrize("autograd", [False, True])
     def test_scan_closure_combine_fn_with_no_grad_init_carries_equal_grad(
         self, reverse, compile_mode, device, autograd
@@ -3329,10 +3439,10 @@ class GraphModule(torch.nn.Module):
             self.check_autograd(result[0], result_exp[0], (x, h1, h2))
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
     @parametrize("autograd", [False, True])
     def test_scan_closure_combine_fn_with_no_grad_additional_inputs_partial(
         self, reverse, compile_mode, device, autograd
@@ -3364,10 +3474,14 @@ class GraphModule(torch.nn.Module):
             self.check_autograd(result[1], result_exp[1], (h, x, W_ih, b_ih))
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
     @parametrize("autograd", [False, True])
     def test_scan_closure_combine_fn_with_no_grad_additional_inputs_all(
         self, reverse, compile_mode, device, autograd
@@ -3603,9 +3717,9 @@ def forward(self, L_init_ : torch.Tensor, L_xs_ : torch.Tensor):
     return (carry, out_1)""",  # noqa: B950
         )
 
-    @requires_cuda
+        device = torch.device("npu")
+    # @requires_cuda
     def test_scan_input_mutation(self):
-        device = torch.device("cuda")
 
         def fct_input_mutation(x, y):
             x.add_(1)
@@ -3644,9 +3758,9 @@ def forward(self, L_init_ : torch.Tensor, L_xs_ : torch.Tensor):
         ):
             scan(fct_input_output_alias, init, inp, dim=0)
 
-    @requires_cuda
     def test_scan_input_output_alias(self):
-        device = torch.device("cuda")
+        device = torch.device("npu")
+    # @requires_cuda
 
         def fct_input_output_alias(x, y):
             return (x[0] + 1, x[1] + y[1]), (x[1], x[1] + y[1] + 2)
@@ -3689,9 +3803,9 @@ def forward(self, L_init_ : torch.Tensor, L_xs_ : torch.Tensor):
             scan(fct_carry_carry_alias, init, inp, dim=0)
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+        device = torch.device("npu")
+    # @requires_cuda
     def test_scan_carry_output_alias(self):
-        device = torch.device("cuda")
 
         def fct_carry_output_alias(x, y):
             c = x[0] + y[1]
@@ -3874,12 +3988,16 @@ class AssociativeScanTests(TestCase):
         return kwargs_fake
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
     @parametrize("autograd", [False, True])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
     # Skipping the combination of combine_mode=pointwise and compile_mode=compile_dynamic_shape
@@ -3930,6 +4048,12 @@ class AssociativeScanTests(TestCase):
                 results_torch.append(op_pt(x, 0))
             self.assertEqual(results, results_torch)
 
+            x = torch.randn(*shapes, device=torch.device("npu"))
+    # @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
+            x = torch.randn(*shapes, device=torch.device("npu"))    # torch.device("cuda")
+    # @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
         # Jax Examples
         x = torch.arange(
             0, 4, device=device, dtype=torch.float32, requires_grad=autograd
@@ -3957,11 +4081,11 @@ class AssociativeScanTests(TestCase):
         self.assertEqual(result, results_torch)
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])
+    # @requires_cuda
     @parametrize("reverse", [False, True])
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("combine_mode", ["pointwise", "generic"])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
     @parametrize("autograd", [False, True])
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
@@ -4077,12 +4201,13 @@ class AssociativeScanTests(TestCase):
             inputs=inp,
             autograd_param=None if not autograd else inp,
         )
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("cuda")
+    @unittest.skipIf(not torch_npu.npu.is_available(), "Test requires NPU.")
+    # @unittest.skipIf(not SM70OrLater, "triton")
 
-    @unittest.skipIf(not SM70OrLater, "triton")
     @requires_cuda
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
     @parametrize("autograd", [False, True])
     def test_associative_scan_expand_in_combine_fn(
         self, compile_mode, reverse, device, autograd
@@ -4796,9 +4921,9 @@ class GraphModule(torch.nn.Module):
             inputs=elements,
             autograd_param=None if not autograd else elements,
         )
+    # @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
     @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
@@ -4809,6 +4934,16 @@ class GraphModule(torch.nn.Module):
         dstate = 7
 
         deltaA = torch.randn(
+            (batch, hidden_dim, length, dstate), device=torch.device("npu")
+            (batch, hidden_dim, length, dstate), device=torch.device("npu")
+        C = torch.randn((batch, dstate, length), device=torch.device("npu"))
+            (batch, hidden_dim, length, dstate), device=torch.device("npu")
+            (batch, hidden_dim, length, dstate), device=torch.device("npu")
+            (batch, hidden_dim, length, dstate), device=torch.device("npu")
+            (batch, hidden_dim, length, dstate), device=torch.device("npu")
+        C = torch.randn((batch, dstate, length), device=torch.device("npu"))
+            (batch, hidden_dim, length, dstate), device=torch.device("npu")
+            (batch, hidden_dim, length, dstate), device=torch.device("npu") # cuda
             (batch, hidden_dim, length, dstate), requires_grad=True, device=device
         )
         deltaB_u = torch.randn(
@@ -4869,6 +5004,8 @@ class GraphModule(torch.nn.Module):
                 combine_mode="pointwise",
             )
 
+        device = torch.device("npu")    # torch.device("cuda")
+    # @requires_cuda
     @unittest.skipIf(not SM70OrLater, "triton")
     @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
@@ -5175,12 +5312,12 @@ class GraphModule(torch.nn.Module):
             autograd_param=None if not autograd else (*pytree.tree_leaves(inp),),
         )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])      # torch.device("cuda")
+    # @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
     # Skipping the combination of combine_mode=pointwise and compile_mode=compile_dynamic_shape
@@ -5235,12 +5372,12 @@ class GraphModule(torch.nn.Module):
                 autograd_param=inp,
             )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+    @parametrize("device", [torch.device("cpu"), torch.device("npu")])    # torch.device("cuda")
+    # @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
     @parametrize("combine_mode", ["pointwise", "generic"])
     @parametrize("compile_mode", ["none", "eager", "compile", "compile_dynamic_shape"])
     @parametrize("reverse", [False, True])
-    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
     # Skipping the combination of combine_mode=pointwise and device=cpu
     # as the current implementation of pointwise does only support CUDA device
     # Skipping the combination of combine_mode=pointwise and compile_mode=compile_dynamic_shape
@@ -5284,7 +5421,7 @@ class GraphModule(torch.nn.Module):
             autograd_param=inp[0:1],
         )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
+    # @unittest.skipIf(not SM70OrLater, "triton")
     def test_associative_scan_sparse_tensor(self):
         x = torch.tensor(
             [[[0.0, 0], [1.0, 2.0]], [[0.0, 0], [3.0, 4.0]], [[0.0, 0], [5.0, 6.0]]]
@@ -5298,10 +5435,10 @@ class GraphModule(torch.nn.Module):
                 get_scan_combine_fn("add", True), x, 0, combine_mode="generic"
             )
 
-    @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+        device = torch.device("npu")  # torch.device("cuda")
+    # @requires_cuda
+    # @unittest.skipIf(not SM70OrLater, "triton")
     def test_associative_scan_combine_fn_wrong_meta_in_combine_fn(self):
-        device = torch.device("cuda")
         B, N, C, H, W = 3, 3, 2, 3, 3
         x = torch.randn(B, N, C, H, W, device=device)
 
@@ -5309,8 +5446,10 @@ class GraphModule(torch.nn.Module):
             return (x + y).to(torch.int64)
 
         def fct_wrong_device(x, y):
+                # torch.device("cpu") if device.type == "npu" else torch.device("npu")   # cuda
+                _device,
+            _device = 'cpu' if device.type == 'npu' else 'npu'
             return (x + y).to(
-                torch.device("cpu") if device.type == "cuda" else torch.device("cuda")
             )
 
         def fct_wrong_stride(x, y):
@@ -5400,9 +5539,9 @@ class GraphModule(torch.nn.Module):
             associative_scan(fct_input_output_alias, inp, 0)
 
     @unittest.skipIf(not SM70OrLater, "triton")
-    @requires_cuda
+        device = torch.device("npu")
+    # @requires_cuda
     def test_associative_scan_output_output_alias(self):
-        device = torch.device("cuda")
 
         def fct_output_output_alias(x, y):
             c = x[0] + y[1]
