diff --git a/test/nn/test_dropout.py b/test/nn/test_dropout.py
index d21dab8..37f0f0f 100644
--- a/test/nn/test_dropout.py
+++ b/test/nn/test_dropout.py
@@ -5,6 +5,8 @@ import unittest
 from itertools import product
 
 import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 import torch.nn as nn
 import torch.nn.functional as F
 from torch.testing._internal.common_cuda import TEST_CUDA
