diff --git a/test/nn/test_init.py b/test/nn/test_init.py
index 40df5b1..2e3b1cf 100644
--- a/test/nn/test_init.py
+++ b/test/nn/test_init.py
@@ -7,6 +7,8 @@ from functools import reduce
 from operator import mul
 
 import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 import torch.nn.functional as F
 import torch.nn.init as init
 from torch.testing._internal.common_utils import (
@@ -319,7 +321,7 @@ class TestNNInit(TestCase):
                 init.xavier_normal_(tensor)
 
     @unittest.skipIf(not TEST_SCIPY, "Scipy not found.")
-    @slowTest
+    # @slowTest
     def test_xavier_uniform(self):
         for use_gain in [True, False]:
             for dims in [2, 4]:
