diff --git a/test/nn/test_lazy_modules.py b/test/nn/test_lazy_modules.py
index 163063c..8b74423 100644
--- a/test/nn/test_lazy_modules.py
+++ b/test/nn/test_lazy_modules.py
@@ -3,6 +3,8 @@ import pickle
 import unittest
 
 import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 import torch.nn as nn
 from torch.nn import Buffer, Parameter
 from torch.nn.parameter import UninitializedBuffer, UninitializedParameter
@@ -748,7 +750,8 @@ class TestLazyModules(TestCase):
         module.register_parameter("test_param", UninitializedParameter())
         module.to(device)
         module.test_param.materialize(10)
-        self.assertTrue(module.test_param.device.type == device)
+        # self.assertTrue(module.test_param.device.type == device)
+        self.assertTrue(module.test_param.device.type == "npu")
 
     @suppress_warnings
     def test_chained_initialization(self):
