diff --git a/test/distributed/checkpoint/fsdp/test_fsdp_dsd.py b/test/distributed/checkpoint/fsdp/test_fsdp_dsd.py
index 5f5ab1ebd39..1ade80d960e 100644
--- a/test/distributed/checkpoint/fsdp/test_fsdp_dsd.py
+++ b/test/distributed/checkpoint/fsdp/test_fsdp_dsd.py
@@ -210,6 +210,11 @@ class TestFullyShardWithDistributedStateDict(FSDPTest):
         # init device mesh
         dp_size = 2
         global_mesh = init_device_mesh(
+                "npu",
+                "npu", (self.world_size,), mesh_dim_names=("tp",)
+            "npu", (dp_size, self.world_size // dp_size), mesh_dim_names=("dp", "tp")
+            "npu", (self.world_size,), mesh_dim_names=("tp",)
+            "npu",
             device_type,
             (dp_size, self.world_size // dp_size),
             mesh_dim_names=("dp", "tp"),
