diff --git a/test/distributed/checkpoint/fsdp/test_fsdp_dsd.py b/test/distributed/checkpoint/fsdp/test_fsdp_dsd.py
index fac49dd..8aea449 100644
--- a/test/distributed/checkpoint/fsdp/test_fsdp_dsd.py
+++ b/test/distributed/checkpoint/fsdp/test_fsdp_dsd.py
@@ -1,3 +1,8 @@
+import torch_npu.testing
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["oncall: distributed"]
 
 import contextlib
@@ -206,7 +211,7 @@ class TestFullyShardWithDistributedStateDict(FSDPTest):
         # init device mesh
         dp_size = 2
         global_mesh = init_device_mesh(
-            "cuda",
+            "npu",
             (dp_size, self.world_size // dp_size),
             mesh_dim_names=("dp", "tp"),
         )
@@ -340,10 +345,10 @@ class TestFullyShardWithDistributedStateDict(FSDPTest):
         # init device mesh
         dp_size = 2
         global_mesh_1d = init_device_mesh(
-            "cuda", (self.world_size,), mesh_dim_names=("tp",)
+            "npu", (self.world_size,), mesh_dim_names=("tp",)
         )
         global_mesh_2d = init_device_mesh(
-            "cuda", (dp_size, self.world_size // dp_size), mesh_dim_names=("dp", "tp")
+            "npu", (dp_size, self.world_size // dp_size), mesh_dim_names=("dp", "tp")
         )
         dp_mesh, tp_mesh = global_mesh_2d["dp"], global_mesh_2d["tp"]
 
@@ -494,10 +499,10 @@ class TestFullyShardWithDistributedStateDict(FSDPTest):
             # init device mesh
             dp_size = 2
             global_mesh_1d = init_device_mesh(
-                "cuda", (self.world_size,), mesh_dim_names=("tp",)
+                "npu", (self.world_size,), mesh_dim_names=("tp",)
             )
             global_mesh_2d = init_device_mesh(
-                "cuda",
+                "npu",
                 (dp_size, self.world_size // dp_size),
                 mesh_dim_names=("dp", "tp"),
             )
