diff --git a/test/distributed/tensor/test_redistribute.py b/test/distributed/tensor/test_redistribute.py
index fc019b53109..1174b97f1af 100644
--- a/test/distributed/tensor/test_redistribute.py
+++ b/test/distributed/tensor/test_redistribute.py
@@ -59,11 +59,12 @@ from torch.testing._internal.distributed._tensor.common_dtensor import (
     map_local_tensor_for_rank,
     patched_distribute_tensor as _distribute_tensor,
     redistribute,
+from torch_npu.testing.common_distributed import with_comms
+    # with_comms,
 )
 from torch.testing._internal.distributed.fake_pg import FakeStore
 from torch.utils._debug_mode import DebugMode
 
-
 funcol = torch.ops.c10d_functional
 
 
