diff --git a/test/distributed/tensor/test_dtensor_compile.py b/test/distributed/tensor/test_dtensor_compile.py
index cd91af3f040..23c73d9ae6e 100644
--- a/test/distributed/tensor/test_dtensor_compile.py
+++ b/test/distributed/tensor/test_dtensor_compile.py
@@ -61,7 +61,12 @@ from torch.testing._internal.common_utils import (
 from torch.testing._internal.distributed._tensor.common_dtensor import (
     DTensorTestBase,
     MLPModule,
-    with_comms,
+from torch_npu.testing.common_distributed import with_comms
+    # with_comms,
+os.environ['TORCH_NPU_DEVICE_CAPABILITY']='9.0'
+import os
+from torch_npu.testing.common_distributed import with_comms
+    # with_comms,
 )
 from torch.testing._internal.distributed.fake_pg import FakeStore
 from torch.testing._internal.inductor_utils import HAS_GPU
