diff --git a/test/distributed/tensor/test_view_ops.py b/test/distributed/tensor/test_view_ops.py
index e709458fdb9..94fe64f78a0 100644
--- a/test/distributed/tensor/test_view_ops.py
+++ b/test/distributed/tensor/test_view_ops.py
@@ -37,11 +37,12 @@ from torch.testing._internal.common_utils import run_tests
 from torch.testing._internal.distributed._tensor.common_dtensor import (
     create_local_tensor_test_class,
     DTensorContinuousTestBase,
+from torch_npu.testing.common_distributed import with_comms
+    # with_comms,
     LocalDTensorContinuousTestBase,
 )
 from torch.utils import _pytree as pytree
 
-
 class TestViewOps(DTensorContinuousTestBase):
     world_size = 6
 
