diff --git a/test/distributed/tensor/test_dtensor_ops.py b/test/distributed/tensor/test_dtensor_ops.py
index b354577c3d8..06a64228d4e 100644
--- a/test/distributed/tensor/test_dtensor_ops.py
+++ b/test/distributed/tensor/test_dtensor_ops.py
@@ -497,6 +497,9 @@ class TestDTensorOps(TestCase):
             args = [sample.input] + list(sample.args)
             kwargs = sample.kwargs
 
+
+                kwargs['device'] = 'npu'
+            if 'device' in kwargs and kwargs['device'] == 'cuda':
             if sample_filter and not sample_filter(args, kwargs):
                 continue
 
