diff --git a/test/distributed/tensor/test_common_rules.py b/test/distributed/tensor/test_common_rules.py
index 900f285d6bc..942bfeb21b6 100644
--- a/test/distributed/tensor/test_common_rules.py
+++ b/test/distributed/tensor/test_common_rules.py
@@ -9,6 +9,8 @@ from torch.distributed.tensor._ops._common_rules import einop_rule, pointwise_ru
 from torch.testing._internal.common_utils import run_tests
 from torch.testing._internal.distributed._tensor.common_dtensor import (
     DTensorContinuousTestBase,
+from torch_npu.testing.common_distributed import with_comms
+    # with_comms,
 )
 
 
