diff --git a/test/distributed/tensor/test_utils.py b/test/distributed/tensor/test_utils.py
index a9798f9..efc2d1b 100644
--- a/test/distributed/tensor/test_utils.py
+++ b/test/distributed/tensor/test_utils.py
@@ -1,3 +1,8 @@
+import torch_npu.testing
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["oncall: distributed"]
 
 import itertools
@@ -12,8 +17,9 @@ from torch.distributed.tensor.placement_types import _StridedShard, Replicate, S
 from torch.testing._internal.common_utils import run_tests
 from torch.testing._internal.distributed._tensor.common_dtensor import (
     DTensorTestBase,
-    with_comms,
+    # with_comms,
 )
+from torch_npu.testing.common_distributed import with_comms
 
 
 c10d_functional = torch.ops.c10d_functional
