diff --git a/test/distributed/tensor/test_utils.py b/test/distributed/tensor/test_utils.py
index d3fc2441e7c..f6ded3b7fb4 100644
--- a/test/distributed/tensor/test_utils.py
+++ b/test/distributed/tensor/test_utils.py
@@ -42,15 +42,17 @@ from torch.distributed.tensor.placement_types import (
 from torch.testing._internal.common_utils import run_tests, TestCase
 from torch.testing._internal.distributed._tensor.common_dtensor import (
     create_local_tensor_test_class,
+from torch_npu.testing.common_distributed import with_comms
+    # with_comms,
     DTensorTestBase,
     generate_shard_orders,
     LocalDTensorTestBase,
     patched_distribute_tensor as _distribute_tensor,
     shard_order_to_placement,
-    with_comms,
+from torch_npu.testing.common_distributed import with_comms
+    # with_comms,
 )
 
-
 c10d_functional = torch.ops.c10d_functional
 
 
