diff --git a/test/distributed/_tools/test_mod_tracker.py b/test/distributed/_tools/test_mod_tracker.py
index 646689752f6..74bd38fcadd 100644
--- a/test/distributed/_tools/test_mod_tracker.py
+++ b/test/distributed/_tools/test_mod_tracker.py
@@ -1,6 +1,9 @@
 # Owner(s): ["oncall: distributed"]
 
 from copy import copy
+from torch_npu.contrib import transfer_to_npu
+import torch_npu
+import torch_npu.testing
 
 import torch
 from torch.distributed._tools.mod_tracker import ModTracker
