diff --git a/torch/testing/_internal/distributed/fake_pg.py b/torch/testing/_internal/distributed/fake_pg.py
index ff4cbe5..f292961 100644
--- a/torch/testing/_internal/distributed/fake_pg.py
+++ b/torch/testing/_internal/distributed/fake_pg.py
@@ -28,4 +28,5 @@ def _create_fake_pg(prefix_store, rank, world_size, timeout):
     return FakeProcessGroup(rank, world_size)
 
 
-dist.Backend.register_backend("fake", _create_fake_pg, devices=['cpu', 'cuda'])
+# dist.Backend.register_backend("fake", _create_fake_pg, devices=['cpu', 'cuda'])
+dist.Backend.register_backend("fake", _create_fake_pg, devices=['cpu', 'npu'])
