diff --git a/test/distributed/test_serialization.py b/test/distributed/test_serialization.py
index e0feb55..6c9e174 100644
--- a/test/distributed/test_serialization.py
+++ b/test/distributed/test_serialization.py
@@ -1,3 +1,8 @@
+# import torch_npu.testing
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["oncall: distributed"]
 
 import os
@@ -153,7 +158,7 @@ class TestSerialization(TestCase):
 
     @requires_cuda
     def test_cuda(self) -> None:
-        device = torch.device("cuda:0")
+        device = torch.device("npu:0")
 
         tensor = torch.tensor(42, dtype=torch.float, device=device)
         state_dict = {"scalar": tensor}
