diff --git a/test/distributed/test_data_parallel.py b/test/distributed/test_data_parallel.py
index 26f64df..ab38e86 100644
--- a/test/distributed/test_data_parallel.py
+++ b/test/distributed/test_data_parallel.py
@@ -1,3 +1,8 @@
+import torch_npu.testing
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["oncall: distributed"]
 
 import contextlib
@@ -27,9 +32,11 @@ from torch.testing._internal.common_utils import (
     skip_but_pass_in_sandcastle_if,
     TestCase,
 )
-
+TEST_MULTIGPU = True
+TEST_CUDA = True
 
 NO_NCCL = not hasattr(torch.distributed, "ProcessGroupNCCL")
+NO_NCCL = False
 
 # batched grad doesn't support data parallel
 gradcheck = functools.partial(gradcheck, check_batched_grad=False)
