diff --git a/test/distributed/test_functional_api.py b/test/distributed/test_functional_api.py
index b31fdeb..7a3a67d 100644
--- a/test/distributed/test_functional_api.py
+++ b/test/distributed/test_functional_api.py
@@ -1,3 +1,8 @@
+import torch_npu.testing
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["oncall: distributed"]
 
 import sys
@@ -15,6 +20,7 @@ from torch.testing import FileCheck
 from torch.testing._internal.common_device_type import instantiate_device_type_tests
 from torch.testing._internal.distributed.fake_pg import FakeStore
 from torch.testing._internal.inductor_utils import HAS_GPU
+HAS_GPU = True
 
 
 if not dist.is_available():
@@ -59,7 +65,7 @@ from torch.testing._internal.common_utils import (
 #     devices.append("new_device")
 #     DEVICE = "new_device"
 
-DEVICE = "cuda"
+DEVICE = "npu"
 devices = ["cpu"]
 if TEST_HPU:
     devices.append("hpu")
