diff --git a/test/distributed/test_pg_wrapper.py b/test/distributed/test_pg_wrapper.py
index d7e59f1..c35f4b9 100644
--- a/test/distributed/test_pg_wrapper.py
+++ b/test/distributed/test_pg_wrapper.py
@@ -1,3 +1,8 @@
+import torch_npu.testing
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["oncall: distributed"]
 
 import os
@@ -44,9 +49,9 @@ class AbstractProcessGroupWrapperTest(MultiProcessTestCase):
                 f"Did not find shapes {list(tensor.shape)} in error {err}",
             )
             # For CUDA, only assert on device type, not index
-            if "cuda" in str(tensor.device):
+            if "npu" in str(tensor.device):
                 self.assertTrue(
-                    "cuda" in err, f"Did not find cuda device in error {err}"
+                    "npu" in err, f"Did not find cuda device in error {err}"
                 )
             else:
                 self.assertTrue(
