diff --git a/test/distributed/_composable/fsdp/test_fully_shard_comm.py b/test/distributed/_composable/fsdp/test_fully_shard_comm.py
index 0d9daab6be7..367dc799a1b 100644
--- a/test/distributed/_composable/fsdp/test_fully_shard_comm.py
+++ b/test/distributed/_composable/fsdp/test_fully_shard_comm.py
@@ -78,9 +78,9 @@ from torch.testing._internal.distributed._tensor.common_dtensor import (
     Transformer,
     TransformerBlock,
 )
+TEST_CUDA = True
 from torch.testing._internal.inductor_utils import skipCUDAIf
 
-
 c10d_ops = torch.ops.c10d
 
 # For recording FSDP events like unshard or post-backward
@@ -120,6 +120,7 @@ class TestFullyShardCollectiveOps(FSDPTestMultiThread):
         torch.manual_seed(42)
         orig_params = [
             nn.Parameter(torch.randn(size, device=self.device)) for size in param_sizes
+        ref_model = copy.deepcopy(model).npu()
         ]
         # Since seed is per process, not per thread, we broadcast to ensure the
         # same original parameters across ranks
@@ -1292,6 +1293,7 @@ class TestFullyShardPrefetch(FSDPTest):
         fully_shard(model)
         optim = torch.optim.AdamW(model.parameters(), lr=1e-2)
 
+        inp = torch.randint(0, model_args.vocab_size, (2, 16), device="npu")
         events: list[EventType] = []
         unshard_with_record = self._get_unshard_with_record(
             FSDPParamGroup.unshard, events
