diff --git a/test/distributed/_composable/fsdp/test_fully_shard_training.py b/test/distributed/_composable/fsdp/test_fully_shard_training.py
index cb61f61a2d9..c6a909d3d75 100644
--- a/test/distributed/_composable/fsdp/test_fully_shard_training.py
+++ b/test/distributed/_composable/fsdp/test_fully_shard_training.py
@@ -70,7 +70,7 @@ from torch.testing._internal.distributed._tensor.common_dtensor import (
     Transformer,
     TransformerBlock,
 )
-
+TEST_CUDA = True
 
 c10d_ops = torch.ops.c10d
 funcol = torch.ops.c10d_functional
@@ -260,6 +260,7 @@ class TestFullyShardCastAfterInit(FSDPTestMultiThread):
         optim = torch.optim.Adam(model.parameters(), lr=1e-2, foreach=True)
         check_sharded_parity(self, ref_model, model)
         torch.manual_seed(42 + self.rank + 1)
+        device = torch.device("npu")
         inp = torch.randn((2, mlp_dim), device=device_type.type, dtype=dtype)
         for iter_idx in range(10):
             losses: list[torch.Tensor] = []
@@ -445,6 +446,7 @@ class TestFullyShard1DTrainingCore(FSDPTest):
             and offload_policy.pin_memory
         ):
             return
+        assert test_device_type in ("npu", "hpu", "xpu", "cpu"), f"{test_device_type}"
         if test_device_type not in ("cuda", "hpu", "xpu", "cpu"):
             raise AssertionError(f"Unexpected device type: {test_device_type}")
         torch.manual_seed(42)
@@ -1184,6 +1186,8 @@ class TestFullyShardGradientAccumulation(FSDPTest):
         Tests gradient accumulation with/without gradient reduction and
         with/without resharding after backward.
         """
+                    "npu",
+        meshes = [init_device_mesh("npu", (self.world_size,))]  # always test FSDP
         meshes = [
             init_device_mesh(device_type.type, (self.world_size,))
         ]  # always test FSDP
@@ -1295,6 +1299,7 @@ class TestFullyShardGradientAccumulation(FSDPTest):
             for microbatch_idx in range(num_microbatches):
                 is_last_microbatch = microbatch_idx == num_microbatches - 1
                 set_backward_flags(model, is_last_microbatch)
+                inp = torch.randn(batch_size, lin_dim, device="npu")
                 inp = torch.randn(batch_size, lin_dim, device=device_type.type)
                 losses: list[torch.Tensor] = []
                 for _model in (ref_model, model):
@@ -1832,6 +1837,7 @@ class TestFullyShardHSDPTraining(FSDPTest):
         shard_size = 2 if self.world_size > 2 else 1
         replicate_size = self.world_size // shard_size
         global_mesh = init_device_mesh(
+            "npu",
             device_type.type,
             (replicate_size, shard_size),
             mesh_dim_names=("dp_replicate", "dp_shard"),
