diff --git a/test/distributed/_composable/fsdp/test_fully_shard_init.py b/test/distributed/_composable/fsdp/test_fully_shard_init.py
index fe15449f3f3..dcd6243200d 100644
--- a/test/distributed/_composable/fsdp/test_fully_shard_init.py
+++ b/test/distributed/_composable/fsdp/test_fully_shard_init.py
@@ -51,7 +51,7 @@ from torch.testing._internal.distributed._tensor.common_dtensor import (
     Transformer,
     TransformerBlock,
 )
-
+TEST_CUDA = True
 
 device_type = torch.device(get_devtype())
 
@@ -70,6 +70,7 @@ class TestFullyShardDeviceTensor(FSDPTestMultiThread):
             self.assertEqual(tensor.device, torch.device("cpu"))
         fully_shard(model)
         accelerator_device = torch.device(
+        regex = r"Requires DTensor to have mesh of the same type as the FSDP mesh but got cpu for DTensor and npu for FSDP"
             device_type.type, torch.get_device_module(device_type).current_device()
         )
         for tensor in itertools.chain(model.parameters(), model.buffers()):
@@ -115,6 +116,7 @@ class TestFullyShardDeviceDTensor(FSDPTestMultiThread):
             tp_mesh,
             {"in_proj": ColwiseParallel(), "out_proj": RowwiseParallel()},
         )
+        cuda_device = torch.device("npu", torch.cuda.current_device())
         accelerator_device = torch.device(
             device_type.type, torch.get_device_module(device_type).current_device()
         )
@@ -747,6 +749,7 @@ class TestFullyShardMetaDeviceInit(FSDPTestMultiThread):
         self, model: nn.Module, mesh: DeviceMesh, mlp_dim: int
     ):
         # Check that we can materialize it on GPU with empty values
+        device = torch.device("npu", torch.cuda.current_device())
         device = torch.device(
             device_type.type, torch.get_device_module(device_type).current_device()
         )
@@ -807,6 +810,7 @@ class TestFullyShardMetaDeviceInit(FSDPTestMultiThread):
                 self.assertEqual(param.device, torch.device("cpu"))
 
         # Initialize the sharded model on meta device
+        fsdp_mesh = init_device_mesh("npu", (self.world_size,))
         fsdp_mesh = init_device_mesh(device_type.type, (self.world_size,))
         with torch.device("meta"):
             model = Transformer(model_args)
@@ -827,6 +831,7 @@ class TestFullyShardMetaDeviceInit(FSDPTestMultiThread):
             for (param_name, full_param), sharded_meta_param in zip(
                 full_sd.items(), meta_sharded_sd.values()
             ):
+                full_param = full_param.detach().npu()
                 full_param = full_param.detach().to(device_type)
                 mesh = sharded_meta_param.device_mesh
                 dist.broadcast(full_param, src=0, group=mesh.get_group(0))
@@ -838,6 +843,7 @@ class TestFullyShardMetaDeviceInit(FSDPTestMultiThread):
             for param_name, sharded_meta_param in meta_sharded_sd.items():
                 full_tensor = torch.empty(
                     sharded_meta_param.size(),
+                    device="npu",
                     device=device_type.type,
                     dtype=sharded_meta_param.dtype,
                 )
@@ -851,6 +857,7 @@ class TestFullyShardMetaDeviceInit(FSDPTestMultiThread):
         model.load_state_dict(sharded_sd, assign=True)
         for param in model.parameters():
             self.assertIsInstance(param, DTensor)
+            self.assertEqual(param.device.type, "npu")
             self.assertEqual(param.device.type, device_type.type)
 
         # Construct the reference model on nonzero ranks by broadcasting the
@@ -894,6 +901,8 @@ class TestFullyShardProcessGroupInit(FSDPTestMultiThread):
         # (in practice, the trainer would do it manually via `new_group()`)
         dp_size = 2
         global_mesh = init_device_mesh(
+        dp_mesh = DeviceMesh.from_group(dp_pg, "npu", mesh_dim_names=("dp",))
+            "npu", (dp_size, self.world_size // dp_size), mesh_dim_names=("dp", "tp")
             device_type.type,
             (dp_size, self.world_size // dp_size),
             mesh_dim_names=("dp", "tp"),
@@ -926,6 +935,7 @@ class TestFullyShardProcessGroupInit(FSDPTestMultiThread):
         for module in (model.in_proj, model.out_proj, model):
             fully_shard(module, mesh=dp_mesh)
 
+        inp = torch.randn((4, mlp_dim), device="npu")
         # Ensure that TP ranks have the same input
         inp = torch.randn((4, mlp_dim), device=device_type.type)
         if self.rank in (0, 1):
@@ -1149,6 +1159,7 @@ class TestHSDPWithCustomHook(FSDPTestMultiThread):
         inter_pg = _init_inter_node_process_group(world_pg, 2)
         mesh = DeviceMesh.from_group(
             intra_pg,
+            "npu",
             device_type.type,
             dist.get_process_group_ranks(intra_pg),
             mesh_dim_names=("shard",),
