diff --git a/test/dynamo/test_package.py b/test/dynamo/test_package.py
index 87b4c088a1a..b7b1850cfbf 100644
--- a/test/dynamo/test_package.py
+++ b/test/dynamo/test_package.py
@@ -7,6 +7,8 @@ import tempfile
 import unittest
 
 import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 import torch._dynamo.testing
 import torch._inductor.config
 import torch._inductor.test_case
@@ -26,6 +28,7 @@ from torch.testing._internal.inductor_utils import (
     HAS_XPU_AND_TRITON,
 )
 
+HAS_CUDA_AND_TRITON = True
 
 def compute_loss_helper(x):
     return reduce_to_scalar_loss(x)
@@ -67,7 +70,7 @@ class TestPackage(torch._inductor.test_case.TestCase):
         class MyModule(torch.nn.Module):
             def __init__(self):
                 super().__init__()
-                self.linear = torch.nn.Linear(10, 10, device="cuda")
+                self.linear = torch.nn.Linear(10, 10, device="npu")
 
             def forward(self, x):
                 return self.linear(x)
@@ -75,13 +78,13 @@ class TestPackage(torch._inductor.test_case.TestCase):
         fn = MyModule()
         package = CompilePackage(fn.forward)
         compiled_fn = torch._dynamo.optimize("inductor", package=package)(fn)
-        x = torch.randn(10, 10, device="cuda")
+        x = torch.randn(10, 10, device="npu")
         compiled_fn(x)
 
     @parametrize("backend", ("eager", "inductor"))
-    @parametrize("device", ("cpu", "cuda", "xpu"))
+    @parametrize("device", ("cpu", "npu", "xpu"))
     def test_basic_fn(self, backend, device):
-        if device == "cuda" and not HAS_CUDA_AND_TRITON:
+        if device == "npu" and not HAS_CUDA_AND_TRITON:
             raise unittest.SkipTest("Requires CUDA/Triton")
         if device == "xpu" and not HAS_XPU_AND_TRITON:
             raise unittest.SkipTest("Requires XPU/Triton")
@@ -123,9 +126,9 @@ class TestPackage(torch._inductor.test_case.TestCase):
             self.assertEqual(expected, compiled_fn(*args))
 
     @parametrize("backend", ("eager", "inductor"))
-    @parametrize("device", ("cpu", "cuda", "xpu"))
+    @parametrize("device", ("cpu", "npu", "xpu"))
     def test_lazy_backward(self, backend, device):
-        if device == "cuda" and not HAS_CUDA_AND_TRITON:
+        if device == "npu" and not HAS_CUDA_AND_TRITON:
             raise unittest.SkipTest("Requires CUDA/Triton")
         if device == "xpu" and not HAS_XPU_AND_TRITON:
             raise unittest.SkipTest("Requires XPU/Triton")
@@ -170,9 +173,9 @@ class TestPackage(torch._inductor.test_case.TestCase):
             self.assertEqual(expected, compiled_fn(*args))
 
     @parametrize("backend", ("eager", "inductor"))
-    @parametrize("device", ("cpu", "cuda", "xpu"))
+    @parametrize("device", ("cpu", "npu", "xpu"))
     def test_graph_break_bomb(self, backend, device):
-        if device == "cuda" and not HAS_CUDA_AND_TRITON:
+        if device == "npu" and not HAS_CUDA_AND_TRITON:
             raise unittest.SkipTest("Requires CUDA/Triton")
         if device == "xpu" and not HAS_XPU_AND_TRITON:
             raise unittest.SkipTest("Requires XPU/Triton")
@@ -234,9 +237,9 @@ class TestPackage(torch._inductor.test_case.TestCase):
                 compiled_fn(torch.tensor(N), 0, N - 1)
 
     @parametrize("backend", ("eager", "inductor"))
-    @parametrize("device", ("cpu", "cuda", "xpu"))
+    @parametrize("device", ("cpu", "npu", "xpu"))
     def test_dynamic_shape(self, backend, device):
-        if device == "cuda" and not HAS_CUDA_AND_TRITON:
+        if device == "npu" and not HAS_CUDA_AND_TRITON:
             raise unittest.SkipTest("Requires CUDA/Triton")
         if device == "xpu" and not HAS_XPU_AND_TRITON:
             raise unittest.SkipTest("Requires XPU/Triton")
@@ -354,9 +357,9 @@ def add(x, y):
             )
             ctx.load_package(fn, self.path())
 
-    @parametrize("device", ("cpu", "cuda", "xpu"))
+    @parametrize("device", ("cpu", "npu", "xpu"))
     def test_dynamo_cache_manual_load(self, device):
-        if device == "cuda" and not HAS_CUDA_AND_TRITON:
+        if device == "npu" and not HAS_CUDA_AND_TRITON:
             raise unittest.SkipTest("Requires CUDA/Triton")
         if device == "xpu" and not HAS_XPU_AND_TRITON:
             raise unittest.SkipTest("Requires XPU/Triton")
@@ -390,10 +393,10 @@ def add(x, y):
             self.assertEqual(expected, [result1, result2])
         self.assertEqual(torch._dynamo.convert_frame.FRAME_COUNTER, total_frames)
 
-    @parametrize("device", ("cpu", "cuda", "xpu"))
+    @parametrize("device", ("cpu", "npu", "xpu"))
     @torch._dynamo.config.patch(caching_precompile=True)
     def test_automatic_dynamo_serialize(self, device):
-        if device == "cuda" and not HAS_CUDA_AND_TRITON:
+        if device == "npu" and not HAS_CUDA_AND_TRITON:
             raise unittest.SkipTest("Requires CUDA/Triton")
         if device == "xpu" and not HAS_XPU_AND_TRITON:
             raise unittest.SkipTest("Requires XPU/Triton")
@@ -424,10 +427,10 @@ def add(x, y):
             self.assertEqual(expected, [result1, result2])
         self.assertEqual(torch._dynamo.convert_frame.FRAME_COUNTER, total_frames)
 
-    @parametrize("device", ("cpu", "cuda", "xpu"))
+    @parametrize("device", ("cpu", "npu", "xpu"))
     @torch._dynamo.config.patch(caching_precompile=True)
     def test_automatic_dynamo_recompiles(self, device):
-        if device == "cuda" and not HAS_CUDA_AND_TRITON:
+        if device == "npu" and not HAS_CUDA_AND_TRITON:
             raise unittest.SkipTest("Requires CUDA/Triton")
         if device == "xpu" and not HAS_XPU_AND_TRITON:
             raise unittest.SkipTest("Requires XPU/Triton")
@@ -457,10 +460,10 @@ def add(x, y):
         self.assertEqual(result2, expected2)
         self.assertEqual(torch._dynamo.convert_frame.FRAME_COUNTER, total_frames)
 
-    @parametrize("device", ("cpu", "cuda", "xpu"))
+    @parametrize("device", ("cpu", "npu", "xpu"))
     @torch._dynamo.config.patch(caching_precompile=True)
     def test_automatic_dynamo_graph_breaks(self, device):
-        if device == "cuda" and not HAS_CUDA_AND_TRITON:
+        if device == "npu" and not HAS_CUDA_AND_TRITON:
             raise unittest.SkipTest("Requires CUDA/Triton")
         if device == "xpu" and not HAS_XPU_AND_TRITON:
             raise unittest.SkipTest("Requires XPU/Triton")
@@ -503,10 +506,10 @@ def add(x, y):
             # Should have same number of frames as on cold start
             self.assertEqual(torch._dynamo.convert_frame.FRAME_COUNTER, total_frames)
 
-    @parametrize("device", ("cpu", "cuda", "xpu"))
+    @parametrize("device", ("cpu", "npu", "xpu"))
     @torch._dynamo.config.patch(caching_precompile=True)
     def test_automatic_dynamo_lazy_backward(self, device):
-        if device == "cuda" and not HAS_CUDA_AND_TRITON:
+        if device == "npu" and not HAS_CUDA_AND_TRITON:
             raise unittest.SkipTest("Requires CUDA/Triton")
         if device == "xpu" and not HAS_XPU_AND_TRITON:
             raise unittest.SkipTest("Requires XPU/Triton")
@@ -532,10 +535,10 @@ def add(x, y):
 
         self.assertEqual(torch._dynamo.convert_frame.FRAME_COUNTER, total_frames)
 
-    @parametrize("device", ("cpu", "cuda", "xpu"))
+    @parametrize("device", ("cpu", "npu", "xpu"))
     @torch._dynamo.config.patch(caching_precompile=True)
     def test_graph_break_partial_backend(self, device):
-        if device == "cuda" and not HAS_CUDA_AND_TRITON:
+        if device == "npu" and not HAS_CUDA_AND_TRITON:
             raise unittest.SkipTest("Requires CUDA/Triton")
         if device == "xpu" and not HAS_XPU_AND_TRITON:
             raise unittest.SkipTest("Requires XPU/Triton")
@@ -579,10 +582,10 @@ def add(x, y):
         # One recompile on a new frame, so total_frames should increase by 1
         self.assertEqual(torch._dynamo.convert_frame.FRAME_COUNTER, total_frames + 1)
 
-    @parametrize("device", ("cpu", "cuda", "xpu"))
+    @parametrize("device", ("cpu", "npu", "xpu"))
     @torch._dynamo.config.patch(caching_precompile=True)
     def test_call_function_from_resume(self, device):
-        if device == "cuda" and not HAS_CUDA_AND_TRITON:
+        if device == "npu" and not HAS_CUDA_AND_TRITON:
             raise unittest.SkipTest("Requires CUDA/Triton")
         if device == "xpu" and not HAS_XPU_AND_TRITON:
             raise unittest.SkipTest("Requires XPU/Triton")
@@ -607,10 +610,10 @@ def add(x, y):
 
         self.assertEqual(torch._dynamo.convert_frame.FRAME_COUNTER, total_frames)
 
-    @parametrize("device", ("cpu", "cuda", "xpu"))
+    @parametrize("device", ("cpu", "npu", "xpu"))
     @torch._dynamo.config.patch(caching_precompile=True)
     def test_code_with_generator(self, device):
-        if device == "cuda" and not HAS_CUDA_AND_TRITON:
+        if device == "npu" and not HAS_CUDA_AND_TRITON:
             raise unittest.SkipTest("Requires CUDA/Triton")
         if device == "xpu" and not HAS_XPU_AND_TRITON:
             raise unittest.SkipTest("Requires XPU/Triton")
@@ -628,10 +631,10 @@ def add(x, y):
         compiled_fn(*args)
         self._save_and_reload(expected_backends=1, expected_dynamo=1)
 
-    @parametrize("device", ("cpu", "cuda", "xpu"))
+    @parametrize("device", ("cpu", "npu", "xpu"))
     @torch._dynamo.config.patch(caching_precompile=True)
     def test_automatic_dynamo_graph_breaks_from_print_model_as_fn(self, device):
-        if device == "cuda" and not HAS_CUDA_AND_TRITON:
+        if device == "npu" and not HAS_CUDA_AND_TRITON:
             raise unittest.SkipTest("Requires CUDA/Triton")
         if device == "xpu" and not HAS_XPU_AND_TRITON:
             raise unittest.SkipTest("Requires XPU/Triton")
@@ -732,10 +735,10 @@ def add(x, y):
             x = self.instance_method_with_args(x)
             return x
 
-    @parametrize("device", ("cpu", "cuda", "xpu"))
+    @parametrize("device", ("cpu", "npu", "xpu"))
     @torch._dynamo.config.patch(caching_precompile=True)
     def test_classmethod_qualname(self, device):
-        if device == "cuda" and not HAS_CUDA_AND_TRITON:
+        if device == "npu" and not HAS_CUDA_AND_TRITON:
             raise unittest.SkipTest("Requires CUDA/Triton")
         if device == "xpu" and not HAS_XPU_AND_TRITON:
             raise unittest.SkipTest("Requires XPU/Triton")
