diff --git a/test/dynamo/test_repros.py b/test/dynamo/test_repros.py
index 043c0cd..2bca76c 100644
--- a/test/dynamo/test_repros.py
+++ b/test/dynamo/test_repros.py
@@ -1,3 +1,6 @@
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 """
 PYTEST_DONT_REWRITE (prevents pytest from rewriting assertions, which interferes
 with test_rewrite_assert_with_msg and test_rewrite_assert_without_msg)
