diff --git a/test/dynamo/test_compiler_bisector.py b/test/dynamo/test_compiler_bisector.py
index a5a350c..53f7e42 100644
--- a/test/dynamo/test_compiler_bisector.py
+++ b/test/dynamo/test_compiler_bisector.py
@@ -1,3 +1,6 @@
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["module: dynamo"]
 
 import unittest
@@ -23,7 +26,7 @@ i64 = torch.int64
 i32 = torch.int32
 
 
-@requires_cuda
+# @requires_cuda
 class TestCompilerBisector(TestCase):
     test_ns = "_test_bisector"
 
