diff --git a/test/dynamo/test_misc.py b/test/dynamo/test_misc.py
index 7541bd3..6b39b3d 100644
--- a/test/dynamo/test_misc.py
+++ b/test/dynamo/test_misc.py
@@ -0,0 +1,3 @@
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
@@ -10277 +10280 @@ def ___make_guard_fn():
-            torch.randn(3, 2), ConstantSource("x")
+            torch.randn(3, 2).npu(), ConstantSource("x")
@@ -10331 +10334 @@ ShapeEnv not equal: field values don't match:
-            torch.randn(3, 2), ConstantSource("x")
+            torch.randn(3, 2).npu(), ConstantSource("x")
@@ -10398 +10401 @@ ShapeEnv not equal: field values don't match:
-            torch.randn(3, 2), ConstantSource("x")
+            torch.randn(3, 2).npu(), ConstantSource("x")
@@ -10401 +10404 @@ ShapeEnv not equal: field values don't match:
-            torch.randn(3, 2), ConstantSource("x")
+            torch.randn(3, 2).npu(), ConstantSource("x")
@@ -10438 +10441 @@ ShapeEnv not equal: field values don't match:
-            torch.randn(3, 2), ConstantSource("x")
+            torch.randn(3, 2).npu(), ConstantSource("x")
@@ -10441 +10444 @@ ShapeEnv not equal: field values don't match:
-            torch.randn(3, 2), ConstantSource("x")
+            torch.randn(3, 2).npu(), ConstantSource("x")
@@ -10481 +10484 @@ ShapeEnv not equal: field values don't match:
-            torch.randn(3, 2), ConstantSource("x")
+            torch.randn(3, 2).npu(), ConstantSource("x")
@@ -10484 +10487 @@ ShapeEnv not equal: field values don't match:
-            torch.randn(3, 2), ConstantSource("x")
+            torch.randn(3, 2).npu(), ConstantSource("x")
