diff --git a/test/dynamo/test_autograd_function.py b/test/dynamo/test_autograd_function.py
index f41434b8dc9..6fef7e8863c 100644
--- a/test/dynamo/test_autograd_function.py
+++ b/test/dynamo/test_autograd_function.py
@@ -1541,6 +1541,7 @@ class GraphModule(torch.nn.Module):
         loss.backward()
         self.assertEqual(x + y, z)
 
+    # @requires_cuda
     @requires_gpu
     def test_triton_kernel_multiple_out(self):
         class Add(torch.autograd.Function):
