diff --git a/test/dynamo/test_backends.py b/test/dynamo/test_backends.py
index 7c4402e..6d5b3bb 100644
--- a/test/dynamo/test_backends.py
+++ b/test/dynamo/test_backends.py
@@ -1,3 +1,6 @@
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["module: dynamo"]
 import sys
 import unittest
@@ -134,7 +137,7 @@ class TestOptimizations(torch._dynamo.test_case.TestCase):
     def test_aot_ts(self, device):
         self._check_backend_works("aot_ts", device)
 
-    @requires_cuda
+    # @requires_cuda
     def test_aot_cudagraphs(self, device):
         self._check_backend_works("cudagraphs", device)
 
