@@ -35,6 +35,7 @@ from unittest.mock import patch
 import numpy as np
 
 import torch
+import torch_npu
 import torch._dynamo.testing
 import torch._inductor.config
 import torch._inductor.test_case
diff --git a/test/dynamo/test_misc.py b/test/dynamo/test_misc.py
index 9bd7730..4fe5709 100644
--- a/test/dynamo/test_misc.py
+++ b/test/dynamo/test_misc.py
@@ -35,6 +35,8 @@ from unittest.mock import patch
 import numpy as np
 
 import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 import torch._dynamo.testing
 import torch._inductor.config
 import torch._inductor.test_case
@@ -4802,9 +4804,9 @@ not ___dict_contains('cccccccc', G['sys'].modules)""",
         def f(x):
             if torch.overrides.is_tensor_like(x):
                 return (x * 2,)
-            return (torch.ones(10) + x,)
+            return (torch.ones(10).npu() + x,)
 
-        x = torch.randn(10)
+        x = torch.randn(10).npu()
         ref0 = f(x)
         ref1 = f(4)
         opt_f = torch.compile(f, backend=cnts, fullgraph=True)
@@ -11308,7 +11310,7 @@ def ___make_guard_fn():
 
         # Call create_symbolic_sizes_strides_storage_offset on both of them.
         r = main.create_symbolic_sizes_strides_storage_offset(
-            torch.randn(3, 2), ConstantSource("x")
+            torch.randn(3, 2).npu(), ConstantSource("x")
         )
 
         # Create a guard: size[0] == 3 (call evaluate_expr)
@@ -11362,7 +11364,7 @@ ShapeEnv not equal: field values don't match:
     def test_shape_env_equal_create_symbolic_sizes_strides_storage_offset(self):
         main, other = ShapeEnv(), ShapeEnv()
         main.create_symbolic_sizes_strides_storage_offset(
-            torch.randn(3, 2), ConstantSource("x")
+            torch.randn(3, 2).npu(), ConstantSource("x")
         )
         self.assertExpectedRaisesInline(
             NotEqualError,
@@ -11432,10 +11434,10 @@ ShapeEnv not equal: field values don't match:
 
         # Call create_symbolic_sizes_strides_storage_offset on both of them.
         r = main.create_symbolic_sizes_strides_storage_offset(
-            torch.randn(3, 2), ConstantSource("x")
+            torch.randn(3, 2).npu(), ConstantSource("x")
         )
         other.create_symbolic_sizes_strides_storage_offset(
-            torch.randn(3, 2), ConstantSource("x")
+            torch.randn(3, 2).npu(), ConstantSource("x")
         )
 
         # Create a guard: size[0] % 3 == 0 (only in the main ShapeEnv)
@@ -11472,10 +11474,10 @@ ShapeEnv not equal: field values don't match:
 
         # Call create_symbolic_sizes_strides_storage_offset on both of them.
         r = main.create_symbolic_sizes_strides_storage_offset(
-            torch.randn(3, 2), ConstantSource("x")
+            torch.randn(3, 2).npu(), ConstantSource("x")
         )
         other.create_symbolic_sizes_strides_storage_offset(
-            torch.randn(3, 2), ConstantSource("x")
+            torch.randn(3, 2).npu(), ConstantSource("x")
         )
 
         # Create a guard: size[0] == 3 (only in the main ShapeEnv)
@@ -11515,10 +11517,10 @@ ShapeEnv not equal: field values don't match:
 
         # Call create_symbolic_sizes_strides_storage_offset on both of them.
         r = main.create_symbolic_sizes_strides_storage_offset(
-            torch.randn(3, 2), ConstantSource("x")
+            torch.randn(3, 2).npu(), ConstantSource("x")
         )
         other.create_symbolic_sizes_strides_storage_offset(
-            torch.randn(3, 2), ConstantSource("x")
+            torch.randn(3, 2).npu(), ConstantSource("x")
         )
 
         # Create a guard: size[0] >= 3 (only in the main ShapeEnv)
