diff --git a/test/dynamo/test_backends.py b/test/dynamo/test_backends.py
index d39df6e2ebe..c667a17a671 100644
--- a/test/dynamo/test_backends.py
+++ b/test/dynamo/test_backends.py
@@ -153,7 +153,8 @@ class TestOptimizations(torch._dynamo.test_case.TestCase):
     def test_aot_ts(self, device):
         self._check_backend_works("aot_ts", device)
 
-    @requires_cuda_and_triton
+    # @requires_cuda_and_triton
+    # @requires_cuda
     def test_aot_cudagraphs(self, device):
         self._check_backend_works("cudagraphs", device)
 
