diff --git a/test/test_functional_optim.py b/test/test_functional_optim.py
index d1c17a9bc8b..f108ec61f80 100644
--- a/test/test_functional_optim.py
+++ b/test/test_functional_optim.py
@@ -8,6 +8,7 @@ import torch.nn as nn
 import torch.nn.functional as F
 from torch import Tensor
 from torch.optim import Adam, AdamW, SGD
+from torch_npu.contrib import transfer_to_npu
 from torch.testing._internal.common_utils import run_tests, TestCase
 
 
