diff --git a/test/test_fake_tensor.py b/test/test_fake_tensor.py
index 34ce0f79b8..1173f9cee2 100644
--- a/test/test_fake_tensor.py
+++ b/test/test_fake_tensor.py
@@ -82,3 +82,29 @@ torch._dynamo.config.fake_tensor_cache_enabled = True
 torch._dynamo.config.fake_tensor_cache_crosscheck_enabled = True
-
+import torch_npu
+import torch_npu._inductor
+from torch_npu.contrib import transfer_to_npu
+
+DEVICE_NAME = torch_npu.npu.get_device_name(0)
+
+device_is_950 = False
+if "Ascend950" in DEVICE_NAME or "Ascend910_95" in DEVICE_NAME:
+    device_is_950 = True
+
+def get_supported_dtypes_by_npu():
+    base_dtypes = [
+        torch.float32, torch.float64,
+        torch.uint8, torch.int8, torch.int16, torch.int32, torch.int64,
+        torch.complex64
+    ]
+
+    # 950额外支持的Float8类型
+    float8_dtypes = [
+        torch.float8_e4m3fn, torch.float8_e4m3fnuz,
+        torch.float8_e5m2, torch.float8_e5m2fnuz
+    ]
+
+    if device_is_950:
+        return base_dtypes + float8_dtypes
+    else:
+        return base_dtypes
 
@@ -99,4 +125,4 @@ class FakeTensorTest(TestCase):
         with FakeTensorMode():
-            p = torch.randn(4, 2, requires_grad=True, device="cuda")
-            x = torch.randn(8, 4, device="cuda")
+            p = torch.randn(4, 2, requires_grad=True, device="npu")
+            x = torch.randn(8, 4, device="npu")
             y = torch.mm(x, p).square().sum()
@@ -166,3 +192,3 @@ class FakeTensorTest(TestCase):
         "dtype",
-        all_types_complex_float8_and(),
+        get_supported_dtypes_by_npu(),
     )
@@ -170,5 +196,5 @@ class FakeTensorTest(TestCase):
         with FakeTensorMode():
-            x = torch.ones([2048], device="cuda", dtype=dtype)
+            x = torch.ones([2048], device="npu", dtype=dtype)
             out = x[torch.zeros([36], dtype=torch.int64)]
-            self.checkType(out, "cuda", [36])
+            self.checkType(out, "npu", [36])
             self.assertEqual(out.dtype, dtype)
@@ -179,3 +205,3 @@ class FakeTensorTest(TestCase):
             x = torch.empty(1, device="cpu")
-            y = torch.empty(8, 8, device="cuda")
+            y = torch.empty(8, 8, device="npu")
             out = x.resize_as_(y)
@@ -205,3 +231,3 @@ class FakeTensorTest(TestCase):
             x = torch.tensor(0.0)
-            y = torch.rand([4, 4], device="cuda")
+            y = torch.rand([4, 4], device="npu")
             out = x + y
@@ -225,3 +251,3 @@ class FakeTensorTest(TestCase):
             x_conv = mode.from_tensor(x)
-            y = torch.rand([4, 4], device="cuda")
+            y = torch.rand([4, 4], device="npu")
             z = torch.rand([4, 4], device="cpu")
@@ -233,5 +259,5 @@ class FakeTensorTest(TestCase):
             x = torch.rand([16, 1], device="cpu")
-            y = torch.rand([4, 4], device="cuda")
+            y = torch.rand([4, 4], device="npu")
             out = x.type_as(y)
-            self.assertEqual(out.device.type, "cuda")
+            self.assertEqual(out.device.type, "npu")
             self.assertTrue(isinstance(out, FakeTensor))
@@ -240,3 +266,3 @@ class FakeTensorTest(TestCase):
     def test_setitem(self):
-        for device in ["cpu", "cuda"]:
+        for device in ["cpu", "npu"]:
             with FakeTensorMode():
@@ -249,5 +275,5 @@ class FakeTensorTest(TestCase):
             x = torch.rand([8, 8], device="cpu")
-            y = torch.rand([8, 8], device="cuda")
+            y = torch.rand([8, 8], device="npu")
             assert x.copy_(y).device.type == "cpu"
-            assert y.copy_(x).device.type == "cuda"
+            assert y.copy_(x).device.type == "npu"
 
@@ -329,4 +355,4 @@ class FakeTensorTest(TestCase):
             self.assertIs(x, y)
-            z = x.to(torch.device("cuda"))
-            self.assertEqual(z.device.type, "cuda")
+            z = x.to(torch.device("npu"))
+            self.assertEqual(z.device.type, "npu")
 
@@ -365,3 +391,3 @@ class FakeTensorTest(TestCase):
                 y = torch.randn(2, 2, 3)
-                x = torch.randn(2, 2, 3).to("cuda")
+                x = torch.randn(2, 2, 3).to("npu")
                 with self.assertRaises(RuntimeError):
@@ -370,3 +396,3 @@ class FakeTensorTest(TestCase):
                 with self.assertRaises(RuntimeError):
-                    torch.ops.aten.index_put(x, torch.tensor([1, 1], device="cuda"), y)
+                    torch.ops.aten.index_put(x, torch.tensor([1, 1], device="npu"), y)
 
@@ -374,6 +400,6 @@ class FakeTensorTest(TestCase):
                 torch.ops.aten.index_put(
-                    x, torch.tensor([1, 1], device="cuda"), torch.tensor(5.0)
+                    x, torch.tensor([1, 1], device="npu"), torch.tensor(5.0)
                 )
                 torch.ops.aten.index_put_(
-                    x, torch.tensor([1, 1], device="cuda"), torch.tensor(5.0)
+                    x, torch.tensor([1, 1], device="npu"), torch.tensor(5.0)
                 )
@@ -387,5 +413,5 @@ class FakeTensorTest(TestCase):
             self.assertEqual(y.device.type, "cpu")
-            z = torch.ones_like(x, device="cuda")
+            z = torch.ones_like(x, device="npu")
             self.assertTrue(isinstance(z, FakeTensor))
-            self.assertEqual(z.device.type, "cuda")
+            self.assertEqual(z.device.type, "npu")
 
@@ -430,3 +456,3 @@ class FakeTensorTest(TestCase):
                 arg0_1 = torch.empty_strided(
-                    (3, 427, 640), (1, 1920, 3), dtype=torch.float32, device="cuda"
+                    (3, 427, 640), (1, 1920, 3), dtype=torch.float32, device="npu"
                 )
@@ -456,6 +482,6 @@ class FakeTensorTest(TestCase):
         with FakeTensorMode(allow_fallback_kernels=False):
-            filters = torch.randn(8, 4, 3, 3).cuda()
-            inputs = torch.randn(1, 4, 5, 5).cuda()
+            filters = torch.randn(8, 4, 3, 3).npu()
+            inputs = torch.randn(1, 4, 5, 5).npu()
             out = torch.nn.functional.conv2d(inputs, filters, padding=1)
-            self.assertEqual(out.device.type, "cuda")
+            self.assertEqual(out.device.type, "npu")
             self.assertEqual(list(out.size()), [1, 8, 5, 5])
@@ -470,7 +496,7 @@ class FakeTensorTest(TestCase):
         with FakeTensorMode(allow_fallback_kernels=True):
-            filters = torch.randn(8, 4, 3, 3).cuda()
-            inputs = torch.randn(1, 4, 5, 5).cuda()
+            filters = torch.randn(8, 4, 3, 3).npu()
+            inputs = torch.randn(1, 4, 5, 5).npu()
 
             out = torch.nn.functional.conv2d(inputs, filters, padding=1)
-            self.assertEqual(out.device.type, "cuda")
+            self.assertEqual(out.device.type, "npu")
             self.assertEqual(list(out.size()), [1, 8, 5, 5])
@@ -481,3 +507,3 @@ class FakeTensorTest(TestCase):
             x = torch.rand([4])
-            y = torch.rand([4], device="cuda")
+            y = torch.rand([4], device="npu")
 
@@ -495,6 +521,6 @@ class FakeTensorTest(TestCase):
         with FakeTensorMode():
-            x = torch.empty(1, device="cuda")
-            y = torch.empty(1, device=f"cuda:{torch.cuda.current_device()}")
+            x = torch.empty(1, device="npu")
+            y = torch.empty(1, device=f"npu:{torch_npu.npu.current_device()}")
             out = x + y
-        self.checkType(out, "cuda", [1])
+        self.checkType(out, "npu", [1])
 
@@ -612,3 +638,3 @@ class FakeTensorTest(TestCase):
                             self.assertTrue(isinstance(ten, FakeTensor))
-                        self.assertEqual(ten.device.type, "cuda")
+                        self.assertEqual(ten.device.type, "npu")
 
@@ -638,8 +664,8 @@ class FakeTensorTest(TestCase):
                     bidirectional=bidir,
-                    device="cuda",
+                    device="npu",
                 )
 
-                h_0 = torch.randn((num_layers * D, N, H_out), device="cuda")
-                c_0 = torch.randn((num_layers * D, N, hidden_size), device="cuda")
-                inp = torch.randn((L, N, H_in), device="cuda")
+                h_0 = torch.randn((num_layers * D, N, H_out), device="npu")
+                c_0 = torch.randn((num_layers * D, N, hidden_size), device="npu")
+                inp = torch.randn((L, N, H_in), device="npu")
                 (output, (h_n, c_n)) = lstm(inp, (h_0, c_0))
@@ -768,4 +794,4 @@ class FakeTensorTest(TestCase):
             self.checkType(a.new([1, 2, 3, 4]), "cpu", [4])
-            b = torch.rand([4, 4], device="cuda")
-            self.checkType(b.new(device="cuda"), "cuda", [0])
+            b = torch.rand([4, 4], device="npu")
+            self.checkType(b.new(device="npu"), "npu", [0])
             self.checkType(a.new(torch.rand([1])), "cpu", [1])
@@ -851,3 +877,3 @@ class FakeTensorTest(TestCase):
             x1 = torch.rand(4, device="cpu")
-            x2 = torch.rand(4, device="cuda")
+            x2 = torch.rand(4, device="npu")
             copy1 = torch.ops.aten.copy.default(x1, x2)
@@ -857,3 +883,3 @@ class FakeTensorTest(TestCase):
         self.checkType(copy1, "cpu", (4,))
-        self.checkType(copy2, "cuda", (4,))
+        self.checkType(copy2, "npu", (4,))
         self.checkType(out, "cpu", (4,))
@@ -867,4 +893,4 @@ class FakeTensorTest(TestCase):
             x1 = torch.rand(4, 4, device="cpu")
-            x2 = torch.rand(4, 4, device="cuda")
-            i1 = torch.tensor([0, 1], device="cuda")
+            x2 = torch.rand(4, 4, device="npu")
+            i1 = torch.tensor([0, 1], device="npu")
             i2 = torch.tensor([0, 1], device="cpu")
@@ -876,4 +902,4 @@ class FakeTensorTest(TestCase):
             y1 = torch.rand(4, device="cpu")
-            y2 = torch.rand(4, device="cuda")
-            j1 = torch.tensor([2], device="cuda")
+            y2 = torch.rand(4, device="npu")
+            j1 = torch.tensor([2], device="npu")
             j2 = torch.tensor([2], device="cpu")
@@ -882,5 +908,5 @@ class FakeTensorTest(TestCase):
         # self.checkType(r1, "cpu", ())
-        self.checkType(r2, "cuda", ())
+        self.checkType(r2, "npu", ())
         self.checkType(r3, "cpu", (4, 4))
-        self.checkType(r4, "cuda", (4, 4))
+        self.checkType(r4, "npu", (4, 4))
 
@@ -893,4 +919,4 @@ class FakeTensorTest(TestCase):
             x1 = torch.rand(4, 4, device="cpu")
-            y1 = torch.rand(2, 4, device="cuda")
-            x2 = torch.rand(4, 4, device="cuda")
+            y1 = torch.rand(2, 4, device="npu")
+            x2 = torch.rand(4, 4, device="npu")
             y2 = torch.rand(2, 4, device="cpu")
@@ -901,3 +927,3 @@ class FakeTensorTest(TestCase):
         self.checkType(r1, "cpu", (4, 4))
-        self.checkType(r2, "cuda", (4, 4))
+        self.checkType(r2, "npu", (4, 4))
         self.checkType(r3, "cpu", (4, 4))
@@ -1106,3 +1132,3 @@ class FakeTensorOpInfoTest(TestCase):
 make_propagate_real_tensors_cls(FakeTensorOpInfoTest)
-instantiate_device_type_tests(FakeTensorOpInfoTest, globals(), only_for=("cpu", "cuda"))
+instantiate_device_type_tests(FakeTensorOpInfoTest, globals(), only_for=("cpu", "npu"))
 instantiate_device_type_tests(
@@ -1157,3 +1183,3 @@ class FakeTensorConverterTest(TestCase):
     def test_dead_weak_ref(self):
-        x = torch.rand(2, 2, 2)
+        x = torch.rand(2, 2, 2).npu()
         y = x[0]
@@ -1170,3 +1196,3 @@ class FakeTensorConverterTest(TestCase):
     def test_dead_key(self):
-        x = torch.rand(2, 2, 2)
+        x = torch.rand(2, 2, 2).npu()
         mode = FakeTensorMode()
@@ -1384,10 +1410,10 @@ class FakeTensorOperatorInvariants(TestCase):
             [
-                ((1, 48, 64, 64), (0, 4096, 64, 1), torch.float16, "cuda"),
-                ((1, 48, 64, 64), (0, 4096, 64, 1), torch.float16, "cuda"),
-                ((1, 48, 64, 64), (0, 4096, 64, 1), torch.float16, "cuda"),
+                ((1, 48, 64, 64), (0, 4096, 64, 1), torch.float16, "npu"),
+                ((1, 48, 64, 64), (0, 4096, 64, 1), torch.float16, "npu"),
+                ((1, 48, 64, 64), (0, 4096, 64, 1), torch.float16, "npu"),
             ],
             [
-                ((4, 2, 16, 32), (1024, 512, 32, 1), torch.float16, "cuda"),
-                ((4, 2, 16, 32), (1024, 512, 32, 1), torch.float16, "cuda"),
-                ((4, 2, 16, 32), (1024, 512, 32, 1), torch.float16, "cuda"),
+                ((4, 2, 16, 32), (1024, 512, 32, 1), torch.float16, "npu"),
+                ((4, 2, 16, 32), (1024, 512, 32, 1), torch.float16, "npu"),
+                ((4, 2, 16, 32), (1024, 512, 32, 1), torch.float16, "npu"),
             ],
@@ -1454,5 +1480,5 @@ class FakeTensorOperatorInvariants(TestCase):
         args_new = [
-            ((16, 1, 128, 128), (16384, 16384, 128, 1), torch.float16, "cuda"),
-            ((16, 64, 128, 128), (1048576, 1, 8192, 64), torch.float16, "cuda"),
-            ((1, 64, 3, 3), (576, 9, 3, 1), torch.float16, "cuda"),
+            ((16, 1, 128, 128), (16384, 16384, 128, 1), torch.float16, "npu"),
+            ((16, 64, 128, 128), (1048576, 1, 8192, 64), torch.float16, "npu"),
+            ((1, 64, 3, 3), (576, 9, 3, 1), torch.float16, "npu"),
         ]
@@ -1645,3 +1671,3 @@ class FakeTensorPropTest(TestCase):
             with fake_mode:
-                sd_loaded = torch.load(f)
+                sd_loaded = torch.load(f, weights_only=False)
             for k in sd:
@@ -1649,5 +1675,5 @@ class FakeTensorPropTest(TestCase):
             with fake_mode:
-                sd_loaded = torch.load(f, map_location="cuda")
+                sd_loaded = torch.load(f, map_location="npu", weights_only=False)
             for k in sd:
-                _read_tensor_and_check(k, sd_loaded, all_bytes, 'cuda')
+                _read_tensor_and_check(k, sd_loaded, all_bytes, 'npu')
 
@@ -1655,3 +1681,3 @@ class FakeTensorPropTest(TestCase):
         for k in sd.keys():
-            sd[k] = sd[k].to('cuda')
+            sd[k] = sd[k].to('npu')
 
@@ -1664,7 +1690,7 @@ class FakeTensorPropTest(TestCase):
             with fake_mode:
-                sd_loaded = torch.load(f)
+                sd_loaded = torch.load(f, weights_only=False)
             for k in sd:
-                _read_tensor_and_check(k, sd_loaded, all_bytes, 'cuda')
+                _read_tensor_and_check(k, sd_loaded, all_bytes, 'npu')
             with fake_mode:
-                sd_loaded = torch.load(f, map_location="cpu")
+                sd_loaded = torch.load(f, map_location="cpu", weights_only=False)
             for k in sd:
@@ -1755,3 +1781,3 @@ class FakeTensorDispatchCache(TestCase):
             y = torch.randn(4, 3)
-            z = x.to(device="cuda")
+            z = x.to(device="npu")
             self._test_cache_key(fm, x, y, z)
@@ -1903,6 +1929,6 @@ class FakeTensorDispatchCache(TestCase):
 
-            torch.set_default_device("cuda")
+            torch.set_default_device("npu")
             x = torch.tensor([1, 2])
             y = x + 1.0
-            self.assertEqual(y.device.type, "cuda")
+            self.assertEqual(y.device.type, "npu")
             self.assertHitsMisses(0, 2)
@@ -2072,3 +2098,3 @@ class FakeTensorDispatchCache(TestCase):
 
-        a = torch.ones(2, 2, 768, device="cuda")
+        a = torch.ones(2, 2, 768, device="npu")
         wrapped_a = DifferentDeviceTensor(a)
