diff --git a/test/test_torch.py b/test/test_torch.py
index 48a463a0d29..b2af8a9489b 100644
--- a/test/test_torch.py
+++ b/test/test_torch.py
@@ -45,14 +45,17 @@ from torch.testing._internal.common_utils import (  # type: ignore[attr-defined]
     AlwaysWarnTypedStorageRemoval, TEST_WITH_TORCHDYNAMO, xfailIfTorchDynamo,
     xfailIfS390X, set_warn_always_context, decorateIf, isRocmArchAnyOf)
 from multiprocessing.reduction import ForkingPickler
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 from torch.testing._internal.common_device_type import (
     expectedFailureMeta,
     expectedFailureXLA,
     instantiate_device_type_tests,
-    onlyCUDA, onlyCPU,
+    onlyPRIVATEUSE1, onlyCPU,
     dtypes, dtypesIfCUDA, dtypesIfCPU, deviceCountAtLeast,
     skipMeta, PYTORCH_CUDA_MEMCHECK, largeTensorTest, onlyNativeDeviceTypes, skipCUDAIfNotRocm,
     get_all_device_types, skipXLA)
+
 import torch.backends.quantized
 import torch.testing._internal.data
 from torch.testing._internal.common_cuda import (
@@ -81,7 +84,7 @@ if torch.get_default_dtype() is not torch.float32:
 # sharding on sandcastle. This line silences flake warnings
 load_tests = load_tests  # noqa: PLW0127
 
-AMPERE_OR_ROCM = TEST_WITH_ROCM or torch.cuda.is_tf32_supported()
+# AMPERE_OR_ROCM = TEST_WITH_ROCM or torch.cuda.is_tf32_supported()
 
 
 is_cuda_sm86 = torch.cuda.is_available() and torch.cuda.get_device_capability(0) == (8, 6)
@@ -133,11 +136,11 @@ class TestTorchDeviceType(TestCase):
 
     # For testing in64 support in upsample_nearest3d
     @skipIfRocmArch(MI200_ARCH)
-    @onlyCUDA
-    @largeTensorTest('56GB', device='cuda')
+    @onlyPRIVATEUSE1
+    @largeTensorTest('56GB', device='npu')
     @dtypes(torch.bfloat16)
     @unittest.skipIf(IS_JETSON, "Large tensor tests are too large for Jetson.")
-    @decorateIf(unittest.expectedFailure, lambda params: isRocmArchAnyOf(MI200_ARCH))
+    @decorateIf(unittest.expectedFailure, lambda params: True)
     def test_int64_upsample3d(self, device, dtype):
         x = torch.ones((1, 256, 16, 720, 1280), dtype=dtype, device=device)
         try:
@@ -182,12 +185,12 @@ class TestTorchDeviceType(TestCase):
     @slowTestIf(IS_WINDOWS)
     def test_storage_setitem(self, device, dtype):
         # Skip quantized dtypes for CUDA, since they're not supported
-        if torch.device(device).type == 'cuda':
+        if torch.device(device).type == 'npu':
             if dtype in [torch.quint8, torch.qint8, torch.qint32, torch.quint4x2]:
                 return
 
         storage_type_name = torch.storage._dtype_to_storage_type_map()[dtype]
-        if torch.device(device).type == 'cuda':
+        if torch.device(device).type == 'npu':
             storage_type = eval('torch.cuda.' + storage_type_name)
         else:
             storage_type = eval('torch.' + storage_type_name)
@@ -227,7 +230,7 @@ class TestTorchDeviceType(TestCase):
     def test_tensor_storage_type(self, device, dtype):
         a = make_tensor((10,), dtype=dtype, device=device, low=-9, high=9)
 
-        module = torch.cuda if (torch.device(device).type == 'cuda') else torch
+        module = torch.cuda if (torch.device(device).type == 'npu') else torch
         expected_storage_type = getattr(module, torch.storage._dtype_to_storage_type_map()[dtype])
 
         self.assertEqual(a.storage_type(), expected_storage_type)
@@ -363,7 +366,7 @@ class TestTorchDeviceType(TestCase):
             with self.assertRaisesRegex(NotImplementedError, r'Cannot copy out'):
                 s0._write_file(f, True, True, s0.element_size())
 
-        for device in ['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']:
+        for device in ['cpu', 'npu'] if torch.cuda.is_available() else ['cpu']:
             s1 = torch.TypedStorage([1, 2, 3, 4], device=device, dtype=dtype)
 
             with self.assertRaisesRegex(NotImplementedError, r'Cannot copy out'):
@@ -378,12 +381,12 @@ class TestTorchDeviceType(TestCase):
         # This is OK, it changes the meta storage size without allocating
         s0.resize_(10)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     def test_module_share_memory(self):
         # Test fix for issue #80733
         # See https://github.com/pytorch/pytorch/issues/80733
         model = torch.nn.Linear(3, 1)
-        _model_cuda = model.to('cuda')
+        _model_cuda = model.to('npu')
         model.share_memory()
 
     @dtypes(torch.float32, torch.complex64)
@@ -827,7 +830,7 @@ class TestTorchDeviceType(TestCase):
         # Creates long string in advance to avoid a too-long Python line
         s = ".+Triggered internally at.+RangeFactories.+"
         # nvfuser deprecation warning filter
-        warnings.filterwarnings("ignore", "torch::jit::fuser::cuda", UserWarning)
+        warnings.filterwarnings("ignore", "torch::jit::fuser::npu", UserWarning)
 
         def cpp_warn_fn():
             out = torch.empty((5,))
@@ -941,9 +944,9 @@ class TestTorchDeviceType(TestCase):
             # t + 1 allocates a new tensor for result using empty
             t + 1
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     def test_dtypetensor_warnings(self, device):
-        msg = 'The torch.cuda.*DtypeTensor constructors are no longer recommended'
+        msg = 'The torch.npu.*DtypeTensor constructors are no longer recommended'
         with self.assertWarnsOnceRegex(UserWarning, msg):
             torch.cuda.FloatTensor([0])
 
@@ -987,8 +990,8 @@ class TestTorchDeviceType(TestCase):
         out.backward(torch.ones_like(out).transpose(-2, -1))
 
     # TODO: this test should be in test_nn.py
-    @onlyCUDA
-    @largeTensorTest('12GB')
+    # @onlyPRIVATEUSE1
+    # @largeTensorTest('12GB')
     def test_conv_transposed_large(self, device):
         # ConvTranspose3d works for large input tensors (gh-32866)
         in_channels = 64
@@ -1296,8 +1299,8 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             lambda: res.backward(grad, retain_graph=True),
-            'avg_pool3d_backward_cuda',
-            torch.device(device).type == 'cuda')
+            'avg_pool3d_backward_npu',
+            torch.device(device).type == 'npu')
 
     @skipIfMPS
     @skipIfTorchInductor("https://github.com/pytorch/pytorch/issues/113707")
@@ -1309,8 +1312,8 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             lambda: res.backward(grad, retain_graph=True),
-            'adaptive_avg_pool2d_backward_cuda',
-            torch.device(device).type == 'cuda')
+            'adaptive_avg_pool2d_backward_npu',
+            torch.device(device).type == 'npu')
 
     @skipIfMPS
     @skipIfTorchInductor("https://github.com/pytorch/pytorch/issues/113707")
@@ -1322,8 +1325,8 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             lambda: res.backward(grad, retain_graph=True),
-            'adaptive_avg_pool3d_backward_cuda',
-            torch.device(device).type == 'cuda')
+            'adaptive_avg_pool3d_backward_npu',
+            torch.device(device).type == 'npu')
 
     @skipIfMPS
     @skipIfTorchInductor("https://github.com/pytorch/pytorch/issues/113707")
@@ -1335,8 +1338,8 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             lambda: res.backward(grad, retain_graph=True),
-            'max_pool3d_with_indices_backward_cuda',
-            torch.device(device).type == 'cuda')
+            'max_pool3d_with_indices_backward_npu',
+            torch.device(device).type == 'npu')
 
     @skipIfMPS
     @skipIfTorchInductor("https://github.com/pytorch/pytorch/issues/113707")
@@ -1348,8 +1351,8 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             lambda: res.backward(grad, retain_graph=True),
-            'adaptive_max_pool2d_backward_cuda',
-            torch.device(device).type == 'cuda')
+            'adaptive_max_pool2d_backward_npu',
+            torch.device(device).type == 'npu')
 
     @skipIfMPS
     @skipIfTorchInductor("https://github.com/pytorch/pytorch/issues/113707")
@@ -1361,8 +1364,8 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             lambda: res.backward(grad, retain_graph=True),
-            'fractional_max_pool2d_backward_cuda',
-            torch.device(device).type == 'cuda')
+            'fractional_max_pool2d_backward_npu',
+            torch.device(device).type == 'npu')
 
     @skipIfMPS
     @skipIfTorchInductor("https://github.com/pytorch/pytorch/issues/113707")
@@ -1374,8 +1377,8 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             lambda: res.backward(grad, retain_graph=True),
-            'fractional_max_pool3d_backward_cuda',
-            torch.device(device).type == 'cuda')
+            'fractional_max_pool3d_backward_npu',
+            torch.device(device).type == 'npu')
 
     @dtypes(*floating_types_and(torch.half))
     @onlyNativeDeviceTypes
@@ -1432,8 +1435,8 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             lambda: res.backward(grad),
-            'upsample_linear1d_backward_out_cuda',
-            torch.device(device).type == 'cuda')
+            'upsample_linear1d_backward_out_npu',
+            torch.device(device).type == 'npu')
 
     @skipIfTorchInductor("https://github.com/pytorch/pytorch/issues/113707")
     def test_nondeterministic_alert_interpolate_bilinear(self, device):
@@ -1447,8 +1450,8 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             lambda: res.backward(grad),
-            'upsample_bilinear2d_backward_out_cuda',
-            torch.device(device).type == 'cuda')
+            'upsample_bilinear2d_backward_out_npu',
+            torch.device(device).type == 'npu')
 
     def test_no_nondeterministic_alert_interpolate_bilinear(self, device):
         input = torch.randn(1, 2, 4, 4, device=device, requires_grad=True)
@@ -1464,7 +1467,7 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             fn,
-            'upsample_bilinear2d_backward_out_cuda',
+            'upsample_bilinear2d_backward_out_npu',
             False)
 
     def test_no_nondeterministic_alert_interpolate_trilinear(self, device):
@@ -1481,7 +1484,7 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             fn,
-            'upsample_trilinear3d_backward_out_cuda',
+            'upsample_trilinear3d_backward_out_npu',
             False)
 
     @skipIfTorchInductor("aot-autograd issue")
@@ -1546,8 +1549,8 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             lambda: res.backward(grad),
-            'upsample_bicubic2d_backward_out_cuda',
-            torch.device(device).type == 'cuda')
+            'upsample_bicubic2d_backward_out_npu',
+            torch.device(device).type == 'npu')
 
     @skipIfMPS
     @skipIfTorchInductor("https://github.com/pytorch/pytorch/issues/113707")
@@ -1562,8 +1565,8 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             lambda: res.backward(grad),
-            'upsample_trilinear3d_backward_out_cuda',
-            torch.device(device).type == 'cuda')
+            'upsample_trilinear3d_backward_out_npu',
+            torch.device(device).type == 'npu')
 
     @skipIfMPS
     @skipIfTorchInductor("https://github.com/pytorch/pytorch/issues/113707")
@@ -1575,8 +1578,8 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             lambda: res.backward(grad, retain_graph=True),
-            'reflection_pad1d_backward_out_cuda',
-            torch.device(device).type == 'cuda')
+            'reflection_pad1d_backward_out_npu',
+            torch.device(device).type == 'npu')
 
     @skipIfMPS
     @skipIfTorchInductor("https://github.com/pytorch/pytorch/issues/113707")
@@ -1588,8 +1591,8 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             lambda: res.backward(grad, retain_graph=True),
-            'reflection_pad3d_backward_out_cuda',
-            torch.device(device).type == 'cuda')
+            'reflection_pad3d_backward_out_npu',
+            torch.device(device).type == 'npu')
 
     @skipIfMPS
     @skipIfTorchInductor("https://github.com/pytorch/pytorch/issues/113707")
@@ -1601,8 +1604,8 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             lambda: res.backward(grad, retain_graph=True),
-            'replication_pad1d_backward_cuda',
-            torch.device(device).type == 'cuda')
+            'replication_pad1d_backward_npu',
+            torch.device(device).type == 'npu')
 
     @skipIfTorchInductor("https://github.com/pytorch/pytorch/issues/113707")
     def test_nondeterministic_alert_ReplicationPad2d(self, device):
@@ -1615,8 +1618,8 @@ class TestTorchDeviceType(TestCase):
         # nondeterministic
         self.check_nondeterministic_alert(
             lambda: res.backward(grad, retain_graph=True),
-            'replication_pad2d_backward_cuda',
-            torch.device(device).type == 'cuda')
+            'replication_pad2d_backward_npu',
+            torch.device(device).type == 'npu')
 
         with DeterministicGuard(True):
             res = module(input)
@@ -1627,7 +1630,7 @@ class TestTorchDeviceType(TestCase):
         # not be raised
         self.check_nondeterministic_alert(
             lambda: res.backward(grad, retain_graph=True),
-            'replication_pad2d_backward_cuda',
+            'replication_pad2d_backward_npu',
             False)
 
     @skipIfMPS
@@ -1640,8 +1643,8 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             lambda: res.backward(grad, retain_graph=True),
-            'replication_pad3d_backward_cuda',
-            torch.device(device).type == 'cuda')
+            'replication_pad3d_backward_npu',
+            torch.device(device).type == 'npu')
 
     @skipIfTorchDynamo("Warning is not raised.")
     def test_nondeterministic_alert_NLLLoss(self, device):
@@ -1652,8 +1655,8 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             lambda: module(input, target),
-            'nll_loss2d_forward_out_cuda_template',
-            torch.device(device).type == 'cuda')
+            'nll_loss2d_forward_out_npu_template',
+            torch.device(device).type == 'npu')
 
     @skipIfTorchInductor("https://github.com/pytorch/pytorch/issues/113707")
     def test_nondeterministic_alert_CTCLoss(self, device):
@@ -1668,7 +1671,7 @@ class TestTorchDeviceType(TestCase):
         self.check_nondeterministic_alert(
             lambda: res.backward(grad, retain_graph=True),
             'ctc_loss_backward_gpu',
-            torch.device(device).type == 'cuda')
+            torch.device(device).type == 'npu')
 
     @skipIfTorchInductor("https://github.com/pytorch/pytorch/issues/113707")
     def test_nondeterministic_alert_EmbeddingBag_max(self, device):
@@ -1681,11 +1684,11 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             lambda: res.backward(grad, retain_graph=True),
-            'embedding_bag_backward_cuda_max',
-            torch.device(device).type == 'cuda')
+            'embedding_bag_backward_npu_max',
+            torch.device(device).type == 'npu')
 
     @skipIfTorchInductor("https://github.com/pytorch/pytorch/issues/113707")
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     def test_deterministic_cumsum(self, device):
         test_cases = [
             # size, dim
@@ -1723,10 +1726,10 @@ class TestTorchDeviceType(TestCase):
             res_cpu = input.cpu().cumsum(dim)
             self.assertEqual(res0, res_cpu, atol=1e-3, rtol=1e-2)
 
-    @onlyCUDA
-    @largeTensorTest('49GB')
+    @onlyPRIVATEUSE1
+    # @largeTensorTest('49GB')
     def test_cumsum_64bit_indexing(self, device):
-        b = torch.ones(2 * 4096 * 8, 100000, dtype=torch.float, device='cuda')
+        b = torch.ones(2 * 4096 * 8, 100000, dtype=torch.float, device='npu')
         b /= 100000
         d = b.cumsum(dim=-1)
         chunk = 2**30 // b.shape[-1]
@@ -1737,7 +1740,7 @@ class TestTorchDeviceType(TestCase):
         self.assertEqual(b[0, :], d[0, :], atol=3e-5, rtol=3e-5)
         self.assertEqual(b[-1, :], d[-1, :], atol=3e-5, rtol=3e-5)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @largeTensorTest('48GB')
     def test_cumsum_outer_dim_64bit_indexing(self, device):
         x = torch.zeros(309504, 1, 16384, device=device)
@@ -1770,7 +1773,7 @@ class TestTorchDeviceType(TestCase):
             self.check_nondeterministic_alert(
                 lambda: op_call(a, indices, values, accumulate=True),
                 'put_',
-                torch.device(device).type == 'cuda')
+                torch.device(device).type == 'npu')
 
     @dtypes(torch.float32)
     @dtypesIfCUDA(torch.float32, torch.int32)
@@ -1780,8 +1783,8 @@ class TestTorchDeviceType(TestCase):
         for op_call in [torch.histc, torch.Tensor.histc]:
             self.check_nondeterministic_alert(
                 lambda: op_call(a, min=0, max=3),
-                '_histc_cuda with floating point input',
-                torch.device(device).type == 'cuda' and dtype.is_floating_point)
+                '_histc_npu with floating point input',
+                torch.device(device).type == 'npu' and dtype.is_floating_point)
 
     @skipIfMPS
     def test_nondeterministic_alert_bincount(self, device):
@@ -1793,12 +1796,12 @@ class TestTorchDeviceType(TestCase):
             # given
             self.check_nondeterministic_alert(
                 lambda: op_call(a, weights),
-                '_bincount_cuda',
-                torch.device(device).type == 'cuda')
+                '_bincount_npu',
+                torch.device(device).type == 'npu')
 
             self.check_nondeterministic_alert(
                 lambda: op_call(a),
-                '_bincount_cuda',
+                '_bincount_npu',
                 False)
 
     @skipIfMPS
@@ -1811,8 +1814,8 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             lambda: res.backward(grad, retain_graph=True),
-            'grid_sampler_2d_backward_cuda',
-            torch.device(device).type == 'cuda')
+            'grid_sampler_2d_backward_npu',
+            torch.device(device).type == 'npu')
 
     @skipIfMPS
     @skipIfTorchInductor("https://github.com/pytorch/pytorch/issues/113707")
@@ -1824,8 +1827,8 @@ class TestTorchDeviceType(TestCase):
 
         self.check_nondeterministic_alert(
             lambda: res.backward(grad, retain_graph=True),
-            'grid_sampler_3d_backward_cuda',
-            torch.device(device).type == 'cuda')
+            'grid_sampler_3d_backward_npu',
+            torch.device(device).type == 'npu')
 
     def test_invalid_shapes_grid_sampler(self, device):
         make_arg = partial(
@@ -1918,13 +1921,13 @@ class TestTorchDeviceType(TestCase):
                 'median CUDA with indices output',
                 should_error)
 
-        is_cuda = torch.device(device).type == 'cuda'
+        is_npu = torch.device(device).type == 'npu'
 
         test_func_expect_error('function', False)
-        test_func_expect_error('function with indices', is_cuda)
+        test_func_expect_error('function with indices', is_npu)
         test_func_expect_error('method', False)
-        test_func_expect_error('method with indices', is_cuda)
-        test_func_expect_error('out with indices', is_cuda)
+        test_func_expect_error('method with indices', is_npu)
+        test_func_expect_error('out with indices', is_npu)
 
     # FIXME: move to test_scatter_gather_ops
     def _test_gather_backward_one_dim(self, device, deterministic: bool = False) -> None:
@@ -1941,7 +1944,7 @@ class TestTorchDeviceType(TestCase):
                 raise AssertionError("expected src.grad to be not None")
             grad = src.grad.detach().clone()
 
-            if torch.device(device).type == 'cuda' or torch.device(device).type == 'mtia':
+            if torch.device(device).type == 'npu' or torch.device(device).type == 'mtia':
                 for _ in range(2):
                     src.grad.data.zero_()
                     res = torch.gather(src, dim, idx)
@@ -2000,7 +2003,7 @@ class TestTorchDeviceType(TestCase):
         result = original.scatter(0, null_index, null_arr)
         self.assertEqual(result, original, atol=0, rtol=0)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @skipIfTorchInductor("FIXME")
     def test_sync_warning(self, device):
 
@@ -2140,7 +2143,7 @@ class TestTorchDeviceType(TestCase):
         t.bernoulli_(0.5)
         self.assertTrue(isBinary(t))
 
-        for p_dtype in floating_types_and(*[torch.half] if device.startswith('cuda') else []):
+        for p_dtype in floating_types_and(*[torch.half] if device.startswith('npu') else []):
             p = torch.rand(10, dtype=p_dtype, device=device).expand(10, 10)
             t.fill_(2)
             t.bernoulli_(p)
@@ -2154,7 +2157,7 @@ class TestTorchDeviceType(TestCase):
             t.bernoulli_(torch.rand_like(t, dtype=p_dtype))
             self.assertTrue(isBinary(t))
 
-    @slowTest
+    # @slowTest
     @dtypes(*floating_types_and(torch.half))
     @dtypesIfCUDA(*floating_types_and(torch.half))
     def test_bernoulli_edge_cases(self, device, dtype):
@@ -2182,7 +2185,7 @@ class TestTorchDeviceType(TestCase):
         with self.assertRaises(RuntimeError):
             torch.empty((1,), device=device, dtype=dtype).exponential_(-0.5)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @dtypes(torch.half, torch.float)
     def test_exponential_no_zero(self, device, dtype):
         # naively, 0 in exponential can be generated with probability 2^-24
@@ -2382,8 +2385,8 @@ class TestTorchDeviceType(TestCase):
                 res = stats.kstest(t.cpu().to(torch.double), 'cauchy', args=(median, sigma))
                 self.assertTrue(res.statistic < 0.1)
 
-    @slowTest
-    @onlyCUDA
+    # @slowTest
+    @onlyPRIVATEUSE1
     @dtypes(torch.bfloat16, torch.float32)
     def test_cauchy_no_inf(self, device, dtype):
         # torch.float16 will have `inf` because of its smaller range.
@@ -2508,8 +2511,8 @@ class TestTorchDeviceType(TestCase):
                             expected = self._brute_cdist(x, y, p=p)
                             self.assertEqual(expected, actual)
 
-    @onlyCUDA
-    def test_cdist_cuda_backward(self, device):
+    @onlyPRIVATEUSE1
+    def test_cdist_npu_backward(self, device):
         for l1 in [1, 511, 513]:
             for l2 in [1, 511, 513]:
                 for p in [0, 1, 2, 3, 1.5, 2.5, float('inf')]:
@@ -2532,7 +2535,7 @@ class TestTorchDeviceType(TestCase):
                         self.assertEqual(y1.grad, y2.grad, rtol=0, atol=0.001)
 
     @skipIfRocmArch(MI300_ARCH)
-    @tf32_on_and_off(0.005)
+    # @tf32_on_and_off(0.005)
     @reduced_f32_on_and_off(0.08)
     def test_cdist_large(self, device):
         for cm in ['use_mm_for_euclid_dist_if_necessary', 'use_mm_for_euclid_dist', 'donot_use_mm_for_euclid_dist']:
@@ -2542,8 +2545,8 @@ class TestTorchDeviceType(TestCase):
             expected = self._brute_cdist(x, y, p=2)
             self.assertEqual(expected, actual)
 
-    @slowTest
-    @tf32_on_and_off(0.01)
+    # @slowTest
+    # @tf32_on_and_off(0.01)
     @reduced_f32_on_and_off(0.08)
     def test_cdist_large_batch(self, device):
         for cm in ['use_mm_for_euclid_dist_if_necessary', 'use_mm_for_euclid_dist', 'donot_use_mm_for_euclid_dist']:
@@ -2553,7 +2556,7 @@ class TestTorchDeviceType(TestCase):
             expected = self._brute_cdist(x, y, p=2)
             self.assertEqual(expected, actual)
 
-    @tf32_on_and_off(0.005)
+    # @tf32_on_and_off(0.005)
     @reduced_f32_on_and_off(0.04)
     def test_cdist_non_contiguous(self, device):
         for cm in ['use_mm_for_euclid_dist', 'donot_use_mm_for_euclid_dist']:
@@ -2581,7 +2584,7 @@ class TestTorchDeviceType(TestCase):
             self.assertTrue(y.is_contiguous())
             self.assertEqual(expected, actual)
 
-    @tf32_on_and_off(0.005)
+    # @tf32_on_and_off(0.005)
     @reduced_f32_on_and_off(0.04)
     def test_cdist_non_contiguous_batch(self, device):
         for cm in ['use_mm_for_euclid_dist', 'donot_use_mm_for_euclid_dist']:
@@ -3140,10 +3143,10 @@ class TestTorchDeviceType(TestCase):
 
     @unittest.skipIf(IS_FBCODE and IS_REMOTE_GPU, "sandcastle OOM with current tpx gpu/re configuration")
     @unittest.skipIf(IS_JETSON, "psutil issue for largeTensorTest. Too large for Jetson.")
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @dtypes(torch.half)  # only small dtype not to get oom
     @largeTensorTest('25GB', device='cpu')
-    @largeTensorTest('4GB', device='cuda')
+    @largeTensorTest('4GB', device='npu')
     def test_large_cumsum(self, device, dtype):
         # initialization to avoid overflow and half caveats
         x = torch.empty(2**30 + 200, device=device, dtype=dtype)
@@ -3152,10 +3155,10 @@ class TestTorchDeviceType(TestCase):
         x[2::3] = 1
         self._test_large_cum_fn_helper(x, lambda x: torch.cumsum(x, 0))
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @dtypes(torch.half)  # only small dtype not to get oom
     @largeTensorTest('25GB', device='cpu')
-    @largeTensorTest('4GB', device='cuda')
+    @largeTensorTest('4GB', device='npu')
     @unittest.skipIf(IS_JETSON, "psutil issue for largeTensorTest. Too large for Jetson.")
     def test_large_cumprod(self, device, dtype):
         # initialization to avoid overflow and half caveats
@@ -3334,7 +3337,7 @@ class TestTorchDeviceType(TestCase):
 
     # FIXME: move to elementwise ternary test suite
     @parametrize("use_cpu_scalar", [True, False])
-    @dtypesIfCUDA(*set(get_all_math_dtypes('cuda')))
+    @dtypesIfCUDA(*set(get_all_math_dtypes('npu')))
     @dtypes(*set(get_all_math_dtypes('cpu')))
     def test_addcmul(self, device, dtype, use_cpu_scalar):
         # Returns floating or integral scalar corresponding to dtype
@@ -3372,15 +3375,15 @@ class TestTorchDeviceType(TestCase):
                 UserWarning, "This overload of addcmul is deprecated"):
             self.assertEqual(actual, torch.addcmul(a, alpha, b, c))
 
-        if self.device_type == 'cuda' and dtype == torch.half:
+        if self.device_type == 'npu' and dtype == torch.half:
             a = torch.tensor([60000.0], device=device, dtype=dtype)
             b = torch.tensor([60000.0], device=device, dtype=dtype)
             c = torch.tensor([2.0], device=device, dtype=dtype)
             out = torch.addcmul(a, b, c, value=-1)
             self.assertTrue(not (out.isnan() or out.isinf()))
 
-    @onlyCUDA
-    def test_addcmul_cuda_errors_with_cpu_scalars(self, device):
+    @onlyPRIVATEUSE1
+    def test_addcmul_npu_errors_with_cpu_scalars(self, device):
         # Logic is dtype agnostic, so dtype isn't tested
         alpha = 0.5
 
@@ -3561,7 +3564,7 @@ class TestTorchDeviceType(TestCase):
     # FIXME: port to test_scatter_gather_ops.py
     def scatter_allow_reduce(self, device, dtype, reduceop):
         device_type = torch.device(device).type
-        return device_type != 'cuda' or (reduceop == 'multiply' and dtype.is_floating_point)
+        return device_type != 'npu' or (reduceop == 'multiply' and dtype.is_floating_point)
 
     @dtypes(*floating_and_complex_types())
     @dtypesIfCPU(*all_types_and_complex_and(torch.half, torch.bool, torch.bfloat16))
@@ -3652,7 +3655,7 @@ class TestTorchDeviceType(TestCase):
             input.scatter_(0, index, src, reduce=operation)
             self.assertEqual(input, result, msg=f"result: {result} input: {input} method: {str(operation)}")
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @dtypes(*complex_types())
     def test_scatter_reduce_multiply_unsupported_dtypes(self, device, dtype):
         height = 2
@@ -3733,7 +3736,7 @@ class TestTorchDeviceType(TestCase):
         # in order to avoid synchronization, but this means
         # we can not clear the failures. So there is no way
         # to test it then recover.
-        if self.device_type != 'cuda':
+        if self.device_type != 'npu':
             # make src smaller. this should fail
             src = torch.zeros(num_copy - 1, dtype=dt, device=device)
             with self.assertRaises(RuntimeError):
@@ -3766,7 +3769,7 @@ class TestTorchDeviceType(TestCase):
 
     # FIXME: find a test suite for the masked scatter operator
     #   test_scatter_gather_ops or test_masked_ops?
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @largeTensorTest('30GB')
     def test_masked_scatter_large_tensor(self, device):
         t_cpu = torch.empty(2**31 + 1, dtype=torch.bool).random_()
@@ -4104,9 +4107,9 @@ class TestTorchDeviceType(TestCase):
     # FIXME: find a test suite for the pdist operator
     @unittest.skipIf(IS_FBCODE and IS_REMOTE_GPU, "sandcastle OOM with current tpx gpu/re configuration")
     @skipIfRocm
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @largeTensorTest('32GB', device='cpu')
-    @largeTensorTest('5GB', device='cuda')
+    @largeTensorTest('5GB', device='npu')
     def test_pdist_norm_large(self, device):
         # use dim0>=46342 for forward, see:
         # https://github.com/pytorch/pytorch/issues/30583
@@ -4120,7 +4123,7 @@ class TestTorchDeviceType(TestCase):
 
     # FIXME: move to elementwise ternary test suite
     @onlyNativeDeviceTypes
-    @dtypesIfCUDA(*set(get_all_math_dtypes('cuda')))
+    @dtypesIfCUDA(*set(get_all_math_dtypes('npu')))
     @dtypes(*set(get_all_math_dtypes('cpu')))
     def test_addcdiv(self, device, dtype):
         # Returns floating or integral scalar corresponding to dtype
@@ -4162,7 +4165,7 @@ class TestTorchDeviceType(TestCase):
         else:
             _test_addcdiv()
 
-        if self.device_type == 'cuda' and dtype == torch.half:
+        if self.device_type == 'npu' and dtype == torch.half:
             a = torch.tensor([60000.0], device=device, dtype=dtype)
             b = torch.tensor([60000.0], device=device, dtype=dtype)
             c = torch.tensor([1.0], device=device, dtype=dtype)
@@ -4196,11 +4199,11 @@ class TestTorchDeviceType(TestCase):
 
         ops = [
             ("addcmul", True, True, 'cpu'),
-            ("addcmul", True, True, 'cuda'),
+            ("addcmul", True, True, 'npu'),
             ("addcdiv", True, True, 'cpu'),
-            ("addcdiv", True, True, 'cuda'),
+            ("addcdiv", True, True, 'npu'),
             ("lerp", True, True, 'cpu'),
-            ("lerp", True, True, 'cuda')
+            ("lerp", True, True, 'npu')
         ]
 
         for (fn, has_input_output_mem_overlap_check,
@@ -4243,7 +4246,7 @@ class TestTorchDeviceType(TestCase):
         with self.assertRaisesRegex(RuntimeError, 'unsupported operation'):
             ind.index_add_(0, ind.clone(), ind)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @skipCUDAIfNotRocm  # This UT throws an OOM error on CUDA
     def test_index_add_large_inputs(self, device):
         D = 6144
@@ -4407,7 +4410,7 @@ class TestTorchDeviceType(TestCase):
             ind.scatter_(0, ind, ind.clone())
 
     # FIXME: move to test distributions
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     def test_multinomial_device_constrain(self, device):
         x = torch.empty(3, device="cpu")
         y = torch.empty(3, device=device)
@@ -4417,7 +4420,7 @@ class TestTorchDeviceType(TestCase):
 
     # FIXME: move to test distributions
     @deviceCountAtLeast(2)
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @skipIfTorchInductor("FIXME: error not thrown")
     def test_multinomial_gpu_device_constrain(self, devices):
         x = torch.empty(3, device=devices[0])
@@ -4428,7 +4431,7 @@ class TestTorchDeviceType(TestCase):
 
     # FIXME: convert this to an automated OpInfo test
     @deviceCountAtLeast(2)
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     def test_device_guard(self, devices):
         # verify that all operators with `device_guard: False` behave properly with multiple devices.
         # TODO: if we had operator introspection we could figure out this set of operators automatically...
@@ -4524,10 +4527,10 @@ class TestTorchDeviceType(TestCase):
 
     def test_tensor_type(self):
         for t in torch._tensor_classes:
-            if 'cuda' in t.__module__:
-                self.assertEqual(t.is_cuda, True)
+            if 'npu' in t.__module__:
+                self.assertEqual(t.is_npu, True)
             else:
-                self.assertEqual(t.is_cuda, False)
+                self.assertEqual(t.is_npu, False)
             if 'xpu' in t.__module__:
                 self.assertEqual(t.is_xpu, True)
             else:
@@ -4536,7 +4539,7 @@ class TestTorchDeviceType(TestCase):
     # Note - reports a leak of 512 bytes on CUDA device 1
     @deviceCountAtLeast(2)
     @skipCUDAMemoryLeakCheckIf(True)
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     def test_tensor_set_errors_multigpu(self, devices):
         f_cuda0 = torch.randn((2, 3), dtype=torch.float32, device=devices[0])
         f_cuda1 = torch.randn((2, 3), dtype=torch.float32, device=devices[1])
@@ -4547,7 +4550,7 @@ class TestTorchDeviceType(TestCase):
         self.assertRaises(RuntimeError, lambda: f_cuda0.set_(f_cuda1))
 
     # FIXME: move to test_serialization
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @deviceCountAtLeast(1)  # Note: Tests works with one but prefers more devices
     def test_serialization(self, devices):
         def _test_serialization(filecontext_lambda):
@@ -4874,7 +4877,7 @@ class TestTorchDeviceType(TestCase):
             for x in xs:
                 _test_helper(x, op, unary=True)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @unittest.skipIf(PYTORCH_CUDA_MEMCHECK, "is_pinned uses failure to detect pointer property")
     @skipIfTorchDynamo("NotImplementedError: PrimTorch does not support pinned memory")
     def test_pin_memory_from_constructor(self, device):
@@ -4910,7 +4913,7 @@ class TestTorchDeviceType(TestCase):
             self.assertFalse(x.is_pinned())
 
     @deviceCountAtLeast(1)
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @parametrize("non_blocking", (True, False))
     def test_storage_all_devices(self, devices, non_blocking):
         for device in devices:
@@ -5196,7 +5199,7 @@ class TestTorchDeviceType(TestCase):
         self.assertEqual(sample_indices.size(1), n_sample, msg="wrong number of samples")
 
     # FIXME: move to test distributions
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @dtypes(torch.float, torch.double, torch.half)
     def test_multinomial_deterministic(self, device, dtype):
         gen = torch.Generator(device=device)
@@ -5399,7 +5402,7 @@ class TestTorchDeviceType(TestCase):
             self._test_memory_format_transformations(
                 device, get_generator(mf, shape, torch.float64), get_fn('float'), mf, default_is_preserve=True)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     def test_memory_format_cpu_and_cuda_ops(self, device):
         def get_generator(memory_format, shape):
             def input_generator_fn(device):
@@ -5418,7 +5421,7 @@ class TestTorchDeviceType(TestCase):
 
         for mf, shape in formats_shapes:
             self._test_memory_format_transformations(
-                'cuda', get_generator(mf, shape), transformation_cpu_fn, mf, default_is_preserve=True)
+                'npu', get_generator(mf, shape), transformation_cpu_fn, mf, default_is_preserve=True)
             self._test_memory_format_transformations(
                 'cpu', get_generator(mf, shape), transformation_cuda_fn, mf, default_is_preserve=True)
 
@@ -5438,7 +5441,7 @@ class TestTorchDeviceType(TestCase):
         GradScaler = partial(torch.GradScaler, device=device.type)
         for lazy_init_scale in try_lazy_inits:
             a = GradScaler(init_scale=3., growth_factor=4., backoff_factor=.5, growth_interval=2)
-            if device.type == "cuda":
+            if device.type == "npu":
                 self.assertTrue(not a.is_enabled() if torch.cuda.amp.common.amp_definitely_not_available() else a.is_enabled())
             else:
                 self.assertTrue(a.is_enabled())
@@ -5483,7 +5486,7 @@ class TestTorchDeviceType(TestCase):
     @dtypes(torch.float, torch.double)
     def test_grad_scaling_unscale(self, device, dtype):
         device = torch.device(device)
-        device0 = "cuda:0" if device.type == "cuda" else "cpu"
+        device0 = "npu:0" if device.type == "npu" else "cpu"
         inv_scale = torch.full((1,), 0.25, dtype=torch.float, device=device0)
         found_inf = torch.full((1,), 0.0, dtype=torch.float, device=device0)
 
@@ -5533,7 +5536,7 @@ class TestTorchDeviceType(TestCase):
 
         # Passing lists with mismatched devices to a raw
         # _amp_foreach_non_finite_check_and_unscale_ call should raise errors.
-        if device.type == "cuda" and TEST_MULTIGPU:
+        if device.type == "npu" and TEST_MULTIGPU:
             with self.assertRaisesRegex(RuntimeError, r"Expected all tensors to be on the same device"):
                 torch._amp_foreach_non_finite_check_and_unscale_([g.clone(), g.to(device="cuda:1")],
                                                                  found_inf,
@@ -5545,7 +5548,7 @@ class TestTorchDeviceType(TestCase):
         # If inject_inf >= 0, writes an inf into one grad for _unscale_grads_ to find.
         def perfect_storm_grads(inject_inf):
             grads = [g.clone(), g.clone()[:, :5], g.to(dtype=torch.float16), g.to(dtype=torch.float16)]
-            if device.type == "cuda" and TEST_MULTIGPU:
+            if device.type == "npu" and TEST_MULTIGPU:
                 grads += [g.to(device="cuda:1"),
                           g.to(device="cuda:1")[:, :5],
                           g.to(device="cuda:1", dtype=torch.float16),
@@ -5674,7 +5677,7 @@ class TestTorchDeviceType(TestCase):
             if lazy_init_scale:
                 # Dummy scale() call to ensure the scale tensor is lazily initialized.
                 s1.scale(torch.full((1,), 4.0, dtype=torch.float32, device=device))
-                if "cuda" == device.type:
+                if "npu" == device.type:
                     self.assertTrue(isinstance(s1._scale, torch.cuda.FloatTensor))
                 else:
                     self.assertTrue(isinstance(s1._scale, torch.FloatTensor))
@@ -6049,7 +6052,7 @@ class TestTorchDeviceType(TestCase):
     @onlyNativeDeviceTypes
     def test_grad_scaler_deprecated_warning(self, device):
         device = torch.device(device)
-        GradScaler = torch.cuda.amp.GradScaler if "cuda" == device.type else torch.cpu.amp.GradScaler
+        GradScaler = torch.cuda.amp.GradScaler if "npu" == device.type else torch.cpu.amp.GradScaler
 
         with self.assertWarnsRegex(
             FutureWarning,
@@ -6123,7 +6126,7 @@ class TestTorchDeviceType(TestCase):
 
                 check_equal(condition, x, y)
                 check_equal(condition, y, x)
-                if self.device_type == "cuda":
+                if self.device_type == "npu":
                     check_equal(condition, torch.tensor(x), y)
                     check_equal(condition, y, torch.tensor(x))
                     if not isinstance(y, torch.Tensor):
@@ -6367,7 +6370,7 @@ class TestDevicePrecision(TestCase):
     exact_dtype = True
 
     # FIXME: move to indexing test suite
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     def test_index_add_bfloat16(self, device):
         inp_tensor = torch.randn(5, 3, device='cpu').bfloat16()
         t = torch.tensor([[1, 2, 3], [4, 5, 6], [7, 8, 9]], dtype=torch.bfloat16, device='cpu')
@@ -6399,7 +6402,6 @@ class TestDevicePrecision(TestCase):
     def test_multidevice_serialization(self, devices):
         x = [torch.randn(4, 4, device=devices[0]),
              torch.randn(4, 4, device=devices[1])]
-
         with tempfile.NamedTemporaryFile() as f:
             torch.save(x, f)
             f.seek(0)
@@ -6830,10 +6832,10 @@ class TestTorch(TestCase):
             # Low-precision types (float16, bfloat16) on GPU have non-deterministic
             # accumulation order, leading to larger rounding differences.
             # See: https://github.com/pytorch/pytorch/issues/91184
-            if device == 'cuda' and dtype in (torch.half, torch.bfloat16):
+            if device == 'npu' and dtype in (torch.half, torch.bfloat16):
                 # Relaxed tolerance for low-precision GPU accumulation
                 atol, rtol = 1e-1, 1e-1
-            elif device == 'cuda':
+            elif device == 'npu':
                 atol, rtol = 1e-2, 1e-2
             else:
                 # scatter_add uses fp32 as accumulate type, while index_add doesn't.
@@ -7055,7 +7057,7 @@ class TestTorch(TestCase):
 
         # change device
         if torch.cuda.is_available():
-            f_cuda = torch.randn((2, 3), dtype=torch.float32, device='cuda')
+            f_cuda = torch.randn((2, 3), dtype=torch.float32, device='npu')
 
             # cpu -> cuda
             self.assertRaises(RuntimeError, lambda: f_cpu.set_(f_cuda.storage()))
@@ -7073,8 +7075,8 @@ class TestTorch(TestCase):
     # NOTE: test_equal will be deprecated in favor of torch.testing.assert_close
     #   once torch.testing is out of beta
     def test_equal(self):
-        for device in ["cpu", "cuda"]:
-            if device == "cuda" and not torch.cuda.is_available():
+        for device in ["cpu", "npu"]:
+            if device == "npu" and not torch.cuda.is_available():
                 continue
 
             # Contiguous, 1D
@@ -7361,7 +7363,7 @@ class TestTorch(TestCase):
     def test_pickle_generator(self) -> None:
         devices = ['cpu']
         if torch.cuda.is_available():
-            devices += ['cuda']
+            devices += ['npu']
 
         for device in devices:
             with self.subTest(device=device):
@@ -7648,10 +7650,10 @@ class TestTorch(TestCase):
             if storage_class in [torch.UntypedStorage, torch.TypedStorage]:
                 continue
 
-            device = 'cuda' if storage_class.__module__ == 'torch.cuda' else 'cpu'
+            device = 'npu' if storage_class.__module__ == 'torch.npu' else 'cpu'
             dtype = storage_class.dtype
 
-            if device == 'cuda' and not torch.cuda.is_available():
+            if device == 'npu' and not torch.cuda.is_available():
                 continue
 
             # Legacy <type>Storage constructor errors
@@ -7718,7 +7720,7 @@ class TestTorch(TestCase):
             if torch.cuda.is_available():
                 if storage_class in quantized_storages:
                     with self.assertRaisesRegex(RuntimeError, r"Cannot create CUDA storage with quantized dtype"):
-                        torch.TypedStorage(dtype=dtype, device='cuda')
+                        torch.TypedStorage(dtype=dtype, device='npu')
 
             with self.assertRaisesRegex(TypeError, r"Argument type not recognized"):
                 torch.TypedStorage(torch.tensor([]), dtype=dtype, device=device)
@@ -7738,13 +7740,13 @@ class TestTorch(TestCase):
             torch.cuda.FloatStorage,
         ]
         for storage_class in storage_classes:
-            with self.assertRaisesRegex(RuntimeError, r'Not available for CUDA storage'):
+            with self.assertRaisesRegex(RuntimeError, r'Not available for NPU storage'):
                 storage_class.from_buffer()
 
-            with self.assertRaisesRegex(RuntimeError, r'Not available for CUDA storage'):
+            with self.assertRaisesRegex(RuntimeError, r'Not available for NPU storage'):
                 storage_class._new_with_weak_ptr()
 
-            with self.assertRaisesRegex(RuntimeError, r'Not available for CUDA storage'):
+            with self.assertRaisesRegex(RuntimeError, r'Not available for NPU storage'):
                 storage_class._new_shared_filename(0, 0, 0)
 
     def test_storage_casts(self):
@@ -7921,13 +7923,13 @@ class TestTorch(TestCase):
         if torch.cuda.is_available():
             s1 = torch.cuda.FloatStorage(10)
             s1_untyped = s1.untyped()
-            t1 = torch.randn(10, device='cuda')
+            t1 = torch.randn(10, device='npu')
 
             funcs += [
                 lambda: torch.cuda.FloatStorage(_internal=True),
                 lambda: torch.TypedStorage(
                     dtype=torch.float,
-                    device='cuda',
+                    device='npu',
                     _internal=True),
                 lambda: torch.TypedStorage(
                     wrap_storage=s1_untyped,
@@ -8085,6 +8087,8 @@ class TestTorch(TestCase):
     def test_print(self):
         default_type = torch.tensor([]).type()
         for t in torch._tensor_classes:
+            if 'npu' in str(t):
+                continue
             if t == torch.HalfTensor:
                 continue  # HalfTensor does not support fill
             if t.is_sparse:
@@ -8200,9 +8204,9 @@ tensor([ 0.0000e+00, 9.8813e-324, 9.8813e-323, 1.0000e+307, 1.0000e+308,
 
         # test device
         if torch.cuda.is_available():
-            x = torch.tensor([123], device='cuda:0')
+            x = torch.tensor([123], device='npu:0')
             self.assertEqual(x.__repr__(), str(x))
-            self.assertExpectedInline(str(x), '''tensor([123], device='cuda:0')''')
+            self.assertExpectedInline(str(x), '''tensor([123], device='npu:0')''')
 
             # test changing default to cuda
             torch.set_default_tensor_type(torch.cuda.FloatTensor)
@@ -8213,7 +8217,7 @@ tensor([ 0.0000e+00, 9.8813e-324, 9.8813e-323, 1.0000e+307, 1.0000e+308,
             if torch.cuda.device_count() >= 2:
                 with torch.cuda.device(1):
                     self.assertEqual(x.__repr__(), str(x))
-                    self.assertExpectedInline(str(x), '''tensor([123], device='cuda:0')''')
+                    self.assertExpectedInline(str(x), '''tensor([123], device='npu:0')''')
 
             # test printing cpu tensor when default device is cuda
             y = torch.tensor([123], device='cpu')
@@ -8552,7 +8556,7 @@ tensor([[[1.+1.j, 1.+1.j, 1.+1.j,  ..., 1.+1.j, 1.+1.j, 1.+1.j],
         self.assertRaises(RuntimeError, lambda: torch.randn(2, 3, 4).t_())
 
     # skip this test for now as it affects all tests
-    @unittest.skipIf(True, "flush_denormal not supported")
+    # @unittest.skipIf(True, "flush_denormal not supported")
     def test_set_flush_denormal(self):
         tiny_float = 1e-42
         tiny_double = 1e-320
@@ -8642,11 +8646,11 @@ tensor([[[1.+1.j, 1.+1.j, 1.+1.j,  ..., 1.+1.j, 1.+1.j, 1.+1.j],
     def test_cuda_not_built(self):
         msg = "Torch not compiled with CUDA enabled"
         self.assertRaisesRegex(AssertionError, msg, lambda: torch.cuda.current_device())
-        self.assertRaisesRegex(AssertionError, msg, lambda: torch.tensor([1], device="cuda"))
+        self.assertRaisesRegex(AssertionError, msg, lambda: torch.tensor([1], device="npu"))
         self.assertRaisesRegex(AssertionError, msg, lambda: torch.tensor([1]).cuda())
         self.assertRaisesRegex(TypeError, msg, lambda: torch.cuda.FloatTensor())
         self.assertRaisesRegex(TypeError, msg, lambda: torch.set_default_tensor_type(torch.cuda.FloatTensor))
-        self.assertRaisesRegex(AssertionError, msg, lambda: torch.tensor([1]).to(device="cuda"))
+        self.assertRaisesRegex(AssertionError, msg, lambda: torch.tensor([1]).to(device="npu"))
 
     def test_has_internal_overlap(self):
         OVERLAP_NO = 0
@@ -9340,11 +9344,11 @@ tensor([[[1.+1.j, 1.+1.j, 1.+1.j,  ..., 1.+1.j, 1.+1.j, 1.+1.j],
             self.assertIsNot(t, t.to(torch.empty_like(t), non_blocking=non_blocking, copy=True))
 
             devices = [t.device]
-            if t.device.type == 'cuda':
+            if t.device.type == 'npu':
                 if t.device.index == -1:
                     devices.append(f'cuda:{torch.cuda.current_device()}')
                 elif t.device.index == torch.cuda.current_device():
-                    devices.append('cuda')
+                    devices.append('npu')
             for device in devices:
                 self.assertIs(t, t.to(device, non_blocking=non_blocking))
                 self.assertIs(t, t.to(device, t.dtype, non_blocking=non_blocking))
@@ -9382,7 +9386,7 @@ tensor([[[1.+1.j, 1.+1.j, 1.+1.j,  ..., 1.+1.j, 1.+1.j, 1.+1.j],
 
         if torch.cuda.is_available():
             for non_blocking in [True, False]:
-                for cuda in ['cuda', 'cuda:0' if torch.cuda.device_count() == 1 else 'cuda:1']:
+                for cuda in ['npu', 'npu:0' if torch.cuda.device_count() == 1 else 'cuda:1']:
                     b = torch.tensor(5., device=cuda)
                     test_copy_behavior(b, non_blocking)
                     self.assertEqual(b.device, b.to(cuda, non_blocking=non_blocking).device)
@@ -9482,7 +9486,7 @@ tensor([[[1.+1.j, 1.+1.j, 1.+1.j,  ..., 1.+1.j, 1.+1.j, 1.+1.j],
         self.assertEqual(x[0:-1:2].tolist(), [[0, 1, 2, 3], [8, 9, 10, 11]])
 
     def test_split_with_sizes_copy_out(self):
-        device = torch.device("cuda:0") if torch.cuda.is_available() else torch.device("cpu")
+        device = torch.device("npu:0") if torch.cuda.is_available() else torch.device("cpu")
         shape = (30, 40, 50)
         x = torch.rand(*shape, device=device)
         cases = [
@@ -9552,7 +9556,7 @@ tensor([[[1.+1.j, 1.+1.j, 1.+1.j,  ..., 1.+1.j, 1.+1.j, 1.+1.j],
         self.assertIn('Unhandled exception caught in c10/util/AbortHandler.h', output)
 
     # FIXME: port to a distributed test suite
-    @slowTest
+    # @slowTest
     def test_multinomial_invalid_probs(self):
         def _spawn_method(self, method, arg):
             try:
@@ -9583,7 +9587,7 @@ tensor([[[1.+1.j, 1.+1.j, 1.+1.j,  ..., 1.+1.j, 1.+1.j, 1.+1.j],
 
         if torch.cuda.is_available():
             for non_blocking in [True, False]:
-                for cuda in ['cuda', 'cuda:0' if torch.cuda.device_count() == 1 else 'cuda:1']:
+                for cuda in ['npu', 'npu:0' if torch.cuda.device_count() == 1 else 'cuda:1']:
                     b = torch.tensor(5., device=cuda)
                     self.assertEqual(b.device, b.to(b, non_blocking=non_blocking).device)
                     self.assertEqual(a.device, b.to(a, non_blocking=non_blocking).device)
@@ -9605,24 +9609,24 @@ tensor([[[1.+1.j, 1.+1.j, 1.+1.j,  ..., 1.+1.j, 1.+1.j, 1.+1.j],
         self.assertEqual('cpu', cpu0.type)
         self.assertEqual(0, cpu0.index)
 
-        cuda = torch.device('cuda')
-        self.assertEqual('cuda', str(cuda))
-        self.assertEqual('cuda', cuda.type)
+        cuda = torch.device('npu')
+        self.assertEqual('npu', str(cuda))
+        self.assertEqual('npu', cuda.type)
         self.assertEqual(None, cuda.index)
 
         cuda1 = torch.device('cuda:1')
         self.assertEqual('cuda:1', str(cuda1))
-        self.assertEqual('cuda', cuda1.type)
+        self.assertEqual('npu', cuda1.type)
         self.assertEqual(1, cuda1.index)
 
-        cuda1 = torch.device('cuda', 1)
+        cuda1 = torch.device('npu', 1)
         self.assertEqual('cuda:1', str(cuda1))
-        self.assertEqual('cuda', cuda1.type)
+        self.assertEqual('npu', cuda1.type)
         self.assertEqual(1, cuda1.index)
 
-        cuda90 = torch.device('cuda', 90)
+        cuda90 = torch.device('npu', 90)
         self.assertEqual('cuda:90', str(cuda90))
-        self.assertEqual('cuda', cuda90.type)
+        self.assertEqual('npu', cuda90.type)
         self.assertEqual(90, cuda90.index)
 
         self.assertRaises(RuntimeError, lambda: torch.device('cpu:-1'))
@@ -9643,7 +9647,7 @@ tensor([[[1.+1.j, 1.+1.j, 1.+1.j,  ..., 1.+1.j, 1.+1.j, 1.+1.j],
         self.assertRaises(RuntimeError, lambda: torch.device('other'))
         self.assertRaises(RuntimeError, lambda: torch.device('other:0'))
 
-        device_set = {'cpu', 'cpu:0', 'cuda', 'cuda:0', 'cuda:1', 'cuda:10', 'cuda:100'}
+        device_set = {'cpu', 'cpu:0', 'npu', 'npu:0', 'cuda:1', 'cuda:10', 'cuda:100'}
         device_hash_set = set()
         device_hash_set.update(hash(torch.device(device)) for device in device_set)
         self.assertEqual(len(device_set), len(device_hash_set))
@@ -10626,8 +10630,8 @@ tensor([[[1.+1.j, 1.+1.j, 1.+1.j,  ..., 1.+1.j, 1.+1.j, 1.+1.j],
 
     @unittest.skipIf(not torch.cuda.is_available(), "CUDA not available")
     def test_bmm_matmul_mixed_dtype_error(self):
-        a = torch.randn(2, 8, 8, device="cuda", dtype=torch.float16)
-        b = torch.randn(2, 8, 64, device="cuda", dtype=torch.float32)
+        a = torch.randn(2, 8, 8, device="npu", dtype=torch.float16)
+        b = torch.randn(2, 8, 64, device="npu", dtype=torch.float32)
 
         with self.assertRaisesRegex(RuntimeError, "expected scalar type .* but found"):
             torch.bmm(a, b)
