diff --git a/test/test_unary_ufuncs.py b/test/test_unary_ufuncs.py
index ca5ab1e..067e950 100644
--- a/test/test_unary_ufuncs.py
+++ b/test/test_unary_ufuncs.py
@@ -9,7 +9,8 @@ import numpy as np
 import torch
 
 from torch import inf, nan
-
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 from torch.testing import make_tensor
 from torch.testing._internal.common_device_type import (
     dtypes,
