diff --git a/test/test_tensor_creation_ops.py b/test/test_tensor_creation_ops.py
index 87e7db57318..af812cbdc06 100644
--- a/test/test_tensor_creation_ops.py
+++ b/test/test_tensor_creation_ops.py
@@ -1,7 +1,7 @@
 # Owner(s): ["module: tensor creation"]
 # ruff: noqa: F841
-
 import torch
+from torch_npu.contrib import transfer_to_npu
 import numpy as np
 
 import sys
@@ -2679,8 +2679,8 @@ class TestTensorCreation(TestCase):
             torch.empty([0, 4, 2305843009213693952], dtype=torch.float32)
 
     def test_eye(self, device):
-        for dtype in all_types_and_complex_and(torch.half, torch.bool, torch.bfloat16):
-            if dtype == torch.bfloat16:
+            if dtype == torch.float64:
+        for dtype in all_types_and(torch.half, torch.bool):
                 continue
             # Test the RuntimeError is raised when either m or n is a negative number
             for n, m in ((-1, 1), (1, -1), (-1, -1)):
