diff --git a/test/test_nn.py b/test/test_nn.py
index 29a1b44..6e6bea0 100644
--- a/test/test_nn.py
+++ b/test/test_nn.py
@@ -24,6 +24,8 @@ from torch import inf, nan
 import torch.autograd.forward_ad as fwAD
 import torch.backends.cudnn as cudnn
 import torch.nn as nn
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 import torch.nn.functional as F
 import torch.nn.utils.rnn as rnn_utils
 from torch.nn.utils import clip_grad_norm_, clip_grad_value_, clip_grads_with_norm_, get_total_norm
@@ -57,6 +59,9 @@ from torch.testing._internal.common_utils import _assertGradAndGradgradChecks, g
     GRADCHECK_NONDET_TOL
 from torch.testing._internal.common_utils import dtype2prec_DONTUSE
 from torch.testing._internal.common_cuda import tf32_on_and_off, tf32_off, tf32_on
+if (acc := torch.accelerator.current_accelerator()) and acc.type == "npu":
+    torch.cuda.get_device_capability = lambda *args, **kwargs: (0, 0)
+    torch.cuda.is_tf32_supported = lambda: False
 from torch.types import _TensorOrTensors
 from torch.testing._internal.common_mkldnn import reduced_f32_on_and_off
 
@@ -150,7 +155,7 @@ class TestNN(NNTestCase):
         )
 
     def test_requires_grad_(self):
-        m = _create_basic_net()[-1]
+        m = _create_basic_net()[-1].npu()
         if len(list(m.buffers())) <= 0:
             raise AssertionError('invalid test: expected buffers')
         if not all(not b.requires_grad for b in m.buffers()):
