diff --git a/test/test_transformers.py b/test/test_transformers.py
index 798095e0657..6c5c698e096 100644
--- a/test/test_transformers.py
+++ b/test/test_transformers.py
@@ -1,11 +1,15 @@
 # Owner(s): ["module: sdpa"]
+import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
 
 import contextlib
 from functools import partial
 from collections import namedtuple
 import os
 import sys
-import torch
 import torch.nn as nn
 import torch.nn.functional as F
 from torch.nn.functional import scaled_dot_product_attention
@@ -3897,10 +3901,7 @@ class TestSDPAXpuOnly(NNTestCase):
         make_tensor = partial(rand_sdpa_tensor, type=type, device=device, dtype=dtype)
         size = SdpaShape(2, 8, 128, 64)
         q, k, v = make_tensor(size), make_tensor(size), make_tensor(size)
-        if dropout > 0.0 or dtype not in [torch.float32, torch.bfloat16, torch.float16]:
-            assert torch._fused_sdp_choice(q, k, v, dropout_p=dropout) == SDPBackend.MATH.value
-        else:
-            assert torch._fused_sdp_choice(q, k, v, dropout_p=dropout) == SDPBackend.OVERRIDEABLE.value
+        assert torch._fused_sdp_choice(q, k, v, dropout_p=dropout) == SDPBackend.OVERRIDEABLE.value
 
     def test_fused_attention_different_dk_dv(self, device):
         dtype = torch.bfloat16
