diff --git a/test/test_optim.py b/test/test_optim.py
index 23094907f94..43f2d38ec8e 100644
--- a/test/test_optim.py
+++ b/test/test_optim.py
@@ -13,6 +13,8 @@ from optim.test_optim import TestDifferentiableOptimizer  # noqa: F401
 from optim.test_swa_utils import TestSWAUtils  # noqa: F401
 
 import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 from torch.nn import Parameter
 from torch.optim import Optimizer, SGD
 from torch.optim.lr_scheduler import ReduceLROnPlateau
@@ -25,7 +27,7 @@ from torch.testing._internal.common_device_type import (
     instantiate_device_type_tests,
     largeTensorTest,
     onlyCPU,
-    onlyCUDA,
+    onlyPRIVATEUSE1,
     onlyNativeDeviceTypes,
     skipMPS,
     TEST_WITH_ROCM,
@@ -143,7 +145,7 @@ class TestOptimRenewed(TestCase):
         * Grads can also be None, empty, or zero-valued, and this should not disrupt training.
     """
 
-    @onlyCPU
+    # @onlyCPU
     @optims(optim_db)
     def test_optim_infos_do_not_specify_global_cliquey_kwargs(
         self, device, dtype, optim_info
@@ -260,8 +262,8 @@ class TestOptimRenewed(TestCase):
                 else:
                     self.assertLess(closure().item(), initial_value)
 
-    @onlyCUDA
-    @unittest.skipIf(not TEST_MULTIGPU, "only one GPU detected")
+    @onlyPRIVATEUSE1
+    # @unittest.skipIf(not TEST_MULTIGPU, "only one GPU detected")
     @parametrize("with_lrsched", [True, False])
     @optims(optim_db, dtypes=[torch.float32])
     def test_forloop_goes_right_direction_multigpu(
@@ -874,8 +876,8 @@ class TestOptimRenewed(TestCase):
     def test_foreach_matches_forloop(self, device, dtype, optim_info):
         self._test_derived_optimizers(device, dtype, optim_info, "foreach")
 
-    @onlyCUDA
-    @unittest.skipIf(not TEST_MULTIGPU, "only one GPU detected")
+    @onlyPRIVATEUSE1
+    # @unittest.skipIf(not TEST_MULTIGPU, "only one GPU detected")
     @parametrize("impl", ["foreach", "fused"])
     @optims(
         [
@@ -967,7 +969,7 @@ class TestOptimRenewed(TestCase):
                     actual = new_p_state[k]
                     self.assertEqual(og_p_state[k], actual, rtol=rtol, atol=atol)
 
-    @onlyCUDA
+    # @onlyPRIVATEUSE1
     @optims(
         [optim for optim in optim_db if "foreach" in optim.supported_impls],
         dtypes=[torch.float64],
@@ -995,7 +997,7 @@ class TestOptimRenewed(TestCase):
             finally:
                 torch.set_default_dtype(old_default_dtype)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @largeTensorTest("72GB", "cuda")
     @serialTest()
     @optims(
@@ -1011,7 +1013,7 @@ class TestOptimRenewed(TestCase):
             optimizer = optim_cls(params, foreach=True, **optim_input.kwargs)
             optimizer.step()
 
-    @onlyCUDA
+    # @onlyPRIVATEUSE1
     @optims(
         [optim for optim in optim_db if "foreach" in optim.supported_impls],
         dtypes=[torch.float32],
@@ -1181,7 +1183,7 @@ class TestOptimRenewed(TestCase):
             optimizer = optim_cls(params, fused=True, **optim_input.kwargs)
             optimizer.step()
 
-    @onlyCUDA
+    # @onlyPRIVATEUSE1
     @optims(
         [optim for optim in optim_db if "fused" in optim.supported_impls],
         dtypes=[torch.float32],
@@ -1812,7 +1814,7 @@ class TestOptimRenewed(TestCase):
             optimizer.load_state_dict(state_dict)
             optimizer.step(closure)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @optims(optim_db, dtypes=[torch.float32])
     def test_state_dict_with_cuda_params(self, device, dtype, optim_info):
         optim_cls = optim_info.optim_cls
@@ -2232,7 +2234,7 @@ class TestOptimRenewed(TestCase):
             res2 = optim_neg_inf.step(closure)
             self.assertEqual(type(res1), type(res2))
 
-    @onlyCUDA
+    # @onlyPRIVATEUSE1
     @optims(
         [
             optim
@@ -2280,7 +2282,7 @@ class TestOptimRenewed(TestCase):
                 optimizers.append(optimizer)
         self._compare_between(inpts, models, optimizers)
 
-    @onlyCUDA
+    # @onlyPRIVATEUSE1
     @optims(
         [
             o
@@ -2348,7 +2350,7 @@ class TestOptimRenewed(TestCase):
             for state in optim.state.values():
                 self.assertGreater(len(state), 0)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @parametrize("amsgrad", [False, True])
     @optims(
         [o for o in optim_db if o.optim_cls.__name__ in ["Adam", "AdamW"]],
@@ -2387,7 +2389,7 @@ class TestOptimRenewed(TestCase):
             if amsgrad:
                 self.assertEqual(state["max_exp_avg_sq"].dtype, torch.bfloat16)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @parametrize("amsgrad", [False, True])
     @optims(
         [o for o in optim_db if o.optim_cls.__name__ in ["Adam", "AdamW"]],
@@ -2444,7 +2446,7 @@ class TestOptimRenewed(TestCase):
             if amsgrad:
                 self.assertEqual(state["max_exp_avg_sq"].dtype, torch.bfloat16)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @optims(
         [o for o in optim_db if o.optim_cls.__name__ in ["Adam", "AdamW"]],
         dtypes=[torch.float32],
