diff --git a/test/test_ops.py b/test/test_ops.py
index 579b592e883..c999b15849e 100644
--- a/test/test_ops.py
+++ b/test/test_ops.py
@@ -22,11 +22,14 @@ from torch._subclasses.fake_utils import outputs_alias_inputs
 from torch.testing import make_tensor
 from torch.testing._internal import composite_compliance, opinfo
 from torch.testing._internal.common_cuda import with_tf32_off
+import torch_npu
+torch_npu.npu.config.allow_internal_format = False
+
 from torch.testing._internal.common_device_type import (
     deviceCountAtLeast,
     instantiate_device_type_tests,
     onlyCPU,
-    onlyCUDA,
+    onlyPRIVATEUSE1,
     onlyNativeDeviceTypesAnd,
     onlyOn,
     OpDTypes,
@@ -240,11 +243,13 @@ class TestCommon(TestCase):
                 raise AssertionError(err_msg)
 
     # Validates that each OpInfo works correctly on different CUDA devices
-    @onlyOn(["cuda", "xpu"])
+    @onlyOn(["npu", "xpu"])
     @deviceCountAtLeast(2)
     @ops(op_db, allowed_dtypes=(torch.float32, torch.long))
     def test_multiple_devices(self, devices, dtype, op):
         for cuda_device_str in devices:
+            if cuda_device_str.startswith('cpu'):
+                raise unittest.SkipTest("onlyPRIVATEUSE1 skip cpu")
             cuda_device = torch.device(cuda_device_str)
             # NOTE: only tests on first sample
             samples = op.sample_inputs(cuda_device, dtype)
@@ -479,7 +484,7 @@ class TestCommon(TestCase):
             and op.formatted_name
             in ("signal_windows_exponential", "signal_windows_bartlett")
             and dtype == torch.float64
-            and ("cuda" in device or "xpu" in device)
+            and ("npu" in device or "xpu" in device)
             or "cpu" in device
         ):  # noqa: E121
             raise unittest.SkipTest("XXX: raises tensor-likes are not close.")
@@ -492,7 +497,7 @@ class TestCommon(TestCase):
                 )
 
     # Tests that the cpu and gpu results are consistent
-    @onlyOn(["cuda", "xpu"])
+    @onlyOn(["npu", "xpu"])
     @suppress_warnings
     @skipCUDAIfNotRocm
     @ops(_ops_and_refs_with_no_numpy_ref, dtypes=OpDTypes.any_common_cpu_cuda_one)
@@ -502,6 +507,8 @@ class TestCommon(TestCase):
                 return arg.to(device="cpu")
             return arg
 
+        if cuda_device_str.startswith('cpu'):
+            raise unittest.SkipTest("onlyPRIVATEUSE1 skip cpu")
         samples = op.reference_inputs(device, dtype)
 
         for sample in samples:
@@ -762,11 +769,13 @@ class TestCommon(TestCase):
             )
         self._ref_test_helper(contextlib.nullcontext, device, dtype, op)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @ops(python_ref_db)
     @parametrize("executor", ["aten"])
     @skipIfTorchInductor("Takes too long for inductor")
     def test_python_ref_executor(self, device, dtype, op, executor):
+        if cuda_device_str.startswith('cpu'):
+            raise unittest.SkipTest("onlyCUDA skip cpu")
         from copy import copy
 
         from torch._prims.executor import make_traced
@@ -997,7 +1006,7 @@ class TestCommon(TestCase):
             # NOTE: only extracts on the CPU and CUDA device types since some
             #   device types don't have storage
             def _extract_data_ptrs(out):
-                if self.device_type != "cpu" and self.device_type != "cuda":
+                if self.device_type != "cpu" and self.device_type != "npu":
                     return ()
 
                 if isinstance(out, torch.Tensor):
@@ -1127,7 +1136,7 @@ class TestCommon(TestCase):
             # NOTE: only extracts on the CPU and CUDA device types since some
             #   device types don't have storage
             def _extract_data_ptrs(out):
-                if self.device_type != "cpu" and self.device_type != "cuda":
+                if self.device_type != "cpu" and self.device_type != "npu":
                     return ()
 
                 if isinstance(out, torch.Tensor):
@@ -2689,7 +2698,8 @@ fake_autocast_device_skips = defaultdict(dict)
 
 # TODO: investigate/fix
 fake_autocast_device_skips["cpu"] = {"linalg.pinv"}
-fake_autocast_device_skips["cuda"] = {"linalg.pinv", "pinverse"}
+fake_autocast_device_skips["npu"] = {"linalg.pinv", "pinverse"}
+fake_autocast_device_skips["npu"] = {"linalg.pinv", "pinverse"}
 
 
 dynamic_output_op_tests = (
@@ -2977,15 +2987,17 @@ class TestFakeTensor(TestCase):
             except torch._subclasses.fake_tensor.UnsupportedOperatorException:
                 pass
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @ops([op for op in op_db if op.supports_autograd], allowed_dtypes=(torch.float,))
     @skipOps(
         "TestFakeTensor", "test_fake_crossref_backward_no_amp", fake_backward_xfails
     )
     def test_fake_crossref_backward_no_amp(self, device, dtype, op):
+        if device.startswith('cpu'):
+            raise unittest.SkipTest("onlyPRIVATEUSE1 skip cpu")
         self._test_fake_crossref_helper(device, dtype, op, contextlib.nullcontext)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @ops([op for op in op_db if op.supports_autograd], allowed_dtypes=(torch.float,))
     @skipOps(
         "TestFakeTensor",
@@ -2993,7 +3005,9 @@ class TestFakeTensor(TestCase):
         fake_backward_xfails | fake_autocast_backward_xfails,
     )
     def test_fake_crossref_backward_amp(self, device, dtype, op):
-        self._test_fake_crossref_helper(device, dtype, op, torch.cuda.amp.autocast)
+        if device.startswith('cpu'):
+            raise unittest.SkipTest("onlyPRIVATEUSE1 skip cpu")
+        self._test_fake_crossref_helper(device, dtype, op, torch_npu.npu.amp.autocast)
 
     @ops([op for op in ops_and_refs if op.is_factory_function])
     def test_strided_layout(self, device, dtype, op):
