diff --git a/test/test_cuda_nvml_based_avail.py b/test/test_cuda_nvml_based_avail.py
index f890c35..849e1ae 100644
--- a/test/test_cuda_nvml_based_avail.py
+++ b/test/test_cuda_nvml_based_avail.py
@@ -7,7 +7,10 @@ import unittest
 from unittest.mock import patch
 
 import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 
+TEST_NPU = True
 
 # NOTE: Each of the tests in this module need to be run in a brand new process to ensure CUDA is uninitialized
 # prior to test initiation.
@@ -58,6 +61,7 @@ class TestExtendedCUDAIsAvail(TestCase):
     # `torch.cuda.is_available()` assessment. The NVML-based assessment should be attempted when
     # `PYTORCH_NVML_BASED_CUDA_CHECK` is set to 1, reverting to the default CUDA Runtime API check otherwise.
     # If the NVML-based assessment is attempted but fails, the CUDA Runtime API check should be executed
+    @unittest.skipIf(TEST_NPU, "NPU doesn't have nvml")
     @unittest.skipIf(IS_WINDOWS, "Needs fork")
     @parametrize("nvml_avail", [True, False])
     @parametrize("avoid_init", ["1", "0", None])
