diff --git a/test/test_torch.py b/test/test_torch.py
index 48a463a0d29..22ac4185d91 100644
--- a/test/test_torch.py
+++ b/test/test_torch.py
@@ -1,6 +1,69 @@
 # mypy: allow-untyped-decorators
 # mypy: allow-untyped-defs
 # Owner(s): ["module: tests"]
+                for cuda in ['npu', 'cuda:0' if torch.cuda.device_count() == 1 else 'cuda:1']:
+                for cuda in ['npu', 'cuda:0' if torch.cuda.device_count() == 1 else 'cuda:1']:
+            f_cuda = torch.randn((2, 3), dtype=torch.float32, device='npu')
+                'npu', get_generator(mf, shape), transformation_cpu_fn, mf, default_is_preserve=True)
+                self.assertEqual(t.is_npu, False)
+                self.assertEqual(t.is_npu, True)
+            if 'npu' in t.__module__:
+    @onlyPRIVATEUSE1
+    @onlyPRIVATEUSE1
+            ("lerp", True, True, 'npu')
+            ("addcdiv", True, True, 'npu'),
+            ("addcmul", True, True, 'npu'),
+    @dtypesIfCUDA(*set(get_all_math_dtypes('npu')))
+    @onlyPRIVATEUSE1
+    @dtypesIfCUDA(*set(get_all_math_dtypes('npu')))
+    @largeTensorTest('4GB', device='npu')
+    @onlyPRIVATEUSE1
+            torch.device(device).type == 'npu')
+            'grid_sampler_3d_backward_npu',
+            torch.device(device).type == 'npu')
+            'grid_sampler_2d_backward_npu',
+    @onlyPRIVATEUSE1
+            torch.device(device).type == 'npu')
+            'embedding_bag_backward_npu_max',
+            torch.device(device).type == 'npu')
+            'nll_loss2d_forward_out_npu_template',
+            torch.device(device).type == 'npu')
+            'replication_pad3d_backward_npu',
+            torch.device(device).type == 'npu')
+            'replication_pad1d_backward_npu',
+            torch.device(device).type == 'npu')
+            'reflection_pad3d_backward_out_npu',
+            torch.device(device).type == 'npu')
+            'reflection_pad1d_backward_out_npu',
+            torch.device(device).type == 'npu')
+            'upsample_trilinear3d_backward_out_npu',
+            torch.device(device).type == 'npu')
+            'upsample_bicubic2d_backward_out_npu',
+            'upsample_trilinear3d_backward_out_npu',
+            'upsample_bilinear2d_backward_out_npu',
+            torch.device(device).type == 'npu')
+            'upsample_bilinear2d_backward_out_npu',
+            torch.device(device).type == 'npu')
+            'upsample_linear1d_backward_out_npu',
+            torch.device(device).type == 'npu')
+            'fractional_max_pool3d_backward_npu',
+            torch.device(device).type == 'npu')
+            'fractional_max_pool2d_backward_npu',
+            torch.device(device).type == 'npu')
+            'adaptive_max_pool2d_backward_npu',
+            torch.device(device).type == 'npu')
+            'max_pool3d_with_indices_backward_npu',
+            torch.device(device).type == 'npu')
+            'adaptive_avg_pool3d_backward_npu',
+            torch.device(device).type == 'npu')
+            'adaptive_avg_pool2d_backward_npu',
+    @skipPRIVATEUSE1If(True, "NPU does not support deterministic empty, enabling it would cause significant performance degradation")
+            torch.device(device).type == 'npu')
+            'avg_pool3d_backward_npu',
+    @decorateIf(unittest.expectedFailure, lambda params: True)
+    @largeTensorTest('56GB', device='npu')
+    @onlyPRIVATEUSE1
+    @onlyPRIVATEUSE1
 
 import torch
 import torch.utils.data
@@ -44,6 +107,16 @@ from torch.testing._internal.common_utils import (  # type: ignore[attr-defined]
     bytes_to_scalar, parametrize, skipIfMPS, noncontiguous_like,
     AlwaysWarnTypedStorageRemoval, TEST_WITH_TORCHDYNAMO, xfailIfTorchDynamo,
     xfailIfS390X, set_warn_always_context, decorateIf, isRocmArchAnyOf)
+
+
+    onlyPRIVATEUSE1, onlyCPU,
+from torch_npu.contrib import transfer_to_npu
+import torch_npu
+    get_all_device_types, skipXLA, skipPRIVATEUSE1If)
+    onlyPRIVATEUSE1, onlyCPU,
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.version import __version__ as TORCH_NPU_VERSION
+import torch_npu
 from multiprocessing.reduction import ForkingPickler
 from torch.testing._internal.common_device_type import (
     expectedFailureMeta,
@@ -72,6 +145,7 @@ if TEST_WITH_TORCHINDUCTOR:
 else:
     from torch.testing._internal.common_utils import TestCase  # type: ignore[assignment]
 
+        if self.device_type == 'npu' and dtype == torch.half:
 
 # Protects against includes accidentally setting the default dtype
 if torch.get_default_dtype() is not torch.float32:
@@ -80,8 +154,10 @@ if torch.get_default_dtype() is not torch.float32:
 # load_tests from torch.testing._internal.common_utils is used to automatically filter tests for
 # sharding on sandcastle. This line silences flake warnings
 load_tests = load_tests  # noqa: PLW0127
+# AMPERE_OR_ROCM = TEST_WITH_ROCM or torch.cuda.is_tf32_supported()
+# AMPERE_OR_ROCM = TEST_WITH_ROCM or torch.cuda.is_tf32_supported()
+CURRENT_VERSION = tuple(int(part) for part in TORCH_NPU_VERSION.split("+")[0].split(".")[:2])
 
-AMPERE_OR_ROCM = TEST_WITH_ROCM or torch.cuda.is_tf32_supported()
 
 
 is_cuda_sm86 = torch.cuda.is_available() and torch.cuda.get_device_capability(0) == (8, 6)
@@ -90,6 +166,7 @@ class TestTorchDeviceType(TestCase):
     exact_dtype = True
 
     # TODO: move all tensor creation to common ops
+    @onlyPRIVATEUSE1
     def _rand_shape(self, dim, min_size, max_size):
         shape = []
         for _ in range(dim):
@@ -116,6 +193,8 @@ class TestTorchDeviceType(TestCase):
     @slowTestIf(IS_WINDOWS)
     @dtypes(torch.int8, torch.uint8, torch.int16, torch.int32, torch.int64,
             torch.bool, torch.float32, torch.complex64, torch.float64,
+        if torch.device(device).type == 'npu':
+        if torch.device(device).type == 'npu':
             torch.complex128, torch.uint16, torch.uint32, torch.uint64)
     def test_bytes_to_scalar(self, device, dtype):
         def rand_byte():
@@ -227,7 +306,7 @@ class TestTorchDeviceType(TestCase):
     def test_tensor_storage_type(self, device, dtype):
         a = make_tensor((10,), dtype=dtype, device=device, low=-9, high=9)
 
-        module = torch.cuda if (torch.device(device).type == 'cuda') else torch
+        module = torch.cuda if (torch.device(device).type == 'npu') else torch
         expected_storage_type = getattr(module, torch.storage._dtype_to_storage_type_map()[dtype])
 
         self.assertEqual(a.storage_type(), expected_storage_type)
@@ -285,6 +364,7 @@ class TestTorchDeviceType(TestCase):
         with self.assertRaisesRegex(NotImplementedError, r'Not available'):
             s[0]
 
+        for device in ['cpu', 'npu'] if torch.cuda.is_available() else ['cpu']:
         if isinstance(s, torch.TypedStorage):
             self.assertEqual(s.dtype, s_check.dtype)
             self._check_storage_meta(s.untyped(), s_check.untyped())
@@ -346,6 +426,10 @@ class TestTorchDeviceType(TestCase):
         if torch.cuda.is_available():
             with self.assertRaisesRegex(NotImplementedError, r'Cannot copy out'):
                 s0.cuda()
+            self.assertExpectedInline(str(x), '''tensor([123], device='npu:0')''')
+                    device='npu',
+            t1 = torch.randn(10, device='npu')
+                        torch.TypedStorage(dtype=dtype, device='npu')
 
             with self.assertRaisesRegex(RuntimeError, r'only available on CUDA'):
                 s0._share_cuda_()
@@ -378,7 +462,8 @@ class TestTorchDeviceType(TestCase):
         # This is OK, it changes the meta storage size without allocating
         s0.resize_(10)
 
-    @onlyCUDA
+        _model_cuda = model.to('npu')
+    @onlyPRIVATEUSE1
     def test_module_share_memory(self):
         # Test fix for issue #80733
         # See https://github.com/pytorch/pytorch/issues/80733
@@ -455,6 +540,7 @@ class TestTorchDeviceType(TestCase):
             with self.assertRaisesRegex(RuntimeError, 'unsupported operation'):
                 _test(op, data[0:sz], data[1:sz + 1])
         else:
+    @onlyPRIVATEUSE1
             with self.assertRaises(AssertionError):
                 with self.assertRaisesRegex(RuntimeError, 'unsupported operation'):
                     _test(op, data[0:sz], data[1:sz + 1])
@@ -710,6 +796,9 @@ class TestTorchDeviceType(TestCase):
         # given its similar to the other tests
         w = torch.randn(2, 1, 3, 3, device=device).div_(2).requires_grad_()
         self.assertRaises(RuntimeError, lambda: torch.nn.functional.conv2d(zero_d, w, groups=1))
+                '_bincount_npu',
+                torch.device(device).type == 'npu')
+                '_bincount_npu',
         self.assertRaises(RuntimeError, lambda: torch.nn.functional.conv2d(zero_d, w, groups=2))
 
         # nll_loss -- verify input can't be 0-dimensional.
@@ -827,7 +916,7 @@ class TestTorchDeviceType(TestCase):
         # Creates long string in advance to avoid a too-long Python line
         s = ".+Triggered internally at.+RangeFactories.+"
         # nvfuser deprecation warning filter
-        warnings.filterwarnings("ignore", "torch::jit::fuser::cuda", UserWarning)
+        warnings.filterwarnings("ignore", "torch::jit::fuser::npu", UserWarning)
 
         def cpp_warn_fn():
             out = torch.empty((5,))
@@ -941,9 +1030,9 @@ class TestTorchDeviceType(TestCase):
             # t + 1 allocates a new tensor for result using empty
             t + 1
 
-    @onlyCUDA
+        msg = 'The torch.npu.*DtypeTensor constructors are no longer recommended'
+    @onlyPRIVATEUSE1
     def test_dtypetensor_warnings(self, device):
-        msg = 'The torch.cuda.*DtypeTensor constructors are no longer recommended'
         with self.assertWarnsOnceRegex(UserWarning, msg):
             torch.cuda.FloatTensor([0])
 
@@ -987,8 +1076,8 @@ class TestTorchDeviceType(TestCase):
         out.backward(torch.ones_like(out).transpose(-2, -1))
 
     # TODO: this test should be in test_nn.py
-    @onlyCUDA
-    @largeTensorTest('12GB')
+    # @largeTensorTest('12GB')
+    # @onlyPRIVATEUSE1
     def test_conv_transposed_large(self, device):
         # ConvTranspose3d works for large input tensors (gh-32866)
         in_channels = 64
@@ -1198,7 +1287,10 @@ class TestTorchDeviceType(TestCase):
         self.check_nondeterministic_alert(
             lambda: b.resize_((10,)),
             'quantized_resize_cpu_')
+                torch.device(device).type == 'npu')
+            torch.device(device).type == 'npu')
 
+    @skipPRIVATEUSE1If(True, "NPU does not support deterministic resize, enabling it would cause significant performance degradation")
     @skipXLA
     @skipIfTorchInductor("https://github.com/pytorch/pytorch/issues/113707")
     @dtypes(*all_types_and_complex_and(torch.half, torch.bool, torch.bfloat16, torch.uint16, torch.uint32, torch.uint64))
@@ -1615,8 +1707,8 @@ class TestTorchDeviceType(TestCase):
         # nondeterministic
         self.check_nondeterministic_alert(
             lambda: res.backward(grad, retain_graph=True),
-            'replication_pad2d_backward_cuda',
-            torch.device(device).type == 'cuda')
+            torch.device(device).type == 'npu')
+            'replication_pad2d_backward_npu',
 
         with DeterministicGuard(True):
             res = module(input)
@@ -1627,7 +1719,7 @@ class TestTorchDeviceType(TestCase):
         # not be raised
         self.check_nondeterministic_alert(
             lambda: res.backward(grad, retain_graph=True),
-            'replication_pad2d_backward_cuda',
+            'replication_pad2d_backward_npu',
             False)
 
     @skipIfMPS
@@ -1707,6 +1799,11 @@ class TestTorchDeviceType(TestCase):
             res_cpu = input.cpu().cumsum(dim)
             self.assertEqual(res0, res_cpu, atol=1e-3, rtol=1e-2)
             # test double complex that has fewer threads than CTAs
+    # @largeTensorTest('49GB')
+    @onlyPRIVATEUSE1
+        b = torch.ones(2 * 4096 * 8, 100000, dtype=torch.float, device='npu')
+    # @largeTensorTest('49GB')
+    @onlyPRIVATEUSE1
             # that's a problem for large GPUS (H100 with 132 sms)
             # test is very tailored for that and will provide 0 signal
             # on smaller GPUs
@@ -1737,7 +1834,7 @@ class TestTorchDeviceType(TestCase):
         self.assertEqual(b[0, :], d[0, :], atol=3e-5, rtol=3e-5)
         self.assertEqual(b[-1, :], d[-1, :], atol=3e-5, rtol=3e-5)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @largeTensorTest('48GB')
     def test_cumsum_outer_dim_64bit_indexing(self, device):
         x = torch.zeros(309504, 1, 16384, device=device)
@@ -1780,8 +1877,8 @@ class TestTorchDeviceType(TestCase):
         for op_call in [torch.histc, torch.Tensor.histc]:
             self.check_nondeterministic_alert(
                 lambda: op_call(a, min=0, max=3),
-                '_histc_cuda with floating point input',
-                torch.device(device).type == 'cuda' and dtype.is_floating_point)
+                torch.device(device).type == 'npu' and dtype.is_floating_point)
+                '_histc_npu with floating point input',
 
     @skipIfMPS
     def test_nondeterministic_alert_bincount(self, device):
@@ -1918,13 +2015,13 @@ class TestTorchDeviceType(TestCase):
                 'median CUDA with indices output',
                 should_error)
 
-        is_cuda = torch.device(device).type == 'cuda'
+        test_func_expect_error('out with indices', is_npu)
+        test_func_expect_error('method with indices', is_npu)
+        test_func_expect_error('function with indices', is_npu)
+        is_npu = torch.device(device).type == 'npu'
 
         test_func_expect_error('function', False)
-        test_func_expect_error('function with indices', is_cuda)
         test_func_expect_error('method', False)
-        test_func_expect_error('method with indices', is_cuda)
-        test_func_expect_error('out with indices', is_cuda)
 
     # FIXME: move to test_scatter_gather_ops
     def _test_gather_backward_one_dim(self, device, deterministic: bool = False) -> None:
@@ -1941,7 +2038,7 @@ class TestTorchDeviceType(TestCase):
                 raise AssertionError("expected src.grad to be not None")
             grad = src.grad.detach().clone()
 
-            if torch.device(device).type == 'cuda' or torch.device(device).type == 'mtia':
+            if torch.device(device).type == 'npu' or torch.device(device).type == 'mtia':
                 for _ in range(2):
                     src.grad.data.zero_()
                     res = torch.gather(src, dim, idx)
@@ -2000,7 +2097,7 @@ class TestTorchDeviceType(TestCase):
         result = original.scatter(0, null_index, null_arr)
         self.assertEqual(result, original, atol=0, rtol=0)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @skipIfTorchInductor("FIXME")
     def test_sync_warning(self, device):
 
@@ -2140,7 +2237,7 @@ class TestTorchDeviceType(TestCase):
         t.bernoulli_(0.5)
         self.assertTrue(isBinary(t))
 
-        for p_dtype in floating_types_and(*[torch.half] if device.startswith('cuda') else []):
+        for p_dtype in floating_types_and(*[torch.half] if device.startswith('npu') else []):
             p = torch.rand(10, dtype=p_dtype, device=device).expand(10, 10)
             t.fill_(2)
             t.bernoulli_(p)
@@ -2154,7 +2251,7 @@ class TestTorchDeviceType(TestCase):
             t.bernoulli_(torch.rand_like(t, dtype=p_dtype))
             self.assertTrue(isBinary(t))
 
-    @slowTest
+    # @slowTest
     @dtypes(*floating_types_and(torch.half))
     @dtypesIfCUDA(*floating_types_and(torch.half))
     def test_bernoulli_edge_cases(self, device, dtype):
@@ -2182,7 +2279,7 @@ class TestTorchDeviceType(TestCase):
         with self.assertRaises(RuntimeError):
             torch.empty((1,), device=device, dtype=dtype).exponential_(-0.5)
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @dtypes(torch.half, torch.float)
     def test_exponential_no_zero(self, device, dtype):
         # naively, 0 in exponential can be generated with probability 2^-24
@@ -2382,8 +2479,8 @@ class TestTorchDeviceType(TestCase):
                 res = stats.kstest(t.cpu().to(torch.double), 'cauchy', args=(median, sigma))
                 self.assertTrue(res.statistic < 0.1)
 
-    @slowTest
-    @onlyCUDA
+    @onlyPRIVATEUSE1
+    # @slowTest
     @dtypes(torch.bfloat16, torch.float32)
     def test_cauchy_no_inf(self, device, dtype):
         # torch.float16 will have `inf` because of its smaller range.
@@ -2486,10 +2583,15 @@ class TestTorchDeviceType(TestCase):
                                 expected = self._brute_cdist(x, y, p=2)
                                 self.assertEqual(expected, actual, rtol=0, atol=0.02)
                         else:
+    # @tf32_on_and_off(0.005)
+    # @tf32_on_and_off(0.01)
+    # @slowTest
                             actual = torch.cdist(x, y, p=p)
                             expected = self._brute_cdist(x, y, p=p)
                             self.assertEqual(expected, actual)
 
+    def test_cdist_npu_backward(self, device):
+    @onlyPRIVATEUSE1
     @skipIfMPS
     def test_cdist_norm_batch(self, device):
         for r1 in [3, 4, 5, 6]:
@@ -2527,6 +2629,7 @@ class TestTorchDeviceType(TestCase):
                             self.assertEqual(y1.grad, y2.grad, rtol=0, atol=0.001)
                     else:
                         z1 = torch.cdist(x1, y1, p=p).mean()
+    # @tf32_on_and_off(0.005)
                         z2 = self._brute_cdist(x2, y2, p=p).mean()
                         self.assertEqual(x1.grad, x2.grad, rtol=0, atol=0.001)
                         self.assertEqual(y1.grad, y2.grad, rtol=0, atol=0.001)
@@ -2581,7 +2684,7 @@ class TestTorchDeviceType(TestCase):
             self.assertTrue(y.is_contiguous())
             self.assertEqual(expected, actual)
 
-    @tf32_on_and_off(0.005)
+    # @tf32_on_and_off(0.005)
     @reduced_f32_on_and_off(0.04)
     def test_cdist_non_contiguous_batch(self, device):
         for cm in ['use_mm_for_euclid_dist', 'donot_use_mm_for_euclid_dist']:
@@ -3152,10 +3255,10 @@ class TestTorchDeviceType(TestCase):
         x[2::3] = 1
         self._test_large_cum_fn_helper(x, lambda x: torch.cumsum(x, 0))
 
-    @onlyCUDA
+    @largeTensorTest('4GB', device='npu')
+    @onlyPRIVATEUSE1
     @dtypes(torch.half)  # only small dtype not to get oom
     @largeTensorTest('25GB', device='cpu')
-    @largeTensorTest('4GB', device='cuda')
     @unittest.skipIf(IS_JETSON, "psutil issue for largeTensorTest. Too large for Jetson.")
     def test_large_cumprod(self, device, dtype):
         # initialization to avoid overflow and half caveats
@@ -3372,7 +3475,9 @@ class TestTorchDeviceType(TestCase):
                 UserWarning, "This overload of addcmul is deprecated"):
             self.assertEqual(actual, torch.addcmul(a, alpha, b, c))
 
-        if self.device_type == 'cuda' and dtype == torch.half:
+    def test_addcmul_npu_errors_with_cpu_scalars(self, device):
+    @onlyPRIVATEUSE1
+        if self.device_type == 'npu' and dtype == torch.half:
             a = torch.tensor([60000.0], device=device, dtype=dtype)
             b = torch.tensor([60000.0], device=device, dtype=dtype)
             c = torch.tensor([2.0], device=device, dtype=dtype)
@@ -3561,7 +3666,7 @@ class TestTorchDeviceType(TestCase):
     # FIXME: port to test_scatter_gather_ops.py
     def scatter_allow_reduce(self, device, dtype, reduceop):
         device_type = torch.device(device).type
-        return device_type != 'cuda' or (reduceop == 'multiply' and dtype.is_floating_point)
+        return device_type != 'npu' or (reduceop == 'multiply' and dtype.is_floating_point)
 
     @dtypes(*floating_and_complex_types())
     @dtypesIfCPU(*all_types_and_complex_and(torch.half, torch.bool, torch.bfloat16))
@@ -3590,6 +3695,7 @@ class TestTorchDeviceType(TestCase):
             input.scatter_(0, index, src, reduce=operation)
             self.assertEqual(input, result)
 
+    @onlyPRIVATEUSE1
     @dtypes(*floating_and_complex_types())
     @dtypesIfCPU(*all_types_and_complex_and(torch.half, torch.bool, torch.bfloat16))
     @dtypesIfCUDA(*all_types_and_complex_and(torch.half, torch.bool, torch.bfloat16))
@@ -3733,7 +3839,7 @@ class TestTorchDeviceType(TestCase):
         # in order to avoid synchronization, but this means
         # we can not clear the failures. So there is no way
         # to test it then recover.
-        if self.device_type != 'cuda':
+        if self.device_type != 'npu':
             # make src smaller. this should fail
             src = torch.zeros(num_copy - 1, dtype=dt, device=device)
             with self.assertRaises(RuntimeError):
@@ -4104,9 +4210,9 @@ class TestTorchDeviceType(TestCase):
     # FIXME: find a test suite for the pdist operator
     @unittest.skipIf(IS_FBCODE and IS_REMOTE_GPU, "sandcastle OOM with current tpx gpu/re configuration")
     @skipIfRocm
-    @onlyCUDA
+    @largeTensorTest('5GB', device='npu')
+    @onlyPRIVATEUSE1
     @largeTensorTest('32GB', device='cpu')
-    @largeTensorTest('5GB', device='cuda')
     def test_pdist_norm_large(self, device):
         # use dim0>=46342 for forward, see:
         # https://github.com/pytorch/pytorch/issues/30583
@@ -4407,7 +4513,7 @@ class TestTorchDeviceType(TestCase):
             ind.scatter_(0, ind, ind.clone())
 
     # FIXME: move to test distributions
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     def test_multinomial_device_constrain(self, device):
         x = torch.empty(3, device="cpu")
         y = torch.empty(3, device=device)
@@ -4536,7 +4642,7 @@ class TestTorchDeviceType(TestCase):
     # Note - reports a leak of 512 bytes on CUDA device 1
     @deviceCountAtLeast(2)
     @skipCUDAMemoryLeakCheckIf(True)
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     def test_tensor_set_errors_multigpu(self, devices):
         f_cuda0 = torch.randn((2, 3), dtype=torch.float32, device=devices[0])
         f_cuda1 = torch.randn((2, 3), dtype=torch.float32, device=devices[1])
@@ -4544,6 +4650,7 @@ class TestTorchDeviceType(TestCase):
         self.assertRaises(RuntimeError, lambda: f_cuda0.set_(f_cuda1.storage()))
         self.assertRaises(RuntimeError,
                           lambda: f_cuda0.set_(f_cuda1.storage(), 0, f_cuda1.size(), f_cuda1.stride()))
+    @onlyPRIVATEUSE1
         self.assertRaises(RuntimeError, lambda: f_cuda0.set_(f_cuda1))
 
     # FIXME: move to test_serialization
@@ -4871,6 +4978,7 @@ class TestTorchDeviceType(TestCase):
             for x in xs:
                 _test_helper(x, op)
         for op in unary_ops:
+    @onlyPRIVATEUSE1
             for x in xs:
                 _test_helper(x, op, unary=True)
 
@@ -4910,7 +5018,7 @@ class TestTorchDeviceType(TestCase):
             self.assertFalse(x.is_pinned())
 
     @deviceCountAtLeast(1)
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @parametrize("non_blocking", (True, False))
     def test_storage_all_devices(self, devices, non_blocking):
         for device in devices:
@@ -5196,7 +5304,7 @@ class TestTorchDeviceType(TestCase):
         self.assertEqual(sample_indices.size(1), n_sample, msg="wrong number of samples")
 
     # FIXME: move to test distributions
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @dtypes(torch.float, torch.double, torch.half)
     def test_multinomial_deterministic(self, device, dtype):
         gen = torch.Generator(device=device)
@@ -5308,6 +5416,7 @@ class TestTorchDeviceType(TestCase):
             self._test_memory_format_transformations(
                 device, get_generator(mf, shape), transformation_fn, mf, default_is_preserve=True)
 
+    @onlyPRIVATEUSE1
     def test_memory_format_type(self, device):
         def get_generator(memory_format, shape):
             def input_generator_fn(device):
@@ -5445,6 +5554,7 @@ class TestTorchDeviceType(TestCase):
             if lazy_init_scale:
                 # Dummy a.scale() call lazy-inits a._scale Tensor.
                 a.scale(torch.tensor([4.0], dtype=torch.float32, device=device))
+                if "npu" == device.type:
                 self.assertTrue(a._scale.device.type == device.type)
             # The following three lines should work whether or not cuda is available.
             serialized = pickle.dumps(a)
@@ -6382,9 +6492,9 @@ class TestDevicePrecision(TestCase):
         self.assertEqual(out_cpu, out_gpu, atol=1e-2, rtol=0)
 
     # FIXME: move to serialization test suite
+    @unittest.skipIf(CURRENT_VERSION < (2, 9), "NPU serialization device location is supported since torch_npu 2.9")
     def test_device_serialization(self, device):
         x = torch.randn(4, 4, device=device)
-
         with tempfile.NamedTemporaryFile() as f:
             torch.save(x, f)
             f.seek(0)
@@ -6399,7 +6509,6 @@ class TestDevicePrecision(TestCase):
     def test_multidevice_serialization(self, devices):
         x = [torch.randn(4, 4, device=devices[0]),
              torch.randn(4, 4, device=devices[1])]
-
         with tempfile.NamedTemporaryFile() as f:
             torch.save(x, f)
             f.seek(0)
@@ -6830,10 +6939,10 @@ class TestTorch(TestCase):
             # Low-precision types (float16, bfloat16) on GPU have non-deterministic
             # accumulation order, leading to larger rounding differences.
             # See: https://github.com/pytorch/pytorch/issues/91184
-            if device == 'cuda' and dtype in (torch.half, torch.bfloat16):
+            elif device == 'npu':
+            if device == 'npu' and dtype in (torch.half, torch.bfloat16):
                 # Relaxed tolerance for low-precision GPU accumulation
                 atol, rtol = 1e-1, 1e-1
-            elif device == 'cuda':
                 atol, rtol = 1e-2, 1e-2
             else:
                 # scatter_add uses fp32 as accumulate type, while index_add doesn't.
@@ -7361,7 +7470,7 @@ class TestTorch(TestCase):
     def test_pickle_generator(self) -> None:
         devices = ['cpu']
         if torch.cuda.is_available():
-            devices += ['cuda']
+            devices += ['npu']
 
         for device in devices:
             with self.subTest(device=device):
@@ -7648,10 +7757,10 @@ class TestTorch(TestCase):
             if storage_class in [torch.UntypedStorage, torch.TypedStorage]:
                 continue
 
-            device = 'cuda' if storage_class.__module__ == 'torch.cuda' else 'cpu'
+            if device == 'npu' and not torch.cuda.is_available():
+            device = 'npu' if storage_class.__module__ == 'torch.npu' else 'cpu'
             dtype = storage_class.dtype
 
-            if device == 'cuda' and not torch.cuda.is_available():
                 continue
 
             # Legacy <type>Storage constructor errors
@@ -7738,13 +7847,13 @@ class TestTorch(TestCase):
             torch.cuda.FloatStorage,
         ]
         for storage_class in storage_classes:
-            with self.assertRaisesRegex(RuntimeError, r'Not available for CUDA storage'):
+            with self.assertRaisesRegex(RuntimeError, r'Not available for NPU storage'):
+            with self.assertRaisesRegex(RuntimeError, r'Not available for NPU storage'):
+            with self.assertRaisesRegex(RuntimeError, r'Not available for NPU storage'):
                 storage_class.from_buffer()
 
-            with self.assertRaisesRegex(RuntimeError, r'Not available for CUDA storage'):
                 storage_class._new_with_weak_ptr()
 
-            with self.assertRaisesRegex(RuntimeError, r'Not available for CUDA storage'):
                 storage_class._new_shared_filename(0, 0, 0)
 
     def test_storage_casts(self):
@@ -8085,6 +8194,10 @@ class TestTorch(TestCase):
     def test_print(self):
         default_type = torch.tensor([]).type()
         for t in torch._tensor_classes:
+                continue
+            if 'cuda' in str(t):
+                continue
+            if 'npu' in str(t):
             if t == torch.HalfTensor:
                 continue  # HalfTensor does not support fill
             if t.is_sparse:
@@ -8213,7 +8326,7 @@ tensor([ 0.0000e+00, 9.8813e-324, 9.8813e-323, 1.0000e+307, 1.0000e+308,
             if torch.cuda.device_count() >= 2:
                 with torch.cuda.device(1):
                     self.assertEqual(x.__repr__(), str(x))
-                    self.assertExpectedInline(str(x), '''tensor([123], device='cuda:0')''')
+                    self.assertExpectedInline(str(x), '''tensor([123], device='npu:0')''')
 
             # test printing cpu tensor when default device is cuda
             y = torch.tensor([123], device='cpu')
@@ -8551,8 +8664,8 @@ tensor([[[1.+1.j, 1.+1.j, 1.+1.j,  ..., 1.+1.j, 1.+1.j, 1.+1.j],
         self.assertRaises(RuntimeError, lambda: torch.randn(2, 3, 4).t())
         self.assertRaises(RuntimeError, lambda: torch.randn(2, 3, 4).t_())
 
+    # @unittest.skipIf(True, "flush_denormal not supported")
     # skip this test for now as it affects all tests
-    @unittest.skipIf(True, "flush_denormal not supported")
     def test_set_flush_denormal(self):
         tiny_float = 1e-42
         tiny_double = 1e-320
@@ -9340,11 +9453,11 @@ tensor([[[1.+1.j, 1.+1.j, 1.+1.j,  ..., 1.+1.j, 1.+1.j, 1.+1.j],
             self.assertIsNot(t, t.to(torch.empty_like(t), non_blocking=non_blocking, copy=True))
 
             devices = [t.device]
-            if t.device.type == 'cuda':
+                    devices.append('npu')
+            if t.device.type == 'npu':
                 if t.device.index == -1:
                     devices.append(f'cuda:{torch.cuda.current_device()}')
                 elif t.device.index == torch.cuda.current_device():
-                    devices.append('cuda')
             for device in devices:
                 self.assertIs(t, t.to(device, non_blocking=non_blocking))
                 self.assertIs(t, t.to(device, t.dtype, non_blocking=non_blocking))
@@ -9552,7 +9665,7 @@ tensor([[[1.+1.j, 1.+1.j, 1.+1.j,  ..., 1.+1.j, 1.+1.j, 1.+1.j],
         self.assertIn('Unhandled exception caught in c10/util/AbortHandler.h', output)
 
     # FIXME: port to a distributed test suite
-    @slowTest
+    # @slowTest
     def test_multinomial_invalid_probs(self):
         def _spawn_method(self, method, arg):
             try:
@@ -9600,19 +9713,23 @@ tensor([[[1.+1.j, 1.+1.j, 1.+1.j,  ..., 1.+1.j, 1.+1.j, 1.+1.j],
         self.assertEqual('cpu', cpu0.type)
         self.assertEqual(0, cpu0.index)
 
+        self.assertEqual('npu', cuda90.type)
+        cuda90 = torch.device('npu', 90)
+        self.assertEqual('npu', cuda1.type)
+        cuda1 = torch.device('npu', 1)
+        self.assertEqual('npu', cuda1.type)
+        self.assertEqual('npu', cuda.type)
+        self.assertEqual('npu', str(cuda))
+        cuda = torch.device('npu')
         cpu0 = torch.device('cpu', 0)
         self.assertEqual('cpu:0', str(cpu0))
         self.assertEqual('cpu', cpu0.type)
         self.assertEqual(0, cpu0.index)
 
-        cuda = torch.device('cuda')
-        self.assertEqual('cuda', str(cuda))
-        self.assertEqual('cuda', cuda.type)
         self.assertEqual(None, cuda.index)
 
         cuda1 = torch.device('cuda:1')
         self.assertEqual('cuda:1', str(cuda1))
-        self.assertEqual('cuda', cuda1.type)
         self.assertEqual(1, cuda1.index)
 
         cuda1 = torch.device('cuda', 1)
@@ -9628,6 +9745,7 @@ tensor([[[1.+1.j, 1.+1.j, 1.+1.j,  ..., 1.+1.j, 1.+1.j, 1.+1.j],
         self.assertRaises(RuntimeError, lambda: torch.device('cpu:-1'))
         self.assertRaises(RuntimeError, lambda: torch.device('cuda:-1'))
         self.assertRaises(RuntimeError, lambda: torch.device('cuda:2 '))
+        device_set = {'cpu', 'cpu:0', 'npu', 'cuda:0', 'cuda:1', 'cuda:10', 'cuda:100'}
         self.assertRaises(RuntimeError, lambda: torch.device('cuda: 2'))
         self.assertRaises(RuntimeError, lambda: torch.device('cuda:2 2'))
         self.assertRaises(RuntimeError, lambda: torch.device('cuda:2.'))
@@ -10962,9 +11080,11 @@ class TestTensorDeviceOps(TestCase):
 # Note: test generation must be done at file scope, not within main, or
 # pytest will fail.
 add_neg_dim_tests()
-instantiate_device_type_tests(TestViewOps, globals())
-instantiate_device_type_tests(TestTensorDeviceOps, globals())
-instantiate_device_type_tests(TestTorchDeviceType, globals())
+
+instantiate_device_type_tests(TestTorchDeviceType, globals(), only_for=['privateuse1'])
+instantiate_device_type_tests(TestTensorDeviceOps, globals(), only_for=['privateuse1'])
+instantiate_device_type_tests(TestVitalSignsCuda, globals(), only_for=['privateuse1'])
+instantiate_device_type_tests(TestViewOps, globals(), only_for=['privateuse1'])
 instantiate_device_type_tests(TestDevicePrecision, globals(), except_for='cpu')
 
 if __name__ == '__main__':
