diff --git a/test/test_jit_fuser_te.py b/test/test_jit_fuser_te.py
index 17c83cc7264..6aac5f10d03 100644
--- a/test/test_jit_fuser_te.py
+++ b/test/test_jit_fuser_te.py
@@ -2460,6 +2460,7 @@ class TestTEFuser(JitTestCase):
         torch._C._jit_pass_inline(g)
         FileCheck().check_count("prim::If", 1, exactly=True).run(g)

+    @unittest.skip("the JIT fuser is not yet adapted, so this test is skipped.")
     def test_dynamic_shapes(self):
         from functools import partial


