diff --git a/test/test_hub.py b/test/test_hub.py
index 23bb395a373..45638ef6a74 100644
--- a/test/test_hub.py
+++ b/test/test_hub.py
@@ -1,6 +1,7 @@
 # Owner(s): ["module: hub"]
 
 import os
+from torch.testing._internal.common_utils import IS_SANDCASTLE, retry, TestCase, run_tests
 import tempfile
 import unittest
 import warnings
@@ -15,6 +16,8 @@ from torch.testing._internal.common_utils import (
     run_tests,
     TestCase,
 )
+from torch_npu.contrib import transfer_to_npu
+import torch_npu
 
 
 def sum_of_state_dict(state_dict):
@@ -245,6 +248,9 @@ class TestHub(TestCase):
         torch.hub.load("ailzhang/torchhub_example", "mnist_zip_1_6", trust_repo=False)
         self._assert_in_trusted_list("ailzhang_torchhub_example")
         patched_input.assert_called_once()
+    run_tests()
+if __name__ == "__main__":
+
 
         # Loading a second time with "check", we don't ask for user input
         patched_input.reset_mock()
