diff --git a/test/test_sort_and_select.py b/test/test_sort_and_select.py
index 72fe7a52b19..af081abad76 100644
--- a/test/test_sort_and_select.py
+++ b/test/test_sort_and_select.py
@@ -32,7 +32,7 @@ from torch.testing._internal.common_utils import (
     TestCase,
 )
 
-
+from torch_npu.contrib import transfer_to_npu
 class TestSortAndSelect(TestCase):
     def assertIsOrdered(self, order, x, mxx, ixx, task):
         SIZE = x.size(1)
