diff --git a/test/test_autograd.py b/test/test_autograd.py
index cdb4f9ed7aa..7be87f2784c 100644
--- a/test/test_autograd.py
+++ b/test/test_autograd.py
@@ -28,6 +28,7 @@ from operator import mul
 from typing import TYPE_CHECKING
 
 import torch
+import torch_npu
 import torch.autograd._functions
 import torch.autograd.forward_ad as fwAD
 from torch import inf, nan, nn
@@ -7670,7 +7671,7 @@ for shape in [(1,), ()]:
 
             return fn
 
-        a = torch.randn(1, requires_grad=True)
+        a = torch.randn(1, device="npu", requires_grad=True)
 
         # Save fewer tensors during recompute
         fn = get_non_det_fn(orig_fn=save_3_tensors, recompute_fn=save_2_tensors)
@@ -8010,7 +8011,7 @@ for shape in [(1,), ()]:
             self.assertEqual(checkpoint_param.grad, param.grad)
 
     def test_checkpoint_valid_reset_on_error(self):
-        a = torch.randn(2, 2, requires_grad=True)
+        a = torch.randn(2, 2, device="npu", requires_grad=True)
 
         with self.assertRaisesRegex(
             Exception, "torch.utils.checkpoint is incompatible"
@@ -15386,8 +15387,8 @@ class TestSelectiveActivationCheckpoint(TestCase):
         def fn(x, y):
             return x.view(-1), y.sin().cos()
 
-        x = torch.tensor([1, 2, 3], dtype=torch.int64)
-        y = torch.randn(3, requires_grad=True)
+        x = torch.tensor([1, 2, 3], dtype=torch.int64, device="npu")
+        y = torch.randn(3, device="npu", requires_grad=True)
 
         context_fn = functools.partial(
             create_selective_checkpoint_contexts,
@@ -15403,7 +15404,7 @@ class TestSelectiveActivationCheckpoint(TestCase):
         #
         # 1. size/strides are dispatched upon
         # 2. our policy saves sizes
-        ta = torch.randn(6, 2)
+        ta = torch.randn(6, 2, device="npu")
 
         class CustomSizeDynamicShapesTensor(torch.Tensor):
             @staticmethod
@@ -15463,7 +15464,7 @@ class TestSelectiveActivationCheckpoint(TestCase):
             self.assertEqual(view_out.grad_fn._saved_self_sym_sizes, [12])
             return view_out.exp()
 
-        x = torch.randn(4, 3, requires_grad=True)
+        x = torch.randn(4, 3, device="npu", requires_grad=True)
         x_wrapper = CustomSizeDynamicShapesTensor(x)
         context_fn = functools.partial(create_selective_checkpoint_contexts, policy_fn)
         out = checkpoint(fn, x_wrapper, use_reentrant=False, context_fn=context_fn)
@@ -15512,7 +15513,7 @@ class TestSelectiveActivationCheckpoint(TestCase):
         def fn(x):
             return x.sin().sin().sin()
 
-        x = torch.randn(3, requires_grad=True)
+        x = torch.randn(3, device="npu", requires_grad=True)
         context_fn = functools.partial(
             create_selective_checkpoint_contexts,
             Policy(),
@@ -15564,7 +15565,7 @@ class TestSelectiveActivationCheckpoint(TestCase):
         with disable_gc():
             # Case 1: If graph goes away without backward, make sure there's no reference cycle
             #         keeping storage alive.
-            x = torch.randn(3, requires_grad=True)
+            x = torch.randn(3, device="npu", requires_grad=True)
             context_fn = functools.partial(
                 create_selective_checkpoint_contexts, policy_fn
             )
@@ -15574,7 +15575,7 @@ class TestSelectiveActivationCheckpoint(TestCase):
             self.assertIsNone(ref())
 
             # Case 2: After backward, even if retain_graph=True, the storage should go away
-            x = torch.randn(3, requires_grad=True)
+            x = torch.randn(3, device="npu", requires_grad=True)
             context_fn = functools.partial(
                 create_selective_checkpoint_contexts, policy_fn
             )
@@ -15596,7 +15597,7 @@ class TestSelectiveActivationCheckpoint(TestCase):
         def fn(x):
             return x.sin().mul_(2).cos().exp()
 
-        x = torch.randn(3, requires_grad=True)
+        x = torch.randn(3, device="npu", requires_grad=True)
         context_fn = functools.partial(create_selective_checkpoint_contexts, policy_fn)
         out = checkpoint(fn, x, use_reentrant=False, context_fn=context_fn)
 
@@ -15604,7 +15605,7 @@ class TestSelectiveActivationCheckpoint(TestCase):
         with self.assertRaisesRegex(RuntimeError, "has been mutated"):
             out.sum().backward()
 
-        x = torch.randn(3, requires_grad=True)
+        x = torch.randn(3, device="npu", requires_grad=True)
         context_fn = functools.partial(
             create_selective_checkpoint_contexts,
             policy_fn,
@@ -15632,7 +15633,7 @@ class TestSelectiveActivationCheckpoint(TestCase):
             a, b = torch.var_mean(x)
             return a * b
 
-        x = torch.randn(3, requires_grad=True)
+        x = torch.randn(3, device="npu", requires_grad=True)
         context_fn = functools.partial(create_selective_checkpoint_contexts, policy_fn)
         out = checkpoint(fn, x, use_reentrant=False, context_fn=context_fn)
         x_grad = torch.autograd.grad(out.sum(), (x,))
@@ -15663,14 +15664,14 @@ class TestSelectiveActivationCheckpoint(TestCase):
                 "mylib::sin_with_extra", backward, setup_context=setup_context
             )
 
-            x = torch.randn(3, requires_grad=True)
+            x = torch.randn(3, device="npu", requires_grad=True)
 
             def fn(x):
                 return (torch.ops.mylib.sin_with_extra(x)[0] * x.sin().exp()).sin()
 
             ops_list = [torch.ops.mylib.sin_with_extra.default]
 
-            x = torch.randn(3, requires_grad=True)
+            x = torch.randn(3, device="npu", requires_grad=True)
             context_fn = functools.partial(
                 create_selective_checkpoint_contexts, ops_list
             )
@@ -15694,7 +15695,7 @@ class TestSelectiveActivationCheckpoint(TestCase):
         def fn(x):
             return x.sin().cos().exp()
 
-        x = torch.randn(3, requires_grad=True)
+        x = torch.randn(3, device="npu", requires_grad=True)
         context_fn = functools.partial(create_selective_checkpoint_contexts, policy_fn)
         out = checkpoint(fn, x, use_reentrant=False, context_fn=context_fn)
         out.sum().backward(retain_graph=True)
