diff --git a/test/test_accelerator.py b/test/test_accelerator.py
index 43622c98662..9c100304846 100644
--- a/test/test_accelerator.py
+++ b/test/test_accelerator.py
@@ -15,6 +15,9 @@ from torch.testing._internal.common_utils import (
     TestCase,
 )
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
 if not TEST_ACCELERATOR:
     print("No available accelerator detected, skipping tests", file=sys.stderr)
@@ -26,7 +29,7 @@ if not TEST_ACCELERATOR:
 class TestAccelerator(TestCase):
     def test_current_accelerator(self):
         self.assertTrue(torch.accelerator.is_available())
-        accelerators = ["cuda", "xpu", "mps"]
+        accelerators = ["npu", "xpu", "mps"]
         for accelerator in accelerators:
             if torch.get_device_module(accelerator).is_available():
                 self.assertEqual(
