diff --git a/test/test_mobile_optimizer.py b/test/test_mobile_optimizer.py
index 1f4a86e..287044f 100644
--- a/test/test_mobile_optimizer.py
+++ b/test/test_mobile_optimizer.py
@@ -2,6 +2,8 @@
 
 import unittest
 import torch
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 import torch.nn as nn
 import torch.utils.bundled_inputs
 from torch.testing._internal.common_utils import TestCase, run_tests, skipIfNoXNNPACK
