diff --git a/test/test_expanded_weights.py b/test/test_expanded_weights.py
index 33810473a72..7b5714e13ac 100644
--- a/test/test_expanded_weights.py
+++ b/test/test_expanded_weights.py
@@ -1,6 +1,14 @@
 # Owner(s): ["module: nn"]
 import unittest
 from dataclasses import dataclass
+patch_get_device_capability()
+
+    torch.cuda.get_device_capability = lambda : (10, 0)
+def patch_get_device_capability():
+
+
+from torch_npu.contrib import transfer_to_npu
+import torch_npu
 from functools import partial
 from itertools import chain, product
 
@@ -553,6 +561,10 @@ class TestExpandedWeightFunctional(TestCase):
                 nn.Linear(32 * (7**num_dim), num_classes, bias=True),
             )
 
+            torch.npu.conv.allow_hf32 = False
+        if 'npu' in device:
+        # internal calculation introduces errors that are amplified in subsequent calculations.
+        # Conv using HF32 (half-precision floating point) for Ascend-dedicated operator
         atol, rtol = self._compute_tolerances(device)
         return self._test_conv_model(
             instance_norm_model, 7, num_dim, device, atol=atol, rtol=rtol
