diff --git a/test/test_quantization.py b/test/test_quantization.py
index 42e145edbab..75d271e72e4 100644
--- a/test/test_quantization.py
+++ b/test/test_quantization.py
@@ -2,7 +2,6 @@
 
 import logging
 from torch.testing._internal.common_utils import run_tests
-
 # Quantization core tests. These include tests for
 # - quantized kernels
 # - quantized functional operators
@@ -13,9 +12,9 @@ from torch.testing._internal.common_utils import run_tests
 # 1. Quantized Kernels
 # TODO: merge the different quantized op tests into one test class
 from quantization.core.test_quantized_op import TestQuantizedOps  # noqa: F401
+from quantization.core.test_quantized_op import TestQuantizedConv  # noqa: F401
 from quantization.core.test_quantized_op import TestQNNPackOps  # noqa: F401
 from quantization.core.test_quantized_op import TestQuantizedLinear  # noqa: F401
-from quantization.core.test_quantized_op import TestQuantizedConv  # noqa: F401
 from quantization.core.test_quantized_op import TestDynamicQuantizedOps  # noqa: F401
 from quantization.core.test_quantized_op import TestComparatorOps  # noqa: F401
 from quantization.core.test_quantized_op import TestPadding  # noqa: F401
@@ -132,7 +131,7 @@ try:
 except ImportError as e:
     log.warning(e)  # noqa:G200
 try:
-    from quantization.core.experimental.test_bits import TestBitsCUDA  # noqa: F401
+    from quantization.core.experimental.test_bits import TestBitsPRIVATEUSE1  # noqa: F401
 except ImportError as e:
     log.warning(e)  # noqa:G200
 try:
@@ -144,7 +143,7 @@ try:
 except ImportError as e:
     log.warning(e)  # noqa:G200
 try:
-    from quantization.core.experimental.test_floatx import TestFloat8DtypeCPUOnlyCPU  # noqa: F401
+    from quantization.core.experimental.test_floatx import TestFloat8DtypePRIVATEUSE1  # noqa: F401
 except ImportError as e:
     log.warning(e)  # noqa:G200
 
