diff --git a/test/test_fx_experimental.py b/test/test_fx_experimental.py
index 434de52..ea938a5 100644
--- a/test/test_fx_experimental.py
+++ b/test/test_fx_experimental.py
@@ -15,6 +15,7 @@ from types import BuiltinFunctionType
 from typing import Callable, NamedTuple, Optional, Union
 
 import torch
+import torch_npu
 import torch.fx.experimental.meta_tracer
 import torch.fx.experimental.optimization as optimization
 from torch.fx._symbolic_trace import symbolic_trace
@@ -740,7 +741,7 @@ terrible spacing
         class MyModule(torch.nn.Module):
             def __init__(self) -> None:
                 super().__init__()
-                self.param = torch.nn.Parameter(torch.rand(3, 4))
+                self.param = torch.nn.Parameter(torch.rand(3, 4).npu())
                 self.linear = torch.nn.Linear(4, 5)
 
             def forward(self, x, y):
@@ -749,7 +750,7 @@ terrible spacing
                 return z + w
 
         # symbolically trace model
-        my_module = MyModule()
+        my_module = MyModule().npu()
         my_module_traced = symbolic_trace(my_module)
 
         # random mod partitioning
@@ -770,7 +771,7 @@ terrible spacing
         # split module in module with submodules
         module_with_submodules = split_module(
             my_module_traced, my_module, mod_partition
-        )
+        ).npu()
 
         # Check that test_meta_info was still on all nodes.
         submodules = dict(module_with_submodules.named_modules())
@@ -783,8 +784,8 @@ terrible spacing
                         stored_op = submod_node.meta.get("test_meta_info")
                         self.assertTrue(stored_op is not None and stored_op)
 
-        x = torch.rand(3, 4)
-        y = torch.rand(3, 4)
+        x = torch.rand(3, 4).npu()
+        y = torch.rand(3, 4).npu()
 
         orig_out = my_module_traced(x, y)
         submodules_out = module_with_submodules(x, y)
@@ -817,7 +818,7 @@ terrible spacing
 
         split = split_module(traced, mod, split_callback)
 
-        x = torch.randn((5,))
+        x = torch.randn((5,)).npu()
         torch.testing.assert_close(
             split(x), traced(x)
         )
@@ -826,7 +827,7 @@ terrible spacing
         def foo(x):
             x.add_(1)
 
-        gm = make_fx(foo, tracing_mode="fake")(torch.randn(3,))
+        gm = make_fx(foo, tracing_mode="fake")(torch.randn(3,).npu())
 
         def cb(_):
             return 1
@@ -859,8 +860,8 @@ terrible spacing
 
         split = split_module(traced, mod, split_callback)
 
-        x = torch.randn(5, 3)
-        foo = torch.randn(5, 3)
+        x = torch.randn(5, 3).npu()
+        foo = torch.randn(5, 3).npu()
         torch.testing.assert_close(split(x, foo=foo), traced(x, foo=foo))
 
     @skipIfNoTorchVision
@@ -869,7 +870,7 @@ terrible spacing
         # There was an issue before causing submodule names to be aliased
         m = resnet18()
         traced = symbolic_trace(m)
-        a = torch.rand(64, 3, 7, 7)
+        a = torch.rand(64, 3, 7, 7).npu()
         module_with_submodules = split_module(traced, m, lambda node: 0)
         module_with_submodules(a)
 
@@ -887,12 +888,12 @@ terrible spacing
 
                 return x
 
-        mtt = ModelToTrace()
+        mtt = ModelToTrace().npu()
         traced = torch.fx.symbolic_trace(mtt, concrete_args={'targets': None})
 
-        split = split_module(traced, mtt, lambda node: 0)
+        split = split_module(traced, mtt, lambda node: 0).npu()
 
-        x = torch.randn(50, 512)
+        x = torch.randn(50, 512).npu()
         torch.testing.assert_close(split(x), traced(x))
 
     def test_split_module_keep_original_order_and_noop_graph(self):
@@ -901,7 +902,7 @@ terrible spacing
         def fn(x):
             return (x,)
 
-        g = make_fx(fn, tracing_mode="fake")(torch.randn(3, 3))
+        g = make_fx(fn, tracing_mode="fake")(torch.randn(3, 3).npu())
 
         # g.graph.print_tabular()
         # opcode       name    target    args       kwargs
@@ -1253,8 +1254,8 @@ class {test_classname}(torch.nn.Module):
                 add_5 = add_3 + add_4
                 return add_5
 
-        a, b, c, d = torch.ones(4), torch.ones(4), torch.ones(4), torch.ones(4)
-        mm = MyModule()
+        a, b, c, d = torch.ones(4).npu(), torch.ones(4).npu(), torch.ones(4).npu(), torch.ones(4).npu()
+        mm = MyModule().npu()
         traced = symbolic_trace(mm)
 
         def split_cb(node: torch.fx.Node):
@@ -1272,8 +1273,8 @@ class {test_classname}(torch.nn.Module):
         class ExampleCode(torch.nn.Module):
             def __init__(self) -> None:
                 super().__init__()
-                self.mm_param = torch.nn.Parameter(torch.randn(d_hid, d_hid))
-                self.mm_param2 = torch.nn.Parameter(torch.randn(d_hid, d_hid))
+                self.mm_param = torch.nn.Parameter(torch.randn(d_hid, d_hid).npu())
+                self.mm_param2 = torch.nn.Parameter(torch.randn(d_hid, d_hid).npu())
                 self.lin = torch.nn.Linear(d_hid, d_hid)
 
             def forward(self, x):
@@ -1286,7 +1287,7 @@ class {test_classname}(torch.nn.Module):
                 x = self.lin(x)
                 return x
 
-        my_module = ExampleCode()
+        my_module = ExampleCode().npu()
         my_module_traced = symbolic_trace(my_module)
 
         part_idx = 0
