diff --git a/test/test_mkldnn.py b/test/test_mkldnn.py
index 9ad7516..03cff6b 100644
--- a/test/test_mkldnn.py
+++ b/test/test_mkldnn.py
@@ -28,6 +28,10 @@ from torch.testing._internal.common_device_type import (
     dtypes,
 )
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
+
 # batched grad doesn't support mkldnn
 gradcheck = functools.partial(gradcheck, check_batched_grad=False)
 gradgradcheck = functools.partial(gradgradcheck, check_batched_grad=False)
