diff --git a/test/test_ao_sparsity.py b/test/test_ao_sparsity.py
index db6b222..a90e059 100644
--- a/test/test_ao_sparsity.py
+++ b/test/test_ao_sparsity.py
@@ -1,5 +1,4 @@
 # Owner(s): ["module: unknown"]
-
 # Kernels
 from ao.sparsity.test_kernels import (  # noqa: F401
     TestQuantizedSparseKernels,
@@ -26,11 +25,11 @@ from ao.sparsity.test_structured_sparsifier import (  # noqa: F401
     TestSaliencyPruner,
 )
 
-from torch.testing._internal.common_utils import IS_ARM64, run_tests
 
+from torch.testing._internal.common_utils import IS_ARM64, run_tests
 
 # Composability
-if not IS_ARM64:
+if IS_ARM64:
     from ao.sparsity.test_composability import (  # noqa: F401
         TestComposability,
         TestFxComposability,
@@ -54,6 +53,8 @@ from ao.sparsity.test_data_sparsifier import (  # noqa: F401
 # Utilities
 from ao.sparsity.test_sparsity_utils import TestSparsityUtilFunctions  # noqa: F401
 
+# Qlinear Packed Params
+from ao.sparsity.test_qlinear_packed_params import TestQlinearPackedParams
 
 if __name__ == "__main__":
     run_tests()
