diff --git a/test/test_serialization.py b/test/test_serialization.py
index 94632b1a0f..a0f0f1a19a 100644
--- a/test/test_serialization.py
+++ b/test/test_serialization.py
@@ -64,2 +64,5 @@ from torch.utils._import_utils import import_dill
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
@@ -515,3 +518,3 @@ class SerializationMixin:
                     f"`{compressed_indices_name}[[]..., 0[]] == 0` is not satisfied."):
-                y = torch.load(f)
+                y = torch.load(f, weights_only=False)
 
@@ -4800,3 +4803,3 @@ class TestSubclassSerialization(TestCase):
     def test_tensor_subclass_wrapper_serialization(self):
-        wrapped_tensor = torch.rand(2)
+        wrapped_tensor = torch.rand(2).npu()
         my_tensor = TestWrapperSubclass(wrapped_tensor)
@@ -4815,2 +4818,3 @@ class TestSubclassSerialization(TestCase):
         self.assertEqual(new_tensor.elem, my_tensor.elem)
+        self.assertEqual(new_tensor.elem.device.type, "npu")
         self.assertEqual(new_tensor.foo, foo_val)
