diff --git a/test/test_multiprocessing.py b/test/test_multiprocessing.py
old mode 100644
new mode 100755
index 9f58f14143a..734d2cc589b
--- a/test/test_multiprocessing.py
+++ b/test/test_multiprocessing.py
@@ -8,7 +8,8 @@ import sys
 import time
 import unittest
 from sys import platform
-
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 import torch
 import torch.cuda
 import torch.multiprocessing as mp
@@ -612,15 +613,16 @@ class TestMultiprocessing(TestCase):
         stderr = TestCase.runWithPytorchAPIUsageStderr(
             """\
 import torch
+import torch_npu
 from torch.multiprocessing import Process
 def run(rank):
-    torch.cuda.set_device(rank)
+    torch.npu.set_device(rank)
 if __name__ == "__main__":
     size = 2
     processes = []
     for rank in range(size):
         # it would work fine without the line below
-        x = torch.rand(20, 2).cuda()
+        x = torch.rand(20, 2).npu()
         p = Process(target=run, args=(rank,))
         p.start()
         processes.append(p)
@@ -993,4 +995,5 @@ if __name__ == "__main__":
 
 
 if __name__ == "__main__":
+    mp.set_start_method('spawn', force=True)
     run_tests()
