diff --git a/test/test_scatter_gather_ops.py b/test/test_scatter_gather_ops.py
index ce2c83c5ef5..cc3fafcc56c 100644
--- a/test/test_scatter_gather_ops.py
+++ b/test/test_scatter_gather_ops.py
@@ -19,6 +19,12 @@ from torch.testing._internal.common_cuda import CDNA3OrLater
 if torch.get_default_dtype() is not torch.float32:
     raise AssertionError("default dtype should be float32")
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
 # Note: test_scatter_gather_ops.py
 # This test file tests scatter and gather operations,
