diff --git a/test/test_sympy_utils.py b/test/test_sympy_utils.py
index f1a706e4276..d8208142523 100644
--- a/test/test_sympy_utils.py
+++ b/test/test_sympy_utils.py
@@ -39,7 +39,7 @@ from torch.utils._sympy.solve import INEQUALITY_TYPES, mirror_rel_op, try_solve
 from torch.utils._sympy.value_ranges import ValueRanges
 from torch._inductor.bounds import ValueRangeAnalysis
 from torch._inductor.index_propagation import TypedExpr
-
+from torch_npu.contrib import transfer_to_npu
 
 UNARY_OPS = [
     "reciprocal",
