diff --git a/test/test_pytree.py b/test/test_pytree.py
index 4560ac6..c392898 100644
--- a/test/test_pytree.py
+++ b/test/test_pytree.py
@@ -23,8 +23,7 @@ from torch.testing._internal.common_utils import (
     subtest,
     TestCase,
 )
-
-
+from torch_npu.contrib import transfer_to_npu
 if IS_FBCODE:
     # optree is not yet enabled in fbcode, so just re-test the python implementation
     cxx_pytree = py_pytree
