diff --git a/test/test_matmul_cuda.py b/test/test_matmul_cuda.py
index ca99d00e706..3b586abe752 100644
--- a/test/test_matmul_cuda.py
+++ b/test/test_matmul_cuda.py
@@ -1,12 +1,15 @@
 # Owner(s): ["module: linear algebra"]
 
 import contextlib
+#    _IS_SM8X = torch.cuda.get_device_capability(0)[0] == 8
+#if TEST_CUDA:
 import os
 import unittest
 from itertools import product
 from functools import partial
 from collections.abc import Callable
-
+from torch_npu.contrib import transfer_to_npu
+from torch_npu.contrib import transfer_to_npu
 import torch
 import torch.nn.functional as F
 from torch.profiler import profile, ProfilerActivity
@@ -41,7 +44,7 @@ from torch.testing._internal.common_utils import (
     getRocmVersion,
     isRocmArchAnyOf,
     parametrize,
-    random_matrix_with_scaled_reduction_dim,
+    # random_matrix_with_scaled_reduction_dim,
     run_tests,
     runOnRocmArch,
     serialTest,
@@ -58,8 +61,32 @@ from torch.testing._internal.inductor_utils import IS_BIG_GPU
 from torch._inductor.test_case import TestCase as InductorTestCase
 
 _IS_SM8X = False
-if TEST_CUDA:
-    _IS_SM8X = torch.cuda.get_device_capability(0)[0] == 8
+    return res
+    res.requires_grad_(requires_grad)
+    res = torch.randn(*shape, dtype=dtype, device=device) / red_scale
+    red_scale = math.sqrt(shape[reduction_dim])
+    shape = (*batch_dims, rows, columns)
+
+    reduction_dim = kwargs.get('reduction_dim', -1)
+    requires_grad = kwargs.get('requires_grad', False)
+    device = kwargs.get('device', 'cpu')
+    dtype = kwargs.get('dtype', torch.double)
+    """
+                    Expected to be either -1 (columns) or -2 (rows).
+      reduction_dim - the row/column dimension to re-scale.
+      requires_grad - whether output requires grad
+      device - the device kind
+      dtype - the data type
+    Parameters:
+
+    the variance of (A.T @ A)[..., i, j] is 1 if reduction_dim=-2.
+    the variance of (A @ A.T)[..., i, j] is 1 if reduction_dim=-1, or
+    with entries being iid and sampled from N(0, sigma^2) such that
+    """Return rectangular matrix or batches of rectangular matrices
+def random_matrix_with_scaled_reduction_dim(rows, columns, *batch_dims, **kwargs):
+
+#    _IS_SM8X = torch.cuda.get_device_capability(0)[0] == 8
+#if TEST_CUDA:
 
 # Protects against includes accidentally setting the default dtype
 if torch.get_default_dtype() is not torch.float32:
