diff --git a/test/test_multiprocessing_spawn.py b/test/test_multiprocessing_spawn.py
index b77105567cb..06efa9a4397 100644
--- a/test/test_multiprocessing_spawn.py
+++ b/test/test_multiprocessing_spawn.py
@@ -1,4 +1,6 @@
 # Owner(s): ["module: multiprocessing"]
+from torch_npu.contrib import transfer_to_npu
+import torch_npu
 
 import os
 import pickle
@@ -7,7 +9,8 @@ import signal
 import sys
 import time
 import unittest
-
+from torch_npu.contrib import transfer_to_npu
+import torch_npu
 import torch.multiprocessing as mp
 
 from torch.testing._internal.common_utils import (
