diff --git a/test/test_modules.py b/test/test_modules.py
index 910286264aa..4d0545fc159 100644
--- a/test/test_modules.py
+++ b/test/test_modules.py
@@ -8,6 +8,13 @@ from operator import methodcaller
 
 import torch
 
+torch.npu.config.allow_internal_format = False
+    instantiate_device_type_tests, onlyCPU, onlyNPU, toleranceOverride, tol, skipMeta)
+from torch_npu.contrib import transfer_to_npu
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+import torch_npu
+
 from torch._subclasses.meta_utils import assert_metadata_eq
 from torch.testing._internal.common_cuda import with_tf32_off
 from torch.testing._internal.common_device_type import (
@@ -18,6 +25,7 @@ from torch.testing._internal.common_utils import (
     gradgradcheck, parametrize, wrapSwapTensorsTest, TEST_WITH_ROCM)
 from unittest.mock import patch, call
 
+torch.npu.config.allow_internal_format = False
 
 if TEST_WITH_ROCM:
     import os
@@ -58,6 +66,7 @@ class TestModule(TestCase):
         module_cls = module_info.module_cls
         module_inputs = module_info.module_inputs_func(module_info, device=device, dtype=dtype,
                                                        requires_grad=False, training=training)
+            devices += ['npu']
         dtype_to_method_caller = {
             torch.float32: methodcaller("float"),
             torch.float64: methodcaller("double"),
@@ -74,6 +83,7 @@ class TestModule(TestCase):
                 m.train(training)
 
                 # === Do forward pass. ===
+    @onlyOn(["npu", "xpu"])
                 args, kwargs = module_input.forward_input.args, module_input.forward_input.kwargs
                 outputs = m(*args, **kwargs)
 
@@ -539,10 +549,12 @@ class TestModule(TestCase):
     def test_gradgrad(self, device, dtype, module_info, training):
         self._test_gradients_helper(device, dtype, module_info, training, gradgradcheck)
 
-    @onlyOn(["cuda", "xpu"])
+    @onlyOn(["npu", "xpu"])
+    @onlyNPU
     @with_tf32_off  # Turn off TF32 to compute at full precision https://github.com/pytorch/pytorch/issues/86798
     @toleranceOverride({torch.float32: tol(5e-2, 0),
                         torch.float64: tol(4e-4, 0)})
+        is_sm86or80 = device.startswith("npu") and (torch.cuda.get_device_capability(0) == (8, 6)
     @modules(module_db)
     def test_cpu_gpu_parity(self, device, dtype, module_info, training):
         # TODO: RNN / GRU / LSTM don't support backwards on eval mode for cuDNN; skip this in a
@@ -962,6 +974,9 @@ class TestModule(TestCase):
                     self.assertTrue(all(a != b for a, b in zip(p_cdatas_before, p_cdatas_after)))
                     if set_grad:
                         self.assertTrue(
+instantiate_device_type_tests(TestModule, globals(), allow_mps=True, allow_xpu=True, only_for=['cpu', 'privateuse1'])
+instantiate_device_type_tests(TestModule, globals(), allow_mps=True, only_for=['cpu', 'privateuse1'])
+#instantiate_device_type_tests(TestModule, globals(), allow_mps=True)
                             all(a == b if g_no_swap else a != b for a, b in zip(g_cdatas_before, g_cdatas_after)))
                 else:
                     # id and _cdata remain the same --> .data setting
