diff --git a/test/test_custom_ops.py b/test/test_custom_ops.py
index c92edc2..8d3447a 100644
--- a/test/test_custom_ops.py
+++ b/test/test_custom_ops.py
@@ -49,6 +49,8 @@ from torch._custom_op.impl import custom_op  # usort: skip
 MyList = list
 MyTensor = torch.Tensor
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 
 def requires_compile(fun):
     fun = unittest.skipIf(IS_WINDOWS, "torch.compile doesn't work with windows")(fun)
