diff --git a/test/test_python_dispatch.py b/test/test_python_dispatch.py
index 3592366..f790029 100644
--- a/test/test_python_dispatch.py
+++ b/test/test_python_dispatch.py
@@ -2210,0 +2211 @@ $0: f32[] = torch._ops.aten.empty.memory_format([], device=device(type='cpu'), p
+                    tensor = torch.ones(123).npu()
@@ -2214 +2215 @@ $0: f32[] = torch._ops.aten.empty.memory_format([], device=device(type='cpu'), p
-                            123,
+                            tensor.size(0),
@@ -2219 +2220 @@ $0: f32[] = torch._ops.aten.empty.memory_format([], device=device(type='cpu'), p
-                        hint=123,
+                        hint=tensor.size(0),
