diff --git a/test/test_type_hints.py b/test/test_type_hints.py
index 0aae54b..36f7fec 100644
--- a/test/test_type_hints.py
+++ b/test/test_type_hints.py
@@ -9,6 +9,7 @@ import unittest
 from pathlib import Path
 
 import torch
+from torch_npu.contrib import transfer_to_npu
 from torch.testing._internal.common_utils import run_tests, set_cwd, TestCase
 
 
@@ -47,11 +48,15 @@ def get_all_examples():
         "import io",
         "import itertools",
         "",
+        "import numpy as np",
         "import numpy",
         "",
         "import torch",
         "import torch.nn.functional as F",
+        "import torch_npu"
         "",
+        "",
+        "from torch._subclasses.fake_tensor import FakeTensorMode",
         # for requires_grad_ example
         # NB: We are parsing this file as Python 2, so we must use
         # Python 2 type annotation syntax
@@ -138,6 +143,7 @@ class TestTypeHints(TestCase):
                     [
                         "--cache-dir=.mypy_cache/doc",
                         "--no-strict-optional",  # needed because of torch.lu_unpack, see gh-36584
+                        "--follow-imports=skip",
                         str(fn),
                     ]
                 )
