diff --git a/test/test_torch.py b/test/test_torch.py
index fa254541d1..5d83de2659 100644
--- a/test/test_torch.py
+++ b/test/test_torch.py
@@ -52,3 +52,3 @@ from torch.testing._internal.common_device_type import (
     instantiate_device_type_tests,
-    onlyCUDA, onlyCPU,
+    onlyCUDA, onlyCPU, onlyPRIVATEUSE1,
     dtypes, dtypesIfCUDA, dtypesIfCPU, deviceCountAtLeast,
@@ -259,3 +259,3 @@ class TestTorchDeviceType(TestCase):
     @skipIfTorchDynamo("Not a suitable test for TorchDynamo")
-    @onlyNativeDeviceTypes
+    @onlyPRIVATEUSE1
     @slowTestIf(IS_WINDOWS)
@@ -9673,6 +9673,8 @@ tensor([[[1.+1.j, 1.+1.j, 1.+1.j,  ..., 1.+1.j, 1.+1.j, 1.+1.j],
         self.assertEqual(cfloat.imag, torch.zeros_like(cfloat.imag))
-        cdouble = x.cdouble()
+        x_npu = x.npu()
+        cdouble = x_npu.cdouble()
+        self.assertEqual(cdouble.device.type, "npu")
         self.assertEqual(cdouble.dtype, torch.complex128)
-        self.assertEqual(cdouble.real, x.double())
-        self.assertEqual(cdouble.imag, torch.zeros_like(cdouble.imag))
+        self.assertEqual(cdouble.real.cpu(), x.double())
+        self.assertEqual(cdouble.imag.cpu(), torch.zeros_like(x.double()))
         chalf = x.chalf()
