diff --git a/test/test_pytree.py b/test/test_pytree.py
index 1b7902c020f..bd18449a317 100644
--- a/test/test_pytree.py
+++ b/test/test_pytree.py
@@ -26,12 +26,12 @@ from torch.testing._internal.common_utils import (
     subtest,
     TEST_WITH_TORCHDYNAMO,
     TestCase,
+from torch_npu.contrib import transfer_to_npu
 )
-
-
 pytree_modules = {
     "python": python_pytree,
 }
+from torch_npu.contrib import transfer_to_npu
 if not IS_FBCODE:
     import torch.utils._cxx_pytree as cxx_pytree
 
@@ -867,6 +867,7 @@ class TestPythonPytree(TestCase):
                 lambda dummy: ([dummy.x, dummy.y], None),
                 lambda xs, _: DummyType(*xs),
             )
+    @unittest.skip("the _pytree functionality is maintained by upstream(native PyTorch)")
 
         with self.assertWarnsRegex(UserWarning, "already registered"):
             python_pytree._register_pytree_node(
