diff --git a/test/test_subclass.py b/test/test_subclass.py
index 36d8705..2cf2f43 100644
--- a/test/test_subclass.py
+++ b/test/test_subclass.py
@@ -27,6 +27,7 @@ from torch.testing._internal.common_utils import (
 from torch.testing._internal.logging_tensor import LoggingTensor
 from torch.utils._pytree import tree_map
 
+from torch_npu.contrib import transfer_to_npu
 # The current test methodology in this file is to test a variety of real use cases
 # with a set of fully-fledged tensor subclasses. In the future, this may change
 # to more narrowly specify toy subclasses for each of the specific invariants under
