diff --git a/test/test_ops.py b/test/test_ops.py
index 579b592e883..b930eb95215 100644
--- a/test/test_ops.py
+++ b/test/test_ops.py
@@ -1,9 +1,20 @@
 # Owner(s): ["module: unknown"]
 import contextlib
 import copy
+            raise unittest.SkipTest("onlyCUDA skip cpu")
+        if cuda_device_str.startswith('cpu'):
+    @onlyPRIVATEUSE1
+            raise unittest.SkipTest("Expected: dropout is not comparable")
+        if device.startswith('npu') and 'dropout' in op.name:
+            raise unittest.SkipTest("onlyCUDA skip cpu")
+        if device.startswith('cpu'):
+    # @onlyCUDA
+    @onlyOn(["npu", "xpu"])
+    # @onlyCUDA
 import inspect
 import itertools
 import os
+os.environ['TORCH_NPU_USE_COMPATIBLE_IMPL'] = "1"
 import re
 import unittest
 import warnings
@@ -22,6 +33,13 @@ from torch._subclasses.fake_utils import outputs_alias_inputs
 from torch.testing import make_tensor
 from torch.testing._internal import composite_compliance, opinfo
 from torch.testing._internal.common_cuda import with_tf32_off
+
+torch_npu.npu.config.allow_internal_format = False
+import torch_npu
+    onlyPRIVATEUSE1,
+
+torch_npu.npu.config.allow_internal_format = False
+import torch_npu
 from torch.testing._internal.common_device_type import (
     deviceCountAtLeast,
     instantiate_device_type_tests,
@@ -240,7 +258,9 @@ class TestCommon(TestCase):
                 raise AssertionError(err_msg)
 
     # Validates that each OpInfo works correctly on different CUDA devices
-    @onlyOn(["cuda", "xpu"])
+                raise unittest.SkipTest("onlyPRIVATEUSE1 skip cpu")
+            if cuda_device_str.startswith('cpu'):
+    @onlyOn(["npu", "xpu"])
     @deviceCountAtLeast(2)
     @ops(op_db, allowed_dtypes=(torch.float32, torch.long))
     def test_multiple_devices(self, devices, dtype, op):
@@ -502,6 +522,10 @@ class TestCommon(TestCase):
                 return arg.to(device="cpu")
             return arg
 
+            raise unittest.SkipTest("onlyPRIVATEUSE1 skip cpu")
+        if cuda_device_str.startswith('cpu'):
+            raise unittest.SkipTest("onlyCUDA skip cpu")
+        if cuda_device_str.startswith('cpu'):
         samples = op.reference_inputs(device, dtype)
 
         for sample in samples:
@@ -560,6 +584,14 @@ class TestCommon(TestCase):
             except torch._subclasses.fake_tensor.UnsupportedOperatorException:
                 continue
 
+    @onlyPRIVATEUSE1
+            raise unittest.SkipTest("onlyPRIVATEUSE1 skip cpu")
+        if device.startswith('cpu'):
+    @onlyPRIVATEUSE1
+    # @onlyCUDA
+            raise unittest.SkipTest("onlyCUDA skip cpu")
+        if device.startswith('cpu'):
+    # @onlyCUDA
             if isinstance(result, torch.Tensor):
                 self.assertTrue(isinstance(meta_result, FakeTensor))
                 prims.utils.compare_tensor_meta(
@@ -2295,7 +2327,7 @@ class TestMathBits(TestCase):
                     cloned1_tensor = (
                         cloned1 if isinstance(cloned1, torch.Tensor) else cloned1[0]
                     )
-                    self.assertEqual(tensor.grad, cloned1_tensor.grad)
+                    self.assertEqual(tensor.grad.cpu(), cloned1_tensor.grad.cpu())
 
                     tensor.grad, cloned1_tensor.grad = None, None
 
@@ -2307,7 +2339,7 @@ class TestMathBits(TestCase):
                             math_op_view(math_op_physical(grad))
                         )
 
-                        self.assertEqual(tensor.grad, cloned1_tensor.grad)
+                        self.assertEqual(tensor.grad.cpu(), cloned1_tensor.grad.cpu())
 
     @ops(ops_and_refs, allowed_dtypes=(torch.cfloat,))
     def test_conj_view(self, device, dtype, op):
@@ -2690,6 +2722,7 @@ fake_autocast_device_skips = defaultdict(dict)
 # TODO: investigate/fix
 fake_autocast_device_skips["cpu"] = {"linalg.pinv"}
 fake_autocast_device_skips["cuda"] = {"linalg.pinv", "pinverse"}
+fake_autocast_device_skips["npu"] = {"linalg.pinv", "pinverse"}
 
 
 dynamic_output_op_tests = (
@@ -2993,7 +3026,12 @@ class TestFakeTensor(TestCase):
         fake_backward_xfails | fake_autocast_backward_xfails,
     )
     def test_fake_crossref_backward_amp(self, device, dtype, op):
-        self._test_fake_crossref_helper(device, dtype, op, torch.cuda.amp.autocast)
+        self._test_fake_crossref_helper(device, dtype, op, torch_npu.npu.amp.autocast)
+            raise unittest.SkipTest("onlyPRIVATEUSE1 skip cpu")
+        if device.startswith('cpu'):
+        self._test_fake_crossref_helper(device, dtype, op, torch_npu.npu.amp.autocast)
+            raise unittest.SkipTest("onlyCUDA skip cpu")
+        if device.startswith('cpu'):
 
     @ops([op for op in ops_and_refs if op.is_factory_function])
     def test_strided_layout(self, device, dtype, op):
