diff --git a/test/test_python_dispatch.py b/test/test_python_dispatch.py
index 2e6bbd4..2548186 100644
--- a/test/test_python_dispatch.py
+++ b/test/test_python_dispatch.py
@@ -50,0 +51 @@ from torch.utils._pytree import tree_map, tree_map_only
+from torch_npu.contrib import transfer_to_npu
@@ -364 +365 @@ class TestPythonRegistration(TestCase):
-
+                print("--->", torch.where(cond, x, y))
@@ -2191,0 +2193 @@ $0: f32[] = torch._ops.aten.empty.memory_format([], device=device(type='cpu'), p
+                    tensor = torch.ones(123).npu()
@@ -2195 +2197 @@ $0: f32[] = torch._ops.aten.empty.memory_format([], device=device(type='cpu'), p
-                            123,
+                            tensor.size(0),
@@ -2200 +2202 @@ $0: f32[] = torch._ops.aten.empty.memory_format([], device=device(type='cpu'), p
-                        hint=123,
+                        hint=tensor.size(0),
