diff --git a/test/test_namedtensor.py b/test/test_namedtensor.py
index 0076da2..1454a22 100644
--- a/test/test_namedtensor.py
+++ b/test/test_namedtensor.py
@@ -1,5 +1,8 @@
 # Owner(s): ["module: named tensor"]
 # ruff: noqa: F841
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 import unittest
 from torch.testing._internal.common_utils import TestCase, run_tests, TEST_NUMPY
 from torch.testing._internal.common_utils import skipIfTorchDynamo
