diff --git a/test/test_out_dtype_op.py b/test/test_out_dtype_op.py
index f17da40..9a4f6be 100644
--- a/test/test_out_dtype_op.py
+++ b/test/test_out_dtype_op.py
@@ -13,6 +13,10 @@ from torch.testing._internal.common_utils import (
 from torch.testing._internal.common_quantization import skipIfNoDynamoSupport
 from torch.testing import FileCheck
 from torch.testing._internal.common_cuda import SM80OrLater, _get_torch_cuda_version
+from torch_npu.contrib import transfer_to_npu
+torch.cuda.get_device_capability = lambda :(10, 0)
+import torch_npu.testing
+TEST_CUDA = True
 
 
 @unittest.skipIf(not torch._dynamo.is_dynamo_supported(), "dynamo isn't support")
@@ -164,9 +168,9 @@ class TestOutDtypeOp(TestCase):
 
     @unittest.skipIf(IS_WINDOWS, "_int_mm unavailable")
     @unittest.skipIf(TEST_WITH_ROCM, "_int_mm unavailable")
-    @unittest.skipIf(not SM80OrLater, "_int_mm unavailable")
+    # @unittest.skipIf(not SM80OrLater, "_int_mm unavailable")
     @unittest.skipIf(IS_FBCODE and IS_REMOTE_GPU, "cublas runtime error")
-    @unittest.skipIf(_get_torch_cuda_version() >= (11, 7), "_int_mm unavailable")
+    # @unittest.skipIf(_get_torch_cuda_version() >= (11, 7), "_int_mm unavailable")
     @unittest.skipIf(not TEST_CUDA, "_int_mm unavailable")
     @skipIfNoDynamoSupport
     def test_out_dtype_inductor_decomp(self) -> None:
