diff --git a/test/test_expanded_weights.py b/test/test_expanded_weights.py
index 7f210bf..3306604 100644
--- a/test/test_expanded_weights.py
+++ b/test/test_expanded_weights.py
@@ -40,6 +40,14 @@ from torch.testing._internal.common_utils import (
 )
 from torch.utils._pytree import tree_map_only
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
+
+def patch_get_device_capability():
+    torch.cuda.get_device_capability = lambda : (10, 0)
+
+patch_get_device_capability()
 
 class TestContext:
     pass
