diff --git a/test/test_functionalization_of_rng_ops.py b/test/test_functionalization_of_rng_ops.py
index 3cc9f27..2731855 100644
--- a/test/test_functionalization_of_rng_ops.py
+++ b/test/test_functionalization_of_rng_ops.py
@@ -7,6 +7,7 @@ from unittest.mock import patch
 import torch
 import torch.utils.checkpoint
 from functorch.compile import aot_function, min_cut_rematerialization_partition, nop
+from torch_npu.contrib import transfer_to_npu
 
 from torch.testing._internal.common_device_type import (
     dtypes,
@@ -15,6 +16,10 @@ from torch.testing._internal.common_device_type import (
 
 from torch.testing._internal.common_utils import IS_CI, IS_WINDOWS, run_tests, TestCase
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
+
 if IS_WINDOWS and IS_CI:
     sys.stderr.write("torch.compile not supported on windows")
     if __name__ == "__main__":
