diff --git a/test/test_transformers_privateuse1.py b/test/test_transformers_privateuse1.py
index 9f03490..7624b1a 100644
--- a/test/test_transformers_privateuse1.py
+++ b/test/test_transformers_privateuse1.py
@@ -5,11 +5,12 @@ import unittest
 from collections import namedtuple
 from functools import partial
 
-import pytorch_openreg  # noqa: F401
+#import pytorch_openreg  # noqa: F401
 
 import torch
 import torch.utils.cpp_extension
 from torch.nn.attention import SDPBackend
+from torch_npu.contrib import transfer_to_npu
 from torch.testing._internal.common_nn import NNTestCase
 from torch.testing._internal.common_utils import (
     IS_FBCODE,
