diff --git a/test/test_type_promotion.py b/test/test_type_promotion.py
index 1548b88..dd34a4e 100644
--- a/test/test_type_promotion.py
+++ b/test/test_type_promotion.py
@@ -1,5 +1,4 @@
 # Owner(s): ["module: type promotion"]
-
 from functools import wraps
 import itertools
 import unittest
@@ -9,6 +8,8 @@ import torch
 from torch.testing._internal.common_utils import (TestCase, run_tests, load_tests, make_tensor,
                                                   TEST_NUMPY, set_default_dtype, torch_to_numpy_dtype_dict,
                                                   numpy_to_torch_dtype_dict, skipIfTorchDynamo)
+import torch_npu
+import torch_npu.testing
 from torch.testing._internal.common_device_type import (instantiate_device_type_tests, onlyNativeDeviceTypes,
                                                         dtypes, onlyCPU, expectedFailureMeta, skipMeta)
 from torch.testing._internal.common_dtype import (
@@ -422,9 +423,8 @@ class TestTypePromotion(TestCase):
     @float_double_default_dtype
     def test_booleans(self, device):
         onedim = torch.tensor([True], device=device)
-
-        self.assertEqual(onedim + onedim, onedim)
-        self.assertEqual(onedim + True, onedim)
+        self.assertEqual((onedim + onedim).cpu(), onedim)
+        self.assertEqual((onedim + True).cpu(), onedim)
         self.assertEqual(torch.add(True, True), True)
         self.assertEqual(torch.add(False, False), False)
         self.assertEqual(torch.add(False, True), True)
@@ -432,8 +432,8 @@ class TestTypePromotion(TestCase):
         self.assertRaisesRegex(RuntimeError, "Boolean alpha only supported",
                                lambda: torch.add(1, 1, alpha=True))
         self.assertEqual(torch.add(torch.tensor(True, device=device),
-                         torch.tensor(True, device=device), True),
-                         torch.tensor(True, device=device))
+                         torch.tensor(True, device=device), True).cpu(),
+                         torch.tensor(True, device=device).cpu())
 
     @skipIfTorchDynamo("Not a TorchDynamo suitable test")
     @float_double_default_dtype
@@ -1184,7 +1184,6 @@ class TestTypePromotion(TestCase):
 
 
 
-
 instantiate_device_type_tests(TestTypePromotion, globals())
 
 if __name__ == '__main__':
