diff --git a/test/test_sympy_utils.py b/test/test_sympy_utils.py
index e804e28..bfce5ee 100644
--- a/test/test_sympy_utils.py
+++ b/test/test_sympy_utils.py
@@ -37,7 +37,7 @@ from torch.utils._sympy.singleton_int import SingletonInt
 from torch.utils._sympy.solve import INEQUALITY_TYPES, mirror_rel_op, try_solve
 from torch.utils._sympy.value_ranges import ValueRanges
 from torch._inductor.bounds import ValueRangeAnalysis
-
+from torch_npu.contrib import transfer_to_npu
 
 UNARY_OPS = [
     "reciprocal",
