diff --git a/test/test_futures.py b/test/test_futures.py
index dd1e79f..caf52f9 100644
--- a/test/test_futures.py
+++ b/test/test_futures.py
@@ -8,6 +8,7 @@ import unittest
 from torch.futures import Future
 from torch.testing._internal.common_utils import IS_WINDOWS, TestCase, TemporaryFileName, run_tests
 from typing import TypeVar
+import torch_npu
 
 T = TypeVar("T")
 
@@ -79,7 +80,7 @@ class TestFuture(TestCase):
         f = Future[torch.Tensor]()
         self.assertFalse(f.done())
 
-        f.set_result(torch.ones(2, 2))
+        f.set_result(torch.ones(2, 2).npu())
         self.assertTrue(f.done())
 
     def test_done_exception(self) -> None:
@@ -90,7 +91,7 @@ class TestFuture(TestCase):
 
         f1 = Future[torch.Tensor]()
         self.assertFalse(f1.done())
-        f1.set_result(torch.ones(2, 2))
+        f1.set_result(torch.ones(2, 2).npu())
         self.assertTrue(f1.done())
 
         f2 = f1.then(raise_exception)
@@ -100,9 +101,9 @@ class TestFuture(TestCase):
 
     def test_wait(self) -> None:
         f = Future[torch.Tensor]()
-        f.set_result(torch.ones(2, 2))
+        f.set_result(torch.ones(2, 2).npu())
 
-        self.assertEqual(f.wait(), torch.ones(2, 2))
+        self.assertEqual(f.wait(), torch.ones(2, 2).npu())
 
     def test_wait_multi_thread(self) -> None:
 
@@ -112,10 +113,10 @@ class TestFuture(TestCase):
 
         f = Future[torch.Tensor]()
 
-        t = threading.Thread(target=slow_set_future, args=(f, torch.ones(2, 2)))
+        t = threading.Thread(target=slow_set_future, args=(f, torch.ones(2, 2).npu()))
         t.start()
 
-        self.assertEqual(f.wait(), torch.ones(2, 2))
+        self.assertEqual(f.wait(), torch.ones(2, 2).npu())
         t.join()
 
     def test_mark_future_twice(self) -> None:
@@ -138,9 +139,9 @@ class TestFuture(TestCase):
         fut = Future[torch.Tensor]()
         then_fut = fut.then(lambda x: x.wait() + 1)
 
-        fut.set_result(torch.ones(2, 2))
-        self.assertEqual(fut.wait(), torch.ones(2, 2))
-        self.assertEqual(then_fut.wait(), torch.ones(2, 2) + 1)
+        fut.set_result(torch.ones(2, 2).npu())
+        self.assertEqual(fut.wait(), torch.ones(2, 2).npu())
+        self.assertEqual(then_fut.wait(), torch.ones(2, 2).npu() + 1)
 
     def test_chained_then(self):
         fut = Future[torch.Tensor]()
@@ -150,10 +151,10 @@ class TestFuture(TestCase):
             last_fut = last_fut.then(add_one)
             futs.append(last_fut)
 
-        fut.set_result(torch.ones(2, 2))
+        fut.set_result(torch.ones(2, 2).npu())
 
         for i in range(len(futs)):
-            self.assertEqual(futs[i].wait(), torch.ones(2, 2) + i + 1)
+            self.assertEqual(futs[i].wait(), torch.ones(2, 2).npu() + i + 1)
 
     def _test_then_error(self, cb, errMsg):
         fut = Future[int]()
@@ -197,8 +198,8 @@ class TestFuture(TestCase):
         fut.add_done_callback(callback)
 
         self.assertFalse(callback_result)
-        fut.set_result(torch.ones(2, 2))
-        self.assertEqual(fut.wait(), torch.ones(2, 2))
+        fut.set_result(torch.ones(2, 2).npu())
+        self.assertEqual(fut.wait(), torch.ones(2, 2).npu())
         self.assertTrue(callback_result)
 
     def test_add_done_callback_maintains_callback_order(self):
@@ -218,8 +219,8 @@ class TestFuture(TestCase):
         fut.add_done_callback(callback_set1)
         fut.add_done_callback(callback_set2)
 
-        fut.set_result(torch.ones(2, 2))
-        self.assertEqual(fut.wait(), torch.ones(2, 2))
+        fut.set_result(torch.ones(2, 2).npu())
+        self.assertEqual(fut.wait(), torch.ones(2, 2).npu())
         # set2 called last, callback_result = 2
         self.assertEqual(callback_result, 2)
 
@@ -269,10 +270,10 @@ class TestFuture(TestCase):
         then_fut = fut.then(callback_then)
         fut.add_done_callback(callback_set2)
 
-        fut.set_result(torch.ones(2, 2))
-        self.assertEqual(fut.wait(), torch.ones(2, 2))
+        fut.set_result(torch.ones(2, 2).npu())
+        self.assertEqual(fut.wait(), torch.ones(2, 2).npu())
         # then_fut's callback is called with callback_result = 1
-        self.assertEqual(then_fut.wait(), torch.ones(2, 2) + 1)
+        self.assertEqual(then_fut.wait(), torch.ones(2, 2).npu() + 1)
         # set2 called last, callback_result = 2
         self.assertEqual(callback_result, 2)
 
@@ -283,11 +284,11 @@ class TestFuture(TestCase):
         fut = Future[torch.Tensor]()
         then_fut = fut.then(raise_value_error)
         fut.add_done_callback(raise_value_error)
-        fut.set_result(torch.ones(2, 2))
+        fut.set_result(torch.ones(2, 2).npu())
 
         # error from add_done_callback's callback is swallowed
         # error from then's callback is not
-        self.assertEqual(fut.wait(), torch.ones(2, 2))
+        self.assertEqual(fut.wait(), torch.ones(2, 2).npu())
         with self.assertRaisesRegex(RuntimeError, "Expected error"):
             then_fut.wait()
 
