diff --git a/test/test_shape_ops.py b/test/test_shape_ops.py
index 48834a80922..b1c71f992c6 100644
--- a/test/test_shape_ops.py
+++ b/test/test_shape_ops.py
@@ -36,6 +36,9 @@ from torch.testing._internal.common_utils import (
 )
 
 
+from torch_npu.contrib import transfer_to_npu
+import torch_npu
+
 # TODO: replace with make_tensor
 def _generate_input(shape, dtype, device, with_extremal):
     if shape == ():
@@ -577,7 +580,7 @@ class TestShapeOps(TestCase):
                     np_fn = partial(np.flip, axis=flip_dim)
                     self.compare_with_numpy(torch_fn, np_fn, data)
 
-    @onlyOn(["cuda", "xpu"])  # CPU is too slow
+    @onlyOn(["npu", "xpu"])  # CPU is too slow
     @largeTensorTest("17GB")  # 4 tensors of 4GB (in, out) x (torch, numpy) + 1GB
     @largeTensorTest(
         "81GB", "cpu"
