diff --git a/test/test_scaled_matmul_cuda.py b/test/test_scaled_matmul_cuda.py
index 5aafa1399ff..0dc3b7fccf6 100644
--- a/test/test_scaled_matmul_cuda.py
+++ b/test/test_scaled_matmul_cuda.py
@@ -10,7 +10,8 @@ import unittest
 
 import torch
 
-
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 from torch.nn.functional import (
     grouped_mm,
     pad,
@@ -34,7 +35,7 @@ from torch.testing._internal.common_cuda import (
 )
 from torch.testing._internal.common_device_type import (
     instantiate_device_type_tests,
-    onlyCUDA,
+    onlyPRIVATEUSE1,
     onlyOn,
     e4m3_type,
     e5m2_type,
@@ -631,7 +632,7 @@ def _build_scaled_grouped_mm_kwargs(scale_a, scale_b, offs, format):
 
 class TestFP8Matmul(TestCase):
 
-    def _test_tautological_mm(self, device: str = "cuda",
+    def _test_tautological_mm(self, device: str = "npu",
                               x_dtype: torch.dtype = e4m3_type,
                               y_dtype: torch.dtype = e4m3_type,
                               out_dtype: torch.dtype | None = None,
@@ -682,7 +683,7 @@ class TestFP8Matmul(TestCase):
         # supported on ROCm but fails on CUDA
         ctx = (
             self.assertRaises(ValueError)
-            if expect_e5m2_cuda_error and torch.version.hip is None and "cuda" in device
+            if expect_e5m2_cuda_error and torch.version.hip is None and "npu" in device
             else contextlib.nullcontext()
         )
         with ctx:
@@ -754,10 +755,10 @@ class TestFP8Matmul(TestCase):
 
         total_K = K  # Alias for clarity, communicating this consists of several groups along this dim
         input_group_end_offsets = generate_jagged_offs(
-            G, total_K, multiple_of=32, device="cuda"
+            G, total_K, multiple_of=32, device="npu"
         )
-        X = torch.randn((M, total_K), dtype=torch.bfloat16, device="cuda") * 0.1
-        W = torch.randn((N, total_K), dtype=torch.bfloat16, device="cuda") * 0.01
+        X = torch.randn((M, total_K), dtype=torch.bfloat16, device="npu") * 0.1
+        W = torch.randn((N, total_K), dtype=torch.bfloat16, device="npu") * 0.01
 
         xh, xq, x_blocked_scales, x_global_scales = _2d_grouped_tensor_to_blocked_scaled(
             X, M, G, input_group_end_offsets, format=format
@@ -826,10 +827,10 @@ class TestFP8Matmul(TestCase):
         # 2D inputs with groups along M, 3D weights.
         block_size = 32
         total_M = M  # Alias for clarity that M dim contains groups.
-        X = torch.randn((total_M, K), dtype=torch.bfloat16, device="cuda") * 0.1
-        W = torch.randn((G, N, K), dtype=torch.bfloat16, device="cuda") * 0.01
+        X = torch.randn((total_M, K), dtype=torch.bfloat16, device="npu") * 0.1
+        W = torch.randn((G, N, K), dtype=torch.bfloat16, device="npu") * 0.01
         input_group_end_offsets = generate_jagged_offs(
-            G, total_M, multiple_of=32, device="cuda"
+            G, total_M, multiple_of=32, device="npu"
         )
 
         # For each constituent 2d subtensor in the 3d weights, quantize and convert scale to blocked format separately,
@@ -1015,7 +1016,7 @@ class TestFP8Matmul(TestCase):
 
     @unittest.skipIf(not PLATFORM_SUPPORTS_FP8, f8_msg)
     @parametrize("base_dtype", [torch.float16, torch.bfloat16, torch.float32])
-    def test_scaled_mm_change_stride(self, base_dtype, device="cuda"):
+    def test_scaled_mm_change_stride(self, base_dtype, device="npu"):
         torch.manual_seed(42)
         input_dtype = e4m3_type
         output_dtype = base_dtype
@@ -1065,7 +1066,7 @@ class TestFP8Matmul(TestCase):
 
         torch.testing.assert_close(out_scaled_mm, out_emulated, atol=atol, rtol=rtol)
 
-    @onlyOn(["cuda", "xpu", "cpu"])
+    @onlyOn(["npu", "xpu", "cpu"])
     @skipCUDAIf(not PLATFORM_SUPPORTS_FP8, f8_msg)
     def test_float8_bias(self, device) -> None:
         (k, l, m) = (16, 48, 32)
@@ -1082,7 +1083,7 @@ class TestFP8Matmul(TestCase):
         difference = torch.abs(out_fp32 - outb_fp32)
         self.assertEqual(difference, torch.tensor(4.0, device=device).expand_as(out_fp32))
 
-    @onlyOn(["cuda", "xpu", "cpu"])
+    @onlyOn(["npu", "xpu", "cpu"])
     @unittest.skipIf(not PLATFORM_SUPPORTS_FP8, f8_msg)
     @parametrize("bias", [True, False])
     def test_non_divisible_leading_dim(self, device, bias: bool) -> None:
@@ -1095,7 +1096,7 @@ class TestFP8Matmul(TestCase):
             input_bias = torch.rand((16,), device=device).to(torch.bfloat16)
         _ = scaled_mm_wrap(x, y, scale_a, scale_b, bias=input_bias)
 
-    @onlyOn(["cuda", "xpu", "cpu"])
+    @onlyOn(["npu", "xpu", "cpu"])
     @unittest.skipIf(not PLATFORM_SUPPORTS_FP8, f8_msg)
     def test_float8_bias_relu_edgecase(self, device) -> None:
         (k, l, m) = (16, 48, 32)
@@ -1108,7 +1109,7 @@ class TestFP8Matmul(TestCase):
         outb_fp32 = outb_fp8.to(torch.float32)
         self.assertEqual(outb_fp32, torch.tensor(-3.0, device=device).expand_as(outb_fp32))
 
-    @onlyOn(["cuda", "xpu", "cpu"])
+    @onlyOn(["npu", "xpu", "cpu"])
     @unittest.skipIf(not PLATFORM_SUPPORTS_FP8, f8_msg)
     def test_float32_output_errors_with_bias(self, device) -> None:
         (k, l, m) = (16, 48, 32)
@@ -1125,7 +1126,7 @@ class TestFP8Matmul(TestCase):
                 lambda: scaled_mm_wrap(x, y, scale_a, scale_b, bias=bias, out_dtype=torch.float32),
             )
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @unittest.skipIf(PLATFORM_SUPPORTS_FP8 or not torch.cuda.is_available(), f8_msg)
     def test_error_message_fp8_pre_sm89(self, device) -> None:
         (k, l, m) = (16, 48, 32)
@@ -1155,7 +1156,7 @@ class TestFP8Matmul(TestCase):
         out_fp8_s = scaled_mm_wrap(x, y, scale_a=scale_a, scale_b=scale_b, out_dtype=e4m3_type, use_fast_accum=True)
         self.assertEqual(out_fp8, out_fp8_s)
 
-    @onlyOn(["cuda", "xpu", "cpu"])
+    @onlyOn(["npu", "xpu", "cpu"])
     @unittest.skipIf(not PLATFORM_SUPPORTS_FP8 or IS_WINDOWS, f8_msg)
     @skipCUDAIf(not SM89OrLater, "rowwise implementation is currently sm89-sm100 specific")
     @parametrize("use_fast_accum", [True, False])
@@ -1186,7 +1187,7 @@ class TestFP8Matmul(TestCase):
             out_fp8.to(torch.float32), torch.full((M, N), K * (fill_value**2), device=device)
         )
 
-    @onlyOn(["cuda", "xpu", "cpu"])
+    @onlyOn(["npu", "xpu", "cpu"])
     @unittest.skipIf(not PLATFORM_SUPPORTS_FP8 or IS_WINDOWS, f8_msg)
     def test_float8_error_messages(self, device) -> None:
         M, K, N = (1024, 512, 2048)
@@ -1523,7 +1524,7 @@ class TestFP8Matmul(TestCase):
             else:
                 scale_shape = M // 128, K // 128
 
-            scale = torch.full(scale_shape, val, device='cuda')
+            scale = torch.full(scale_shape, val, device='npu')
 
             return scale
 
@@ -1538,20 +1539,20 @@ class TestFP8Matmul(TestCase):
         if test_case == "x_eye_b_eye":
             if M != K or M != N:
                 return unittest.skip("a_eye_b_eye only defined for M = N = K")
-            x = torch.eye(M, device='cuda')
-            y = torch.eye(M, device='cuda')
+            x = torch.eye(M, device='npu')
+            y = torch.eye(M, device='npu')
 
             x_hp, x_recipe, x_fp8, x_scales, x_scales_original = _build_lhs(x, lhs_block)
             y_hp, y_recipe, y_fp8, y_scales, y_scales_original = _build_lhs(y, rhs_block)
         elif test_case == "x_ones_y_ones_calc_scales":
-            x = torch.full((M, K), 1.0, device='cuda')
-            y = torch.full((N, K), 1.0, device='cuda')
+            x = torch.full((M, K), 1.0, device='npu')
+            y = torch.full((N, K), 1.0, device='npu')
 
             x_hp, x_recipe, x_fp8, x_scales, x_scales_original = _build_lhs(x, lhs_block)
             y_hp, y_recipe, y_fp8, y_scales, y_scales_original = _build_lhs(y, rhs_block)
         elif test_case in ["x_ones_y_ones_set_scales", "x_ones_y_ones_modify_scales"]:
-            x = torch.full((M, K), 1.0, device='cuda')
-            y = torch.full((N, K), 1.0, device='cuda')
+            x = torch.full((M, K), 1.0, device='npu')
+            y = torch.full((N, K), 1.0, device='npu')
 
             x_scales = _build_constant_scale(x, lhs_block, 1.)
             y_scales = _build_constant_scale(y, rhs_block, 1.)
@@ -1566,8 +1567,8 @@ class TestFP8Matmul(TestCase):
             x_hp, x_recipe, x_scales, x_scales_original = _adjust_lhs_scale(x_fp8, x_scales, lhs_block)
             y_hp, y_recipe, y_scales, y_scales_original = _adjust_rhs_scale(y_fp8, y_scales, rhs_block)
         elif test_case == "data_random_scales_one":
-            x = torch.randint(0, 255, (M, K), device='cuda', dtype=torch.uint8).to(torch.bfloat16)
-            y = torch.randint(0, 255, (N, K), device='cuda', dtype=torch.uint8).to(torch.bfloat16)
+            x = torch.randint(0, 255, (M, K), device='npu', dtype=torch.uint8).to(torch.bfloat16)
+            y = torch.randint(0, 255, (N, K), device='npu', dtype=torch.uint8).to(torch.bfloat16)
 
             x_scales = _build_constant_scale(x, lhs_block, 1.)
             y_scales = _build_constant_scale(y, rhs_block, 1.)
@@ -1579,8 +1580,8 @@ class TestFP8Matmul(TestCase):
             y_hp, y_recipe, y_scales, y_scales_original = _adjust_rhs_scale(y_fp8, y_scales, rhs_block)
         elif test_case == "data_random_calc_scales":
             # Note: Old test_scaled_mm_vs_emulated_block_wise test case
-            x = torch.randn(M, K, device="cuda", dtype=output_dtype)
-            y = torch.randn(N, K, device="cuda", dtype=output_dtype) * 1e-3
+            x = torch.randn(M, K, device="npu", dtype=output_dtype)
+            y = torch.randn(N, K, device="npu", dtype=output_dtype) * 1e-3
 
             x_hp, x_recipe, x_fp8, x_scales, x_scales_original = _build_lhs(x, lhs_block)
             y_hp, y_recipe, y_fp8, y_scales, y_scales_original = _build_lhs(y, rhs_block)
@@ -1605,8 +1606,8 @@ class TestFP8Matmul(TestCase):
     ):
         torch.manual_seed(42)
 
-        x = torch.randn(M, K, device="cuda", dtype=output_dtype).pow(3)
-        y = torch.randn(N, K, device="cuda", dtype=output_dtype).pow(3)
+        x = torch.randn(M, K, device="npu", dtype=output_dtype).pow(3)
+        y = torch.randn(N, K, device="npu", dtype=output_dtype).pow(3)
 
         x_fp8, x_scales = tensor_to_scale_block(x, e4m3_type, lhs_block, 128)
         y_fp8, y_scales = tensor_to_scale_block(y, e4m3_type, rhs_block, 128)
@@ -1661,7 +1662,7 @@ class TestFP8Matmul(TestCase):
             output_dtype
         )
 
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @unittest.skipIf(not PLATFORM_SUPPORTS_FP8 or IS_WINDOWS, f8_msg)
     @unittest.skipIf(IS_SM90, "DeepSeek style (1x128, 128x128) blockwise scaling works on SM90 (Hopper)")
     @unittest.skipIf(
@@ -1678,8 +1679,8 @@ class TestFP8Matmul(TestCase):
 
         torch.manual_seed(42)
 
-        x = torch.randn(M, K, device="cuda", dtype=output_dtype).pow(3)
-        y = torch.randn(N, K, device="cuda", dtype=output_dtype).pow(3)
+        x = torch.randn(M, K, device="npu", dtype=output_dtype).pow(3)
+        y = torch.randn(N, K, device="npu", dtype=output_dtype).pow(3)
 
         x_fp8, x_scales = tensor_to_scale_block(x, e4m3_type, lhs_block, 128)
         y_fp8, y_scales = tensor_to_scale_block(y, e4m3_type, rhs_block, 128)
@@ -1752,8 +1753,8 @@ class TestFP8Matmul(TestCase):
     def test_honor_sm_carveout(self) -> None:
         torch.manual_seed(42)
 
-        x = torch.randn(8192, 2048, device="cuda", dtype=torch.float32)
-        y = torch.randn(8192, 2048, device="cuda", dtype=torch.float32).t()
+        x = torch.randn(8192, 2048, device="npu", dtype=torch.float32)
+        y = torch.randn(8192, 2048, device="npu", dtype=torch.float32).t()
         x_scales = tensor_to_scale(x, e4m3_type, dim=1).reciprocal()
         y_scales = tensor_to_scale(y, e4m3_type, dim=0).reciprocal()
         x_fp8 = to_fp8_saturated(x / x_scales, e4m3_type)
@@ -1837,7 +1838,7 @@ class TestFP8Matmul(TestCase):
         torch.testing.assert_close(lp_data_actual, lp_data_expected, atol=0, rtol=0)
 
     @skipIfRocm
-    @onlyCUDA
+    @onlyPRIVATEUSE1
     @unittest.skipIf(not PLATFORM_SUPPORTS_MX_GEMM, mx_skip_msg)
     @parametrize("mkn", [
         # Nice shapes
@@ -1858,7 +1859,7 @@ class TestFP8Matmul(TestCase):
         (1025, 128, 96)
     ], name_fn=lambda mkn: f"{mkn[0]}_{mkn[1]}_{mkn[2]}")
     def test_blockwise_nvfp4_with_global_scale(self, mkn) -> None:
-        device = 'cuda'
+        device = 'npu'
         M, K, N = mkn
         BLOCK_SIZE = 16
         # Note: SQNR target from `test_blockwise_mxfp8_nvfp4_mxfp4_numerics` test
@@ -1940,7 +1941,7 @@ class TestFP8Matmul(TestCase):
         if recipe == "mxfp4" and SM120OrLater:
             raise unittest.SkipTest("MXFP4 on CUDA only supported on B200/B300")
 
-        device = "cuda"
+        device = "npu"
         M, K, N = mkn
         if recipe == "nvfp4" and K % 32 != 0:
             raise unittest.SkipTest("K must be divisible by 32 for nvfp4 cublas gemm, skipping")
@@ -2390,7 +2391,7 @@ class TestFP8Matmul(TestCase):
     # AMD does not support NVFP4
     @parametrize("wrap_v2", [True, False])
     def test_scaled_grouped_gemm_2d_2d(self, fast_accum, strided, wrap_v2):
-        device = "cuda"
+        device = "npu"
         fp8_dtype = e4m3_type
         m, n, k, n_groups = 16, 32, 64, 4
         a = torch.randn(m, k * n_groups + k * int(strided), device=device).to(fp8_dtype)[:, :k * n_groups]
@@ -2426,7 +2427,7 @@ class TestFP8Matmul(TestCase):
     @parametrize("strided", [False] + ([True] if torch.version.cuda else []))
     @parametrize("wrap_v2", [True, False])
     def test_scaled_grouped_gemm_2d_3d(self, fast_accum, strided, wrap_v2):
-        device = "cuda"
+        device = "npu"
         fp8_dtype = e4m3_type
         m, n, k, n_groups = 16, 32, 64, 4
         s_int = int(strided)
@@ -2438,11 +2439,11 @@ class TestFP8Matmul(TestCase):
             if check_zero_size and n_groups <= 1:
                 continue
 
-            offs = torch.arange(m, n_groups * m + 1, m, device="cuda", dtype=torch.int32)
+            offs = torch.arange(m, n_groups * m + 1, m, device="npu", dtype=torch.int32)
             if check_zero_size:
                 offs[0] = offs[1]
-            scale_a = torch.rand(n_groups * m, device="cuda", dtype=torch.float32)
-            scale_b = torch.rand(n_groups * n, device="cuda", dtype=torch.float32).view(n_groups, n)
+            scale_a = torch.rand(n_groups * m, device="npu", dtype=torch.float32)
+            scale_b = torch.rand(n_groups * n, device="npu", dtype=torch.float32).view(n_groups, n)
             f = scaled_grouped_mm_wrap
             out = f(a, b.transpose(-2, -1),
                     scale_a,
@@ -2470,7 +2471,7 @@ class TestFP8Matmul(TestCase):
     # AMD does not support non-contiguous inputs yet
     @parametrize("strided", [False] + ([True] if torch.version.cuda else []))
     def test_scaled_grouped_gemm_3d_3d(self, fast_accum, strided):
-        device = "cuda"
+        device = "npu"
         fp8_dtype = e4m3_type
         m, n, k, n_groups = 16, 32, 64, 4
         s_int = int(strided)
@@ -2478,8 +2479,8 @@ class TestFP8Matmul(TestCase):
         b = torch.randn(n_groups * (1 + s_int), n, k * (1 + s_int), device=device).to(fp8_dtype)[::(1 + s_int), :, :k]
         self.assertTrue(a.is_contiguous() is not strided)
         self.assertTrue(b.is_contiguous() is not strided)
-        scale_a = torch.rand(n_groups * m, device="cuda", dtype=torch.float32).view(n_groups, m)
-        scale_b = torch.rand(n_groups * n, device="cuda", dtype=torch.float32).view(n_groups, n)
+        scale_a = torch.rand(n_groups * m, device="npu", dtype=torch.float32).view(n_groups, m)
+        scale_b = torch.rand(n_groups * n, device="npu", dtype=torch.float32).view(n_groups, n)
 
         f = torch._scaled_grouped_mm
         out = f(a, b.transpose(-2, -1), scale_a, scale_b,
@@ -2493,7 +2494,7 @@ class TestFP8Matmul(TestCase):
     # AMD does not support non-contiguous inputs yet
     @parametrize("strided", [False] + ([True] if torch.version.cuda else []))
     def test_scaled_grouped_gemm_3d_2d(self, fast_accum, strided):
-        device = "cuda"
+        device = "npu"
         fp8_dtype = e4m3_type
         m, n, k, n_groups = 16, 32, 64, 4
         s_int = int(strided)
@@ -2501,13 +2502,13 @@ class TestFP8Matmul(TestCase):
         b = torch.randn(n * n_groups, k * (1 + s_int), device=device).to(fp8_dtype)[:, :k]
         self.assertTrue(a.is_contiguous() is not strided)
         self.assertTrue(b.is_contiguous() is not strided)
-        scale_a = torch.rand(n_groups * m, device="cuda", dtype=torch.float32).view(n_groups, m)
-        scale_b = torch.rand(n_groups * n, device="cuda", dtype=torch.float32)
+        scale_a = torch.rand(n_groups * m, device="npu", dtype=torch.float32).view(n_groups, m)
+        scale_b = torch.rand(n_groups * n, device="npu", dtype=torch.float32)
         for check_zero_size in (True, False):
             if check_zero_size and n_groups <= 1:
                 continue
 
-            offs = torch.arange(n, n_groups * n + 1, n, device="cuda", dtype=torch.int32)
+            offs = torch.arange(n, n_groups * n + 1, n, device="npu", dtype=torch.int32)
             if check_zero_size:
                 offs[0] = offs[1]
 
@@ -2528,7 +2529,7 @@ class TestFP8Matmul(TestCase):
     @unittest.skipIf(not PLATFORM_SUPPORTS_MX_GEMM, mx_skip_msg)
     def test_blockwise_mxfp8_compile(self) -> None:
 
-        device = "cuda"
+        device = "npu"
         M, K, N = 128, 128, 128
         BLOCK_SIZE = 32
 
@@ -2557,7 +2558,7 @@ class TestFP8Matmul(TestCase):
     @unittest.skipIf(not PLATFORM_SUPPORTS_MX_GEMM, mx_skip_msg)
     def test_blockwise_nvfp4_compile(self) -> None:
 
-        device = "cuda"
+        device = "npu"
         M, K, N = 128, 128, 128
         BLOCK_SIZE = 32 if torch.version.hip else 16
         fp4_scaling_dtype = torch.float8_e8m0fnu if torch.version.hip else torch.float8_e4m3fn
