diff --git a/test/test_datapipe.py b/test/test_datapipe.py
index da335db..86763b7 100644
--- a/test/test_datapipe.py
+++ b/test/test_datapipe.py
@@ -71,6 +71,8 @@ skipIfNoDataFrames = skipIf(not HAS_PANDAS, "no dataframes (pandas)")
 skipTyping = skipIf(True, "TODO: Fix typing bug")
 T_co = TypeVar("T_co", covariant=True)
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
 
 def create_temp_dir_and_files():
     # The temp dir and files within it will be released and deleted in tearDown().
