diff --git a/test/test_pytree.py b/test/test_pytree.py
index 4560ac6e69..7aeda2acd5 100644
--- a/test/test_pytree.py
+++ b/test/test_pytree.py
@@ -196,7 +196,7 @@ class TestGenericPytree(TestCase):
         run_test_with_leaf(1.0)
         run_test_with_leaf(None)
         run_test_with_leaf(bool)
-        run_test_with_leaf(torch.randn(3, 3))
+        run_test_with_leaf(torch.randn(3, 3).npu())
 
     @parametrize(
         "pytree_impl,gen_expected_fn",
@@ -231,7 +231,7 @@ class TestGenericPytree(TestCase):
         run_test(())
         run_test((1.0,))
         run_test((1.0, 2))
-        run_test((torch.tensor([1.0, 2]), 2, 10, 9, 11))
+        run_test((torch.tensor([1.0, 2]).npu(), 2, 10, 9, 11))
 
     @parametrize(
         "pytree_impl,gen_expected_fn",
@@ -265,7 +265,7 @@ class TestGenericPytree(TestCase):
 
         run_test([])
         run_test([1.0, 2])
-        run_test([torch.tensor([1.0, 2]), 2, 10, 9, 11])
+        run_test([torch.tensor([1.0, 2]).npu(), 2, 10, 9, 11])
 
     @parametrize(
         "pytree_impl,gen_expected_fn",
@@ -304,9 +304,9 @@ class TestGenericPytree(TestCase):
 
         run_test({})
         run_test({"a": 1})
-        run_test({"abcdefg": torch.randn(2, 3)})
-        run_test({1: torch.randn(2, 3)})
-        run_test({"a": 1, "b": 2, "c": torch.randn(2, 3)})
+        run_test({"abcdefg": torch.randn(2, 3).npu()})
+        run_test({1: torch.randn(2, 3).npu()})
+        run_test({"a": 1, "b": 2, "c": torch.randn(2, 3).npu()})
 
     @parametrize(
         "pytree_impl,gen_expected_fn",
@@ -349,7 +349,7 @@ class TestGenericPytree(TestCase):
         run_test(od)
 
         od["b"] = 1
-        od["a"] = torch.tensor(3.14)
+        od["a"] = torch.tensor(3.14).npu()
         run_test(od)
 
     @parametrize(
@@ -392,9 +392,9 @@ class TestGenericPytree(TestCase):
 
         run_test(defaultdict(list, {}))
         run_test(defaultdict(int, {"a": 1}))
-        run_test(defaultdict(int, {"abcdefg": torch.randn(2, 3)}))
-        run_test(defaultdict(int, {1: torch.randn(2, 3)}))
-        run_test(defaultdict(int, {"a": 1, "b": 2, "c": torch.randn(2, 3)}))
+        run_test(defaultdict(int, {"abcdefg": torch.randn(2, 3).npu()}))
+        run_test(defaultdict(int, {1: torch.randn(2, 3).npu()}))
+        run_test(defaultdict(int, {"a": 1, "b": 2, "c": torch.randn(2, 3).npu()}))
 
     @parametrize(
         "pytree_impl,gen_expected_fn",
@@ -436,7 +436,7 @@ class TestGenericPytree(TestCase):
 
         run_test(deque([]))
         run_test(deque([1.0, 2]))
-        run_test(deque([torch.tensor([1.0, 2]), 2, 10, 9, 11], maxlen=8))
+        run_test(deque([torch.tensor([1.0, 2]).npu(), 2, 10, 9, 11], maxlen=8))
 
     @parametrize(
         "pytree_impl",
@@ -465,7 +465,7 @@ class TestGenericPytree(TestCase):
             self.assertIsInstance(unflattened, Point)
 
         run_test(Point(1.0, 2))
-        run_test(Point(torch.tensor(1.0), 2))
+        run_test(Point(torch.tensor(1.0).npu(), 2))
 
     @parametrize(
         "op",
@@ -482,7 +482,7 @@ class TestGenericPytree(TestCase):
         ],
     )
     def test_flatten_unflatten_return_types(self, pytree_impl, op):
-        x = torch.randn(3, 3)
+        x = torch.randn(3, 3).npu()
         expected = op(x, dim=0)
 
         values, spec = pytree_impl.tree_flatten(expected)
@@ -517,7 +517,11 @@ class TestGenericPytree(TestCase):
             ([],),
             {"a": ()},
             {"a": 0, "b": [{"c": 1}]},
-            {"a": 0, "b": [1, {"c": 2}, torch.randn(3)], "c": (torch.randn(2, 3), 1)},
+            {
+                "a": 0,
+                "b": [1, {"c": 2}, torch.randn(3).npu()],
+                "c": (torch.randn(2, 3).npu(), 1),
+            },
         ]
         for case in cases:
             run_test(case)
@@ -558,10 +562,10 @@ class TestGenericPytree(TestCase):
             (
                 {
                     "a": 0,
-                    "b": [1, {"c": 2}, torch.ones(3)],
-                    "c": (torch.zeros(2, 3), 1),
+                    "b": [1, {"c": 2}, torch.ones(3).npu()],
+                    "c": (torch.zeros(2, 3).npu(), 1),
                 },
-                [0, [1, {"c": 2}, torch.ones(3)], (torch.zeros(2, 3), 1)],
+                [0, [1, {"c": 2}, torch.ones(3).npu()], (torch.zeros(2, 3).npu(), 1)],
             ),
         ]
         for case in cases:
@@ -755,6 +759,7 @@ class TestPythonPytree(TestCase):
                 lambda xs, _: DummyType(*xs),
             )
 
+    @unittest.skip("the _pytree functionality is maintained by upstream(native PyTorch)")
     def test_import_pytree_doesnt_import_optree(self):
         # importing torch.utils._pytree shouldn't import optree.
         # only importing torch.utils._cxx_pytree should.
@@ -1174,10 +1179,10 @@ if "optree" in sys.modules:
 
         py_pytree.register_dataclass(Point)
 
-        point = Point(torch.tensor(0), torch.tensor(1))
+        point = Point(torch.tensor(0).npu(), torch.tensor(1).npu())
         point = py_pytree.tree_map(lambda x: x + 1, point)
-        self.assertEqual(point.x, torch.tensor(1))
-        self.assertEqual(point.y, torch.tensor(2))
+        self.assertEqual(point.x, torch.tensor(1).npu())
+        self.assertEqual(point.y, torch.tensor(2).npu())
 
     def test_constant(self):
         # Either use `frozen=True` or `unsafe_hash=True` so we have a
@@ -1272,7 +1277,7 @@ if "optree" in sys.modules:
         SOME_PYTREES = [
             (None,),
             ["hello", [1, 2], {"foo": [(3)]}],
-            [ANamedTuple(x=torch.rand(2, 3), y=1, z="foo")],
+            [ANamedTuple(x=torch.rand(2, 3).npu(), y=1, z="foo")],
             [ACustomPytree(x=12, y={"cin": [1, 4, 10], "bar": 18}, z="leaf"), 5],
         ]
         for pytree in SOME_PYTREES:
@@ -1302,7 +1307,7 @@ if "optree" in sys.modules:
         SOME_PYTREES = [
             (None,),
             ["hello", [1, 2], {"foo": [(3)]}],
-            [ANamedTuple(x=torch.rand(2, 3), y=1, z="foo")],
+            [ANamedTuple(x=torch.rand(2, 3).npu(), y=1, z="foo")],
             [ACustomPytree(x=12, y={"cin": [1, 4, 10], "bar": 18}, z="leaf"), 5],
         ]
         for pytree in SOME_PYTREES:
