diff --git a/test/distributed/elastic/test_control_plane.py b/test/distributed/elastic/test_control_plane.py
index cfa2211..9b9f18b 100644
--- a/test/distributed/elastic/test_control_plane.py
+++ b/test/distributed/elastic/test_control_plane.py
@@ -1,3 +1,8 @@
+import torch_npu.testing
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 #!/usr/bin/env python3
 # Owner(s): ["oncall: distributed"]
 
@@ -74,80 +79,80 @@ class WorkerServerTest(TestCase):
             self.assertEqual(resp.status, 404)
             self.assertIn(b"Handler nonexistant not found:", resp.data)
 
-    @requires_cuda
+    # @requires_cuda
     def test_dump_nccl_trace_pickle(self) -> None:
         with local_worker_server() as pool:
-            resp = pool.request("POST", "/handler/dump_nccl_trace_pickle")
+            resp = pool.request("POST", "/handler/dump_hccl_trace_pickle")
             self.assertEqual(resp.status, 200)
             out = pickle.loads(resp.data)
             self.assertIsInstance(out, dict)
             self.assertIn("version", out)
 
-    @requires_cuda
+    # @requires_cuda
     def test_dump_nccl_trace_pickle_with_params(self) -> None:
         with local_worker_server() as pool:
             # bad key - not lower case
             resp = pool.request(
-                "POST", "/handler/dump_nccl_trace_pickle?includeCollectives=true"
+                "POST", "/handler/dump_hccl_trace_pickle?includeCollectives=true"
             )
             self.assertEqual(resp.status, 400)
             # unknown key
             resp = pool.request(
-                "POST", "/handler/dump_nccl_trace_pickle?unknownkey=true"
+                "POST", "/handler/dump_hccl_trace_pickle?unknownkey=true"
             )
             self.assertEqual(resp.status, 400)
             # bad value - not a bool
             resp = pool.request(
-                "POST", "/handler/dump_nccl_trace_pickle?includecollectives=notabool"
+                "POST", "/handler/dump_hccl_trace_pickle?includecollectives=notabool"
             )
             self.assertEqual(resp.status, 400)
             # bad value - value not lowercase
             resp = pool.request(
-                "POST", "/handler/dump_nccl_trace_pickle?includecollectives=True"
+                "POST", "/handler/dump_hccl_trace_pickle?includecollectives=True"
             )
             self.assertEqual(resp.status, 400)
             # good key and value
             resp = pool.request(
-                "POST", "/handler/dump_nccl_trace_pickle?includecollectives=true"
+                "POST", "/handler/dump_hccl_trace_pickle?includecollectives=true"
             )
             self.assertEqual(resp.status, 200)
             # multiple good keys and values
             resp = pool.request(
                 "POST",
-                "/handler/dump_nccl_trace_pickle?includecollectives=true&includestacktraces=false&onlyactive=true",
+                "/handler/dump_hccl_trace_pickle?includecollectives=true&includestacktraces=false&onlyactive=true",
             )
             self.assertEqual(resp.status, 200)
 
-    @requires_cuda
+    # @requires_cuda
     def test_dump_nccl_trace_pickle_with_json(self) -> None:
         with local_worker_server() as pool:
             # bad key - not lower case
             resp = pool.request(
-                "POST", "/handler/dump_nccl_trace_json?includeCollectives=true"
+                "POST", "/handler/dump_hccl_trace_json?includeCollectives=true"
             )
             self.assertEqual(resp.status, 400)
             # unknown key
-            resp = pool.request("POST", "/handler/dump_nccl_trace_json?unknownkey=true")
+            resp = pool.request("POST", "/handler/dump_hccl_trace_json?unknownkey=true")
             self.assertEqual(resp.status, 400)
             # bad value - not a bool
             resp = pool.request(
-                "POST", "/handler/dump_nccl_trace_json?includecollectives=notabool"
+                "POST", "/handler/dump_hccl_trace_json?includecollectives=notabool"
             )
             self.assertEqual(resp.status, 400)
             # bad value - value not lowercase
             resp = pool.request(
-                "POST", "/handler/dump_nccl_trace_json?includecollectives=True"
+                "POST", "/handler/dump_hccl_trace_json?includecollectives=True"
             )
             self.assertEqual(resp.status, 400)
             # good key and value
             resp = pool.request(
-                "POST", "/handler/dump_nccl_trace_json?includecollectives=true"
+                "POST", "/handler/dump_hccl_trace_json?includecollectives=true"
             )
             self.assertEqual(resp.status, 200)
             # multiple good keys and values
             resp = pool.request(
                 "POST",
-                "/handler/dump_nccl_trace_json?includecollectives=true&onlyactive=true",
+                "/handler/dump_hccl_trace_json?includecollectives=true&onlyactive=true",
             )
             self.assertEqual(resp.status, 200)
 
