diff --git a/torch/_inductor/codecache.py b/torch/_inductor/codecache.py
index 75f3c6316be..2d6e9926806 100644
--- a/torch/_inductor/codecache.py
+++ b/torch/_inductor/codecache.py
@@ -208,7 +208,8 @@ class CacheBase:
                 system["device"]["name"] = device_properties.name
                 system["version"]["cuda"] = torch.version.cuda
             else:
-                system["device"]["name"] = device_properties.gcnArchName
+                # system["device"]["name"] = device_properties.gcnArchName
+                system["device"]["name"] = "arm"
                 system["version"]["hip"] = torch.version.hip
         except (AssertionError, RuntimeError):
             # If cuda is not installed, none of the above config is relevant.
