diff --git a/torch/_inductor/utils.py b/torch/_inductor/utils.py
index 8e92150..6c8a831 100644
--- a/torch/_inductor/utils.py
+++ b/torch/_inductor/utils.py
@@ -71,7 +71,7 @@ if TYPE_CHECKING:
     from .scheduler import BaseSchedulerNode, SchedulerBuffer
 
 
-GPU_TYPES = ["cuda", "mps", "xpu"]
+GPU_TYPES = ["cuda", "mps", "xpu", "npu"]
 T = TypeVar("T")
 
 
@@ -79,10 +79,7 @@ T = TypeVar("T")
 # when get_gpu_type is imported from dynamo
 @functools.lru_cache(None)
 def get_gpu_type() -> str:
-    avail_gpus = [x for x in GPU_TYPES if getattr(torch, x).is_available()]
-    assert len(avail_gpus) <= 1
-    gpu_type = "cuda" if len(avail_gpus) == 0 else avail_gpus.pop()
-    return gpu_type
+    return "npu"
 
 
 from torch._dynamo.device_interface import get_interface_for_device
