diff --git a/test/profiler/test_profiler_tree.py b/test/profiler/test_profiler_tree.py
index 7dac5fb..7231d2f 100644
--- a/test/profiler/test_profiler_tree.py
+++ b/test/profiler/test_profiler_tree.py
@@ -22,6 +22,9 @@ from torch.testing._internal.common_utils import (
 )
 from torch.utils._pytree import tree_map
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
 # These functions can vary from based on platform and build (e.g. with CUDA)
 # and generally distract from rather than adding to the test.
