diff --git a/test/profiler/test_memory_profiler.py b/test/profiler/test_memory_profiler.py
index 304587f..6dcbe69 100644
--- a/test/profiler/test_memory_profiler.py
+++ b/test/profiler/test_memory_profiler.py
@@ -21,6 +21,9 @@ from torch.testing._internal.common_utils import (
 )
 from torch.utils import _pytree as pytree
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
 profile = functools.partial(
     torch.profiler.profile, record_shapes=True, profile_memory=True, with_stack=True
