diff --git a/test/profiler/test_torch_tidy.py b/test/profiler/test_torch_tidy.py
index 119db5b..1edc369 100644
--- a/test/profiler/test_torch_tidy.py
+++ b/test/profiler/test_torch_tidy.py
@@ -29,6 +29,9 @@ from torch._C._profiler import _TensorMetadata
 from torch.profiler import _utils, profile
 from torch.testing._internal.common_utils import run_tests, TestCase
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
 Json = dict[str, Any]
 
