diff --git a/test/profiler/test_execution_trace.py b/test/profiler/test_execution_trace.py
index 072b5b1..c8743cf 100644
--- a/test/profiler/test_execution_trace.py
+++ b/test/profiler/test_execution_trace.py
@@ -52,6 +52,9 @@ from torch.testing._internal.common_utils import (
 )
 from torch.utils._triton import has_triton
 
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 
 Json = dict[str, Any]
 
