diff --git a/test/lazy/test_ts_opinfo.py b/test/lazy/test_ts_opinfo.py
index 7c467dc..1d83ca9 100644
--- a/test/lazy/test_ts_opinfo.py
+++ b/test/lazy/test_ts_opinfo.py
@@ -1,5 +1,6 @@
 # Owner(s): ["oncall: jit"]
-
+# import torch_npu
+# from torch_npu.contrib import transfer_to_npu
 import functools
 import itertools
 import os
@@ -28,7 +29,7 @@ torch._lazy.ts_backend.init()
 
 
 def get_test_device():
-    return "cuda" if "LTC_TS_CUDA" in os.environ else "cpu"
+    return "npu" if "LTC_TS_CUDA" in os.environ else "cpu"
 
 
 def remove_suffixes(l):
@@ -332,7 +333,7 @@ class TestLazyOpInfo(TestCase):
 
 # TODO: after we move to master, add Lazy as a new Device here:
 # https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_device_type.py#L532
-instantiate_device_type_tests(TestLazyOpInfo, globals(), only_for="cpu")
+instantiate_device_type_tests(TestLazyOpInfo, globals(), only_for=["cpu"])
 
 
 class TestLazyDynamicOps(TestCase):
