diff --git a/test/onnx/test_models_onnxruntime.py b/test/onnx/test_models_onnxruntime.py
index 9c38644..dac68ce 100644
--- a/test/onnx/test_models_onnxruntime.py
+++ b/test/onnx/test_models_onnxruntime.py
@@ -24,6 +24,15 @@ from torchvision.models.detection import (
 )
 
 import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
+
 from torch import nn
 from torch.testing._internal import common_utils
 
@@ -38,30 +47,38 @@ def exportTest(
     acceptable_error_percentage=None,
 ):
     opset_versions = opset_versions if opset_versions else [7, 8, 9, 10, 11, 12, 13, 14]
-
-    for opset_version in opset_versions:
-        self.opset_version = opset_version
-        self.onnx_shape_inference = True
-        onnx_test_common.run_model_test(
-            self,
-            model,
-            input_args=inputs,
-            rtol=rtol,
-            atol=atol,
-            acceptable_error_percentage=acceptable_error_percentage,
-        )
-
-        if self.is_script_test_enabled and opset_version > 11:
-            script_model = torch.jit.script(model)
+    
+    original_hf32 = torch_npu.npu.conv.allow_hf32
+    try:
+        # Conv using HF32 (half-precision floating point) for Ascend-dedicated operator
+        # internal calculation introduces errors that are amplified in subsequent calculations.
+        torch.npu.conv.allow_hf32 = False
+
+        for opset_version in opset_versions:
+            self.opset_version = opset_version
+            self.onnx_shape_inference = True
             onnx_test_common.run_model_test(
                 self,
-                script_model,
+                model,
                 input_args=inputs,
                 rtol=rtol,
                 atol=atol,
                 acceptable_error_percentage=acceptable_error_percentage,
             )
 
+            if self.is_script_test_enabled and opset_version > 11:
+                script_model = torch.jit.script(model)
+                onnx_test_common.run_model_test(
+                    self,
+                    script_model,
+                    input_args=inputs,
+                    rtol=rtol,
+                    atol=atol,
+                    acceptable_error_percentage=acceptable_error_percentage,
+                )
+    finally:
+        torch.npu.conv.allow_hf32 = original_hf32
+
 
 TestModels = type(
     "TestModels",
