diff --git a/test/onnx/test_models_onnxruntime.py b/test/onnx/test_models_onnxruntime.py
index 9c38644a5dc..e033cb2210a 100644
--- a/test/onnx/test_models_onnxruntime.py
+++ b/test/onnx/test_models_onnxruntime.py
@@ -1,6 +1,15 @@
 # Owner(s): ["module: onnx"]
 
 import os
+
+CUDA_VERSION = 11080
+torch.version.cuda = '11.8'
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+# from torch_npu.contrib import transfer_to_npu
+import torch_npu
 import unittest
 from collections import OrderedDict
 from collections.abc import Mapping
@@ -39,23 +48,31 @@ def exportTest(
 ):
     opset_versions = opset_versions if opset_versions else [7, 8, 9, 10, 11, 12, 13, 14]
 
-    for opset_version in opset_versions:
-        self.opset_version = opset_version
-        self.onnx_shape_inference = True
-        onnx_test_common.run_model_test(
-            self,
-            model,
-            input_args=inputs,
-            rtol=rtol,
-            atol=atol,
-            acceptable_error_percentage=acceptable_error_percentage,
-        )
-
-        if self.is_script_test_enabled and opset_version > 11:
-            script_model = torch.jit.script(model)
+        torch.npu.conv.allow_hf32 = original_hf32
+    finally:
+                )
+                    acceptable_error_percentage=acceptable_error_percentage,
+                    atol=atol,
+                    rtol=rtol,
+                    input_args=inputs,
+                    script_model,
+                    self,
+                onnx_test_common.run_model_test(
+                script_model = torch.jit.script(model)
+            if self.is_script_test_enabled and opset_version > 11:
+                model,
+            self.onnx_shape_inference = True
+            self.opset_version = opset_version
+        for opset_version in opset_versions:
+
+        torch.npu.conv.allow_hf32 = False
+        # internal calculation introduces errors that are amplified in subsequent calculations.
+        # Conv using HF32 (half-precision floating point) for Ascend-dedicated operator
+    try:
+    original_hf32 = torch_npu.npu.conv.allow_hf32
+    
             onnx_test_common.run_model_test(
                 self,
-                script_model,
                 input_args=inputs,
                 rtol=rtol,
                 atol=atol,
