diff --git a/test/onnx/test_onnxscript_no_runtime.py b/test/onnx/test_onnxscript_no_runtime.py
index fcac54d..92b4975 100644
--- a/test/onnx/test_onnxscript_no_runtime.py
+++ b/test/onnx/test_onnxscript_no_runtime.py
@@ -10,6 +10,15 @@ import onnxscript
 from onnxscript.onnx_types import FLOAT
 
 import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
+
 from torch.onnx._internal import jit_utils
 from torch.testing._internal import common_utils
 
@@ -160,3 +169,7 @@ class TestONNXScriptExport(common_utils.TestCase):
         )
         loop_selu_proto = onnx.load(io.BytesIO(saved_model.getvalue()))
         self.assertEqual(len(loop_selu_proto.functions), 1)
+
+
+if __name__ == "__main__":
+    common_utils.run_tests()
\ No newline at end of file
