diff --git a/test/onnx/test_pytorch_onnx_onnxruntime_cuda.py b/test/onnx/test_pytorch_onnx_onnxruntime_cuda.py
index 85aeafc..5968c8f 100644
--- a/test/onnx/test_pytorch_onnx_onnxruntime_cuda.py
+++ b/test/onnx/test_pytorch_onnx_onnxruntime_cuda.py
@@ -15,7 +15,16 @@ from pytorch_test_common import (
 from test_pytorch_onnx_onnxruntime import _parameterized_class_attrs_and_values
 
 import torch
-from torch.cuda.amp import autocast
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
+
+from torch_npu.npu.amp import autocast
 from torch.testing._internal import common_utils
 
 
@@ -27,7 +36,7 @@ from torch.testing._internal import common_utils
 )
 class TestONNXRuntime_cuda(onnx_test_common._TestONNXRuntime):
     @skipIfUnsupportedMinOpsetVersion(9)
-    @skipIfNoCuda
+    # @skipIfNoCuda
     def test_gelu_fp16(self):
         class GeluModel(torch.nn.Module):
             def forward(self, x):
@@ -40,12 +49,12 @@ class TestONNXRuntime_cuda(onnx_test_common._TestONNXRuntime):
             6,
             requires_grad=True,
             dtype=torch.float16,
-            device=torch.device("cuda"),
+            device=torch.device("npu"),
         )
         self.run_test(GeluModel(), x, rtol=1e-3, atol=1e-5)
 
     @skipIfUnsupportedMinOpsetVersion(9)
-    @skipIfNoCuda
+    # @skipIfNoCuda
     @skipScriptTest()
     def test_layer_norm_fp16(self):
         class LayerNormModel(torch.nn.Module):
@@ -64,12 +73,12 @@ class TestONNXRuntime_cuda(onnx_test_common._TestONNXRuntime):
             10,
             requires_grad=True,
             dtype=torch.float16,
-            device=torch.device("cuda"),
+            device=torch.device("npu"),
         )
-        self.run_test(LayerNormModel().cuda(), x, rtol=1e-3, atol=1e-5)
+        self.run_test(LayerNormModel().npu(), x, rtol=1e-3, atol=1e-5)
 
     @skipIfUnsupportedMinOpsetVersion(12)
-    @skipIfNoCuda
+    # @skipIfNoCuda
     @skipScriptTest()
     def test_softmaxCrossEntropy_fusion_fp16(self):
         class FusionModel(torch.nn.Module):
@@ -84,8 +93,8 @@ class TestONNXRuntime_cuda(onnx_test_common._TestONNXRuntime):
                 return output
 
         N, C = 5, 4
-        input = torch.randn(N, 16, dtype=torch.float16, device=torch.device("cuda"))
-        target = torch.empty(N, dtype=torch.long, device=torch.device("cuda")).random_(
+        input = torch.randn(N, 16, dtype=torch.float16, device=torch.device("npu"))
+        target = torch.empty(N, dtype=torch.long, device=torch.device("npu")).random_(
             0, C
         )
 
@@ -93,7 +102,7 @@ class TestONNXRuntime_cuda(onnx_test_common._TestONNXRuntime):
         target[target == 1] = -100
         self.run_test(FusionModel(), (input, target))
 
-    @skipIfNoCuda
+    # @skipIfNoCuda
     @skipScriptTest()
     def test_apex_o2(self):
         class LinearModel(torch.nn.Module):
@@ -108,29 +117,29 @@ class TestONNXRuntime_cuda(onnx_test_common._TestONNXRuntime):
             from apex import amp
         except Exception as e:
             raise unittest.SkipTest("Apex is not available") from e
-        input = torch.randn(3, 3, device=torch.device("cuda"))
+        input = torch.randn(3, 3, device=torch.device("npu"))
         model = amp.initialize(LinearModel(), opt_level="O2")
         self.run_test(model, input)
 
     # ONNX supports bfloat16 for opsets >= 13
     # Add, Sub and Mul ops don't support bfloat16 cpu in onnxruntime.
     @skipIfUnsupportedMinOpsetVersion(13)
-    @skipIfNoBFloat16Cuda
+    # @skipIfNoBFloat16Cuda
     def test_arithmetic_bfp16(self):
         class MyModule(torch.nn.Module):
             def forward(self, x):
-                y = torch.ones(3, 4, dtype=torch.bfloat16, device=torch.device("cuda"))
+                y = torch.ones(3, 4, dtype=torch.bfloat16, device=torch.device("npu"))
                 x = x.type_as(y)
                 return torch.mul(torch.add(x, y), torch.sub(x, y)).to(
                     dtype=torch.float16
                 )
 
         x = torch.ones(
-            3, 4, requires_grad=True, dtype=torch.float16, device=torch.device("cuda")
+            3, 4, requires_grad=True, dtype=torch.float16, device=torch.device("npu")
         )
         self.run_test(MyModule(), x, rtol=1e-3, atol=1e-5)
 
-    @skipIfNoCuda
+    # @skipIfNoCuda
     def test_deduplicate_initializers_diff_devices(self):
         class Model(torch.nn.Module):
             def __init__(self) -> None:
@@ -138,13 +147,13 @@ class TestONNXRuntime_cuda(onnx_test_common._TestONNXRuntime):
                 self.w = torch.nn.Parameter(
                     torch.ones(2, 3, device=torch.device("cpu"))
                 )
-                self.b = torch.nn.Parameter(torch.ones(3, device=torch.device("cuda")))
+                self.b = torch.nn.Parameter(torch.ones(3, device=torch.device("npu")))
 
             def forward(self, x, y):
                 return torch.matmul(self.w, x), y + self.b
 
         x = torch.randn(3, 3, device=torch.device("cpu"))
-        y = torch.randn(3, 3, device=torch.device("cuda"))
+        y = torch.randn(3, 3, device=torch.device("npu"))
         self.run_test(Model(), (x, y))
 
 
