diff --git a/test/distributed/_composable/test_replicate_with_compiler.py b/test/distributed/_composable/test_replicate_with_compiler.py
index 5936a729cbe..e4987371ab7 100644
--- a/test/distributed/_composable/test_replicate_with_compiler.py
+++ b/test/distributed/_composable/test_replicate_with_compiler.py
@@ -35,10 +35,11 @@ from torch.testing._internal.common_utils import run_tests
 from torch.testing._internal.distributed.fake_pg import FakeStore
 from torch.testing._internal.inductor_utils import HAS_GPU
 from torch.utils.checkpoint import checkpoint
+device_type = "npu"
+# device_type = str(get_devtype())
+HAS_GPU = True
 
 
-device_type = str(get_devtype())
-
 DIM = 2000
 
 
@@ -95,6 +96,9 @@ class ReplicateTest(MultiProcessInductorTestCase):
         checkpoint: bool = False,
         device: str | torch.device,
     ):
+        model = Net(checkpoint=checkpoint)
+        torch.npu.set_device(self.rank)
+        torch.npu.set_device(self.rank)
         self.create_pg(device)
         torch._dynamo.config.optimize_ddp = "python_reducer"
         torch.manual_seed(123)
