diff --git a/test/distributed/tensor/debug/test_op_coverage.py b/test/distributed/tensor/debug/test_op_coverage.py
index 2b19415aa89..99a52e12132 100644
--- a/test/distributed/tensor/debug/test_op_coverage.py
+++ b/test/distributed/tensor/debug/test_op_coverage.py
@@ -1,3 +1,8 @@
+import torch_npu.testing
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["oncall: distributed"]
 
 import torch
