diff --git a/test/quantization/core/test_workflow_module.py b/test/quantization/core/test_workflow_module.py
index 95c08a3..3320572 100644
--- a/test/quantization/core/test_workflow_module.py
+++ b/test/quantization/core/test_workflow_module.py
@@ -1,3 +1,8 @@
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
+from torch.testing._internal.common_utils import run_tests
+
 # Owner(s): ["oncall: quantization"]
 # ruff: noqa: F841
 
@@ -11,6 +16,14 @@ import unittest
 
 import numpy as np
 import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
 
 import torch.nn as nn
 import torch.testing._internal.hypothesis_utils as hu
@@ -41,7 +54,8 @@ from torch.ao.quantization import (
     RecordingObserver,
 )
 from torch.ao.quantization.quantize import _get_observer_dict
-
+settings.register_profile("disable_deadline", deadline=None)
+settings.load_profile("disable_deadline")
 hu.assert_deadline_disabled()
 from torch.testing._internal.common_cuda import TEST_CUDA, TEST_MULTIGPU
 
@@ -298,8 +312,8 @@ class TestObserver(QuantizationTestCase):
             loaded = torch.jit.load(buf)
             self.assertEqual(obs.calculate_qparams(), loaded.calculate_qparams())
 
-    @unittest.skipIf(not TEST_MULTIGPU, "multi-GPU not supported")
-    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    # @unittest.skipIf(not TEST_MULTIGPU, "multi-GPU not supported")
+    # @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
     @override_qengines
     def test_state_dict_respects_device_affinity(self):
         """
@@ -307,7 +321,8 @@ class TestObserver(QuantizationTestCase):
         device.
         """
         device_cpu = torch.device('cpu')
-        device_cuda = torch.device('cuda:0')
+        # device_cuda = torch.device('cuda:0')
+        device_cuda = torch.device('npu:0')
         test_cases = itertools.product(
             [device_cpu, device_cuda],
             [device_cpu, device_cuda],
@@ -431,8 +446,8 @@ class TestObserver(QuantizationTestCase):
             self.assertEqual(scripted.state_dict(), scripted_2.state_dict())
 
 
-    @unittest.skipIf(not TEST_MULTIGPU, "multi-GPU not supported")
-    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    # @unittest.skipIf(not TEST_MULTIGPU, "multi-GPU not supported")
+    # @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
     def test_observer_qparams_respects_device_affinity(self):
         """
         Ensure that the scale and zero_point returned by the observer
@@ -443,7 +458,7 @@ class TestObserver(QuantizationTestCase):
                         PerChannelMinMaxObserver(),
                         MovingAveragePerChannelMinMaxObserver()]
         for obs in observerList:
-            device = torch.device('cuda:1')
+            device = torch.device('npu:1')
             x = torch.randn(1, 2, device=device)
             obs.to(device)
             result = obs(x)
@@ -854,7 +869,7 @@ class TestHistogramObserver(QuantizationTestCase):
         self.assertEqual(myobs.histogram, [1., 0., 1., 2., 1., 0., 0., 1., 1., 1.])
 
 class TestFakeQuantize(TestCase):
-    @given(device=st.sampled_from(['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']),
+    @given(device=st.sampled_from(['cpu', 'npu'] if torch_npu.npu.is_available() else ['cpu']),
            X=hu.per_channel_tensor(shapes=hu.array_shapes(2, 5,),
            qparams=hu.qparams(dtypes=torch.qint8)))
     def test_fq_module_per_channel(self, device, X):
@@ -1044,8 +1059,8 @@ class TestDistributed(QuantizationTestCase):
             buffer_ids_after,
             msg="FakeQuant: Buffers must be modified in place")
 
-    @unittest.skipIf(not TEST_MULTIGPU, "multi-GPU not supported")
-    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    # @unittest.skipIf(not TEST_MULTIGPU, "multi-GPU not supported")
+    # @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
     def test_qat_data_parallel(self):
         """
         Tests that doing QAT in nn.DataParallel does not crash.
@@ -1053,7 +1068,7 @@ class TestDistributed(QuantizationTestCase):
         if 'fbgemm' not in torch.backends.quantized.supported_engines:
             return
         with override_quantized_engine('fbgemm'):
-            device = torch.device('cuda')
+            device = torch.device('npu')
 
             model = nn.Sequential(
                 torch.ao.quantization.QuantStub(),
@@ -1136,8 +1151,8 @@ class TestDistributed(QuantizationTestCase):
             hasattr(m[1], "qconfig"),
             "missing qconfig after SyncBatchNorm conversion")
 
-    @unittest.skipIf(not TEST_MULTIGPU, "multi-GPU not supported")
-    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    # @unittest.skipIf(not TEST_MULTIGPU, "multi-GPU not supported")
+    # @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
     @override_qengines
     def test_device_affinity(self):
         """
@@ -1159,7 +1174,8 @@ class TestDistributed(QuantizationTestCase):
 
         model = Model()
         model.qconfig = torch.ao.quantization.get_default_qat_qconfig(torch.backends.quantized.engine)
-        device = torch.device('cuda:0')
+        # device = torch.device('cuda:0')
+        device = torch.device('npu:0')
         model.to(device)
         torch.ao.quantization.prepare_qat(model, inplace=True)
         model_devices = {p.device for p in model.parameters()} | \
@@ -1175,7 +1191,7 @@ class TestDistributed(QuantizationTestCase):
 class TestFusedObsFakeQuantModule(TestCase):
     @given(
         device=st.sampled_from(
-            ["cpu", "cuda"] if torch.cuda.is_available() else ["cpu"]
+            ["cpu", "npu"] if torch_npu.npu.is_available() else ["cpu"]
         )
     )
     @settings(deadline=None)
@@ -1224,7 +1240,7 @@ class TestFusedObsFakeQuantModule(TestCase):
 
     @given(
         device=st.sampled_from(
-            ["cpu", "cuda"] if torch.cuda.is_available() else ["cpu"]
+            ["cpu", "npu"] if torch_npu.npu.is_available() else ["cpu"]
         )
     )
     @settings(deadline=None)
@@ -1279,7 +1295,7 @@ class TestFusedObsFakeQuantModule(TestCase):
 
     @given(
         device=st.sampled_from(
-            ["cpu", "cuda"] if torch.cuda.is_available() else ["cpu"]
+            ["cpu", "npu"] if torch_npu.npu.is_available() else ["cpu"]
         )
     )
     @settings(deadline=None)
@@ -1309,7 +1325,7 @@ class TestFusedObsFakeQuantModule(TestCase):
             )
 
     def test_fused_mod_per_channel(self):
-        devices = ["cpu", "cuda"] if torch.cuda.is_available() else ["cpu"]
+        devices = ["cpu", "npu"] if torch_npu.npu.is_available() else ["cpu"]
         m = 5
         n = 10
         for device in devices:
@@ -1502,6 +1518,9 @@ class TestFusedObsFakeQuantModule(TestCase):
                              obs2match)
 
 if __name__ == '__main__':
+    run_tests()
     raise RuntimeError("This test file is not meant to be run directly, use:\n\n"
                        "\tpython test/test_quantization.py TESTNAME\n\n"
                        "instead.")
+    
+    
