diff --git a/test/quantization/core/test_quantized_module.py b/test/quantization/core/test_quantized_module.py
index c31fe44..25fb4be 100644
--- a/test/quantization/core/test_quantized_module.py
+++ b/test/quantization/core/test_quantized_module.py
@@ -1,6 +1,15 @@
 # Owner(s): ["oncall: quantization"]
 
 import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
+
 import torch.nn as nn
 import torch.ao.nn.intrinsic as nni
 import torch.ao.nn.intrinsic.quantized as nniq
@@ -32,9 +41,11 @@ from torch.testing._internal.common_quantized import (
     qengine_is_onednn,
 )
 import torch.fx
-from hypothesis import assume, given
+from hypothesis import assume, given, settings
 from hypothesis import strategies as st
 import torch.testing._internal.hypothesis_utils as hu
+settings.register_profile("disable_deadline", deadline=None)
+settings.load_profile("disable_deadline")
 hu.assert_deadline_disabled()
 
 import copy
