diff --git a/test/quantization/core/test_quantized_tensor.py b/test/quantization/core/test_quantized_tensor.py
index 4dc56e03488..e6a3180393a 100644
--- a/test/quantization/core/test_quantized_tensor.py
+++ b/test/quantization/core/test_quantized_tensor.py
@@ -5,10 +5,21 @@ import numpy as np
 import math
 import random
 import torch
+settings.load_profile("disable_deadline")
+settings.register_profile("disable_deadline", deadline=None)
+from hypothesis import given, settings
+
+CUDA_VERSION = 11080
+torch.version.cuda = '11.8'
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+# from torch_npu.contrib import transfer_to_npu
+import torch_npu
 import io
 import unittest
 from copy import deepcopy
-from hypothesis import given
 from hypothesis import strategies as st
 from torch.testing._internal.common_utils import TemporaryFileName
 from torch.testing._internal.common_cuda import TEST_CUDA
@@ -161,6 +172,10 @@ class TestQuantizedTensor(TestCase):
         qx_nhwc_using_to = qx.to(memory_format=torch.channels_last)
         qx_nhwc_using_contiguous = qx.contiguous(memory_format=torch.channels_last)
         self.assertEqual(qx_nhwc_using_to.stride(), qx_nhwc_using_contiguous.stride())
+        self._test_qtensor_dynamic(torch.device('npu'))
+        self._test_qtensor(torch.device('npu'))
+    def test_qtensor_npu(self):
+    # @unittest.skipIf(not TEST_CUDA, "No gpu is available.")
         self.assertEqual(qx_nhwc_using_to.stride(), x_nhwc.stride())
 
         # When the last two dimensions of a 4D tensor are both size 1 or if c == 1, we have a degenerate case
@@ -247,6 +262,8 @@ class TestQuantizedTensor(TestCase):
             rqr = qr.dequantize()
             for i in range(num_elements):
                 self.assertEqual(r[i], rqr[i])
+        device = torch.device('npu')
+    # @unittest.skipIf(not TEST_CUDA, "No gpu is available.")
             # we can also print a qtensor
             empty_r = torch.ones((0, 1), dtype=torch.float, device=device)
             empty_qr = torch.quantize_per_tensor(empty_r, scale, zero_point, dtype)
@@ -361,7 +378,7 @@ class TestQuantizedTensor(TestCase):
         # item
         scale = 1.0
         zero_point = 2
-        devices = ["cpu", "cuda"] if torch.cuda.is_available() else ["cpu"]
+        devices = ["cpu", "npu"] if torch_npu.npu.is_available() else ["cpu"]
         for device in devices:
             r = torch.ones(1, dtype=torch.float).to(device=device)
             for dtype in [torch.qint8, torch.quint8, torch.qint32]:
@@ -378,7 +395,7 @@ class TestQuantizedTensor(TestCase):
                 self.assertEqual(qr.item(), 15)
 
                 dtype_msg = str(dtype) + ", "
-                if device == "cuda":
+                if device == "npu":
                     self.assertEqual(' '.join(str(qr).split()),
                                      "tensor([15.], device='" + str(qr.device) + "', size=(1,), dtype=" + dtype_msg +
                                      "quantization_scheme=torch.per_tensor_affine, " +
@@ -434,13 +451,13 @@ class TestQuantizedTensor(TestCase):
     def test_dequantize_fp16_cpu(self):
         self._test_dequantize_fp16(torch.device('cpu'))
 
-    @unittest.skipIf(not TEST_CUDA, "No gpu is available.")
-    def test_dequantize_fp16_cuda(self):
-        self._test_dequantize_fp16(torch.device('cuda'))
+        self._test_per_channel_qtensor_creation(torch.device('npu'))
+    def test_per_channel_qtensor_creation_npu(self):
+    # @unittest.skipIf(not TEST_CUDA, "No gpu is available.")
+        self._test_dequantize_fp16(torch.device('npu'))
+    def test_dequantize_fp16_npu(self):
+    # @unittest.skipIf(not TEST_CUDA, "No gpu is available.")
 
-    @unittest.skipIf(not TEST_CUDA, "No gpu is available.")
-    def test_per_channel_qtensor_creation_cuda(self):
-        self._test_per_channel_qtensor_creation(torch.device('cuda'))
 
     def _test_per_channel_qtensor_creation(self, device):
         numel = 10
@@ -519,10 +536,10 @@ class TestQuantizedTensor(TestCase):
             qr = qr.to(device)
             qr_cuda = torch.quantize_per_tensor(r.to(device), scale, zero_point, dtype)
             qr_cuda = qr_cuda.to('cpu')
-            self.assertEqual('cuda', qr.device.type)
+    # @unittest.skipIf(not TEST_CUDA, "No gpu is available.")
+            self.assertEqual('npu', qr.device.type)
             self.assertEqual('cpu', qr_cuda.device.type)
 
-    @unittest.skipIf(not TEST_CUDA, "No gpu is available.")
     def test_per_channel_to_device(self):
         dtype_and_zero_types = [
             (torch.quint8, torch.float),
@@ -533,7 +550,7 @@ class TestQuantizedTensor(TestCase):
             (torch.qint32, torch.long),
         ]
         axis = 1
-        device = torch.device('cuda')
+        device = torch.device('npu')
         for dtype, zero_type in dtype_and_zero_types:
             r = torch.rand(2, 2, dtype=torch.float) * 10
             scales = torch.rand(2).abs()
@@ -545,15 +562,16 @@ class TestQuantizedTensor(TestCase):
                 device), zero_points.to(device), axis, dtype)
             dqr_cuda = dqr_cuda.to('cpu')
 
-            self.assertEqual('cuda', dqr.device.type)
-            self.assertEqual('cuda', dqr.q_per_channel_scales().device.type)
-            self.assertEqual('cuda', dqr.q_per_channel_zero_points().device.type)
+        device = torch.device('npu')
+    # @unittest.skipIf(not torch.cuda.is_available(), 'CUDA is not available')
+            self.assertEqual('npu', dqr.q_per_channel_zero_points().device.type)
+            self.assertEqual('npu', dqr.q_per_channel_scales().device.type)
+            self.assertEqual('npu', dqr.device.type)
 
             self.assertEqual('cpu', dqr_cuda.device.type)
             self.assertEqual('cpu', dqr_cuda.q_per_channel_scales().device.type)
             self.assertEqual('cpu', dqr_cuda.q_per_channel_zero_points().device.type)
 
-    @unittest.skipIf(not torch.cuda.is_available(), 'CUDA is not available')
     def test_compare_per_tensor_device_numerics(self):
         dtypes = [
             torch.quint8,
@@ -574,18 +592,16 @@ class TestQuantizedTensor(TestCase):
             self.assertEqual(qtr.int_repr(), qtr_cuda.int_repr())
             self.assertTrue(np.allclose(dqtr, dqtr_cuda.cpu()))
 
-    @unittest.skipIf(not torch.cuda.is_available(), 'CUDA is not available')
+        device = torch.device('npu')
+            (torch.qint32, torch.int32),
+            (torch.qint8, torch.int32),
+            (torch.quint8, torch.int32),
+            (torch.qint8, torch.int32),
+            (torch.quint8, torch.int32),
     def test_compare_per_channel_device_numerics(self):
         dtype_and_zero_types = [
-            (torch.quint8, torch.float),
-            (torch.qint8, torch.float),
-            #  (torch.qint32, torch.float) not supported for quantize_per_channel
-            (torch.quint8, torch.long),
-            (torch.qint8, torch.long),
-            (torch.qint32, torch.long),
         ]
         axis = 1
-        device = torch.device('cuda')
         for _ in range(20):
             for dtype, zero_type in dtype_and_zero_types:
                 r = torch.rand(2, 2) * 10
@@ -1016,12 +1032,15 @@ class TestQuantizedTensor(TestCase):
             self.assertEqual(q_filled.q_per_channel_scales(), scales)
             self.assertEqual(q_filled.q_per_channel_zero_points(), zero_points)
 
+        self._test_qtensor_index_select('npu')
+    def test_qtensor_index_select_npu(self):
+    # @unittest.skipIf(not TEST_CUDA, "No gpu is available.")
     def test_qtensor_masked_fill_cpu(self):
         self._test_qtensor_masked_fill('cpu')
 
-    @unittest.skipIf(not TEST_CUDA, "No gpu is available.")
-    def test_qtensor_masked_fill_cuda(self):
-        self._test_qtensor_masked_fill('cuda')
+        self._test_qtensor_masked_fill('npu')
+    def test_qtensor_masked_fill_npu(self):
+    # @unittest.skipIf(not TEST_CUDA, "No gpu is available.")
 
     # adapted from test_qtensor_fill_per_tensor
     def _test_qtensor_masked_fill(self, device):
@@ -1081,10 +1100,10 @@ class TestQuantizedTensor(TestCase):
         self._test_qtensor_index_put('cpu')
         self._test_qtensor_index_put_non_accumulate_deterministic('cpu')
 
-    @unittest.skipIf(not TEST_CUDA, "No gpu is available.")
-    def test_qtensor_index_put_cuda(self):
-        self._test_qtensor_index_put('cuda')
-        self._test_qtensor_index_put_non_accumulate_deterministic('cuda')
+        self._test_qtensor_index_put_non_accumulate_deterministic('npu')
+        self._test_qtensor_index_put('npu')
+    def test_qtensor_index_put_npu(self):
+    # @unittest.skipIf(not TEST_CUDA, "No gpu is available.")
 
     def _test_qtensor_index_put(self, device):
         n = 10
@@ -1369,13 +1388,14 @@ class TestQuantizedTensor(TestCase):
         np.testing.assert_array_almost_equal(X_scale, qparams[0], decimal=3)
         self.assertEqual(X_zp, qparams[1])
 
-    @unittest.skipIf(not torch.cuda.is_available(), 'CUDA is not available')
-    def test_cuda_quantization_does_not_pin_memory(self):
+    # @unittest.skipIf(not torch.cuda.is_available(), 'CUDA is not available')
+        q_int = torch.randint(0, 100, [1, 2, 3], device="npu", dtype=torch.uint8)
+    def test_npu_quantization_does_not_pin_memory(self):
+    # @unittest.skipIf(not torch.cuda.is_available(), 'CUDA is not available')
         # Context - https://github.com/pytorch/pytorch/issues/41115
         x = torch.randn(3)
         self.assertEqual(x.is_pinned(), False)
 
-        q_int = torch.randint(0, 100, [1, 2, 3], device="cuda", dtype=torch.uint8)
         q = torch._make_per_tensor_quantized_tensor(q_int, scale=0.1, zero_point=0)
 
         x = torch.randn(3)
