diff --git a/test/quantization/core/test_workflow_ops.py b/test/quantization/core/test_workflow_ops.py
index 4b7a658..0c4ae34 100644
--- a/test/quantization/core/test_workflow_ops.py
+++ b/test/quantization/core/test_workflow_ops.py
@@ -1,7 +1,23 @@
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
+from torch.testing._internal.common_utils import run_tests
 # Owner(s): ["oncall: quantization"]
 # ruff: noqa: F841
 
 import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
+
 import math
 from torch.ao.quantization import (
     FakeQuantize,
@@ -28,6 +44,8 @@ import numpy as np
 from hypothesis import given, settings
 from hypothesis import strategies as st
 import torch.testing._internal.hypothesis_utils as hu
+settings.register_profile("disable_deadline", deadline=None)
+settings.load_profile("disable_deadline")
 hu.assert_deadline_disabled()
 from torch.testing._internal.common_cuda import TEST_CUDA
 from torch.testing._internal.common_utils import TestCase, skipIfTorchDynamo
@@ -37,7 +55,8 @@ from torch.testing._internal.common_utils import TestCase, skipIfTorchDynamo
 def _fake_quantize_per_tensor_affine_reference(X, scale, zero_point, quant_min, quant_max):
     dtype = X.dtype
     res = ((torch.clamp(torch.round(X.to(torch.float32) * (1.0 / scale) + zero_point), quant_min, quant_max) - zero_point) * scale)
-    return res.to(dtype)
+    # return res.to(dtype)
+    return torch.tensor(res, dtype=dtype)
 
 # Reference method for the gradient of the fake quantize operator
 # Note: because scale/zero_point are left as float in the actual kernel, this mimics how fake_quant works for float16/64
@@ -47,7 +66,8 @@ def _fake_quantize_per_tensor_affine_grad_reference(dY, X, scale, zero_point, qu
     mask = (Xq >= quant_min) * (Xq <= quant_max)
     res = torch.zeros_like(dY)
     res[mask] = dY[mask]
-    return res.to(dtype)
+    # return res.to(dtype)
+    return torch.tensor(res, dtype=dtype)
 
 # Reference method for the gradients of the fake quantize operator
 def _fake_quantize_learnable_per_tensor_affine_grad_reference(dY, X, scale, zero_point, quant_min, quant_max, device):
@@ -253,7 +273,7 @@ NP_RANDOM_SEED = 19
 tolerance = 1e-6
 
 class TestFakeQuantizeOps(TestCase):
-    @given(device=st.sampled_from(['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']),
+    @given(device=st.sampled_from(['cpu', 'npu'] if torch_npu.npu.is_available() else ['cpu']),
            X=hu.tensor(shapes=hu.array_shapes(1, 5,),
                        qparams=hu.qparams(dtypes=torch.quint8)))
     def test_forward_per_tensor(self, device, X):
@@ -270,7 +290,7 @@ class TestFakeQuantizeOps(TestCase):
             X, scale, zero_point, quant_min, quant_max)
         np.testing.assert_allclose(Y, Y_prime.cpu(), rtol=tolerance, atol=tolerance)
 
-    @given(device=st.sampled_from(['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']),
+    @given(device=st.sampled_from(['cpu', 'npu'] if torch_npu.npu.is_available() else ['cpu']),
            X=hu.tensor(shapes=hu.array_shapes(1, 5,),
                        qparams=hu.qparams(dtypes=torch.quint8)))
     @unittest.skip("temporarily disable the test")
@@ -298,7 +318,7 @@ class TestFakeQuantizeOps(TestCase):
         net.qconfig = torch.ao.quantization.get_default_qat_qconfig('fbgemm')
         net_prep = torch.ao.quantization.prepare_qat(net)
 
-        with torch.cuda.amp.autocast():
+        with torch_npu.npu.amp.autocast():
             x = torch.randn(4, 1, 5, 5)
             out = net_prep(x).sum()
             out.backward()
@@ -331,13 +351,14 @@ class TestFakeQuantizeOps(TestCase):
         self.assertEqual(Y3, Y3r, rtol=tolerance, atol=tolerance)
 
     def _test_forward_per_tensor_cachemask_impl(self, device):
-        float_types = (torch.float32, torch.float16, torch.float64, torch.bfloat16)
+        float_types = (torch.float64, torch.float32, torch.float16, torch.bfloat16)
         torch_types = (torch.qint8, torch.quint8)
         Xs = (torch.randn(4, 8, device=device), torch.randn(4, 16, device=device)[:, ::2])
         tensor_qparam = (True, False)
         for float_type, torch_type, X, tensor_qparams in itertools.product(float_types, torch_types, Xs, tensor_qparam):
             # pick the scale + zp so that some values get clipped
-            X = X.to(float_type)
+            # X = X.to(float_type)
+            X = torch.tensor(X, dtype=float_type)
             obs = torch.ao.quantization.MinMaxObserver(torch_type)
             obs.to(device)
             obs(X * 0.75)
@@ -356,17 +377,18 @@ class TestFakeQuantizeOps(TestCase):
         device = torch.device('cpu')
         self._test_forward_per_tensor_cachemask_impl(device)
 
-    @unittest.skipIf(not TEST_CUDA, "No gpu is not available.")
-    def test_forward_per_tensor_cachemask_cuda(self):
-        device = torch.device('cuda')
+    # @unittest.skipIf(not TEST_CUDA, "No gpu is not available.")
+    def test_forward_per_tensor_cachemask_npu(self):
+        device = torch.device('npu')
         self._test_forward_per_tensor_cachemask_impl(device)
 
     def _test_backward_per_tensor_cachemask_impl(self, device):
-        float_types = (torch.float32, torch.float16, torch.float64)
+        float_types = (torch.float64, torch.float32, torch.float16)
         torch_types = (torch.qint8, torch.quint8)
         tensor_qparams = (True, False)
         for float_type, torch_type, tensor_qparam in itertools.product(float_types, torch_types, tensor_qparams):
-            X = torch.randn(4, 8).to(device).to(float_type)
+            # X = torch.randn(4, 8).to(device).to(float_type)
+            X = torch.randn(4, 8, dtype=float_type).to(device)
             X.requires_grad_()
             # pick the scale + zp so that some values get clipped
             obs = torch.ao.quantization.MinMaxObserver(torch_type)
@@ -385,7 +407,8 @@ class TestFakeQuantizeOps(TestCase):
             self.assertEqual(Y_test, Y_ref, rtol=tolerance, atol=tolerance)
 
             # backward pass
-            dout = torch.rand_like(X, dtype=torch.float).to(device)
+            # dout = torch.rand_like(X, dtype=torch.float).to(device)
+            dout = torch.rand_like(X, dtype=float_type).to(device)
             dX = _fake_quantize_per_tensor_affine_grad_reference(
                 dout, X, scale, zero_point, quant_min, quant_max)
             Y_test.backward(dout)
@@ -396,9 +419,9 @@ class TestFakeQuantizeOps(TestCase):
         device = torch.device('cpu')
         self._test_backward_per_tensor_cachemask_impl(device)
 
-    @unittest.skipIf(not TEST_CUDA, "No gpu is not available.")
-    def test_backward_per_tensor_cachemask_cuda(self):
-        device = torch.device('cuda')
+    # @unittest.skipIf(not TEST_CUDA, "No gpu is not available.")
+    def test_backward_per_tensor_cachemask_npu(self):
+        device = torch.device('npu')
         self._test_backward_per_tensor_cachemask_impl(device)
 
     def _test_learnable_forward_per_tensor(self, X, device, scale_base, zero_point_base):
@@ -438,13 +461,13 @@ class TestFakeQuantizeOps(TestCase):
     @given(X=hu.tensor(shapes=hu.array_shapes(1, 5,),
                        elements=hu.floats(-1e3, 1e3, allow_nan=False, allow_infinity=False),
                        qparams=hu.qparams(dtypes=torch.quint8)))
-    @unittest.skipIf(not TEST_CUDA, "No gpu is not available.")
-    def test_learnable_forward_per_tensor_cuda(self, X):
+    # @unittest.skipIf(not TEST_CUDA, "No gpu is not available.")
+    def test_learnable_forward_per_tensor_npu(self, X):
         X, (_, _, _) = X
         scale_base = torch.normal(mean=0, std=1, size=(1,)).clamp(1e-4, 100)
         zero_point_base = torch.normal(mean=0, std=128, size=(1,))
         self._test_learnable_forward_per_tensor(
-            X, 'cuda', scale_base, zero_point_base)
+            X, 'npu', scale_base, zero_point_base)
 
     def _test_learnable_backward_per_tensor(self, X, device, scale_base, zero_point_base):
         r"""Tests the backward method with additional backprop support for scale and zero point.
@@ -507,16 +530,16 @@ class TestFakeQuantizeOps(TestCase):
     @given(X=hu.tensor(shapes=hu.array_shapes(1, 5,),
                        elements=hu.floats(-1e3, 1e3, allow_nan=False, allow_infinity=False),
                        qparams=hu.qparams(dtypes=torch.quint8)))
-    @unittest.skipIf(not TEST_CUDA, "No gpu is not available.")
-    def test_learnable_backward_per_tensor_cuda(self, X):
+    # @unittest.skipIf(not TEST_CUDA, "No gpu is not available.")
+    def test_learnable_backward_per_tensor_npu(self, X):
         torch.random.manual_seed(NP_RANDOM_SEED)
         X, (_, _, _) = X
         scale_base = torch.normal(mean=0, std=1, size=(1,)).clamp(1e-4, 100)
         zero_point_base = torch.normal(mean=0, std=128, size=(1,))
         self._test_learnable_backward_per_tensor(
-            X, 'cuda', scale_base, zero_point_base)
+            X, 'npu', scale_base, zero_point_base)
 
-    @given(device=st.sampled_from(['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']),
+    @given(device=st.sampled_from(['cpu', 'npu'] if torch_npu.npu.is_available() else ['cpu']),
            X=hu.tensor(shapes=hu.array_shapes(1, 5,),
                        qparams=hu.qparams(dtypes=[torch.quint8])),
            )
@@ -541,7 +564,7 @@ class TestFakeQuantizeOps(TestCase):
         dX = _fake_quantize_per_tensor_affine_grad_reference(dout, X, fq_module.scale, fq_module.zero_point, quant_min, quant_max)
         np.testing.assert_allclose(dX.cpu().numpy(), X.grad.cpu().detach().numpy(), rtol=tolerance, atol=tolerance)
 
-    @given(device=st.sampled_from(['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']),
+    @given(device=st.sampled_from(['cpu', 'npu'] if torch_npu.npu.is_available() else ['cpu']),
            X=hu.tensor(shapes=hu.array_shapes(1, 5,),
                        qparams=hu.qparams(dtypes=torch.quint8)))
     def test_fixed_qparams_fq_module(self, device, X):
@@ -677,7 +700,7 @@ class TestFakeQuantizeOps(TestCase):
             self.assertEqual(fq_module.calculate_qparams(), loaded_module.calculate_qparams())
 
 
-    @given(device=st.sampled_from(['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']),
+    @given(device=st.sampled_from(['cpu', 'npu'] if torch_npu.npu.is_available() else ['cpu']),
            X=hu.per_channel_tensor(shapes=hu.array_shapes(1, 5,),
            qparams=hu.qparams(dtypes=torch.quint8)))
     def test_forward_per_channel(self, device, X):
@@ -722,9 +745,9 @@ class TestFakeQuantizeOps(TestCase):
     def test_forward_per_channel_cachemask_cpu(self):
         self._test_forward_per_channel_cachemask_impl('cpu')
 
-    @unittest.skipIf(not TEST_CUDA, "No gpu is not available.")
-    def test_forward_per_channel_cachemask_cuda(self):
-        self._test_forward_per_channel_cachemask_impl('cuda')
+    # @unittest.skipIf(not TEST_CUDA, "No gpu is not available.")
+    def test_forward_per_channel_cachemask_npu(self):
+        self._test_forward_per_channel_cachemask_impl('npu')
 
     def test_forward_per_channel_half_precision_numerics(self):
         scale = torch.randn(5).abs()
@@ -762,7 +785,7 @@ class TestFakeQuantizeOps(TestCase):
         quant_min = torch.iinfo(torch_type).min
         quant_max = torch.iinfo(torch_type).max
 
-        for device in ['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']:
+        for device in ['cpu', 'npu'] if torch_npu.npu.is_available() else ['cpu']:
             X = to_tensor(X, device)
             scale = to_tensor(scale, device)
 
@@ -817,18 +840,18 @@ class TestFakeQuantizeOps(TestCase):
 
     @given(X=hu.per_channel_tensor(shapes=hu.array_shapes(1, 5,),
                                    qparams=hu.qparams(dtypes=torch.quint8)))
-    @unittest.skipIf(not TEST_CUDA, "No gpu is not available.")
-    def test_learnable_forward_per_channel_cuda(self, X):
+    # @unittest.skipIf(not TEST_CUDA, "No gpu is not available.")
+    def test_learnable_forward_per_channel_npu(self, X):
         torch.random.manual_seed(NP_RANDOM_SEED)
         X, (_, _, axis, _) = X
-        X_base = torch.tensor(X).to('cuda')
+        X_base = torch.tensor(X).to('npu')
         channel_size = X_base.size(axis)
         scale_base = torch.normal(mean=0, std=1, size=(channel_size,)).clamp(1e-4, 100)
         zero_point_base = torch.normal(mean=0, std=128, size=(channel_size,))
         self._test_learnable_forward_per_channel(
-            X_base, 'cuda', scale_base, zero_point_base, axis)
+            X_base, 'npu', scale_base, zero_point_base, axis)
 
-    @given(device=st.sampled_from(['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']),
+    @given(device=st.sampled_from(['cpu', 'npu'] if torch_npu.npu.is_available() else ['cpu']),
            X=hu.per_channel_tensor(shapes=hu.array_shapes(1, 5,),
            qparams=hu.qparams(dtypes=torch.quint8)))
     @unittest.skip(
@@ -886,9 +909,9 @@ class TestFakeQuantizeOps(TestCase):
     def test_backward_per_channel_cachemask_cpu(self):
         self._test_backward_per_channel_cachemask_impl('cpu')
 
-    @unittest.skipIf(not TEST_CUDA, "No gpu is not available.")
-    def test_backward_per_channel_cachemask_cuda(self):
-        self._test_backward_per_channel_cachemask_impl('cuda')
+    # @unittest.skipIf(not TEST_CUDA, "No gpu is not available.")
+    def test_backward_per_channel_cachemask_npu(self):
+        self._test_backward_per_channel_cachemask_impl('npu')
 
     def _test_learnable_backward_per_channel(self, X_base, device, scale_base, zero_point_base, axis):
         r"""Tests the backward path of the learnable FakeQuantizePerTensorAffine op.
@@ -953,15 +976,15 @@ class TestFakeQuantizeOps(TestCase):
 
     @given(X=hu.per_channel_tensor(shapes=hu.array_shapes(2, 5,),
                                    qparams=hu.qparams(dtypes=torch.quint8)))
-    @unittest.skipIf(not TEST_CUDA, "No gpu is not available.")
-    def test_learnable_backward_per_channel_cuda(self, X):
+    # @unittest.skipIf(not TEST_CUDA, "No gpu is not available.")
+    def test_learnable_backward_per_channel_npu(self, X):
         torch.random.manual_seed(NP_RANDOM_SEED)
         X, (scale, zero_point, axis, torch_type) = X
-        X_base = torch.tensor(X).to('cuda')
-        scale_base = to_tensor(scale, 'cuda')
-        zero_point_base = to_tensor(zero_point, 'cuda')
+        X_base = torch.tensor(X).to('npu')
+        scale_base = to_tensor(scale, 'npu')
+        zero_point_base = to_tensor(zero_point, 'npu')
         self._test_learnable_backward_per_channel(
-            X_base, 'cuda', scale_base, zero_point_base, axis)
+            X_base, 'npu', scale_base, zero_point_base, axis)
 
     def test_numerical_consistency_per_tensor(self):
         self._test_numerical_consistency('per_tensor')
@@ -979,7 +1002,7 @@ class TestFakeQuantizeOps(TestCase):
             zero_types = [torch.int, torch.float, torch.float16]
         else:
             zero_types = [torch.int]
-        devices = [torch.device('cpu'), torch.device('cuda')] if torch.cuda.is_available() else [torch.device('cpu')]
+        devices = [torch.device('cpu'), torch.device('npu')] if torch_npu.npu.is_available() else [torch.device('cpu')]
         axis = 1
         for i in range(20):
             for torch_type, float_type, device, zero_type in itertools.product(torch_types, float_types, devices, zero_types):
@@ -1034,7 +1057,7 @@ class TestFakeQuantizeOps(TestCase):
 
 
 class TestFusedObsFakeQuant(TestCase):
-    @given(device=st.sampled_from(['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']),
+    @given(device=st.sampled_from(['cpu', 'npu'] if torch_npu.npu.is_available() else ['cpu']),
            symmetric_quant=st.booleans())
     @settings(deadline=None)
     def test_fused_obs_fake_quant_moving_avg(self, device, symmetric_quant) -> None:
@@ -1125,7 +1148,7 @@ class TestFusedObsFakeQuant(TestCase):
         output_shape = (0, 5)
         self.assertEqual(out.shape, output_shape)
 
-    @given(device=st.sampled_from(['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']),
+    @given(device=st.sampled_from(['cpu', 'npu'] if torch_npu.npu.is_available() else ['cpu']),
            symmetric_quant=st.booleans())
     @settings(deadline=None)
     def test_fused_obs_fake_quant_moving_avg_per_channel(self, device, symmetric_quant) -> None:
@@ -1198,7 +1221,7 @@ class TestFusedObsFakeQuant(TestCase):
                 self.assertEqual(in_running_max_ref, in_running_max_op)
                 torch.testing.assert_close(out, x_in)
 
-    @given(device=st.sampled_from(['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']),)
+    @given(device=st.sampled_from(['cpu', 'npu'] if torch_npu.npu.is_available() else ['cpu']),)
     @settings(deadline=None)
     def test_fused_obs_fake_quant_backward_op(self, device) -> None:
         n = m = k = 10
@@ -1249,7 +1272,7 @@ class TestFusedObsFakeQuant(TestCase):
         self.assertEqual(dX, x.grad)
         self.assertTrue(x.grad.dtype == torch.float32)
 
-    @given(device=st.sampled_from(['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']),)
+    @given(device=st.sampled_from(['cpu', 'npu'] if torch_npu.npu.is_available() else ['cpu']),)
     @settings(deadline=None)
     def test_fused_backward_op_fake_quant_off(self, device) -> None:
         n = m = 4
@@ -1296,6 +1319,7 @@ class TestFusedObsFakeQuant(TestCase):
         self.assertTrue(x.grad.dtype == torch.float32)
 
 if __name__ == '__main__':
+    run_tests()
     raise RuntimeError("This test file is not meant to be run directly, use:\n\n"
                        "\tpython test/test_quantization.py TESTNAME\n\n"
                        "instead.")
