diff --git a/test/distributed/_composable/fsdp/test_fully_shard_state_dict.py b/test/distributed/_composable/fsdp/test_fully_shard_state_dict.py
index 6422462..43114c1 100644
--- a/test/distributed/_composable/fsdp/test_fully_shard_state_dict.py
+++ b/test/distributed/_composable/fsdp/test_fully_shard_state_dict.py
@@ -1,3 +1,8 @@
+import torch_npu.testing
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["oncall: distributed"]
 
 import copy
@@ -25,7 +30,7 @@ from torch.testing._internal.distributed._tensor.common_dtensor import (
     Transformer,
     TransformerBlock,
 )
-
+TEST_CUDA = True
 
 class TestFullyShardStateDictMultiProcess(FSDPTest):
     @property
@@ -148,12 +153,12 @@ class TestFullyShardStateDictMultiProcess(FSDPTest):
             model.load_state_dict(sd, assign=True, strict=False)
 
         # lazy init without error
-        inp = torch.rand((mlp_dim, mlp_dim), device="cuda")
+        inp = torch.rand((mlp_dim, mlp_dim), device="npu")
 
         context = (
             self.assertRaisesRegex(
                 RuntimeError,
-                r"Found following parameters on non-CPU device: \[\('0.weight', device\(type='cuda'",
+                r"Found following parameters on non-CPU device: \[\('0.weight', device\(type='npu'",
             )
             if not cpu_state_dict
             else nullcontext()
