diff --git a/test/distributed/fsdp/test_fsdp_misc.py b/test/distributed/fsdp/test_fsdp_misc.py
index a1a317f..70c9786 100644
--- a/test/distributed/fsdp/test_fsdp_misc.py
+++ b/test/distributed/fsdp/test_fsdp_misc.py
@@ -1,3 +1,8 @@
+import torch_npu.testing
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["oncall: distributed"]
 
 import functools
@@ -95,7 +100,7 @@ class TestFSDPMiscMultiProcess(FSDPTest):
         dev_id = (
             torch.cuda.current_device()
             if use_index
-            else torch.device("cuda", torch.cuda.current_device())
+            else torch.device("npu", torch.cuda.current_device())
         )
 
         def _check_device_matches(module, device_id):
@@ -108,7 +113,7 @@ class TestFSDPMiscMultiProcess(FSDPTest):
             self.assertEqual(1, len(devices))
             found_device = devices.pop()
             if use_index and not isinstance(device_id, torch.device):
-                device = torch.device("cuda", device_id)
+                device = torch.device("npu", device_id)
             else:
                 device = device_id
             self.assertEqual(found_device, device)
@@ -140,10 +145,10 @@ class TestFSDPMiscMultiProcess(FSDPTest):
                 self.process_group,
                 FSDPInitMode.RECURSIVE,
                 DEVICEInitMode.DEVICE_BEFORE,
-                fsdp_kwargs={"device_id": torch.device("cuda")},
+                fsdp_kwargs={"device_id": torch.device("npu")},
             )
         _check_device_matches(
-            nested_wrapped_module, torch.device("cuda", torch.cuda.current_device())
+            nested_wrapped_module, torch.device("npu", torch.cuda.current_device())
         )
 
     @skip_if_lt_x_gpu(2)
@@ -613,12 +618,12 @@ class TestFSDPMiscMultiProcess(FSDPTest):
         # should be on GPU since we specified `device_id`
         self.assertEqual(
             nested_wrapped_module.buf.device,
-            torch.device("cuda", torch.cuda.current_device()),
+            torch.device("npu", torch.cuda.current_device()),
         )
         self.assertEqual(nested_wrapped_module.buf, torch.zeros((2, 2)))
         self.assertEqual(
             nested_wrapped_module.module.module[0].buf.device,
-            torch.device("cuda", torch.cuda.current_device()),
+            torch.device("npu", torch.cuda.current_device()),
         )
         self.assertEqual(
             nested_wrapped_module.module.module[0].buf, torch.zeros((3, 2))
@@ -694,7 +699,7 @@ class TestFSDPMiscMultiThread(FSDPTestMultiThread):
         for fsdp_module in FSDP.fsdp_modules(fsdp_model):
             self.assertEqual(
                 fsdp_module.compute_device,
-                torch.device("cuda", torch.cuda.current_device()),
+                torch.device("npu", torch.cuda.current_device()),
             )
 
     @skip_if_lt_x_gpu(2)
