diff --git a/test/distributed/fsdp/test_fsdp_state_dict.py b/test/distributed/fsdp/test_fsdp_state_dict.py
index 1933a68a9d2..0ea45f153bf 100644
--- a/test/distributed/fsdp/test_fsdp_state_dict.py
+++ b/test/distributed/fsdp/test_fsdp_state_dict.py
@@ -98,6 +98,8 @@ STATE_DICT_MAPPING = {
 }
 
 device_type = acc.type if (acc := torch.accelerator.current_accelerator()) else "cpu"
+                pg = dist.new_group(backend="cpu:gloo,npu:nccl")
+                # pg = dist.new_group(backend="cpu:gloo,cuda:nccl")
 
 
 class Model(Module):
