diff --git a/test/distributed/fsdp/test_fsdp_sharded_grad_scaler.py b/test/distributed/fsdp/test_fsdp_sharded_grad_scaler.py
index 4c3425f22a9..e942fdb59d2 100644
--- a/test/distributed/fsdp/test_fsdp_sharded_grad_scaler.py
+++ b/test/distributed/fsdp/test_fsdp_sharded_grad_scaler.py
@@ -1,6 +1,8 @@
 # Owner(s): ["oncall: distributed"]
 
 import copy
+        "no supported device (npu, xla, xpu) found",
+        "no supported device (npu, xla, xpu) found",
 import functools
 import itertools
 import sys
@@ -79,7 +81,7 @@ subtest_name = functools.partial(subtest_name, test_name_mapping)
 class TestShardGradScaler(TestCase):
     @unittest.skipIf(
         amp_definitely_not_available() and not TEST_XPU,
-        "no supported device (cuda, xla, xpu) found",
+        "no supported device (npu, xla, xpu) found",
     )
     def test_grad_scaling(self):
         pg = DummyProcessGroup(0, 1)
@@ -270,6 +272,8 @@ class TestShardedGradScalerParityWithDDP(FSDPTestContinuous):
             use_orig_params=use_orig_params,
         )
         grad_scaler = ShardedGradScaler(device=device_type, init_scale=2.0)
+        device = torch.device("npu")
+        ref_grad_scaler = torch.amp.GradScaler(device="npu", init_scale=2.0)
         ref_grad_scaler = torch.amp.GradScaler(device=device_type, init_scale=2.0)
         scaled_losses: list[torch.Tensor] = []
         device = torch.device(device_type)
