diff --git a/test/distributed/fsdp/test_hsdp_dtensor_state_dict.py b/test/distributed/fsdp/test_hsdp_dtensor_state_dict.py
index 0b7a6f1..c1320ee 100644
--- a/test/distributed/fsdp/test_hsdp_dtensor_state_dict.py
+++ b/test/distributed/fsdp/test_hsdp_dtensor_state_dict.py
@@ -1,3 +1,8 @@
+import torch_npu.testing
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["oncall: distributed"]
 
 import io
@@ -22,8 +27,9 @@ from torch.testing._internal.common_utils import parametrize, run_tests
 from torch.testing._internal.distributed._tensor.common_dtensor import (
     DTensorTestBase,
     skip_if_lt_x_gpu,
-    with_comms,
+    # with_comms,
 )
+from torch_npu.testing.common_distributed import with_comms
 
 
 device_type = torch.device(get_devtype())
