diff --git a/test/quantization/jit/test_ondevice_quantization.py b/test/quantization/jit/test_ondevice_quantization.py
index f9d43f1..2b384ec 100644
--- a/test/quantization/jit/test_ondevice_quantization.py
+++ b/test/quantization/jit/test_ondevice_quantization.py
@@ -3,6 +3,15 @@
 import io
 
 import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
+
 import torch._C
 from torch.ao.quantization import default_dynamic_qconfig, per_channel_dynamic_qconfig
 from torch.ao.quantization.quantize_jit import (
