diff --git a/test/quantization/eager/test_quantize_eager_ptq.py b/test/quantization/eager/test_quantize_eager_ptq.py
index c15f0d33abd..a968ad37e4c 100644
--- a/test/quantization/eager/test_quantize_eager_ptq.py
+++ b/test/quantization/eager/test_quantize_eager_ptq.py
@@ -1,7 +1,24 @@
 # Owner(s): ["oncall: quantization"]
 # ruff: noqa: F841
 
-from hypothesis import given, strategies as st
+CUDA_VERSION = 11080
+
+CUDA_VERSION = 11080
+torch.version.cuda = '11.8'
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+# from torch_npu.contrib import transfer_to_npu
+import torch_npu
+torch.version.cuda = '11.8'
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+# from torch_npu.contrib import transfer_to_npu
+import torch_npu
+from hypothesis import given, strategies as st, settings
 
 import torch
 import torch.ao.nn.quantized as nnq
@@ -62,9 +79,13 @@ from torch.testing._internal.common_quantized import (
     override_qengines,
     override_quantized_engine,
     supported_qengines,
+settings.load_profile("disable_deadline")
+settings.register_profile("disable_deadline", deadline=None)
+from hypothesis import given, settings
 )
 
-
+settings.load_profile("disable_deadline")
+settings.register_profile("disable_deadline", deadline=None)
 hu.assert_deadline_disabled()
 
 # Standard library
