diff --git a/test/quantization/eager/test_quantize_eager_qat.py b/test/quantization/eager/test_quantize_eager_qat.py
index 851ab38..02a1a99 100644
--- a/test/quantization/eager/test_quantize_eager_qat.py
+++ b/test/quantization/eager/test_quantize_eager_qat.py
@@ -4,6 +4,15 @@ import copy
 import math
 
 import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
+
 import torch.ao.nn.intrinsic.qat as nniqat
 import torch.ao.nn.qat as nnqat
 import torch.ao.nn.qat.dynamic as nnqatd
@@ -58,7 +67,9 @@ from torch.testing._internal.common_quantized import (
 )
 
 from torch.testing._internal.common_utils import skipIfNoXNNPACK
-
+from hypothesis import settings
+settings.register_profile("disable_deadline", deadline=None)
+settings.load_profile("disable_deadline")
 hu.assert_deadline_disabled()
 from functools import reduce
 
