diff --git a/test/onnx/test_symbolic_helper.py b/test/onnx/test_symbolic_helper.py
index b7358fc..980299e 100644
--- a/test/onnx/test_symbolic_helper.py
+++ b/test/onnx/test_symbolic_helper.py
@@ -2,6 +2,15 @@
 """Unit tests on `torch.onnx.symbolic_helper`."""
 
 import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
+
 from torch.onnx import symbolic_helper
 from torch.onnx._globals import GLOBALS
 from torch.testing._internal import common_utils
