diff --git a/test/onnx/test_utility_funs.py b/test/onnx/test_utility_funs.py
index 387a898..c6e3c4b 100644
--- a/test/onnx/test_utility_funs.py
+++ b/test/onnx/test_utility_funs.py
@@ -20,6 +20,15 @@ from pytorch_test_common import (
 )
 
 import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
+
 import torch.onnx
 import torch.utils.cpp_extension
 from torch.onnx import _constants, OperatorExportTypes, TrainingMode, utils
@@ -1840,7 +1849,7 @@ class TestUtilityFuns(_BaseTestCase):
     def test_deduplicate_initializers_torchscript(self):
         self._test_deduplicate_initializers(torchscript=True)
 
-    @skipIfNoCuda
+    # @skipIfNoCuda
     def test_deduplicate_initializers_diff_devices(self):
         class Model(torch.nn.Module):
             def __init__(self) -> None:
@@ -1848,15 +1857,15 @@ class TestUtilityFuns(_BaseTestCase):
                 self.w_cpu = torch.nn.Parameter(
                     torch.ones(3, device=torch.device("cpu"))
                 )
-                self.w_cuda = torch.nn.Parameter(
-                    torch.ones(3, device=torch.device("cuda"))
+                self.w_npu = torch.nn.Parameter(
+                    torch.ones(3, device=torch.device("npu"))
                 )
 
             def forward(self, x, y):
-                return x + self.w_cpu, y + self.w_cuda
+                return x + self.w_cpu, y + self.w_npu
 
         x = torch.randn(3, 3, device=torch.device("cpu"))
-        y = torch.randn(3, 3, device=torch.device("cuda"))
+        y = torch.randn(3, 3, device=torch.device("npu"))
         f = io.BytesIO()
         torch.onnx.export(Model(), (x, y), f, opset_version=self.opset_version)
         graph = onnx.load(io.BytesIO(f.getvalue()))
