diff --git a/test/onnx/test_op_consistency.py b/test/onnx/test_op_consistency.py
index 762279b..2f2196a 100644
--- a/test/onnx/test_op_consistency.py
+++ b/test/onnx/test_op_consistency.py
@@ -34,13 +34,21 @@ import parameterized
 from onnx_test_common import skip, xfail
 
 import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
+
 from torch.testing._internal import (
     common_device_type,
     common_methods_invocations,
     common_utils,
 )
 
-
 OPS_DB = copy.deepcopy(common_methods_invocations.op_db)
 
 # Modify this section ##########################################################
@@ -281,6 +289,7 @@ class TestOnnxModelOutputConsistency(onnx_test_common._TestONNXRuntime):
     def test_output_match(self, device: str, dtype: torch.dtype, op):
         """Test the ONNX exporter."""
         # device is provided by instantiate_device_type_tests, but we only want to run in cpu.
+        print(f'device: {device}')
         assert device == "cpu"
 
         samples = op.sample_inputs(
@@ -334,7 +343,7 @@ for opset in onnx_test_common.TESTED_OPSETS:
         skip_or_xfails=EXPECTED_SKIPS_OR_FAILS,
     )
     common_device_type.instantiate_device_type_tests(
-        globals()[test_class_name], globals(), only_for="cpu"
+        globals()[test_class_name], globals(), only_for=['cpu']
     )
 
 
