diff --git a/test/onnx/test_verification.py b/test/onnx/test_verification.py
index 4d2b467..e438304 100644
--- a/test/onnx/test_verification.py
+++ b/test/onnx/test_verification.py
@@ -13,6 +13,15 @@ import pytorch_test_common
 from packaging import version
 
 import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
+
 from torch.onnx import _constants, _experimental, verification
 from torch.testing._internal import common_utils
 
