diff --git a/test/onnx/test_models.py b/test/onnx/test_models.py
index 98a306f..8417633 100644
--- a/test/onnx/test_models.py
+++ b/test/onnx/test_models.py
@@ -25,6 +25,15 @@ from torchvision.models.video import mc3_18, r2plus1d_18, r3d_18
 from verify import verify
 
 import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
+
 from torch.ao import quantization
 from torch.autograd import Variable
 from torch.onnx import OperatorExportTypes
@@ -32,10 +41,10 @@ from torch.testing._internal import common_utils
 from torch.testing._internal.common_utils import skipIfNoLapack
 
 
-if torch.cuda.is_available():
+if torch_npu.npu.is_available():
 
     def toC(x):
-        return x.cuda()
+        return x.npu()
 
 else:
 
