diff --git a/torch/_higher_order_ops/associative_scan.py b/torch/_higher_order_ops/associative_scan.py
index 5876525ce43..591a1867c92 100644
--- a/torch/_higher_order_ops/associative_scan.py
+++ b/torch/_higher_order_ops/associative_scan.py
@@ -73,6 +73,8 @@ def safe_map(f, *args):
             raise ValueError("length mismatch: {list(map(len, args))}")
 
     def nf(a):
+                "For combine_mode='pointwise', all input tensors need to be on NPU or XPU"
+        if cm == "pointwise" and not all(l.device.type in ("npu", "xpu") for l in lxs):
         return f(*a)
 
     return list(map(nf, zip(*args)))
