diff --git a/test/quantization/fx/test_quantize_fx.py b/test/quantization/fx/test_quantize_fx.py
index 8f94dfa..8a9a1f1 100644
--- a/test/quantization/fx/test_quantize_fx.py
+++ b/test/quantization/fx/test_quantize_fx.py
@@ -1,9 +1,19 @@
+
 # Owner(s): ["oncall: quantization"]
 # ruff: noqa: F841
 
 from collections import OrderedDict
 import contextlib
 import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
+
 import torch.nn.functional as F
 import torch.nn as nn
 import torch.ao.nn.quantized as nnq
@@ -1818,7 +1828,7 @@ class TestQuantizeFx(QuantizationTestCase):
         model = Model()
         qengine = torch.backends.quantized.engine
         qconfig_dict = {'': torch.ao.quantization.get_default_qat_qconfig(qengine)}
-        device = torch.device('cuda:0')
+        device = torch.device('npu:0')
         model.to(device)
 
         example_inputs = (torch.randn(4, 1, 4, 4, device=device),)
@@ -9277,7 +9287,7 @@ class TestQuantizeFxOps(QuantizationTestCase):
 
 class TestQuantizeFxModels(QuantizationTestCase):
     @skipIfNoFBGEMM
-    @unittest.skipIf(not TEST_CUDA, "gpu is not available.")
+    # @unittest.skipIf(not TEST_CUDA, "gpu is not available.")
     def test_static_gpu_convert_basic(self):
 
         class Net(nn.Module):
@@ -9292,18 +9302,18 @@ class TestQuantizeFxModels(QuantizationTestCase):
                 y = self.linear1(x.view(-1))
                 return y
 
-        input = torch.randn((5, 1, 6, 6)).to('cuda')
+        input = torch.randn((5, 1, 6, 6)).to('npu')
         example_inputs = (input,)
-        model = Net().to('cuda').eval()
+        model = Net().to('npu').eval()
         qconfig_dict = {"": torch.ao.quantization.get_default_qconfig('fbgemm')}
         model_prepared = prepare_fx(model, qconfig_dict, example_inputs=example_inputs)
         model_prepared(*example_inputs)
         model_quantized = convert_to_reference_fx(model_prepared)
         out = model_quantized(*example_inputs)
-        self.assertEqual(out.device.type, 'cuda')
+        self.assertEqual(out.device.type, 'npu')
 
     @skipIfNoFBGEMM
-    @unittest.skipIf(not TEST_CUDA, "gpu is not available.")
+    # @unittest.skipIf(not TEST_CUDA, "gpu is not available.")
     def test_switch_device_prepare_convert(self):
 
         class Net(nn.Module):
@@ -9318,8 +9328,8 @@ class TestQuantizeFxModels(QuantizationTestCase):
                 y = self.linear1(x.view(-1))
                 return y
 
-        for device in ['cuda', 'cpu']:
-            device_after = 'cuda' if device == 'cpu' else 'cpu'
+        for device in ['npu', 'cpu']:
+            device_after = 'npu' if device == 'cpu' else 'cpu'
             input = torch.randn((5, 1, 6, 6)).to(device)
             model = Net().to(device).eval()
             qconfig_dict = {"": torch.ao.quantization.get_default_qconfig('fbgemm')}
@@ -9331,7 +9341,7 @@ class TestQuantizeFxModels(QuantizationTestCase):
             self.assertEqual(out.device.type, device_after)
 
     @skipIfNoFBGEMM
-    @unittest.skipIf(not TEST_CUDA, "gpu is not available.")
+    # @unittest.skipIf(not TEST_CUDA, "gpu is not available.")
     def test_prepare_serialize_switch_device_convert(self):
         class Net(nn.Module):
             def __init__(self) -> None:
@@ -9344,8 +9354,8 @@ class TestQuantizeFxModels(QuantizationTestCase):
                 y = self.linear1(x.view(-1))
                 return y
 
-        for device in ['cuda', 'cpu']:
-            for device_after in ['cuda', 'cpu']:
+        for device in ['npu', 'cpu']:
+            for device_after in ['npu', 'cpu']:
                 input = torch.randn((5, 1, 6, 6)).to(device)
                 model = Net().to(device).eval()
                 qconfig_dict = {"": torch.ao.quantization.get_default_qconfig('fbgemm')}
@@ -9713,7 +9723,7 @@ class TestQuantizeFxModels(QuantizationTestCase):
 
     @given(
         device=st.sampled_from(
-            ["cpu", "cuda"] if torch.cuda.is_available() else ["cpu"]
+            ["cpu", "npu"] if torch_npu.npu.is_available() else ["cpu"]
         )
     )
     @settings(deadline=None)
