diff --git a/test/quantization/fx/test_numeric_suite_fx.py b/test/quantization/fx/test_numeric_suite_fx.py
index 84c4f84..5a4c203 100644
--- a/test/quantization/fx/test_numeric_suite_fx.py
+++ b/test/quantization/fx/test_numeric_suite_fx.py
@@ -1,3 +1,5 @@
+
+
 # Owner(s): ["oncall: quantization"]
 # ruff: noqa: F841
 
@@ -7,6 +9,15 @@ import operator
 import unittest
 
 import torch
+import torch_npu
+# from torch_npu.contrib import transfer_to_npu
+torch._C._cuda_setStream = torch_npu._C._npu_setStream
+torch.cuda.get_device_capability = lambda *args, **kwargs:(10,0)
+torch._C._cuda_setDevice = torch_npu._C._npu_setDevice
+torch._C._cuda_getCompiledVersion = lambda:11080
+torch.version.cuda = '11.8'
+CUDA_VERSION = 11080
+
 import torch.nn as nn
 import torch.nn.functional as F
 from torch.ao.quantization import (
@@ -1949,26 +1960,26 @@ class TestFXNumericSuiteCoreAPIs(FXNumericSuiteQuantizationTestCase):
         ref_shadow = mc_shadows_mp(*example_inputs)
         self.assertEqual(ref_fp32, ref_shadow)
 
-    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
-    def test_extract_weights_cuda(self):
+    # @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    def test_extract_weights_npu(self):
         # Note: this is not using quantization because quantized kernels do not
         # work on cuda yet.
-        m1 = nn.Sequential(nn.Conv2d(1, 1, 1)).cuda()
-        m2 = nn.Sequential(nn.Conv2d(1, 1, 1)).cuda()
+        m1 = nn.Sequential(nn.Conv2d(1, 1, 1)).npu()
+        m2 = nn.Sequential(nn.Conv2d(1, 1, 1)).npu()
         results = extract_weights('a', m1, 'b', m2)
         extend_logger_results_with_comparison(
             results, 'a', 'b', compute_sqnr, 'sqnr')
         self.assert_ns_compare_dict_valid(results)
 
-    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
-    def test_add_loggers_cuda(self):
+    # @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    def test_add_loggers_npu(self):
         # Note: this is not using quantization because quantized kernels do not
         # work on cuda yet.
-        m1 = nn.Sequential(nn.Conv2d(1, 1, 1)).cuda()
-        m2 = nn.Sequential(nn.Conv2d(1, 1, 1)).cuda()
+        m1 = nn.Sequential(nn.Conv2d(1, 1, 1)).npu()
+        m2 = nn.Sequential(nn.Conv2d(1, 1, 1)).npu()
         m1_ns, m2_ns = add_loggers('a', m1, 'b', m2, OutputLogger)
         datum = torch.randn(1, 1, 1, 1)
-        datum = datum.cuda()
+        datum = datum.npu()
 
         m1_ns(datum)
         m2_ns(datum)
@@ -1977,15 +1988,15 @@ class TestFXNumericSuiteCoreAPIs(FXNumericSuiteQuantizationTestCase):
         extend_logger_results_with_comparison(
             act_compare_dict, 'a', 'b', compute_sqnr, 'sqnr')
 
-    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
-    def test_add_shadow_loggers_cuda(self):
+    # @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    def test_add_shadow_loggers_npu(self):
         # Note: this is not using quantization because quantized kernels do not
         # work on cuda yet.
-        m1 = nn.Sequential(nn.Conv2d(1, 1, 1)).cuda()
-        m2 = nn.Sequential(nn.Conv2d(1, 1, 1)).cuda()
+        m1 = nn.Sequential(nn.Conv2d(1, 1, 1)).npu()
+        m2 = nn.Sequential(nn.Conv2d(1, 1, 1)).npu()
         m1_shadows_m2 = add_shadow_loggers('a', m1, 'b', m2, OutputLogger)
         datum = torch.randn(1, 1, 1, 1)
-        datum = datum.cuda()
+        datum = datum.npu()
 
         m1_shadows_m2(datum)
 
