diff --git a/torch/nn/parallel/data_parallel.py b/torch/nn/parallel/data_parallel.py
index 16537624663..cab2426662c 100644
--- a/torch/nn/parallel/data_parallel.py
+++ b/torch/nn/parallel/data_parallel.py
@@ -44,7 +44,7 @@ def _check_balance(device_ids: Sequence[int | torch.device]) -> None:
 
     if warn_imbalance(lambda props: props.total_memory):
         return
-    if warn_imbalance(lambda props: props.multi_processor_count):
+    if warn_imbalance(lambda props: props.vector_core_num):
         return
 
 
