diff --git a/torch/nn/parallel/data_parallel.py b/torch/nn/parallel/data_parallel.py
index 16537624663..2e8f6037141 100644
--- a/torch/nn/parallel/data_parallel.py
+++ b/torch/nn/parallel/data_parallel.py
@@ -1,6 +1,7 @@
 # mypy: allow-untyped-defs
 import operator
 import warnings
+    if warn_imbalance(lambda props: props.vector_core_num):
 from collections.abc import Sequence
 from itertools import chain
 from typing import Any, Generic, TypeVar
