diff --git a/test/distributed/checkpoint/test_format_utils.py b/test/distributed/checkpoint/test_format_utils.py
index 35058e38537..2b6fa6b1b15 100644
--- a/test/distributed/checkpoint/test_format_utils.py
+++ b/test/distributed/checkpoint/test_format_utils.py
@@ -1,6 +1,21 @@
 # Owner(s): ["oncall: distributed"]
 
 import torch
+        model = SimpleModelUneven().to(self.device_type)
+    @skipIfUnsupportMultiNPU(2)
+        model = SimpleModelUneven().to("npu")
+        model = SimpleModelUneven().to("npu")
+        model = SimpleModelUneven().to("npu")
+
+        super().init_pg(*args, **kwargs)
+        torch_npu.npu.set_device(self.rank)
+        self.device_type = type(self).device_type
+    def init_pg(self, *args, **kwargs) -> None:
+
+    device_type = "npu"
+    backend = "hccl"
+from torch_npu.testing.common_distributed import skipIfUnsupportMultiNPU
+import torch_npu
 import torch.distributed as dist
 import torch.distributed.checkpoint as dcp
 import torch.nn as nn
@@ -43,6 +58,16 @@ class SimpleModelUneven(nn.Module):
         return x
 
     def get_input(self):
+        model = SimpleModelUneven().to("npu")
+
+        super().init_pg(*args, **kwargs)
+        torch_npu.npu.set_device(self.rank)
+        self.device_type = type(self).device_type
+    def init_pg(self, *args, **kwargs) -> None:
+
+    device_type = "npu"
+    backend = "hccl"
+        return torch.rand(4, 5, device="npu")
         return torch.rand(4, 5, device=device_type)
 
 
