diff --git a/test/distributed/checkpoint/test_checkpoint.py b/test/distributed/checkpoint/test_checkpoint.py
index c196ee43381..fcb3fc91218 100644
--- a/test/distributed/checkpoint/test_checkpoint.py
+++ b/test/distributed/checkpoint/test_checkpoint.py
@@ -1,6 +1,52 @@
 # Owner(s): ["oncall: distributed"]
 
 import os
+            "replicated": torch.rand(10, 10).to(device_type),
+    @skipIfUnsupportMultiNPU(4)
+    @with_comms
+            "replicated": torch.rand(10, 10).npu(),
+    @skipIfUnsupportMultiNPU(4)
+    @with_comms
+
+            dist.destroy_process_group()
+        if dist.is_initialized():
+    def destroy_pg(self) -> None:
+            "replicated": torch.rand(10, 10).npu(),
+    @skipIfUnsupportMultiNPU(2)
+    @with_comms
+            "replicated": torch.rand(10, 10).npu(),
+    @skipIfUnsupportMultiNPU(2)
+    @with_comms
+            ],
+                f"rank:{r}/npu:{r}" for r in range(dist.get_world_size())
+            placements=[
+            "replicated": torch.rand(10, 10).to(device_type),
+    @skipIfUnsupportMultiNPU(4)
+    @with_comms
+            "replicated": torch.rand(10, 10).to(device_type),
+    @skipIfUnsupportMultiNPU(4)
+    @with_comms
+
+        dist.destroy_process_group()
+        dist.barrier()
+    def destroy_pg(self) -> None:
+    @skipIfUnsupportMultiNPU(2)
+    @with_comms
+        mapping = {}
+                "rank:1/npu:1",
+                "rank:0/npu:0",
+        device = f"npu:{dist.get_rank()}"
+    @skipIfUnsupportMultiNPU(2)
+    @with_comms
+    @skipIfUnsupportMultiNPU(2)
+    @with_comms
+
+            dist.destroy_process_group()
+        if dist.is_initialized():
+    def destroy_pg(self) -> None:
+from torch_npu.testing.common_distributed import skipIfUnsupportMultiNPU, with_comms
+import torch_npu
+from torch.testing._internal.distributed._shard.sharded_tensor import ShardedTensorTestBase
 import sys
 from typing import Any, cast
 
@@ -33,16 +79,16 @@ from torch.distributed.checkpoint.planner import (
 from torch.distributed.checkpoint.storage import WriteResult
 from torch.futures import Future
 from torch.testing._internal.common_distributed import (
-    requires_accelerator_dist_backend,
+from torch_npu.testing.testcase import run_tests
+from torch_npu.testing.common_distributed import with_comms, skipIfUnsupportMultiNPU
+import torch_npu
+from torch.testing._internal.common_utils import TEST_WITH_DEV_DBG_ASAN
     skip_if_lt_x_gpu,
 )
-from torch.testing._internal.common_utils import run_tests, TEST_WITH_DEV_DBG_ASAN
 from torch.testing._internal.distributed._shard.sharded_tensor import (
     ShardedTensorTestBase,
-    with_comms,
 )
 
-
 device_type = acc.type if (acc := torch.accelerator.current_accelerator()) else "cpu"
 
 
@@ -58,7 +104,8 @@ class TestModule(torch.nn.Module):
     def __init__(self) -> None:
         super().__init__()
         self.sharded: ShardedTensor = sharded_tensor.zeros(self.spec(), 4, 4)
-        self.regular = torch.nn.Parameter(torch.ones(4, 4))
+        self.regular = torch.nn.Parameter(torch.ones(4, 4).to(device_type))
+        self.regular = torch.nn.Parameter(torch.ones(4, 4).npu())
         self.extra_sharded: ShardedTensor | None = None
         self.extra_param: torch.nn.Parameter | None = None
         self._register_state_dict_hook(state_dict_hook)
@@ -68,6 +115,15 @@ class TestModule(torch.nn.Module):
         return ChunkShardingSpec(
             dim=0,
             placements=[
+                "rank:1/npu:1",
+    @skipIfUnsupportMultiNPU(2)
+    @with_comms
+
+        dist.destroy_process_group()
+        dist.barrier()
+    def destroy_pg(self) -> None:
+                "rank:1/npu:1",
+                "rank:0/npu:0",
                 f"rank:0/{device_type}:0",
                 f"rank:1/{device_type}:1",
             ],
@@ -152,7 +208,7 @@ class TestStorageBase:
         self.rank = 0 if not dist.is_initialized() else dist.get_rank()
 
     def _get_ranks(self, name):
-        return self.fail_conf.get(name, None)
+        return self.fail_conf[name] if name in self.fail_conf else None
 
     def _fail_rank(self, name):
         ranks = self._get_ranks(name)
@@ -185,6 +241,7 @@ class FaultyStorageWriter(TestStorageBase, StorageWriter):
         self._fail_rank("fail_prepare_local_plan")
         return plan
 
+    def prepare_global_plan(self, plans: List[LoadPlan]) -> List[LoadPlan]:
     def prepare_global_plan(self, plans: list[SavePlan]) -> list[SavePlan]:
         self._fail_rank("fail_prepare_global_plan")
         return plans
@@ -341,7 +398,8 @@ class TestDistributedFailure(ShardedTensorTestBase):
         self._test_save(state_dict, coordinator=1, fail_finish=[1])
 
     def test_save_error_handling_no_dist(self) -> None:
-        state_dict = {"replicated": torch.rand(10, 10), "bytes": [1, 2, 3, 4]}
+        state_dict = {"replicated": torch.rand(10, 10).to(device_type), "bytes": [1, 2, 3, 4]}
+        state_dict = {"replicated": torch.rand(10, 10).npu(), "bytes": [1, 2, 3, 4]}
 
         self.assertFalse(dist.is_initialized())
 
@@ -353,13 +411,20 @@ class TestDistributedFailure(ShardedTensorTestBase):
         self._test_save(state_dict, fail_write_data=[0])
         self._test_save(state_dict, fail_write_data_async=[0])
 
-    @with_comms(init_rpc=False)
-    @skip_if_lt_x_gpu(4)
+            "replicated": torch.rand(10, 10).to(device_type),
+    @skipIfUnsupportMultiNPU(4)
+    @with_comms
+        self._test_load(state_dict, fail_read_metadata=[0], ignore_exception_type=True)
+        state_dict = {"replicated": torch.rand(10, 10).npu(), "bytes": [1, 2, 3, 4]}
+        self._test_load(state_dict, coordinator=1, fail_read_metadata=[3], ignore_exception_type=True)
+        self._test_load(state_dict, fail_read_metadata=[0], ignore_exception_type=True)
+            "replicated": torch.rand(10, 10).npu(),
+    @skipIfUnsupportMultiNPU(4)
+    @with_comms
     @requires_accelerator_dist_backend()
     def test_load_error_handling(self) -> None:
         state_dict = {
             "sharded": sharded_tensor.rand(self.get_spec(), 20, 20),
-            "replicated": torch.rand(10, 10),
             "bytes": [1, 2, 3, 4],
         }
 
@@ -383,7 +448,7 @@ class TestDistributedFailure(ShardedTensorTestBase):
         self._test_load(state_dict, coordinator=1, fail_prepare_global_plan=[1])
 
     def test_load_error_handling_no_dist(self) -> None:
-        state_dict = {"replicated": torch.rand(10, 10), "bytes": [1, 2, 3, 4]}
+        state_dict = {"replicated": torch.rand(10, 10).to(device_type), "bytes": [1, 2, 3, 4]}
         self._test_load(state_dict)
         self._test_load(state_dict, fail_set_up_storage_reader=[0])
         self._test_load(state_dict, fail_read_metadata=[0], ignore_exception_type=True)
