diff --git a/test/distributed/checkpoint/test_state_dict.py b/test/distributed/checkpoint/test_state_dict.py
index 6bfea439468..aa0ff117595 100644
--- a/test/distributed/checkpoint/test_state_dict.py
+++ b/test/distributed/checkpoint/test_state_dict.py
@@ -51,7 +51,7 @@ from torch.testing._internal.common_utils import run_tests, TEST_WITH_DEV_DBG_AS
 from torch.testing._internal.distributed._tensor.common_dtensor import (
     DTensorTestBase,
     MultiProcessTestCase,
-    with_comms,
+    # with_comms,
 )
 from torch.testing._internal.distributed.common_state_dict import (
     FusionEmbedding,
@@ -60,6 +60,8 @@ from torch.testing._internal.distributed.common_state_dict import (
     VerifyStateDictMixin,
 )
 from torch.utils._pytree import tree_all, tree_all_only
+from torch_npu.testing.common_distributed import with_comms
+from torch_npu.testing.common_distributed import with_comms, skipIfUnsupportMultiNPU
 
 
 device_type = acc.type if (acc := torch.accelerator.current_accelerator()) else "cpu"
@@ -823,7 +825,8 @@ class TestStateDict(DTensorTestBase, VerifyStateDictMixin):
         self._test_deprecate_fsdp_api()
 
     @with_comms
-    @skip_if_lt_x_gpu(2)
+    @skipIfUnsupportMultiNPU(2)
+    # @skip_if_lt_x_gpu(2)
     def test_shared_weight(self):
         class TiedEmbeddingModel(nn.Module):
             def __init__(self, vocab_size, embedding_dim):
