diff --git a/test/distributed/checkpoint/test_format_utils.py b/test/distributed/checkpoint/test_format_utils.py
index dd83da0..27df89d 100644
--- a/test/distributed/checkpoint/test_format_utils.py
+++ b/test/distributed/checkpoint/test_format_utils.py
@@ -1,6 +1,7 @@
 # Owner(s): ["oncall: distributed"]
 
 import torch
+import torch_npu
 import torch.distributed as dist
 import torch.distributed.checkpoint as dcp
 import torch.nn as nn
@@ -13,7 +14,7 @@ from torch.distributed.checkpoint.format_utils import (
     torch_save_to_dcp,
 )
 from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
-from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+from torch_npu.testing.common_distributed import skipIfUnsupportMultiNPU
 from torch.testing._internal.common_utils import run_tests
 from torch.testing._internal.distributed._tensor.common_dtensor import (
     DTensorTestBase,
@@ -40,13 +41,21 @@ class SimpleModelUneven(nn.Module):
         return x
 
     def get_input(self):
-        return torch.rand(4, 5, device="cuda")
+        return torch.rand(4, 5, device="npu")
 
 
 class TestFormatUtils(DTensorTestBase):
+    backend = "hccl"
+    device_type = "npu"
+
+    def init_pg(self, *args, **kwargs) -> None:
+        self.device_type = type(self).device_type
+        torch_npu.npu.set_device(self.rank)
+        super().init_pg(*args, **kwargs)
+
     @with_temp_dir
     def test_dcp_to_torch_save(self) -> None:
-        model = SimpleModelUneven()
+        model = SimpleModelUneven().to("npu")
         dcp.save({"model": model}, checkpoint_id=self.temp_dir)
 
         torch_path = self.temp_dir + "/model.pt"
@@ -57,21 +66,21 @@ class TestFormatUtils(DTensorTestBase):
 
     @with_temp_dir
     def test_torch_save_to_dcp(self) -> None:
-        model = SimpleModelUneven()
+        model = SimpleModelUneven().to("npu")
         sd = {"model": model.state_dict()}
         torch_path = self.temp_dir + "/model.pt"
         torch.save(sd, torch_path)
 
         torch_save_to_dcp(torch_path, self.temp_dir)
 
-        model = SimpleModelUneven()
+        model = SimpleModelUneven().to("npu")
         dcp.load({"model": model}, checkpoint_id=self.temp_dir)
 
         self.assertEqual({"model": model.state_dict()}, sd)
 
     @with_comms
     @with_temp_dir
-    @skip_if_lt_x_gpu(2)
+    @skipIfUnsupportMultiNPU(2)
     def test_online_torch_save_to_dcp(self) -> None:
         """Tests loading a model saved by torch.save directly into a sharded model
         using dcp.load
@@ -87,7 +96,7 @@ class TestFormatUtils(DTensorTestBase):
 
         # Load into a sharded model
         device_mesh = init_device_mesh(self.device_type, (self.world_size,))
-        model = SimpleModelUneven().cuda()
+        model = SimpleModelUneven().to(self.device_type)
         model = FSDP(
             model,
             device_mesh=device_mesh,
