diff --git a/test/distributed/checkpoint/test_format_utils.py b/test/distributed/checkpoint/test_format_utils.py
index dd83da0..c01b713 100644
--- a/test/distributed/checkpoint/test_format_utils.py
+++ b/test/distributed/checkpoint/test_format_utils.py
@@ -1,3 +1,8 @@
+import torch_npu.testing
+
+import torch_npu
+from torch_npu.contrib import transfer_to_npu
+
 # Owner(s): ["oncall: distributed"]
 
 import torch
@@ -17,9 +22,10 @@ from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
 from torch.testing._internal.common_utils import run_tests
 from torch.testing._internal.distributed._tensor.common_dtensor import (
     DTensorTestBase,
-    with_comms,
+    # with_comms,
 )
 from torch.testing._internal.distributed.checkpoint_utils import with_temp_dir
+from torch_npu.testing.common_distributed import with_comms
 
 
 class SimpleModelUneven(nn.Module):
